{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BRCA Exchange Literature Search Stats & Export\n",
    "\n",
    "Ingest artifacts from correlate.py, generate literature.json for import into BRCA Exchange as well as various statistics.\n",
    "\n",
    "Notes:\n",
    "* pmid is coerced into a string throughout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import json\n",
    "import sqlite3\n",
    "import pandas as pd\n",
    "\n",
    "os.chdir(os.path.expanduser(\"/notebooks/data/pubmunch/crawl-14-11-2018/\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download Stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Attempted to download 16980 PMIDs with BRCA in the title or abstract\n"
     ]
    }
   ],
   "source": [
    "with open(\"download/pmids.txt\") as f:\n",
    "    pmids = f.read().splitlines()\n",
    "print(\"Attempted to download {} PMIDs with BRCA in the title or abstract\".format(len(pmids)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed to download 2753(16%) of 14088 papers.\n",
      "Reasons:\n",
      "OK                     14088\n",
      "invalidPdf               978\n",
      "noCrawlerSuccess         817\n",
      "httpError                302\n",
      "noLicense                301\n",
      "HighwirePdfNotValid      254\n",
      "invalidHostname           55\n",
      "pageErrorMessage          28\n",
      "no_meta                    9\n",
      "noOutlinkOrDoi             6\n",
      "HtmlParseError             1\n",
      "BeautifulSoupError         1\n",
      "tooManySupplFiles          1\n",
      "Name: status, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "download = pd.read_table(\"download/docStatus.tab\", header=None, index_col=0, dtype=str,\n",
    "                         names=[\"pmid\", \"status\", \"msg\", \"crawler\", \"journal\", \"year\", \"numFiles\", \"detail\"])\n",
    "print(\"Failed to download {}({}%) of {} papers.\".format(\n",
    "    download[download.status != \"OK\"].shape[0], \n",
    "    round(100 * download[download.status != \"OK\"].shape[0]/len(pmids)),\n",
    "      download[download.status == \"OK\"].shape[0]\n",
    "     ))\n",
    "print(\"Reasons:\")\n",
    "print(download.status.value_counts())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ingest\n",
    "Ingest artifacts from various pieces of the pipeline indexed by normalized genomic hgvs and integer pmid "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14044 articles loaded from the articles sqlite database\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>articleId</th>\n",
       "      <th>externalId</th>\n",
       "      <th>source</th>\n",
       "      <th>publisher</th>\n",
       "      <th>origFile</th>\n",
       "      <th>journal</th>\n",
       "      <th>printIssn</th>\n",
       "      <th>eIssn</th>\n",
       "      <th>journalUniqueId</th>\n",
       "      <th>year</th>\n",
       "      <th>...</th>\n",
       "      <th>issue</th>\n",
       "      <th>page</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pmcId</th>\n",
       "      <th>doi</th>\n",
       "      <th>fulltextUrl</th>\n",
       "      <th>time</th>\n",
       "      <th>offset</th>\n",
       "      <th>size</th>\n",
       "      <th>chunkId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5011585672</td>\n",
       "      <td>PMID11585672</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Trends in genetics : TIG</td>\n",
       "      <td>0168-9525</td>\n",
       "      <td>0168-9525</td>\n",
       "      <td>8507085</td>\n",
       "      <td>2001</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>S18</td>\n",
       "      <td>11585672</td>\n",
       "      <td></td>\n",
       "      <td>10.1016/s0168-9525(01)02451-9</td>\n",
       "      <td>https://linkinghub.elsevier.com/retrieve/pii/S...</td>\n",
       "      <td>2018-11-14T16:29:09+0000</td>\n",
       "      <td>78</td>\n",
       "      <td>5258</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5019688261</td>\n",
       "      <td>PMID19688261</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Breast cancer research and treatment</td>\n",
       "      <td>0167-6806</td>\n",
       "      <td>1573-7217</td>\n",
       "      <td>8111104</td>\n",
       "      <td>2010</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>575</td>\n",
       "      <td>19688261</td>\n",
       "      <td></td>\n",
       "      <td>10.1007/s10549-009-0501-3</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:29:16+0000</td>\n",
       "      <td>5048</td>\n",
       "      <td>100825</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5012228710</td>\n",
       "      <td>PMID12228710</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Science (New York, N.Y.)</td>\n",
       "      <td>0036-8075</td>\n",
       "      <td>1095-9203</td>\n",
       "      <td>0404511</td>\n",
       "      <td>2002</td>\n",
       "      <td>...</td>\n",
       "      <td>5588</td>\n",
       "      <td>1837</td>\n",
       "      <td>12228710</td>\n",
       "      <td></td>\n",
       "      <td>10.1126/science.297.5588.1837</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:29:40+0000</td>\n",
       "      <td>1436851129</td>\n",
       "      <td>148950</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5029369605</td>\n",
       "      <td>PMID29369605</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Genetika</td>\n",
       "      <td>0016-6758</td>\n",
       "      <td>0016-6758</td>\n",
       "      <td>0047354</td>\n",
       "      <td>2016</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>1215</td>\n",
       "      <td>29369605</td>\n",
       "      <td></td>\n",
       "      <td>10.1134/s102279541609012x</td>\n",
       "      <td>https://link.springer.com/article/10.1134%2FS1...</td>\n",
       "      <td>2018-11-14T16:30:43+0000</td>\n",
       "      <td>4458079378</td>\n",
       "      <td>22264</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5022084640</td>\n",
       "      <td>PMID22084640</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Therapeutic advances in medical oncology</td>\n",
       "      <td>1758-8340</td>\n",
       "      <td>1758-8359</td>\n",
       "      <td>101510808</td>\n",
       "      <td>2011</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>257</td>\n",
       "      <td>22084640</td>\n",
       "      <td>3210467</td>\n",
       "      <td>10.1177/1758834011417039</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:30:58+0000</td>\n",
       "      <td>6079002966</td>\n",
       "      <td>93229</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    articleId    externalId source publisher origFile  \\\n",
       "0  5011585672  PMID11585672         download            \n",
       "1  5019688261  PMID19688261         download            \n",
       "2  5012228710  PMID12228710         download            \n",
       "3  5029369605  PMID29369605         download            \n",
       "4  5022084640  PMID22084640         download            \n",
       "\n",
       "                                    journal  printIssn      eIssn  \\\n",
       "0                  Trends in genetics : TIG  0168-9525  0168-9525   \n",
       "1      Breast cancer research and treatment  0167-6806  1573-7217   \n",
       "2                  Science (New York, N.Y.)  0036-8075  1095-9203   \n",
       "3                                  Genetika  0016-6758  0016-6758   \n",
       "4  Therapeutic advances in medical oncology  1758-8340  1758-8359   \n",
       "\n",
       "  journalUniqueId  year   ...    issue  page      pmid    pmcId  \\\n",
       "0         8507085  2001   ...       10   S18  11585672            \n",
       "1         8111104  2010   ...        3   575  19688261            \n",
       "2         0404511  2002   ...     5588  1837  12228710            \n",
       "3         0047354  2016   ...       10  1215  29369605            \n",
       "4       101510808  2011   ...        6   257  22084640  3210467   \n",
       "\n",
       "                             doi  \\\n",
       "0  10.1016/s0168-9525(01)02451-9   \n",
       "1      10.1007/s10549-009-0501-3   \n",
       "2  10.1126/science.297.5588.1837   \n",
       "3      10.1134/s102279541609012x   \n",
       "4       10.1177/1758834011417039   \n",
       "\n",
       "                                         fulltextUrl  \\\n",
       "0  https://linkinghub.elsevier.com/retrieve/pii/S...   \n",
       "1                                                      \n",
       "2                                                      \n",
       "3  https://link.springer.com/article/10.1134%2FS1...   \n",
       "4                                                      \n",
       "\n",
       "                       time      offset    size  chunkId  \n",
       "0  2018-11-14T16:29:09+0000          78    5258  0_00000  \n",
       "1  2018-11-14T16:29:16+0000        5048  100825  0_00000  \n",
       "2  2018-11-14T16:29:40+0000  1436851129  148950  0_00000  \n",
       "3  2018-11-14T16:30:43+0000  4458079378   22264  0_00000  \n",
       "4  2018-11-14T16:30:58+0000  6079002966   93229  0_00000  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "connection = sqlite3.connect(\"file:text/articles.db?mode=ro\", uri=True)\n",
    "articles = pd.read_sql_query(\"SELECT * FROM articles\", connection)\n",
    "articles.pmid = articles.pmid.astype(str)\n",
    "print(\"{} articles loaded from the articles sqlite database\".format(articles.shape[0]))\n",
    "articles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20934 variants in the BRCA Exchange database\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>pyhgvs_Genomic_Coordinate_38</th>\n",
       "      <th>pyhgvs_cDNA</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>norm_g_hgvs</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32314943A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32314943</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32314943:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.-764A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32315226G&gt;A</th>\n",
       "      <td>13</td>\n",
       "      <td>32315226</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr13:g.32315226:G&gt;A</td>\n",
       "      <td>NM_000059.3:c.-481G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32315490C&gt;T</th>\n",
       "      <td>13</td>\n",
       "      <td>32315490</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr13:g.32315490:C&gt;T</td>\n",
       "      <td>NM_000059.3:c.-217C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32315519C&gt;T</th>\n",
       "      <td>13</td>\n",
       "      <td>32315519</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr13:g.32315519:C&gt;T</td>\n",
       "      <td>NM_000059.3:c.-188C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32315532C&gt;T</th>\n",
       "      <td>13</td>\n",
       "      <td>32315532</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr13:g.32315532:C&gt;T</td>\n",
       "      <td>NM_000059.3:c.-175C&gt;T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Chr       Pos Ref Alt  \\\n",
       "norm_g_hgvs                                         \n",
       "NC_000013.11:g.32314943A>G   13  32314943   A   G   \n",
       "NC_000013.11:g.32315226G>A   13  32315226   G   A   \n",
       "NC_000013.11:g.32315490C>T   13  32315490   C   T   \n",
       "NC_000013.11:g.32315519C>T   13  32315519   C   T   \n",
       "NC_000013.11:g.32315532C>T   13  32315532   C   T   \n",
       "\n",
       "                           pyhgvs_Genomic_Coordinate_38            pyhgvs_cDNA  \n",
       "norm_g_hgvs                                                                     \n",
       "NC_000013.11:g.32314943A>G         chr13:g.32314943:A>G  NM_000059.3:c.-764A>G  \n",
       "NC_000013.11:g.32315226G>A         chr13:g.32315226:G>A  NM_000059.3:c.-481G>A  \n",
       "NC_000013.11:g.32315490C>T         chr13:g.32315490:C>T  NM_000059.3:c.-217C>T  \n",
       "NC_000013.11:g.32315519C>T         chr13:g.32315519:C>T  NM_000059.3:c.-188C>T  \n",
       "NC_000013.11:g.32315532C>T         chr13:g.32315532:C>T  NM_000059.3:c.-175C>T  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants = pd.read_table(\"variants-normalized.tsv\", index_col=\"norm_g_hgvs\")\n",
    "print(\"{} variants in the BRCA Exchange database\".format(variants.shape[0]))\n",
    "variants.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total normalized mentions: 442198\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>norm_c_hgvs</th>\n",
       "      <th>pmid</th>\n",
       "      <th>snippet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>norm_g_hgvs</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NC_000007.14:g.96709760A&gt;C</th>\n",
       "      <td>NM_006304.1:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000004.12:g.56655997A&gt;C</th>\n",
       "      <td>NM_001145459.1:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000004.12:g.56655997A&gt;C</th>\n",
       "      <td>NM_139212.3:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000004.12:g.56655997A&gt;C</th>\n",
       "      <td>NM_139211.4:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000007.14:g.96709760A&gt;C</th>\n",
       "      <td>NM_006304.1:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      norm_c_hgvs      pmid  \\\n",
       "norm_g_hgvs                                                   \n",
       "NC_000007.14:g.96709760A>C     NM_006304.1:c.4T>G  12228710   \n",
       "NC_000004.12:g.56655997A>C  NM_001145459.1:c.4T>G  12228710   \n",
       "NC_000004.12:g.56655997A>C     NM_139212.3:c.4T>G  12228710   \n",
       "NC_000004.12:g.56655997A>C     NM_139211.4:c.4T>G  12228710   \n",
       "NC_000007.14:g.96709760A>C     NM_006304.1:c.4T>G  12228710   \n",
       "\n",
       "                                                                      snippet  \n",
       "norm_g_hgvs                                                                    \n",
       "NC_000007.14:g.96709760A>C  ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "NC_000004.12:g.56655997A>C  ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "NC_000004.12:g.56655997A>C  ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "NC_000004.12:g.56655997A>C  ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "NC_000007.14:g.96709760A>C  ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mentions = pd.read_table(\"mentions-normalized.tsv\", index_col=\"norm_g_hgvs\")\n",
    "mentions.pmid = mentions.pmid.astype(str)\n",
    "print(\"Total normalized mentions: {}\".format(mentions.shape[0]))\n",
    "mentions.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Wrangle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "After dropping duplicates of hgvs+pmid+snippet: 296420\n",
      "Total unique genomic hgvs variants: 188514\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>norm_c_hgvs</th>\n",
       "      <th>pmid</th>\n",
       "      <th>snippet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>norm_g_hgvs</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NC_000007.14:g.96709760A&gt;C</th>\n",
       "      <td>NM_006304.1:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000004.12:g.56655997A&gt;C</th>\n",
       "      <td>NM_001145459.1:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000007.14:g.96709760A&gt;C</th>\n",
       "      <td>NM_006304.1:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000004.12:g.56655997A&gt;C</th>\n",
       "      <td>NM_001145459.1:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32326569del</th>\n",
       "      <td>NM_000059.3:c.587del</td>\n",
       "      <td>29369605</td>\n",
       "      <td>Gene  Gene function in the cell and involveme...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      norm_c_hgvs      pmid  \\\n",
       "norm_g_hgvs                                                   \n",
       "NC_000007.14:g.96709760A>C     NM_006304.1:c.4T>G  12228710   \n",
       "NC_000004.12:g.56655997A>C  NM_001145459.1:c.4T>G  12228710   \n",
       "NC_000007.14:g.96709760A>C     NM_006304.1:c.4T>G  12228710   \n",
       "NC_000004.12:g.56655997A>C  NM_001145459.1:c.4T>G  12228710   \n",
       "NC_000013.11:g.32326569del   NM_000059.3:c.587del  29369605   \n",
       "\n",
       "                                                                      snippet  \n",
       "norm_g_hgvs                                                                    \n",
       "NC_000007.14:g.96709760A>C  ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "NC_000004.12:g.56655997A>C  ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "NC_000007.14:g.96709760A>C  ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...  \n",
       "NC_000004.12:g.56655997A>C  ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...  \n",
       "NC_000013.11:g.32326569del   Gene  Gene function in the cell and involveme...  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mentions = mentions.reset_index().drop_duplicates([\"norm_g_hgvs\", \"pmid\", \"snippet\"]).set_index(\"norm_g_hgvs\")\n",
    "print(\"After dropping duplicates of hgvs+pmid+snippet: {}\".format(mentions.shape[0]))\n",
    "print(\"Total unique genomic hgvs variants: {}\".format(mentions.index.unique().shape[0]))\n",
    "mentions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "34737 mentions of variants in BRCA Exchange\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>pyhgvs_Genomic_Coordinate_38</th>\n",
       "      <th>pyhgvs_cDNA</th>\n",
       "      <th>norm_c_hgvs</th>\n",
       "      <th>pmid</th>\n",
       "      <th>snippet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>norm_g_hgvs</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32316461A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32316461:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>tions: in EPCAM, c.412C&gt;T p.(Arg138*), and coe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32316461A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32316461:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>EPCAM, c.412C&gt;T p.(Arg138*), and coexisting mu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32316461A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32316461:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>22006311</td>\n",
       "      <td>nd in-frame deletions were classified as delet...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32316461A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32316461:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>22006311</td>\n",
       "      <td>and in-frame deletions were classiﬁed as dele...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32316461A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32316461:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>24302565</td>\n",
       "      <td>and clinical data. Two cancer syndrome gene v...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Chr       Pos Ref Alt  \\\n",
       "norm_g_hgvs                                         \n",
       "NC_000013.11:g.32316461A>G   13  32316461   A   G   \n",
       "NC_000013.11:g.32316461A>G   13  32316461   A   G   \n",
       "NC_000013.11:g.32316461A>G   13  32316461   A   G   \n",
       "NC_000013.11:g.32316461A>G   13  32316461   A   G   \n",
       "NC_000013.11:g.32316461A>G   13  32316461   A   G   \n",
       "\n",
       "                           pyhgvs_Genomic_Coordinate_38         pyhgvs_cDNA  \\\n",
       "norm_g_hgvs                                                                   \n",
       "NC_000013.11:g.32316461A>G         chr13:g.32316461:A>G  NM_000059.3:c.1A>G   \n",
       "NC_000013.11:g.32316461A>G         chr13:g.32316461:A>G  NM_000059.3:c.1A>G   \n",
       "NC_000013.11:g.32316461A>G         chr13:g.32316461:A>G  NM_000059.3:c.1A>G   \n",
       "NC_000013.11:g.32316461A>G         chr13:g.32316461:A>G  NM_000059.3:c.1A>G   \n",
       "NC_000013.11:g.32316461A>G         chr13:g.32316461:A>G  NM_000059.3:c.1A>G   \n",
       "\n",
       "                                   norm_c_hgvs      pmid  \\\n",
       "norm_g_hgvs                                                \n",
       "NC_000013.11:g.32316461A>G  NM_000059.3:c.1A>G  30103829   \n",
       "NC_000013.11:g.32316461A>G  NM_000059.3:c.1A>G  30103829   \n",
       "NC_000013.11:g.32316461A>G  NM_000059.3:c.1A>G  22006311   \n",
       "NC_000013.11:g.32316461A>G  NM_000059.3:c.1A>G  22006311   \n",
       "NC_000013.11:g.32316461A>G  NM_000059.3:c.1A>G  24302565   \n",
       "\n",
       "                                                                      snippet  \n",
       "norm_g_hgvs                                                                    \n",
       "NC_000013.11:g.32316461A>G  tions: in EPCAM, c.412C>T p.(Arg138*), and coe...  \n",
       "NC_000013.11:g.32316461A>G  EPCAM, c.412C>T p.(Arg138*), and coexisting mu...  \n",
       "NC_000013.11:g.32316461A>G  nd in-frame deletions were classified as delet...  \n",
       "NC_000013.11:g.32316461A>G   and in-frame deletions were classiﬁed as dele...  \n",
       "NC_000013.11:g.32316461A>G   and clinical data. Two cancer syndrome gene v...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Prune mentions down to only those with a variant in BRCA Excchange\n",
    "variant_mentions = pd.merge(variants, mentions, left_index=True, right_index=True)\n",
    "print(\"{} mentions of variants in BRCA Exchange\".format(variant_mentions.shape[0]))\n",
    "variant_mentions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unpacking 11804 of 34737 snippets with multiple phrases seaparated by '|'\n",
      "48941 individual snippets after expanding and de-duplicating snippets\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>norm_g_hgvs</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>pyhgvs_cDNA</th>\n",
       "      <th>norm_c_hgvs</th>\n",
       "      <th>pmid</th>\n",
       "      <th>snippet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pyhgvs_Genomic_Coordinate_38</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr13:g.32316461:A&gt;G</th>\n",
       "      <td>NC_000013.11:g.32316461A&gt;G</td>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>tions: in EPCAM, c.412C&gt;T p.(Arg138*), and coe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:g.32316461:A&gt;G</th>\n",
       "      <td>NC_000013.11:g.32316461A&gt;G</td>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>EPCAM, c.412C&gt;T p.(Arg138*), and coexisting mu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:g.32316461:A&gt;G</th>\n",
       "      <td>NC_000013.11:g.32316461A&gt;G</td>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>in EPCAM, c.412C&gt;T p.(Arg138*), and coexisting...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:g.32316461:A&gt;G</th>\n",
       "      <td>NC_000013.11:g.32316461A&gt;G</td>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>22006311</td>\n",
       "      <td>nd in-frame deletions were classified as delet...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:g.32316461:A&gt;G</th>\n",
       "      <td>NC_000013.11:g.32316461A&gt;G</td>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>22006311</td>\n",
       "      <td>BRCA1 p.M1I,&lt;&lt;&lt; p.M1V&gt;&gt;&gt;, and p.C61G and TP53 ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             norm_g_hgvs  Chr       Pos Ref  \\\n",
       "pyhgvs_Genomic_Coordinate_38                                                  \n",
       "chr13:g.32316461:A>G          NC_000013.11:g.32316461A>G   13  32316461   A   \n",
       "chr13:g.32316461:A>G          NC_000013.11:g.32316461A>G   13  32316461   A   \n",
       "chr13:g.32316461:A>G          NC_000013.11:g.32316461A>G   13  32316461   A   \n",
       "chr13:g.32316461:A>G          NC_000013.11:g.32316461A>G   13  32316461   A   \n",
       "chr13:g.32316461:A>G          NC_000013.11:g.32316461A>G   13  32316461   A   \n",
       "\n",
       "                             Alt         pyhgvs_cDNA         norm_c_hgvs  \\\n",
       "pyhgvs_Genomic_Coordinate_38                                               \n",
       "chr13:g.32316461:A>G           G  NM_000059.3:c.1A>G  NM_000059.3:c.1A>G   \n",
       "chr13:g.32316461:A>G           G  NM_000059.3:c.1A>G  NM_000059.3:c.1A>G   \n",
       "chr13:g.32316461:A>G           G  NM_000059.3:c.1A>G  NM_000059.3:c.1A>G   \n",
       "chr13:g.32316461:A>G           G  NM_000059.3:c.1A>G  NM_000059.3:c.1A>G   \n",
       "chr13:g.32316461:A>G           G  NM_000059.3:c.1A>G  NM_000059.3:c.1A>G   \n",
       "\n",
       "                                  pmid  \\\n",
       "pyhgvs_Genomic_Coordinate_38             \n",
       "chr13:g.32316461:A>G          30103829   \n",
       "chr13:g.32316461:A>G          30103829   \n",
       "chr13:g.32316461:A>G          30103829   \n",
       "chr13:g.32316461:A>G          22006311   \n",
       "chr13:g.32316461:A>G          22006311   \n",
       "\n",
       "                                                                        snippet  \n",
       "pyhgvs_Genomic_Coordinate_38                                                     \n",
       "chr13:g.32316461:A>G          tions: in EPCAM, c.412C>T p.(Arg138*), and coe...  \n",
       "chr13:g.32316461:A>G          EPCAM, c.412C>T p.(Arg138*), and coexisting mu...  \n",
       "chr13:g.32316461:A>G          in EPCAM, c.412C>T p.(Arg138*), and coexisting...  \n",
       "chr13:g.32316461:A>G          nd in-frame deletions were classified as delet...  \n",
       "chr13:g.32316461:A>G          BRCA1 p.M1I,<<< p.M1V>>>, and p.C61G and TP53 ...  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Some of the snippets are multiple mentions separated by | so unpack these,\n",
    "# but limit to 3 as some have as many as 168!\n",
    "print(\"Unpacking {} of {} snippets with multiple phrases seaparated by '|'\".format(\n",
    "    variant_mentions[variant_mentions.snippet.str.contains(\"\\|\")].shape[0], variant_mentions.shape[0]))\n",
    "\n",
    "# https://stackoverflow.com/questions/17116814/pandas-how-do-i-split-text-in-a-column-into-multiple-rows/21032532\n",
    "\n",
    "# Resest index so each row id is unique vs. norm_g_hgvs\n",
    "df = variant_mentions.reset_index()\n",
    "# df = variant_mentions[variant_mentions.snippet.str.contains(\"\\|\")].iloc[0:100].reset_index()\n",
    "\n",
    "# Generate a new dataframe splitting each snippet segment into its own row\n",
    "# Limit to max of 3 as some of them have > 100\n",
    "snippets = df.apply(lambda x: pd.Series(x.snippet.split(\"|\")[:3]), axis=1).stack()\n",
    "\n",
    "# To line up with the original index\n",
    "snippets.index = snippets.index.droplevel(-1)\n",
    "\n",
    "# Join back to the original dataframe replaceing the old \"snippet\" columnd\n",
    "snippets.name = \"snippet\"\n",
    "del df[\"snippet\"]\n",
    "exploded = df.join(snippets).drop_duplicates([\"pyhgvs_Genomic_Coordinate_38\", \"snippet\"]).set_index(\"pyhgvs_Genomic_Coordinate_38\")\n",
    "print(\"{} individual snippets after expanding and de-duplicating snippets\".format(exploded.shape[0]))\n",
    "\n",
    "exploded.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Combined 48941 separate snippets down to 18025 after grouping by pmid\n"
     ]
    }
   ],
   "source": [
    "# Variants by pyhgvs_Genomic_Coordinate_38 by pmid with all snippets in a list\n",
    "combined = exploded.groupby([\"pyhgvs_Genomic_Coordinate_38\", \"pmid\"])[\"snippet\"].apply(lambda s: s.tolist())\n",
    "print(\"Combined {} separate snippets down to {} after grouping by pmid\".format(exploded.shape[0], combined.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>articleId</th>\n",
       "      <th>externalId</th>\n",
       "      <th>source</th>\n",
       "      <th>publisher</th>\n",
       "      <th>origFile</th>\n",
       "      <th>journal</th>\n",
       "      <th>printIssn</th>\n",
       "      <th>eIssn</th>\n",
       "      <th>journalUniqueId</th>\n",
       "      <th>year</th>\n",
       "      <th>...</th>\n",
       "      <th>issue</th>\n",
       "      <th>page</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pmcId</th>\n",
       "      <th>doi</th>\n",
       "      <th>fulltextUrl</th>\n",
       "      <th>time</th>\n",
       "      <th>offset</th>\n",
       "      <th>size</th>\n",
       "      <th>chunkId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5011585672</td>\n",
       "      <td>PMID11585672</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Trends in genetics : TIG</td>\n",
       "      <td>0168-9525</td>\n",
       "      <td>0168-9525</td>\n",
       "      <td>8507085</td>\n",
       "      <td>2001</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>S18</td>\n",
       "      <td>11585672</td>\n",
       "      <td></td>\n",
       "      <td>10.1016/s0168-9525(01)02451-9</td>\n",
       "      <td>https://linkinghub.elsevier.com/retrieve/pii/S...</td>\n",
       "      <td>2018-11-14T16:29:09+0000</td>\n",
       "      <td>78</td>\n",
       "      <td>5258</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5019688261</td>\n",
       "      <td>PMID19688261</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Breast cancer research and treatment</td>\n",
       "      <td>0167-6806</td>\n",
       "      <td>1573-7217</td>\n",
       "      <td>8111104</td>\n",
       "      <td>2010</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>575</td>\n",
       "      <td>19688261</td>\n",
       "      <td></td>\n",
       "      <td>10.1007/s10549-009-0501-3</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:29:16+0000</td>\n",
       "      <td>5048</td>\n",
       "      <td>100825</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5012228710</td>\n",
       "      <td>PMID12228710</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Science (New York, N.Y.)</td>\n",
       "      <td>0036-8075</td>\n",
       "      <td>1095-9203</td>\n",
       "      <td>0404511</td>\n",
       "      <td>2002</td>\n",
       "      <td>...</td>\n",
       "      <td>5588</td>\n",
       "      <td>1837</td>\n",
       "      <td>12228710</td>\n",
       "      <td></td>\n",
       "      <td>10.1126/science.297.5588.1837</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:29:40+0000</td>\n",
       "      <td>1436851129</td>\n",
       "      <td>148950</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5029369605</td>\n",
       "      <td>PMID29369605</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Genetika</td>\n",
       "      <td>0016-6758</td>\n",
       "      <td>0016-6758</td>\n",
       "      <td>0047354</td>\n",
       "      <td>2016</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>1215</td>\n",
       "      <td>29369605</td>\n",
       "      <td></td>\n",
       "      <td>10.1134/s102279541609012x</td>\n",
       "      <td>https://link.springer.com/article/10.1134%2FS1...</td>\n",
       "      <td>2018-11-14T16:30:43+0000</td>\n",
       "      <td>4458079378</td>\n",
       "      <td>22264</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5022084640</td>\n",
       "      <td>PMID22084640</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Therapeutic advances in medical oncology</td>\n",
       "      <td>1758-8340</td>\n",
       "      <td>1758-8359</td>\n",
       "      <td>101510808</td>\n",
       "      <td>2011</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>257</td>\n",
       "      <td>22084640</td>\n",
       "      <td>3210467</td>\n",
       "      <td>10.1177/1758834011417039</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:30:58+0000</td>\n",
       "      <td>6079002966</td>\n",
       "      <td>93229</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    articleId    externalId source publisher origFile  \\\n",
       "0  5011585672  PMID11585672         download            \n",
       "1  5019688261  PMID19688261         download            \n",
       "2  5012228710  PMID12228710         download            \n",
       "3  5029369605  PMID29369605         download            \n",
       "4  5022084640  PMID22084640         download            \n",
       "\n",
       "                                    journal  printIssn      eIssn  \\\n",
       "0                  Trends in genetics : TIG  0168-9525  0168-9525   \n",
       "1      Breast cancer research and treatment  0167-6806  1573-7217   \n",
       "2                  Science (New York, N.Y.)  0036-8075  1095-9203   \n",
       "3                                  Genetika  0016-6758  0016-6758   \n",
       "4  Therapeutic advances in medical oncology  1758-8340  1758-8359   \n",
       "\n",
       "  journalUniqueId  year   ...    issue  page      pmid    pmcId  \\\n",
       "0         8507085  2001   ...       10   S18  11585672            \n",
       "1         8111104  2010   ...        3   575  19688261            \n",
       "2         0404511  2002   ...     5588  1837  12228710            \n",
       "3         0047354  2016   ...       10  1215  29369605            \n",
       "4       101510808  2011   ...        6   257  22084640  3210467   \n",
       "\n",
       "                             doi  \\\n",
       "0  10.1016/s0168-9525(01)02451-9   \n",
       "1      10.1007/s10549-009-0501-3   \n",
       "2  10.1126/science.297.5588.1837   \n",
       "3      10.1134/s102279541609012x   \n",
       "4       10.1177/1758834011417039   \n",
       "\n",
       "                                         fulltextUrl  \\\n",
       "0  https://linkinghub.elsevier.com/retrieve/pii/S...   \n",
       "1                                                      \n",
       "2                                                      \n",
       "3  https://link.springer.com/article/10.1134%2FS1...   \n",
       "4                                                      \n",
       "\n",
       "                       time      offset    size  chunkId  \n",
       "0  2018-11-14T16:29:09+0000          78    5258  0_00000  \n",
       "1  2018-11-14T16:29:16+0000        5048  100825  0_00000  \n",
       "2  2018-11-14T16:29:40+0000  1436851129  148950  0_00000  \n",
       "3  2018-11-14T16:30:43+0000  4458079378   22264  0_00000  \n",
       "4  2018-11-14T16:30:58+0000  6079002966   93229  0_00000  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "literature = {\n",
    "    \"papers\": articles[articles.pmid.isin(variant_mentions.pmid)].set_index(\"pmid\", drop=False).to_dict(orient=\"index\"),\n",
    "    \"variants\": {\n",
    "        k: {kk: vv[0] for kk, vv in v.unstack().transpose().iterrows()}\n",
    "        for k, v in combined.groupby(\"pyhgvs_Genomic_Coordinate_38\")},\n",
    "}\n",
    "\n",
    "# with open(\"literature.json\", \"w\") as output:\n",
    "#     output.write(json.dumps(literature, sort_keys=True))\n",
    "    \n",
    "# print(\"Exported {} variants in {} papers\".format(\n",
    "#     len(literature[\"variants\"].keys()), len(literature[\"papers\"].keys())))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Max mentions for a single variant: chr17:g.43106487:A>C 1411\n",
      "Max mentions for a single paper from one variant: ('chr13:g.32319082:G>A', '27490902') 19\n",
      "Paper referenced from the most variants: 21990134 1633\n"
     ]
    }
   ],
   "source": [
    "print(\"Max mentions for a single variant: {} {}\".format(\n",
    "    exploded.groupby([\"pyhgvs_Genomic_Coordinate_38\"]).size().idxmax(),\n",
    "    exploded.groupby([\"pyhgvs_Genomic_Coordinate_38\"]).size().max()))\n",
    "\n",
    "print(\"Max mentions for a single paper from one variant: {} {}\".format(\n",
    "    exploded.groupby([\"pyhgvs_Genomic_Coordinate_38\", \"pmid\"]).size().idxmax(),\n",
    "    exploded.groupby([\"pyhgvs_Genomic_Coordinate_38\", \"pmid\"]).size().max()))\n",
    "\n",
    "print(\"Paper referenced from the most variants: {} {}\".format(\n",
    "    exploded.groupby([\"pmid\"]).size().idxmax(),\n",
    "    exploded.groupby([\"pmid\"]).size().max()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variant mention stats:\n",
      "count    3754.000000\n",
      "mean       13.037027\n",
      "std        39.876334\n",
      "min         1.000000\n",
      "25%         2.000000\n",
      "50%         4.000000\n",
      "75%        11.000000\n",
      "max      1411.000000\n",
      "Name: pyhgvs_Genomic_Coordinate_38, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "print(\"Variant mention stats:\")\n",
    "print(exploded.reset_index().groupby([\"pyhgvs_Genomic_Coordinate_38\"])\n",
    "      [\"pyhgvs_Genomic_Coordinate_38\"].count().describe())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Paper mentions stats:\n",
      "count    2145.000000\n",
      "mean       22.816317\n",
      "std        58.991191\n",
      "min         1.000000\n",
      "25%         3.000000\n",
      "50%         6.000000\n",
      "75%        19.000000\n",
      "max      1633.000000\n",
      "Name: pmid, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "print(\"Paper mentions stats:\")\n",
    "print(exploded.reset_index().groupby([\"pmid\"])\n",
    "      [\"pmid\"].count().describe())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Failed To Correlate\n",
    "Papers where pubMunch found mutations but that failed to correlate after normalization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>hgvsCoding</th>\n",
       "      <th>mutSnippets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>12228710</td>\n",
       "      <td>NM_006304.1:c.4T&gt;G</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>12228710</td>\n",
       "      <td>NM_001145459.1:c.4T&gt;G|NM_139212.3:c.4T&gt;G|NM_13...</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12228710</td>\n",
       "      <td>NM_006304.1:c.4T&gt;G</td>\n",
       "      <td>ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>12228710</td>\n",
       "      <td>NM_001145459.1:c.4T&gt;G|NM_139212.3:c.4T&gt;G|NM_13...</td>\n",
       "      <td>ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>29369605</td>\n",
       "      <td>NM_000059.3:c.587G&gt;None</td>\n",
       "      <td>Gene  Gene function in the cell and involveme...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        pmid                                         hgvsCoding  \\\n",
       "5   12228710                                 NM_006304.1:c.4T>G   \n",
       "6   12228710  NM_001145459.1:c.4T>G|NM_139212.3:c.4T>G|NM_13...   \n",
       "11  12228710                                 NM_006304.1:c.4T>G   \n",
       "12  12228710  NM_001145459.1:c.4T>G|NM_139212.3:c.4T>G|NM_13...   \n",
       "21  29369605                            NM_000059.3:c.587G>None   \n",
       "\n",
       "                                          mutSnippets  \n",
       "5   ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "6   ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "11  ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...  \n",
       "12  ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...  \n",
       "21   Gene  Gene function in the cell and involveme...  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutations = pd.read_table(\"mutations.tsv\", header=0,\n",
    "                          usecols=[\"docId\", \"hgvsCoding\", \"mutSnippets\"],\n",
    "                          dtype={\"docId\": \"str\", \"hgvsCoding\": \"str\", \"mutSnippets\": \"str\"},\n",
    "                         ).dropna(subset=[\"hgvsCoding\"])\n",
    "mutations = mutations.rename({\"docId\": \"pmid\"}, axis=\"columns\")\n",
    "mutations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14044"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mutations.pmid.values))\n",
    "\n",
    "len(set(articles.index.values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[0,\n",
       " 1000,\n",
       " 2000,\n",
       " 3000,\n",
       " 4000,\n",
       " 5000,\n",
       " 6000,\n",
       " 7000,\n",
       " 8000,\n",
       " 9000,\n",
       " 10000,\n",
       " 11000,\n",
       " 12000,\n",
       " 13000,\n",
       " 14000]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(set(articles.index.values) - set(mutations.pmid.values))[::1000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['9988281',\n",
       " '9988226',\n",
       " '9973246',\n",
       " '9950543',\n",
       " '9950212',\n",
       " '9930365',\n",
       " '9928546',\n",
       " '9928543',\n",
       " '9927063',\n",
       " '9927062']"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "papers_with_no_variants = sorted(set(pmids) - set(exploded.pmid.values), reverse=True)\n",
    "papers_with_no_variants[0:10]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Export Stats\n",
    "Stats generated directly from the exported literature.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(os.path.expanduser(\"/notebooks/data/pubmunch/crawl-14-11-2018/literature-05-12-2018-v1.json\")) as f:\n",
    "    lit = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2227 Papers, 3754 Variants\n"
     ]
    }
   ],
   "source": [
    "print(\"{} Papers, {} Variants\".format(len(lit[\"papers\"]), len(lit[\"variants\"])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'chr17:g.43092596:G>A'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(lit[\"variants\"].keys())[0]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
