{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BRCA Exchange Literature Search Export\n",
    "\n",
    "Ingest artifacts from the pipeline, wrangle, and generate literature.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import datetime\n",
    "import json\n",
    "import sqlite3\n",
    "import pprint\n",
    "import pandas as pd\n",
    "\n",
    "os.chdir(os.path.expanduser(\"/home/jovyan/data/pubmunch/crawl/\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ingest\n",
    "Ingest artifacts from various pieces of the pipeline indexed by normalized genomic hgvs and integer pmid "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14044 articles loaded from the articles sqlite database\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>articleId</th>\n",
       "      <th>externalId</th>\n",
       "      <th>source</th>\n",
       "      <th>publisher</th>\n",
       "      <th>origFile</th>\n",
       "      <th>journal</th>\n",
       "      <th>printIssn</th>\n",
       "      <th>eIssn</th>\n",
       "      <th>journalUniqueId</th>\n",
       "      <th>year</th>\n",
       "      <th>...</th>\n",
       "      <th>issue</th>\n",
       "      <th>page</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pmcId</th>\n",
       "      <th>doi</th>\n",
       "      <th>fulltextUrl</th>\n",
       "      <th>time</th>\n",
       "      <th>offset</th>\n",
       "      <th>size</th>\n",
       "      <th>chunkId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5011585672</td>\n",
       "      <td>PMID11585672</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Trends in genetics : TIG</td>\n",
       "      <td>0168-9525</td>\n",
       "      <td>0168-9525</td>\n",
       "      <td>8507085</td>\n",
       "      <td>2001</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>S18</td>\n",
       "      <td>11585672</td>\n",
       "      <td></td>\n",
       "      <td>10.1016/s0168-9525(01)02451-9</td>\n",
       "      <td>https://linkinghub.elsevier.com/retrieve/pii/S...</td>\n",
       "      <td>2018-11-14T16:29:09+0000</td>\n",
       "      <td>78</td>\n",
       "      <td>5258</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5019688261</td>\n",
       "      <td>PMID19688261</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Breast cancer research and treatment</td>\n",
       "      <td>0167-6806</td>\n",
       "      <td>1573-7217</td>\n",
       "      <td>8111104</td>\n",
       "      <td>2010</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>575</td>\n",
       "      <td>19688261</td>\n",
       "      <td></td>\n",
       "      <td>10.1007/s10549-009-0501-3</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:29:16+0000</td>\n",
       "      <td>5048</td>\n",
       "      <td>100825</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5012228710</td>\n",
       "      <td>PMID12228710</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Science (New York, N.Y.)</td>\n",
       "      <td>0036-8075</td>\n",
       "      <td>1095-9203</td>\n",
       "      <td>0404511</td>\n",
       "      <td>2002</td>\n",
       "      <td>...</td>\n",
       "      <td>5588</td>\n",
       "      <td>1837</td>\n",
       "      <td>12228710</td>\n",
       "      <td></td>\n",
       "      <td>10.1126/science.297.5588.1837</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:29:40+0000</td>\n",
       "      <td>1436851129</td>\n",
       "      <td>148950</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5029369605</td>\n",
       "      <td>PMID29369605</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Genetika</td>\n",
       "      <td>0016-6758</td>\n",
       "      <td>0016-6758</td>\n",
       "      <td>0047354</td>\n",
       "      <td>2016</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>1215</td>\n",
       "      <td>29369605</td>\n",
       "      <td></td>\n",
       "      <td>10.1134/s102279541609012x</td>\n",
       "      <td>https://link.springer.com/article/10.1134%2FS1...</td>\n",
       "      <td>2018-11-14T16:30:43+0000</td>\n",
       "      <td>4458079378</td>\n",
       "      <td>22264</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5022084640</td>\n",
       "      <td>PMID22084640</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Therapeutic advances in medical oncology</td>\n",
       "      <td>1758-8340</td>\n",
       "      <td>1758-8359</td>\n",
       "      <td>101510808</td>\n",
       "      <td>2011</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>257</td>\n",
       "      <td>22084640</td>\n",
       "      <td>3210467</td>\n",
       "      <td>10.1177/1758834011417039</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:30:58+0000</td>\n",
       "      <td>6079002966</td>\n",
       "      <td>93229</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    articleId    externalId source publisher origFile  \\\n",
       "0  5011585672  PMID11585672         download            \n",
       "1  5019688261  PMID19688261         download            \n",
       "2  5012228710  PMID12228710         download            \n",
       "3  5029369605  PMID29369605         download            \n",
       "4  5022084640  PMID22084640         download            \n",
       "\n",
       "                                    journal  printIssn      eIssn  \\\n",
       "0                  Trends in genetics : TIG  0168-9525  0168-9525   \n",
       "1      Breast cancer research and treatment  0167-6806  1573-7217   \n",
       "2                  Science (New York, N.Y.)  0036-8075  1095-9203   \n",
       "3                                  Genetika  0016-6758  0016-6758   \n",
       "4  Therapeutic advances in medical oncology  1758-8340  1758-8359   \n",
       "\n",
       "  journalUniqueId  year   ...    issue  page      pmid    pmcId  \\\n",
       "0         8507085  2001   ...       10   S18  11585672            \n",
       "1         8111104  2010   ...        3   575  19688261            \n",
       "2         0404511  2002   ...     5588  1837  12228710            \n",
       "3         0047354  2016   ...       10  1215  29369605            \n",
       "4       101510808  2011   ...        6   257  22084640  3210467   \n",
       "\n",
       "                             doi  \\\n",
       "0  10.1016/s0168-9525(01)02451-9   \n",
       "1      10.1007/s10549-009-0501-3   \n",
       "2  10.1126/science.297.5588.1837   \n",
       "3      10.1134/s102279541609012x   \n",
       "4       10.1177/1758834011417039   \n",
       "\n",
       "                                         fulltextUrl  \\\n",
       "0  https://linkinghub.elsevier.com/retrieve/pii/S...   \n",
       "1                                                      \n",
       "2                                                      \n",
       "3  https://link.springer.com/article/10.1134%2FS1...   \n",
       "4                                                      \n",
       "\n",
       "                       time      offset    size  chunkId  \n",
       "0  2018-11-14T16:29:09+0000          78    5258  0_00000  \n",
       "1  2018-11-14T16:29:16+0000        5048  100825  0_00000  \n",
       "2  2018-11-14T16:29:40+0000  1436851129  148950  0_00000  \n",
       "3  2018-11-14T16:30:43+0000  4458079378   22264  0_00000  \n",
       "4  2018-11-14T16:30:58+0000  6079002966   93229  0_00000  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "connection = sqlite3.connect(\"file:text/articles.db?mode=ro\", uri=True)\n",
    "articles = pd.read_sql_query(\"SELECT * FROM articles\", connection)\n",
    "articles.pmid = articles.pmid.astype(str)\n",
    "print(\"{} articles loaded from the articles sqlite database\".format(articles.shape[0]))\n",
    "articles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total matched mentions: 174932\n",
      "After dropping duplicates: 82548\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pyhgvs_Genomic_Coordinate_38</th>\n",
       "      <th>pmid</th>\n",
       "      <th>snippets</th>\n",
       "      <th>points</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17:g.43124027:ACT&gt;A</td>\n",
       "      <td>10030809</td>\n",
       "      <td>Novel inherited mutations and variable express...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17:g.43124027:ACT&gt;A</td>\n",
       "      <td>10030809</td>\n",
       "      <td>Novel inherited mutations and variable express...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17:g.43124027:ACT&gt;A</td>\n",
       "      <td>10036974</td>\n",
       "      <td>umber of large series of young Ashkenzai Jewis...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17:g.43057059:T&gt;TG</td>\n",
       "      <td>10036974</td>\n",
       "      <td>0n99 0n99  1n01 11n80 15n86 18n87  0n01 0n92 ...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17:g.43057062:T&gt;TG</td>\n",
       "      <td>10036974</td>\n",
       "      <td>0n99 0n99  1n01 11n80 15n86 18n87  0n01 0n92 ...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  pyhgvs_Genomic_Coordinate_38      pmid  \\\n",
       "0       chr17:g.43124027:ACT>A  10030809   \n",
       "1       chr17:g.43124027:ACT>A  10030809   \n",
       "2       chr17:g.43124027:ACT>A  10036974   \n",
       "3        chr17:g.43057059:T>TG  10036974   \n",
       "4        chr17:g.43057062:T>TG  10036974   \n",
       "\n",
       "                                            snippets  points  \n",
       "0  Novel inherited mutations and variable express...       3  \n",
       "1  Novel inherited mutations and variable express...       3  \n",
       "2  umber of large series of young Ashkenzai Jewis...       3  \n",
       "3   0n99 0n99  1n01 11n80 15n86 18n87  0n01 0n92 ...       3  \n",
       "4   0n99 0n99  1n01 11n80 15n86 18n87  0n01 0n92 ...       3  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mentions = pd.read_csv(\"mentions-matched.tsv\", sep=\"\\t\", encoding=\"utf-8\")\n",
    "mentions.pmid = mentions.pmid.astype(str)\n",
    "print(\"Total matched mentions: {}\".format(mentions.shape[0]))\n",
    "mentions = mentions.drop_duplicates([\"pyhgvs_Genomic_Coordinate_38\", \"pmid\", \"snippets\"])\n",
    "print(\"After dropping duplicates: {}\".format(mentions.shape[0]))\n",
    "mentions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([['22962691', 40],\n",
       "       ['21990134', 40],\n",
       "       ['24323938', 40],\n",
       "       ['25146914', 40],\n",
       "       ['20215541', 30],\n",
       "       ['18424508', 30],\n",
       "       ['30039884', 20],\n",
       "       ['21990165', 20],\n",
       "       ['23586058', 20],\n",
       "       ['20690207', 20],\n",
       "       ['20522429', 20],\n",
       "       ['20507642', 20],\n",
       "       ['19471317', 20],\n",
       "       ['18607349', 20],\n",
       "       ['18451181', 20],\n",
       "       ['18273839', 20],\n",
       "       ['26913838', 20],\n",
       "       ['27060066', 20],\n",
       "       ['28324225', 20],\n",
       "       ['15026808', 20],\n",
       "       ['29884841', 20],\n",
       "       ['29988080', 20],\n",
       "       ['23108138', 20],\n",
       "       ['28339459', 15],\n",
       "       ['25003164', 10],\n",
       "       ['24817641', 10],\n",
       "       ['24122022', 10],\n",
       "       ['10464631', 10],\n",
       "       ['22753008', 10],\n",
       "       ['21735045', 10],\n",
       "       ['21638052', 10],\n",
       "       ['12145750', 10],\n",
       "       ['21309043', 10],\n",
       "       ['17924331', 10],\n",
       "       ['17899372', 10],\n",
       "       ['16792514', 10],\n",
       "       ['15744044', 10],\n",
       "       ['12845657', 10],\n",
       "       ['21344236', 10],\n",
       "       ['19332451', 2],\n",
       "       ['16825284', 2],\n",
       "       ['12915465', 2]], dtype=object)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def top_papers(mentions, pyhgvs):\n",
    "    \"\"\" Return (pmid, points) tuple sorted by points against pyhgvs \"\"\"\n",
    "    top = mentions[mentions.pyhgvs_Genomic_Coordinate_38 == pyhgvs] \\\n",
    "        .groupby([\"pyhgvs_Genomic_Coordinate_38\", \"pmid\"]) \\\n",
    "        .agg({\"points\": sum}) \\\n",
    "        .sort_values(\"points\", ascending=False)\n",
    "    return top.reset_index()[[\"pmid\", \"points\"]].values\n",
    "\n",
    "top_papers(mentions, \"chr13:g.32363367:C>G\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['BRCA2<<< T2722R>>> is a deleterious allele that causes exon skipping.',\n",
       " 'Of these, the exonic variant BRCA2 c.8162T→C in exon 18 affects a position only three nucleotides upstream of the mutation BRCA2 c.8165C→G (predicting<<< p.T2722R>>>), which is known to cause exon skipping by disrupting several ESE sites.20  Two variants, BRCA2 c.316+5G→C (IVS3+5G→C) and c.7805G→C, at the last base of exon 16, induced strong effects on splicing (figs 2A,B and 2C,D, respectively).',\n",
       " 'Of these, the exonic variant BRCA2 c.8162TRC in exon 18 affects a position only three nucleotides upstream of the mutation BRCA2<<< c.8165CRG>>> (predicting p.T2722R), which is known to cause exon skipping by disrupting several ESE sites.20 Two variants, BRCA2 c.316+5GRC (IVS3+5GRC) and c.7805']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def top_snippets(mentions, pyhgvs, pmid):\n",
    "    \"\"\" Return list of top snippets for this variant and paper by points \"\"\"\n",
    "    paper = mentions[(mentions.pyhgvs_Genomic_Coordinate_38 == pyhgvs) & (mentions.pmid == pmid)]\n",
    "    return [snippet \n",
    "     for snippets in paper.sort_values(\"points\", ascending=False).snippets.values \n",
    "     for snippet in snippets.split(\"|\")][:3]\n",
    "\n",
    "top_snippets(mentions, \"chr13:g.32363367:C>G\", '18424508')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr13:g.32363367:C>G\n",
      "[{'pmid': '22962691',\n",
      "  'points': 40,\n",
      "  'snippets': [', in the BRCA2 gene, besides the exon 7 mutations described '\n",
      "               'above, only one mutation in exon 3, c.231T>G (p.Thr77Thr),16 '\n",
      "               'and two mutations in exon 18,<<< c.8165C>G>>> (p.Thr2722Arg)28 '\n",
      "               'and c.7992T>A (p.Ile2664Ile),16 have been reported to induce '\n",
      "               'exon skipping by altering splicing regulatory elements.',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'CA2 gene, besides the exon 7 mutations described above, only '\n",
      "               'one mutation in exon 3, c.231T>G (p.Thr77Thr),16 and two '\n",
      "               'mutations in exon 18, c.8165C>G <<<(p.Thr2722Arg>>>)28 and '\n",
      "               'c.7992T>A (p.Ile2664Ile),16 have been reported to induce exon '\n",
      "               'skipping by altering splicing regulatory elements.']},\n",
      " {'pmid': '21990134',\n",
      "  'points': 40,\n",
      "  'snippets': ['0.81  0.95  4   18  A2717S  p.Ala2717Ser  8377G>T  c.8149G>T  '\n",
      "               '1.60×10−4  Spurdle et al., 2008  0.03  4.95×10−6  1   18  '\n",
      "               'T2722R  p.Thr2722Arg  8393C>G <<< c.8165C>G>>>  93.50  '\n",
      "               'Farrugia et al., 2008  0.81  1.00  5   18  D2723H  '\n",
      "               'p.Asp2723His  8395G>C  c.8167G>C  5.62×1011  Farrugia et al., '\n",
      "               '2008  0.81  1.00  5   18  #D',\n",
      "               '057T>C  4.00  This manuscript  0.81  0.95  4   18  A2717S  '\n",
      "               'p.Ala2717Ser  8377G>T  c.8149G>T  1.60×10−4  Spurdle et al., '\n",
      "               '2008  0.03  4.95×10−6  1   18 <<< T2722R>>>  p.Thr2722Arg  '\n",
      "               '8393C>G  c.8165C>G  93.50  Farrugia et al., 2008  0.81  1.00  '\n",
      "               '5   18  D2723H  p.Asp2723His  8395G>C  c.8167G>C  5.62×1011  '\n",
      "               'Farrugia et',\n",
      "               '4.00  This manuscript  0.81  0.95  4   18  A2717S  '\n",
      "               'p.Ala2717Ser  8377G>T  c.8149G>T  1.60×10−4  Spurdle et al., '\n",
      "               '2008  0.03  4.95×10−6  1   18  T2722R <<< p.Thr2722Arg>>>  '\n",
      "               '8393C>G  c.8165C>G  93.50  Farrugia et al., 2008  0.81  1.00  '\n",
      "               '5   18  D2723H  p.Asp2723His  8395G>C  c.8167G>C  5.62×1011  '\n",
      "               'Farrugia et al., 2008  0.']},\n",
      " {'pmid': '24323938',\n",
      "  'points': 40,\n",
      "  'snippets': ['.8149G>T  8377G>T  OB1  1  Edwards SM et al., 2003; Diez et '\n",
      "               'al., 2003; Salazar et al., 2006; Spurdle et al., 2008; Lindor '\n",
      "               'et al., 2012   p.Thr2722Arg <<< c.8165C>G>>>  8393C>G  OB1  5  '\n",
      "               'Easton et al., 2007; Farrugia et al 2008; Lindor et al., '\n",
      "               '2012   p.Asp2723His  c.8167G>C  8395G>C  OB1  5  Goldgar et '\n",
      "               'al. 2004; Wu e',\n",
      "               '.Ala2717Ser  c.8149G>T  8377G>T  OB1  1  Edwards SM et al., '\n",
      "               '2003; Diez et al., 2003; Salazar et al., 2006; Spurdle et al., '\n",
      "               '2008; Lindor et al., 2012  <<< p.Thr2722Arg>>>  c.8165C>G  '\n",
      "               '8393C>G  OB1  5  Easton et al., 2007; Farrugia et al 2008; '\n",
      "               'Lindor et al., 2012   p.Asp2723His  c.8167G>C  8395G>C  OB1  '\n",
      "               '5  Goldgar et al.',\n",
      "               ' +/−  p.Ile2490Thr  +  +  +  +  +  p.Ala2643Gly  +  '\n",
      "               'p.Ser2006Arg  −  p.Gly2748Asp  −   p.Glu462Gly  +  '\n",
      "               'p.Ser2807Leu  +  p.Leu2510Pro  +/−  −  −  −  − <<< '\n",
      "               'p.Thr2722Arg>>>  −  p.Arg2108Cys  −  p.Ala2951Thr  +   '\n",
      "               'p.Thr1302del  +  p.Gly2812Glu  +/−  p.Trp2626Cys  +/−  −  −  '\n",
      "               '−  −  p.Lys2729Asn  +  p.Gly2353Arg  −    p.Glu13']}]\n",
      "chr17:g.43124027:ACT>A\n",
      "[{'pmid': '22923021',\n",
      "  'points': 40,\n",
      "  'snippets': ['of positive families  BRCA1 c.66_67delAG <<< '\n",
      "               'p.Glu23Valfs*17>>>  Frameshift  1   c.116G>A  p.Cys39Tyr  '\n",
      "               'Missense  8  c.181T>G  p.Cys61Gly  Missense  31   c.181T>A  '\n",
      "               'p.Cys61Ser  Missense  5   c.191G>A  p.Cys64Tyr  Mi',\n",
      "               't - OC (65) Maternal aunt - BC (70) Paternal aunt - OC (41)  '\n",
      "               'Paternal aunt - BC (42)  and OC (56)  Paternal aunt - OC (71) '\n",
      "               'c.66_68delAG  185_186delAG <<< p.Glu23Valfs*17>>>  1  BC (39), '\n",
      "               'OC (42)  /  c.3436_3439delTGTT  3555_3558delTGTT  '\n",
      "               'p.Cys1146Leufs*8  1  OC (55)  Healthy daughter,        age '\n",
      "               '31         c.3718C>T  3837C',\n",
      "               'OC families.  1624 NOVAKOVIĆ et al: NOVEL BRCA1 AND BRCA2 '\n",
      "               'MUTATIONS  INTERNATIONAL JOURNAL OF ONCOLOGY 41: 1619-1627, '\n",
      "               '2012  The mutation c.66_67delAG <<<(p.Glu23Valfs*17>>>) is the '\n",
      "               'most common BRCA1 mutation worldwide which occurs at a '\n",
      "               'frequency of 1.1% in the Ashkenazi Jews\\xa0(26).']},\n",
      " {'pmid': '14507240',\n",
      "  'points': 38,\n",
      "  'snippets': ['     Mother – B (40 y)      12     BRCA2     1D     '\n",
      "               'c.5130_5133delTGTA; p.Tyr1693X     Breast     27     No family '\n",
      "               'history      13     BRCA1     2    <<< c.66_67delAG>>>; p.',\n",
      "               'elCTTAA; p.Ser2072Ser fsX4 *  Breast  38  Mother – B (40 y)  '\n",
      "               '12  BRCA2 1D  c.5130_5133delTGTA; p.Tyr1693X  Breast  27  No '\n",
      "               'family history  13  BRCA1 2 <<< c.66_67delAG>>>; p.',\n",
      "               'AY706913]     Ovarian     61     Daughter – O (39 y)      '\n",
      "               '7     BRCA1     2    <<< c.68_69delAG>>>; p.Glu23Val fsX16     '\n",
      "               'Breast     33     No family history      8     BRCA1     2     '\n",
      "               'c.68_69delAG; p.Glu23Val fsX16     Breast     33     No family '\n",
      "               'h']},\n",
      " {'pmid': '19656415',\n",
      "  'points': 38,\n",
      "  'snippets': ['     Mother – B (40 y)      12     BRCA2     1D     '\n",
      "               'c.5130_5133delTGTA; p.Tyr1693X     Breast     27     No family '\n",
      "               'history      13     BRCA1     2    <<< c.66_67delAG>>>; p.',\n",
      "               'elCTTAA; p.Ser2072Ser fsX4 *  Breast  38  Mother – B (40 y)  '\n",
      "               '12  BRCA2 1D  c.5130_5133delTGTA; p.Tyr1693X  Breast  27  No '\n",
      "               'family history  13  BRCA1 2 <<< c.66_67delAG>>>; p.',\n",
      "               'AY706913]     Ovarian     61     Daughter – O (39 y)      '\n",
      "               '7     BRCA1     2    <<< c.68_69delAG>>>; p.Glu23Val fsX16     '\n",
      "               'Breast     33     No family history      8     BRCA1     2     '\n",
      "               'c.68_69delAG; p.Glu23Val fsX16     Breast     33     No family '\n",
      "               'h']}]\n",
      "chr13:g.32340526:AT>A\n",
      "[{'pmid': '30040829',\n",
      "  'points': 30,\n",
      "  'snippets': [' Pharoah PD, Narod S, Risch HA, Eyfjord JE, Hopper JL, et al. '\n",
      "               'Breast and ovarian cancer risks to carriers of the BRCA1 '\n",
      "               '5382insC and 185delAG and BRCA2<<< 6174delT>>> mutations: a '\n",
      "               'combined analysis of 22 population based studies.',\n",
      "               'Breast and ovarian cancer risks to carriers of the BRCA1 '\n",
      "               '5382insC and 185delAG and BRCA2<<< 6174delT>>> mutations: a '\n",
      "               'combined analysis of 22 population based studies.',\n",
      "               '83_4484delGT,p.Val1495Profs,11,F,Pathogenic,lack,Yes,TNBC,44,BC,1xOC,HOBCsusp '\n",
      "               'BRCA2,31,c.5946delT,p.S1982fs,11,F,\"Pending (BIC), Pathogenic '\n",
      "               '(ClinVar)\"<<<,6174delT>>>,Yes,Serous,52,OC,1xGO,\"HOC susp\" '\n",
      "               'BRCA2,32,c.4481_4482delGT,p.S1494fs,11,F,Pathogenic,lack,No,Serous,65,OC,1xBC,\" '\n",
      "               'HBOC susp\" BRCA2,33,c.6405_6409delCT']},\n",
      " {'pmid': '23966579',\n",
      "  'points': 24,\n",
      "  'snippets': ['A1/2 mutations; 43 had BRCA1 mutations (n = 34 with 185delAG, '\n",
      "               'n = 8 with 5382insC, and n = 1 with ex18-20dup); and 37 had '\n",
      "               'BRCA2 mutations (n = 33 with<<< 6174delT>>>, n = 2 with '\n",
      "               '1vs+1G>A, n = 1 with 8675del, and n = 1 with G.6024dupG).',\n",
      "               'ad BRCA1 mutations (n = 13 with 185delAG and n = 1 with '\n",
      "               '5382insC) and 21 had BRCA2 mutations (n = 18 with 174delT, n = '\n",
      "               '2 with 1vs+1G>A, and n = 1 with<<< 6174delT>>>). This group '\n",
      "               'had undergone preventive BSOs before the enrollment (mean time '\n",
      "               'since surgery, 4 years; range, 1–11 years).  All participants '\n",
      "               'were fully informed about the study, provided written consent, '\n",
      "               'and donated a blood sample.',\n",
      "               ' cancer with mean time to diagnosis of 19 months (range, 13–22 '\n",
      "               'months); 3 had a founder BRCA1 mutation (185delAG), and 1 had '\n",
      "               'a founder BRCA2 mutation <<<(6174delT>>>).']},\n",
      " {'pmid': '9743298',\n",
      "  'points': 21,\n",
      "  'snippets': ['ncer  Search E-alert Submit My Account Login  Clinical '\n",
      "               'Oncology     Published: 01 September 1998  # Absence of '\n",
      "               '185delAG mutation of the BRCA1 gene and<<< 6174delT>>> '\n",
      "               'mutation of the BRCA2 gene in Ashkenazi Jewish men with '\n",
      "               'prostate cancer    * S Lehrer1   * , F Fodor   * , RG Stock   '\n",
      "               '* , NN Stone   * , C Eng   * , ',\n",
      "               'We tested for the BRCA1 185delAG frameshift mutation, found in '\n",
      "               '0.9% of Ashkenazi Jews, and the BRCA2<<< 6174delT>>> mutation, '\n",
      "               'found in 1% of Ashkenazi Jews, in Ashkenazi Jewish men with '\n",
      "               'prostate cancer.',\n",
      "               'None of the 60 men carried the 185delAG BRCA1 or<<< '\n",
      "               '6174delT>>> BRCA2 mutations.']}]\n"
     ]
    }
   ],
   "source": [
    "def top_papers_and_snippets(mentions, pyhgvs):\n",
    "    return [{\"pmid\": str(p[0]), \"points\": int(p[1]), \"snippets\": top_snippets(mentions, pyhgvs, p[0])}\n",
    "            for p in top_papers(mentions, pyhgvs)]\n",
    "\n",
    "for pyhgvs in [\"chr13:g.32363367:C>G\", \"chr17:g.43124027:ACT>A\", \"chr13:g.32340526:AT>A\"]:\n",
    "    print(pyhgvs)\n",
    "    pprint.pprint(top_papers_and_snippets(mentions, pyhgvs)[:3])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-02-27 22:58:23.671037\n",
      "Remaining: 1000\r"
     ]
    }
   ],
   "source": [
    "variants = {}\n",
    "\n",
    "remaining = mentions.pyhgvs_Genomic_Coordinate_38.unique().shape[0]\n",
    "print(datetime.datetime.now())\n",
    "for pyhgvs in mentions.pyhgvs_Genomic_Coordinate_38.unique():\n",
    "    sys.stdout.write(\"Remaining: {}\\r\".format(remaining))\n",
    "    sys.stdout.flush()\n",
    "    variants[pyhgvs] = top_papers_and_snippets(mentions, pyhgvs)\n",
    "    remaining -= 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exported 9783 variants in 3615 papers\n",
      "3615 Papers and 9783 Variants exported\n"
     ]
    }
   ],
   "source": [
    "lit = {\n",
    "    \"date\": open(\"pubs-date.txt\").read().strip(),\n",
    "    \"papers\": articles[articles.pmid.isin(mentions.pmid)].set_index(\"pmid\", drop=False).to_dict(orient=\"index\"),\n",
    "    \"variants\": variants\n",
    "}\n",
    "\n",
    "with open(\"literature.json\", \"w\") as output:\n",
    "    output.write(json.dumps(lit, sort_keys=True))\n",
    "    \n",
    "print(\"Exported {} variants in {} papers\".format(\n",
    "    len(lit[\"variants\"].keys()), len(lit[\"papers\"].keys())))\n",
    "\n",
    "with open(\"literature.json\") as f:\n",
    "    lit = json.loads(f.read())\n",
    "print(\"{} Papers and {} Variants exported\".format(len(lit[\"papers\"]), len(lit[\"variants\"])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'pmid': '22962691',\n",
      "  'points': 40,\n",
      "  'snippets': [', in the BRCA2 gene, besides the exon 7 mutations described '\n",
      "               'above, only one mutation in exon 3, c.231T>G (p.Thr77Thr),16 '\n",
      "               'and two mutations in exon 18,<<< c.8165C>G>>> (p.Thr2722Arg)28 '\n",
      "               'and c.7992T>A (p.Ile2664Ile),16 have been reported to induce '\n",
      "               'exon skipping by altering splicing regulatory elements.',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'CA2 gene, besides the exon 7 mutations described above, only '\n",
      "               'one mutation in exon 3, c.231T>G (p.Thr77Thr),16 and two '\n",
      "               'mutations in exon 18, c.8165C>G <<<(p.Thr2722Arg>>>)28 and '\n",
      "               'c.7992T>A (p.Ile2664Ile),16 have been reported to induce exon '\n",
      "               'skipping by altering splicing regulatory elements.']},\n",
      " {'pmid': '21990134',\n",
      "  'points': 40,\n",
      "  'snippets': ['0.81  0.95  4   18  A2717S  p.Ala2717Ser  8377G>T  c.8149G>T  '\n",
      "               '1.60×10−4  Spurdle et al., 2008  0.03  4.95×10−6  1   18  '\n",
      "               'T2722R  p.Thr2722Arg  8393C>G <<< c.8165C>G>>>  93.50  '\n",
      "               'Farrugia et al., 2008  0.81  1.00  5   18  D2723H  '\n",
      "               'p.Asp2723His  8395G>C  c.8167G>C  5.62×1011  Farrugia et al., '\n",
      "               '2008  0.81  1.00  5   18  #D',\n",
      "               '057T>C  4.00  This manuscript  0.81  0.95  4   18  A2717S  '\n",
      "               'p.Ala2717Ser  8377G>T  c.8149G>T  1.60×10−4  Spurdle et al., '\n",
      "               '2008  0.03  4.95×10−6  1   18 <<< T2722R>>>  p.Thr2722Arg  '\n",
      "               '8393C>G  c.8165C>G  93.50  Farrugia et al., 2008  0.81  1.00  '\n",
      "               '5   18  D2723H  p.Asp2723His  8395G>C  c.8167G>C  5.62×1011  '\n",
      "               'Farrugia et',\n",
      "               '4.00  This manuscript  0.81  0.95  4   18  A2717S  '\n",
      "               'p.Ala2717Ser  8377G>T  c.8149G>T  1.60×10−4  Spurdle et al., '\n",
      "               '2008  0.03  4.95×10−6  1   18  T2722R <<< p.Thr2722Arg>>>  '\n",
      "               '8393C>G  c.8165C>G  93.50  Farrugia et al., 2008  0.81  1.00  '\n",
      "               '5   18  D2723H  p.Asp2723His  8395G>C  c.8167G>C  5.62×1011  '\n",
      "               'Farrugia et al., 2008  0.']},\n",
      " {'pmid': '24323938',\n",
      "  'points': 40,\n",
      "  'snippets': ['.8149G>T  8377G>T  OB1  1  Edwards SM et al., 2003; Diez et '\n",
      "               'al., 2003; Salazar et al., 2006; Spurdle et al., 2008; Lindor '\n",
      "               'et al., 2012   p.Thr2722Arg <<< c.8165C>G>>>  8393C>G  OB1  5  '\n",
      "               'Easton et al., 2007; Farrugia et al 2008; Lindor et al., '\n",
      "               '2012   p.Asp2723His  c.8167G>C  8395G>C  OB1  5  Goldgar et '\n",
      "               'al. 2004; Wu e',\n",
      "               '.Ala2717Ser  c.8149G>T  8377G>T  OB1  1  Edwards SM et al., '\n",
      "               '2003; Diez et al., 2003; Salazar et al., 2006; Spurdle et al., '\n",
      "               '2008; Lindor et al., 2012  <<< p.Thr2722Arg>>>  c.8165C>G  '\n",
      "               '8393C>G  OB1  5  Easton et al., 2007; Farrugia et al 2008; '\n",
      "               'Lindor et al., 2012   p.Asp2723His  c.8167G>C  8395G>C  OB1  '\n",
      "               '5  Goldgar et al.',\n",
      "               ' +/−  p.Ile2490Thr  +  +  +  +  +  p.Ala2643Gly  +  '\n",
      "               'p.Ser2006Arg  −  p.Gly2748Asp  −   p.Glu462Gly  +  '\n",
      "               'p.Ser2807Leu  +  p.Leu2510Pro  +/−  −  −  −  − <<< '\n",
      "               'p.Thr2722Arg>>>  −  p.Arg2108Cys  −  p.Ala2951Thr  +   '\n",
      "               'p.Thr1302del  +  p.Gly2812Glu  +/−  p.Trp2626Cys  +/−  −  −  '\n",
      "               '−  −  p.Lys2729Asn  +  p.Gly2353Arg  −    p.Glu13']},\n",
      " {'pmid': '25146914',\n",
      "  'points': 40,\n",
      "  'snippets': ['   +   +   ND   No   NA   NA   17   c.7879A>T   p.Ile2627Phe   '\n",
      "               'Class 5   Impairs protein function   Oligo targeting   +   +   '\n",
      "               'ND   No   NA   NA   18  <<< c.8165C>G>>>   p.Thr2722Arg   '\n",
      "               'Class 5   Impairs protein function   Exon mutagenesis   +   '\n",
      "               '+   ND   No   NA   NA   18   c.8167G>C   p.Asp2723His   Class '\n",
      "               '5   Impair',\n",
      "               'D   No   NA   NA   17   c.7879A>T   p.Ile2627Phe   Class 5   '\n",
      "               'Impairs protein function   Oligo targeting   +   +   ND   No   '\n",
      "               'NA   NA   18   c.8165C>G  <<< p.Thr2722Arg>>>   Class 5   '\n",
      "               'Impairs protein function   Exon mutagenesis   +   +   ND   '\n",
      "               'No   NA   NA   18   c.8167G>C   p.Asp2723His   Class 5   '\n",
      "               'Impairs protein funct',\n",
      "               'In agreement with their established functional impairment, '\n",
      "               'none of the pathogenic hBRCA2 variants (p.Trp2626Cys, '\n",
      "               'p.Ile2627Phe,<<< p.Thr2722Arg>>>, p.Asp2723His; p.Asp2723Gly, '\n",
      "               'p.Gly2748Asp and p.Arg3052Trp, c.7617+2T>G, c.8754+5G>A) could '\n",
      "               'complement the lethality of mBrca2‐deficient mES cells, a']},\n",
      " {'pmid': '20215541',\n",
      "  'points': 30,\n",
      "  'snippets': ['None of the five missense mutations directed to conserved ESE '\n",
      "               'significantly altered splicing, although variant<<< '\n",
      "               'c.8165C>G>>> (34) showed a weak exon 18 skipping in minigene '\n",
      "               'context, also detected by other authors (33, 35).  Six '\n",
      "               'positive splicing mutations showed simultaneou',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'None of the five missense mutations directed to conserved ESE '\n",
      "               'significantly altered splicing, although variant<<< '\n",
      "               'c.8165C>G>>> (34) showed a weak exon 18 skipping in minigene '\n",
      "               'context, also detected by other authors (33, 35). Six positive '\n",
      "               'splicing mutations showed simultaneous']},\n",
      " {'pmid': '18424508',\n",
      "  'points': 30,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'Of these, the exonic variant BRCA2 c.8162T→C in exon 18 '\n",
      "               'affects a position only three nucleotides upstream of the '\n",
      "               'mutation BRCA2 c.8165C→G (predicting<<< p.T2722R>>>), which is '\n",
      "               'known to cause exon skipping by disrupting several ESE '\n",
      "               'sites.20  Two variants, BRCA2 c.316+5G→C (IVS3+5G→C) and '\n",
      "               'c.7805G→C, at the last base of exon 16, induced strong effects '\n",
      "               'on splicing (figs 2A,B and 2C,D, respectively).',\n",
      "               'Of these, the exonic variant BRCA2 c.8162TRC in exon 18 '\n",
      "               'affects a position only three nucleotides upstream of the '\n",
      "               'mutation BRCA2<<< c.8165CRG>>> (predicting p.T2722R), which is '\n",
      "               'known to cause exon skipping by disrupting several ESE '\n",
      "               'sites.20 Two variants, BRCA2 c.316+5GRC (IVS3+5GRC) and '\n",
      "               'c.7805']},\n",
      " {'pmid': '30039884',\n",
      "  'points': 20,\n",
      "  'snippets': ['Another missense mutation (BRCA2<<< p.Thr2722Arg>>>) was '\n",
      "               'interestingly reported as pathogenic in the Clinvar database.',\n",
      "               'Therefore, BRCA2<<< p.Thr2722Arg>>> has a high probability to '\n",
      "               'be pathogenic.  The positional relationship between possible '\n",
      "               'disease‐causing mutations and the functional domains of the BR',\n",
      "               'l2050fs were located in the BRC repeat region; p.Trp2990Ter, '\n",
      "               'p.Ile2822fs, and p.Leu2904fs were located in the '\n",
      "               'oligonucleotide‐binding (OB) region; and<<< p.Thr2722Arg>>> '\n",
      "               'was located in the DNA binding domain of the BRCA2 protein.']},\n",
      " {'pmid': '21990165',\n",
      "  'points': 20,\n",
      "  'snippets': ['plus BRCA2 p.D3095E) and eight were reclassified as Class 5 '\n",
      "               '(BRCA1 p.T1685A, p.T1685I, p.S1715R, p.G1738R, p.L1764P, '\n",
      "               'p.I1766S, plus BRCA2 p.W2626C and<<< p.T2722R>>>).  A group of '\n",
      "               'four BRCA2 missense substitutions assessed by Spearman et al. '\n",
      "               '(Spearman et al., 2008) are now classified as Class 2.',\n",
      "               'plus BRCA2 p.D3095E) and eight were reclassified as Class 5 '\n",
      "               '(BRCA1 p.T1685A, p.T1685I, p.S1715R, p.G1738R, p.L1764P, '\n",
      "               'p.I1766S, plus BRCA2 p.W2626C and<<< p.T2722R>>>).  Hum '\n",
      "               'Mutat.']},\n",
      " {'pmid': '23586058',\n",
      "  'points': 20,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'Olopade, “BRCA2<<< T2722R>>> is a deleterious allele that '\n",
      "               'causes exon skipping,” The American Journal of Human Genetics, '\n",
      "               'vol. 71, no. 3, pp. 625–631, 2002. [38] C.']},\n",
      " {'pmid': '20690207',\n",
      "  'points': 20,\n",
      "  'snippets': ['28.  Fackenthal JD, Cartegni L, Krainer AR, Olopade OI: '\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               ' TI, Olah E: Emerging roles of BRCA1 alternative splicing. Mol '\n",
      "               'Pathol 2003; 56: 191 – 197. 28 Fackenthal JD, Cartegni L, '\n",
      "               'Krainer AR, Olopade OI: BRCA2<<< T2722R>>> is a deleterious '\n",
      "               'allele that causes exon skipping.']},\n",
      " {'pmid': '20522429',\n",
      "  'points': 20,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'In the BRCA2 gene, exon skipping has been associated with the '\n",
      "               'missense mutations c.8165C→G <<<(p.Thr2722Arg>>>) in exon 1824 '\n",
      "               'and c.6853A→G (p.Ile2285Val) in exon 12,25 but the molecular '\n",
      "               'bases of these splicing defects have not been described.  '\n",
      "               'Retrospectively,',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.']},\n",
      " {'pmid': '20507642',\n",
      "  'points': 20,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'Fackenthal JD, Cartegni L, Krainer AR, Olopade OI: BRCA2<<< '\n",
      "               'T2722R>>> is a deleterious allele that causes exon skipping.']},\n",
      " {'pmid': '19471317',\n",
      "  'points': 20,\n",
      "  'snippets': ['28.  Fackenthal JD, Cartegni L, Krainer AR, Olopade OI: '\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               ' TI, Olah E: Emerging roles of BRCA1 alternative splicing. Mol '\n",
      "               'Pathol 2003; 56: 191 – 197. 28 Fackenthal JD, Cartegni L, '\n",
      "               'Krainer AR, Olopade OI: BRCA2<<< T2722R>>> is a deleterious '\n",
      "               'allele that causes exon skipping.']},\n",
      " {'pmid': '18607349',\n",
      "  'points': 20,\n",
      "  'snippets': ['Here, we tested a BRCA2 missense variant<<< T2722R>>> that was '\n",
      "               'recently reported to cause aberrant splicing and results in '\n",
      "               'deletion of exon 18 in the transcript22.',\n",
      "               'Nevertheless, the full-length<<< T2722R>>> BRCA2 did not '\n",
      "               'rescue ES cell viability, corroborating our conclusion that '\n",
      "               'the mutation is deleterious.  Figure 3: Analysis of BRCA2 '\n",
      "               'mutants affecting',\n",
      "               'ing our conclusion that the mutation is deleterious.  Figure '\n",
      "               '3: Analysis of BRCA2 mutants affecting splicing or domain '\n",
      "               'structure.  Figure 3  (a) BRCA2<<< T2722R>>> transcript '\n",
      "               'reveals skipping of exon 18 (Δ18) not detected for WT BRCA2.']},\n",
      " {'pmid': '18451181',\n",
      "  'points': 20,\n",
      "  'snippets': ['Two other variants (E2663V, D2723G) reached odds in favor of '\n",
      "               'disease causality of at least 100:1 and<<< T2722R>>> reached '\n",
      "               'odds of 93:1 ( Table 1).',\n",
      "               'In contrast, I2627F, L2647P, L2653P,<<< T2722R>>>, D2723H, '\n",
      "               'D2723G, D2723A, G2748D, R2784W, R3052W, and D3095E displayed '\n",
      "               'reduced HDR activity ( Fig. 3B).',\n",
      "               'The limits of these groups were defined by R2520Q (4.0-fold) '\n",
      "               'and<<< T2722R>>> (2.5-fold).']},\n",
      " {'pmid': '18273839',\n",
      "  'points': 20,\n",
      "  'snippets': ['In the other major breast cancer predisposing gene BRCA2, the '\n",
      "               'missense mutation<<< T2722R>>> (8393C>G) which is predicted to '\n",
      "               'disrupt three potential ESE sites also causes exon skipping, '\n",
      "               'leading to an out‐of‐frame fusion of BRCA2 exons 17 and ',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'In the other major breast cancer predisposing gene BRCA2, the '\n",
      "               'missense mutation<<< T2722R>>> (8393C>G) which is predicted to '\n",
      "               'disrupt three potential ESE sites also causes exon skipping, '\n",
      "               'leading to an out-of-frame fusion of BRCA2 exons 17 and ']},\n",
      " {'pmid': '26913838',\n",
      "  'points': 20,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'BRCA2<<< T2722R>>> is a delete- rious allele that causes exon '\n",
      "               'skipping.']},\n",
      " {'pmid': '27060066',\n",
      "  'points': 20,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'C c.7976+3_7974+4del c.7976G>A c.7975A>G c.7806-9T>G '\n",
      "               'c.7806-2A>G c.7977-1G>C c.7988A>T c.8257_8259del c.7977-1G>C '\n",
      "               'c.7988A>T c.8257_8259del c.8331+6G>T<<< c.8165C>G>>> c.8331G>A '\n",
      "               'c.7977-7C>G  27 28 29 4 30 31 31 31 32 50 15 33  Not '\n",
      "               'applicable 34  Not applicable 35, 36 31 37 12 31 31 31 37 14 '\n",
      "               '12 36  Not applicable Not']},\n",
      " {'pmid': '28324225',\n",
      "  'points': 20,\n",
      "  'snippets': ['In BRCA2, p.(Asp2723His) was found in three, p.(Ile262Phe) in '\n",
      "               'two index patients, and p.(Met1Ile) and p.(Asn3124Ile) and '\n",
      "               'p.<<<(Thr2722Arg>>>) were seen in one index patient, '\n",
      "               'respectively.  More VUS were diagnosed in BRCA2 (n\\u2009=\\u2009'\n",
      "               '57) compared to BRCA1 (n\\u2009=\\u200916) (Tables 3, 4).',\n",
      "               'RCA2     Ex 18     c.8172_8175dupGTGG     p.Tyr2726Valfs*5     '\n",
      "               'p.2730X     Frame shift      1     BRCA2     Ex 18     '\n",
      "               'c.8165C\\u2009>\\u2009G     p.Thr2722Arg    <<< p.T2722R>>>     '\n",
      "               'Missense      1     BRCA2     In 20     c.8633-2A\\u2009>\\u2009'\n",
      "               'G     Splice-2     Splice-2     Splice defect      1     '\n",
      "               'BRCA2     Ex 23     c.9025dupT     p',\n",
      "               'hift      1     BRCA2     Ex 18     c.8172_8175dupGTGG     '\n",
      "               'p.Tyr2726Valfs*5     p.2730X     Frame shift      1     '\n",
      "               'BRCA2     Ex 18     c.8165C\\u2009>\\u2009G    <<< '\n",
      "               'p.Thr2722Arg>>>     p.T2722R     Missense      1     BRCA2     '\n",
      "               'In 20     c.8633-2A\\u2009>\\u2009G     Splice-2     '\n",
      "               'Splice-2     Splice defect      1     BRCA2     Ex 23     '\n",
      "               'c.9']},\n",
      " {'pmid': '15026808',\n",
      "  'points': 20,\n",
      "  'snippets': ['Fackenthal JD, Cartegni L, Krainer AR, Olopade OI (2002) '\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes Exon '\n",
      "               'skipping.',\n",
      "               'Dis Markers 15: 191 – 195 Fackenthal JD, Cartegni L, Krainer '\n",
      "               'AR, Olopade OI (2002) BRCA2<<< T2722R>>> is a deleterious '\n",
      "               'allele that causes Exon skipping.']},\n",
      " {'pmid': '29884841',\n",
      "  'points': 20,\n",
      "  'snippets': ['5T>C  C45  1.07  ± 0.08  1.00  2.02E-07  Guidugli et al.,36  '\n",
      "               'D3095E  c.9285C>G  C35  Class 4  1.07  ± 0.12  1.00  1.89E-07  '\n",
      "               'Guidugli et al.,6  T2722R <<< c.8165C>G>>>  C65  Class 5  '\n",
      "               '1.08  ± 0.08  1.00  2.78E-07  Lindor et al.,19  G2508R  '\n",
      "               'c.7522G>C  C65  1.09  ± 0.15  1.00  2.92E-07  Current study  '\n",
      "               'S2691F  c.8072C>T',\n",
      "               'S  c.7925T>C  C45  1.07  ± 0.08  1.00  2.02E-07  Guidugli et '\n",
      "               'al.,36  D3095E  c.9285C>G  C35  Class 4  1.07  ± 0.12  1.00  '\n",
      "               '1.89E-07  Guidugli et al.,6 <<< T2722R>>>  c.8165C>G  C65  '\n",
      "               'Class 5  1.08  ± 0.08  1.00  2.78E-07  Lindor et al.,19  '\n",
      "               'G2508R  c.7522G>C  C65  1.09  ± 0.15  1.00  2.92E-07  Current '\n",
      "               'study  S2691F',\n",
      "               '50L L2972W F2562L D2733V R2842C L2654P N2622S Y2660D Y2624C '\n",
      "               'D3073G Y2726C K2498E G2609D G2585R R2784Q L2792P A3028P W2626C '\n",
      "               'V2687F W2626R D2723G S2691F<<< T2722R>>> F2642S N3124K W2788S '\n",
      "               'D2723V G3076V A2730P N2781I R2824G L2510P L3125H G3076E L2647P '\n",
      "               'N3124I L3125R Y2624D R3052W L2653P G2748D  Normalized fold '\n",
      "               'change']},\n",
      " {'pmid': '29988080',\n",
      "  'points': 20,\n",
      "  'snippets': ['Leu2647Pro  No  NA  NA  17  c.7958T>C  p.Leu2653Pro  No  NA  '\n",
      "               'NA  18  c.8057T>C  p.Leu2686Pro  No  NA  NA  18  c.8063T>C  '\n",
      "               'p.Leu2688Pro  No  NA  NA  18 <<< c.8165C>G>>>  p.Thr2722Arg  '\n",
      "               'No  NA  NA  18  c.8167G>C  p.Asp2723His  No  NA  NA  18  '\n",
      "               'c.8168A>G  p.Asp2723Gly  No  NA  NA  18  c.8243G>A  '\n",
      "               'p.Gly2748Asp  No  NA  NA ',\n",
      "               ' No  NA  NA  17  c.7958T>C  p.Leu2653Pro  No  NA  NA  18  '\n",
      "               'c.8057T>C  p.Leu2686Pro  No  NA  NA  18  c.8063T>C  '\n",
      "               'p.Leu2688Pro  No  NA  NA  18  c.8165C>G <<< p.Thr2722Arg>>>  '\n",
      "               'No  NA  NA  18  c.8167G>C  p.Asp2723His  No  NA  NA  18  '\n",
      "               'c.8168A>G  p.Asp2723Gly  No  NA  NA  18  c.8243G>A  '\n",
      "               'p.Gly2748Asp  No  NA  NA  24  c.9154C>T']},\n",
      " {'pmid': '23108138',\n",
      "  'points': 20,\n",
      "  'snippets': ['In addition, the T2643R mutation (human<<< T2722R>>>) in OB1 '\n",
      "               'potentially leads to a steric clash with W2547 in the helical '\n",
      "               'domain.',\n",
      "               ' 35.96 24.06 0.99 5 D2665G 0.81 0.01 0.13 0.24 3.34   10 4 '\n",
      "               '1.42   10 3 2 M2676T 0.01 1.00 1.16 0.83 0.96 0.01 2 L2688P '\n",
      "               '0.81 1.00 1.19 3.44 4.08 0.95 4<<< T2722R>>> 0.81 10.80 1.24 '\n",
      "               '6.95 93.47 1.00 5 D2723G 0.81 1.00 1.59 75.96 120.57 1.00 5 '\n",
      "               'D2723H 0.81 1.38   104 4.17 9.77   106 5.62   1011 1.00 5 '\n",
      "               'K2729N 0.01 0.0',\n",
      "               'In addition, the T2643Rmutation (human<<< T2722R>>>) in OB1 '\n",
      "               'potentially leads to a steric clash with W2547 in the helical '\n",
      "               'domain.']},\n",
      " {'pmid': '28339459',\n",
      "  'points': 15,\n",
      "  'snippets': ['TATTGCTAAAAGTTTCAGATATATTTGCGC                                  '\n",
      "               'c.8165C>G                                    '\n",
      "               'Missense                                 <<< '\n",
      "               'p.Thr2722Arg>>>                                       '\n",
      "               'VUS                                        -  [-] 2 SF2 '\n",
      "               '(77.62;79.62); [-] SRp40 (83.89); [-] SRp55 (80.71); [+',\n",
      "               '               [-] Sironi ESS '\n",
      "               'GCGCAAATATATCTGAAACTTTTAGCAATAAAACTAGTAGTGC '\n",
      "               'GCACTACTAGTTTTATTGCTAAAAGTTTCAGATATATTTGCGC                                 '\n",
      "               '<<< c.8165C>G>>>                                    '\n",
      "               'Missense                                  '\n",
      "               'p.Thr2722Arg                                       '\n",
      "               'VUS                  ',\n",
      "               'A\\xa0 c.8009C>G\\xa0 c.8009C>T\\xa0 c.8010_8032del\\xa0 '\n",
      "               'c.8019A>G\\xa0 c.8023A>G\\xa0 c.8027T>C\\xa0 c.8027T>A\\xa0 '\n",
      "               'c.8035G>T\\xa0 c.8039A>G\\xa0 c.8042C>G\\xa0 c.8072C>T\\xa0 '\n",
      "               'c.8084C>T\\xa0 c.8111C>T\\xa0<<< c.8165C>G>>>\\xa0 c.8168A>G\\xa0 '\n",
      "               'c.8191C>T\\xa0  VUS VUS VUS VUS VUS VUS VUS/causal Causal '\n",
      "               '(frameshift) VUS VUS VUS VUS VUS Causal (nonsense) VUS VUS '\n",
      "               'Causal (frameshift) VUS ']},\n",
      " {'pmid': '25003164',\n",
      "  'points': 10,\n",
      "  'snippets': ['Nature 434:598–604 Fackenthal JD, Cartegni L, Krainer AR, '\n",
      "               'Olopade OI (2002) BRCA2<<< T2722R>>> is a deleterious allele '\n",
      "               'that causes exon skipping.']},\n",
      " {'pmid': '24817641',\n",
      "  'points': 10,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.']},\n",
      " {'pmid': '24122022',\n",
      "  'points': 10,\n",
      "  'snippets': ['Nature 434:598–604 Fackenthal JD, Cartegni L, Krainer AR, '\n",
      "               'Olopade OI (2002) BRCA2<<< T2722R>>> is a deleterious allele '\n",
      "               'that causes exon skipping.']},\n",
      " {'pmid': '10464631',\n",
      "  'points': 10,\n",
      "  'snippets': ['6        * BRCA2<<< T2722R>>> Is a Deleterious Allele That '\n",
      "               'Causes Exon Skipping  The American Journal of Human Genetics, '\n",
      "               'Vol. 71, No. 3        * Novel germline CDH1 mutations in h']},\n",
      " {'pmid': '22753008',\n",
      "  'points': 10,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.']},\n",
      " {'pmid': '21735045',\n",
      "  'points': 10,\n",
      "  'snippets': ['Fackenthal JD, Cartegni L, Krainer AR, Olopade OI (2002) '\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping. Am J Hum Genet 71:625–631 2.']},\n",
      " {'pmid': '21638052',\n",
      "  'points': 10,\n",
      "  'snippets': ['Fackenthal JD, Cartegni L, Krainer AR, Olopade OI (2002) '\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping. Am J Hum Genet 71(3):625–631 982 Breast Cancer Res '\n",
      "               'Treat (2011) 129:971–982 123']},\n",
      " {'pmid': '12145750',\n",
      "  'points': 10,\n",
      "  'snippets': ['71:625–631, 2002  Report  BRCA2<<< T2722R>>> Is a Deleterious '\n",
      "               'Allele That Causes Exon Skipping James D.',\n",
      "               'These predictions revealed that BRCA2<<< T2722R>>> (8393CrG), '\n",
      "               'which segregates with affected individuals in a family with '\n",
      "               'breast cancer, disrupts three potential ESE sites.',\n",
      "               'Of these, only one allele, BRCA2<<< T2722R>>>, segregated with '\n",
      "               'affected members of a family with breast cancer (family number '\n",
      "               '98-11).']},\n",
      " {'pmid': '21309043',\n",
      "  'points': 10,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.']},\n",
      " {'pmid': '17924331',\n",
      "  'points': 10,\n",
      "  'snippets': ['9R S1715R IVS12⫹2del21insA M18T A1623G (4987CrG) IVS11-1GrA '\n",
      "               'IVS18-1GrC BRCA2: R2659K (8204GrA) G2748D I2627F E2663V P3039P '\n",
      "               '(9345GrA) D2723G IVS15-1GrA<<< T2722R>>> IVS24-1GrC R2336H '\n",
      "               '(7235GrA) W2626C IVS21⫹4ArG L2653P D3095E IVS5-2ArG R2502C '\n",
      "               'IVS19⫹1GrA  FamilyHistory LLR  Co-Occurrence  Cosegregation  '\n",
      "               'Combined LL']},\n",
      " {'pmid': '17899372',\n",
      "  'points': 10,\n",
      "  'snippets': ['Fackenthal JD, Cartegni L, Krainer AR et al (2002) BRCA2<<< '\n",
      "               'T2722R>>> is a deleterious allele that causes exon skipping.']},\n",
      " {'pmid': '16792514',\n",
      "  'points': 10,\n",
      "  'snippets': ['Am J Hum Genet 56:265–271. Fackenthal JD, Cartegni L, Krainer '\n",
      "               'AR, Olopade OI (2002) BRCA2<<< T2722R>>> is a deleterious '\n",
      "               'allele that causes exon skipping.']},\n",
      " {'pmid': '15744044',\n",
      "  'points': 10,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.']},\n",
      " {'pmid': '12845657',\n",
      "  'points': 10,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.']},\n",
      " {'pmid': '21344236',\n",
      "  'points': 10,\n",
      "  'snippets': ['doi:10.1038/ nature05609 Fackenthal JD, Cartegni L, Krainer '\n",
      "               'AR, Olopade OI (2002) BRCA2<<< T2722R>>> is a deleterious '\n",
      "               'allele that causes exon skipping. Am J Hum Genet 71:625–631. '\n",
      "               'doi:10.1086/342192 Fox D 3rd, Le Trong I, Rajagopal P, Brzovic '\n",
      "               'PS, Sten']},\n",
      " {'pmid': '19332451',\n",
      "  'points': 2,\n",
      "  'snippets': ['Human Genetics113551–554.      * Search Google Scholar     * '\n",
      "               'Export Citation    * '\n",
      "               'FackenthalJDCartegniLKrainerAROlopadeOI2002BRCA2<<< T2722R>>> '\n",
      "               'is a deleterious allele that causes exon skipping.',\n",
      "               'Human Genetics113551–554.    * Search Google Scholar   * '\n",
      "               'Export Citation  '\n",
      "               'FackenthalJDCartegniLKrainerAROlopadeOI2002BRCA2<<< T2722R>>> '\n",
      "               'is a deleterious allele that causes exon skipping.',\n",
      "               'Human Genetics 113 551–554. Fackenthal JD, Cartegni L, Krainer '\n",
      "               'AR & Olopade OI 2002 BRCA2<<< T2722R>>> is a deleterious '\n",
      "               'allele that causes exon skipping.']},\n",
      " {'pmid': '16825284',\n",
      "  'points': 2,\n",
      "  'snippets': ['\\xa0(pg.  12614  -  12619  )  Google Scholar  Crossref  Search '\n",
      "               'ADS    29  Fackenthal  J.D.  ,  Cartegni  L.  ,  Krainer  '\n",
      "               'A.R.  ,  Olopade  O.I.  .  BRCA2<<< T2722R>>> is a deleterious '\n",
      "               'allele that causes exon skipping  ,  Am.',\n",
      "               '(2002) BRCA2<<< T2722R>>> is a deleterious allele that causes '\n",
      "               'exon skipping. Am.']},\n",
      " {'pmid': '12915465',\n",
      "  'points': 2,\n",
      "  'snippets': ['(  2002  ) BRCA2<<< T2722R>>> is a deleterious allele that '\n",
      "               'causes exon skipping.  Am.',\n",
      "               '(2002) BRCA2<<< T2722R>>> is a deleterious allele that causes '\n",
      "               'exon skipping.']}]\n"
     ]
    }
   ],
   "source": [
    "pprint.pprint(lit[\"variants\"][\"chr13:g.32363367:C>G\"])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
