{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BRCA Exchange Literature Search Export\n",
    "\n",
    "Ingest artifacts from the pipeline, wrangle, and generate literature.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import json\n",
    "import sqlite3\n",
    "import pprint\n",
    "import pandas as pd\n",
    "\n",
    "os.chdir(os.path.expanduser(\"/home/jovyan/data/pubmunch/crawl/\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ingest\n",
    "Ingest artifacts from various pieces of the pipeline indexed by normalized genomic hgvs and integer pmid "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "159 articles loaded from the articles sqlite database\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>articleId</th>\n",
       "      <th>externalId</th>\n",
       "      <th>source</th>\n",
       "      <th>publisher</th>\n",
       "      <th>origFile</th>\n",
       "      <th>journal</th>\n",
       "      <th>printIssn</th>\n",
       "      <th>eIssn</th>\n",
       "      <th>journalUniqueId</th>\n",
       "      <th>year</th>\n",
       "      <th>...</th>\n",
       "      <th>issue</th>\n",
       "      <th>page</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pmcId</th>\n",
       "      <th>doi</th>\n",
       "      <th>fulltextUrl</th>\n",
       "      <th>time</th>\n",
       "      <th>offset</th>\n",
       "      <th>size</th>\n",
       "      <th>chunkId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5019423647</td>\n",
       "      <td>PMID19423647</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Annals of oncology : official journal of the E...</td>\n",
       "      <td>0923-7534</td>\n",
       "      <td>1569-8041</td>\n",
       "      <td>9007735</td>\n",
       "      <td>2009</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>1143</td>\n",
       "      <td>19423647</td>\n",
       "      <td></td>\n",
       "      <td>10.1093/annonc/mdp241</td>\n",
       "      <td></td>\n",
       "      <td>2019-02-21T00:07:31+0000</td>\n",
       "      <td>78</td>\n",
       "      <td>21596</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5009126734</td>\n",
       "      <td>PMID9126734</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Nature</td>\n",
       "      <td>0028-0836</td>\n",
       "      <td>0028-0836</td>\n",
       "      <td>0410462</td>\n",
       "      <td>1997</td>\n",
       "      <td>...</td>\n",
       "      <td>6627</td>\n",
       "      <td>772</td>\n",
       "      <td>9126734</td>\n",
       "      <td></td>\n",
       "      <td>10.1038/386772a0</td>\n",
       "      <td></td>\n",
       "      <td>2019-02-21T00:07:58+0000</td>\n",
       "      <td>22305</td>\n",
       "      <td>19931</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5022889855</td>\n",
       "      <td>PMID22889855</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Journal of medical genetics</td>\n",
       "      <td>0022-2593</td>\n",
       "      <td>1468-6244</td>\n",
       "      <td>2985087R</td>\n",
       "      <td>2012</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>525</td>\n",
       "      <td>22889855</td>\n",
       "      <td>3810416</td>\n",
       "      <td>10.1136/jmedgenet-2012-101037</td>\n",
       "      <td></td>\n",
       "      <td>2019-02-21T00:08:35+0000</td>\n",
       "      <td>42718</td>\n",
       "      <td>118856</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5012473589</td>\n",
       "      <td>PMID12473589</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td>1078-0432</td>\n",
       "      <td>1078-0432</td>\n",
       "      <td>9502500</td>\n",
       "      <td>2002</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>3776</td>\n",
       "      <td>12473589</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2019-02-21T00:08:56+0000</td>\n",
       "      <td>2919658796</td>\n",
       "      <td>72920</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5018285836</td>\n",
       "      <td>PMID18285836</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>European journal of human genetics : EJHG</td>\n",
       "      <td>1018-4813</td>\n",
       "      <td>1018-4813</td>\n",
       "      <td>9302235</td>\n",
       "      <td>2008</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>820</td>\n",
       "      <td>18285836</td>\n",
       "      <td>3905962</td>\n",
       "      <td>10.1038/ejhg.2008.13</td>\n",
       "      <td></td>\n",
       "      <td>2019-02-21T00:09:18+0000</td>\n",
       "      <td>4427189208</td>\n",
       "      <td>113381</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    articleId    externalId source publisher origFile  \\\n",
       "0  5019423647  PMID19423647         download            \n",
       "1  5009126734   PMID9126734         download            \n",
       "2  5022889855  PMID22889855         download            \n",
       "3  5012473589  PMID12473589         download            \n",
       "4  5018285836  PMID18285836         download            \n",
       "\n",
       "                                             journal  printIssn      eIssn  \\\n",
       "0  Annals of oncology : official journal of the E...  0923-7534  1569-8041   \n",
       "1                                             Nature  0028-0836  0028-0836   \n",
       "2                        Journal of medical genetics  0022-2593  1468-6244   \n",
       "3  Clinical cancer research : an official journal...  1078-0432  1078-0432   \n",
       "4          European journal of human genetics : EJHG  1018-4813  1018-4813   \n",
       "\n",
       "  journalUniqueId  year   ...    issue  page      pmid    pmcId  \\\n",
       "0         9007735  2009   ...        6  1143  19423647            \n",
       "1         0410462  1997   ...     6627   772   9126734            \n",
       "2        2985087R  2012   ...        8   525  22889855  3810416   \n",
       "3         9502500  2002   ...       12  3776  12473589            \n",
       "4         9302235  2008   ...        7   820  18285836  3905962   \n",
       "\n",
       "                             doi fulltextUrl                      time  \\\n",
       "0          10.1093/annonc/mdp241              2019-02-21T00:07:31+0000   \n",
       "1               10.1038/386772a0              2019-02-21T00:07:58+0000   \n",
       "2  10.1136/jmedgenet-2012-101037              2019-02-21T00:08:35+0000   \n",
       "3                                             2019-02-21T00:08:56+0000   \n",
       "4           10.1038/ejhg.2008.13              2019-02-21T00:09:18+0000   \n",
       "\n",
       "       offset    size  chunkId  \n",
       "0          78   21596  0_00000  \n",
       "1       22305   19931  0_00000  \n",
       "2       42718  118856  0_00000  \n",
       "3  2919658796   72920  0_00000  \n",
       "4  4427189208  113381  0_00000  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "connection = sqlite3.connect(\"file:text/articles.db?mode=ro\", uri=True)\n",
    "articles = pd.read_sql_query(\"SELECT * FROM articles\", connection)\n",
    "articles.pmid = articles.pmid.astype(str)\n",
    "print(\"{} articles loaded from the articles sqlite database\".format(articles.shape[0]))\n",
    "articles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total matched mentions: 9804\n",
      "After dropping duplicates: 5153\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pyhgvs_Genomic_Coordinate_38</th>\n",
       "      <th>pmid</th>\n",
       "      <th>snippets</th>\n",
       "      <th>points</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17:g.43092632:T&gt;A</td>\n",
       "      <td>10373534</td>\n",
       "      <td>1051-1863, using a Clontech transformer site-d...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17:g.43092632:T&gt;A</td>\n",
       "      <td>10373534</td>\n",
       "      <td>1051-1863, using a Clontech transformer site-...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17:g.43076488:C&gt;A</td>\n",
       "      <td>10373534</td>\n",
       "      <td>A mutation in close proximity has been compile...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr17:g.43074517:A&gt;C</td>\n",
       "      <td>10373534</td>\n",
       "      <td>Mutants&lt;&lt;&lt; S1497A&gt;&gt;&gt; and S1497T were generated...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr17:g.43074517:A&gt;C</td>\n",
       "      <td>10373534</td>\n",
       "      <td>Mutants&lt;&lt;&lt; S1497A&gt;&gt;&gt; and S1497T were generated...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  pyhgvs_Genomic_Coordinate_38      pmid  \\\n",
       "0         chr17:g.43092632:T>A  10373534   \n",
       "1         chr17:g.43092632:T>A  10373534   \n",
       "2         chr17:g.43076488:C>A  10373534   \n",
       "6         chr17:g.43074517:A>C  10373534   \n",
       "7         chr17:g.43074517:A>C  10373534   \n",
       "\n",
       "                                            snippets  points  \n",
       "0  1051-1863, using a Clontech transformer site-d...      10  \n",
       "1   1051-1863, using a Clontech transformer site-...      10  \n",
       "2  A mutation in close proximity has been compile...      10  \n",
       "6  Mutants<<< S1497A>>> and S1497T were generated...      10  \n",
       "7  Mutants<<< S1497A>>> and S1497T were generated...      10  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mentions = pd.read_csv(\"mentions-matched.tsv\", sep=\"\\t\", encoding=\"utf-8\")\n",
    "mentions.pmid = mentions.pmid.astype(str)\n",
    "print(\"Total matched mentions: {}\".format(mentions.shape[0]))\n",
    "mentions = mentions.drop_duplicates([\"pyhgvs_Genomic_Coordinate_38\", \"pmid\", \"snippets\"])\n",
    "print(\"After dropping duplicates: {}\".format(mentions.shape[0]))\n",
    "mentions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([['22962691', 40],\n",
       "       ['18424508', 30],\n",
       "       ['20215541', 30],\n",
       "       ['18451181', 20],\n",
       "       ['18607349', 20],\n",
       "       ['20507642', 20],\n",
       "       ['20522429', 20],\n",
       "       ['12145750', 10],\n",
       "       ['16792514', 10],\n",
       "       ['17899372', 10],\n",
       "       ['17924331', 10],\n",
       "       ['12915465', 2]], dtype=object)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def top_papers(mentions, pyhgvs):\n",
    "    \"\"\" Return (pmid, points) tuple sorted by points against pyhgvs \"\"\"\n",
    "    top = mentions[mentions.pyhgvs_Genomic_Coordinate_38 == pyhgvs] \\\n",
    "        .groupby([\"pyhgvs_Genomic_Coordinate_38\", \"pmid\"]) \\\n",
    "        .agg({\"points\": sum}) \\\n",
    "        .sort_values(\"points\", ascending=False)\n",
    "    return top.reset_index()[[\"pmid\", \"points\"]].values\n",
    "\n",
    "top_papers(mentions, \"chr13:g.32363367:C>G\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['BRCA2<<< T2722R>>> is a deleterious allele that causes exon skipping.',\n",
       " 'Of these, the exonic variant BRCA2 c.8162T→C in exon 18 affects a position only three nucleotides upstream of the mutation BRCA2 c.8165C→G (predicting<<< p.T2722R>>>), which is known to cause exon skipping by disrupting several ESE sites.20  Two variants, BRCA2 c.316+5G→C (IVS3+5G→C) and c.7805G→C, at the last base of exon 16, induced strong effects on splicing (figs 2A,B and 2C,D, respectively).',\n",
       " 'Of these, the exonic variant BRCA2 c.8162TRC in exon 18 affects a position only three nucleotides upstream of the mutation BRCA2<<< c.8165CRG>>> (predicting p.T2722R), which is known to cause exon skipping by disrupting several ESE sites.20 Two variants, BRCA2 c.316+5GRC (IVS3+5GRC) and c.7805']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def top_snippets(mentions, pyhgvs, pmid):\n",
    "    \"\"\" Return list of top snippets for this variant and paper by points \"\"\"\n",
    "    paper = mentions[(mentions.pyhgvs_Genomic_Coordinate_38 == pyhgvs) & (mentions.pmid == pmid)]\n",
    "    return [snippet \n",
    "     for snippets in paper.sort_values(\"points\", ascending=False).snippets.values \n",
    "     for snippet in snippets.split(\"|\")][:3]\n",
    "\n",
    "top_snippets(mentions, \"chr13:g.32363367:C>G\", '18424508')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr13:g.32363367:C>G\n",
      "[{'pmid': '22962691',\n",
      "  'points': 40,\n",
      "  'snippets': [', in the BRCA2 gene, besides the exon 7 mutations described '\n",
      "               'above, only one mutation in exon 3, c.231T>G (p.Thr77Thr),16 '\n",
      "               'and two mutations in exon 18,<<< c.8165C>G>>> (p.Thr2722Arg)28 '\n",
      "               'and c.7992T>A (p.Ile2664Ile),16 have been reported to induce '\n",
      "               'exon skipping by altering splicing regulatory elements.',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'CA2 gene, besides the exon 7 mutations described above, only '\n",
      "               'one mutation in exon 3, c.231T>G (p.Thr77Thr),16 and two '\n",
      "               'mutations in exon 18, c.8165C>G <<<(p.Thr2722Arg>>>)28 and '\n",
      "               'c.7992T>A (p.Ile2664Ile),16 have been reported to induce exon '\n",
      "               'skipping by altering splicing regulatory elements.']},\n",
      " {'pmid': '18424508',\n",
      "  'points': 30,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'Of these, the exonic variant BRCA2 c.8162T→C in exon 18 '\n",
      "               'affects a position only three nucleotides upstream of the '\n",
      "               'mutation BRCA2 c.8165C→G (predicting<<< p.T2722R>>>), which is '\n",
      "               'known to cause exon skipping by disrupting several ESE '\n",
      "               'sites.20  Two variants, BRCA2 c.316+5G→C (IVS3+5G→C) and '\n",
      "               'c.7805G→C, at the last base of exon 16, induced strong effects '\n",
      "               'on splicing (figs 2A,B and 2C,D, respectively).',\n",
      "               'Of these, the exonic variant BRCA2 c.8162TRC in exon 18 '\n",
      "               'affects a position only three nucleotides upstream of the '\n",
      "               'mutation BRCA2<<< c.8165CRG>>> (predicting p.T2722R), which is '\n",
      "               'known to cause exon skipping by disrupting several ESE '\n",
      "               'sites.20 Two variants, BRCA2 c.316+5GRC (IVS3+5GRC) and '\n",
      "               'c.7805']},\n",
      " {'pmid': '20215541',\n",
      "  'points': 30,\n",
      "  'snippets': ['None of the five missense mutations directed to conserved ESE '\n",
      "               'significantly altered splicing, although variant<<< '\n",
      "               'c.8165C>G>>> (34) showed a weak exon 18 skipping in minigene '\n",
      "               'context, also detected by other authors (33, 35).  Six '\n",
      "               'positive splicing mutations showed simultaneou',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'None of the five missense mutations directed to conserved ESE '\n",
      "               'significantly altered splicing, although variant<<< '\n",
      "               'c.8165C>G>>> (34) showed a weak exon 18 skipping in minigene '\n",
      "               'context, also detected by other authors (33, 35). Six positive '\n",
      "               'splicing mutations showed simultaneous']}]\n",
      "chr17:g.43124027:ACT>A\n",
      "[{'pmid': '20608970',\n",
      "  'points': 15,\n",
      "  'snippets': ['Library CAS PubMed Web of Science®Google ScholarUC-eLinks    * '\n",
      "               '29 Buisson M, Anczukow O, Zetoune AB, Ware MD & Mazoyer S '\n",
      "               '(2006) The 185delAG mutation <<<(c.68_69delAG>>>) in the BRCA1 '\n",
      "               'gene triggers translation reinitiation at a downstream AUG '\n",
      "               'codon.',\n",
      "               'lar, cellular and clinical impact for mutation carriers.  ### '\n",
      "               'Abbreviations  * BARD1      * BRCA1‐associated RING domain '\n",
      "               'protein 1 * BRAT      * BRCA1<<< 185delAG>>> truncation * '\n",
      "               'BRCA1      * breast cancer susceptibility gene 1 * BRCT      * '\n",
      "               'BRCA1 C‐terminus  ##  Introduction  Family history is the '\n",
      "               'strongest risk ',\n",
      "               'n Y1853X, which lacks the last 11 amino acids, is only missing '\n",
      "               'a small portion of the second BRCT (BRCA1 C‐terminus) repeat, '\n",
      "               'whereas the 39 amino acid<<< 185delAG>>> mutant lacks all of '\n",
      "               'BRCA1’s known functional domains.  image  Figure 1  Open in '\n",
      "               'figure viewerPowerPoint  BRCA1 mutations and their cellular '\n",
      "               'and physi']},\n",
      " {'pmid': '16267036',\n",
      "  'points': 12,\n",
      "  'snippets': ['It has previously been reported that this effect is '\n",
      "               'responsible for the finding that the most prevalent BRCA1 '\n",
      "               'mutation,<<< 187delAG>>>, is found on two haplotypes ( 12, '\n",
      "               '21).',\n",
      "               'The<<< 185delAG>>> BRCA1 mutation originated before the '\n",
      "               'dispersion of Jews in the diaspora and is not limited to '\n",
      "               'Ashkenazim.',\n",
      "               'It has previously been reported that this effect is '\n",
      "               'responsible for the finding that the most prevalent BRCA1 '\n",
      "               'mutation,<<< 187delAG>>>, is found on two haplotypes (12, '\n",
      "               '21).']},\n",
      " {'pmid': '15235020',\n",
      "  'points': 12,\n",
      "  'snippets': ['(Under this convention, the two mutations commonly referred to '\n",
      "               'as “185delAG” and “5382insC” are named<<< 187delAG>>> and '\n",
      "               '5385insC, respectively.',\n",
      "               'mations of the expected number of Ashkenazi homozygotes and '\n",
      "               'compound heterozygotes  There are two BRCA1 founder mutations '\n",
      "               'in the Ashkenazi population,<<< 185delAG>>> and 5382insC.',\n",
      "               'ividuals is unlikely to produce interesting results. However, '\n",
      "               'in the Ashkenazi Jewish population, there are two founder '\n",
      "               'frameshift mutations in BRCA1:<<< 185delAG>>> and 5382insC.']}]\n",
      "chr13:g.32340526:AT>A\n",
      "[{'pmid': '10638982',\n",
      "  'points': 20,\n",
      "  'snippets': ['n L, Tran T, McGuire P, Swensen J, Hampel H, Borgen P, Brown '\n",
      "               'K, Skolnick M, Shattuck-Eidens D, Jhanwar S, Goldgar D and '\n",
      "               'Offit K (1996) Recurrent BRCA2<<< 6174delT>>> mutations in '\n",
      "               'Askenazi Jewish women affected by breast cancer.',\n",
      "               'Few major founder mutations have so far been detected: the '\n",
      "               '999del5 mutation in the Icelandic population, the<<< '\n",
      "               '6174delT>>> mutation in the Ashkenazi Jewish population, and '\n",
      "               'the 8765delAG mutation in the French-Canadian population '\n",
      "               '(Neuhausen et al, 1996, 1998; Thorlacius et al, 1996; Tonin et '\n",
      "               'al, 1998).',\n",
      "               'n L, Tran T, McGuire P, Swensen J, Hampel H, Borgen P, Brown '\n",
      "               'K, Skolnick M, Shattuck-Eidens D, Jhanwar S, Goldgar D and '\n",
      "               'Offit K (1996) Recurrent BRCA2<<< 6174delT>>> mutations in '\n",
      "               'Askenazi Jewish women affected by breast cancer.']},\n",
      " {'pmid': '15695382',\n",
      "  'points': 20,\n",
      "  'snippets': ['The BRCA2 truncating<<< 6174delT>>> Ashkenazi Jewish founder '\n",
      "               'mutation associated with a breast cancer risk of 70% by age 70 '\n",
      "               '(32) was also included as a positive/inactivating control, wh',\n",
      "               'The localization of BRCA2 and<<< 6174delT>>> to the nucleus '\n",
      "               'and cytoplasm ( Table 2 ; Fig. 1C), respectively, was '\n",
      "               'consistent with the C-terminal location of the human BRCA2 '\n",
      "               'nuclear localization signals (4).',\n",
      "               'VC8 cells transiently transfected with GFP-tagged wtBRCA2 '\n",
      "               'and<<< 6174delT>>> BRCA2 and flow sorted for GFP were '\n",
      "               'evaluated for MMC sensitivity by clonogenic survival assay and '\n",
      "               'trypan blue exclusion. % surviving colonies and % viable cells '\n",
      "               'in treated relative to untreated cells was plotted against MMC '\n",
      "               'concentration.']},\n",
      " {'pmid': '25192939',\n",
      "  'points': 20,\n",
      "  'snippets': ['CA2 mutation carrier only after she develops cancer.  Among '\n",
      "               'Ashkenazi (European) Jews (AJ), three mutations, BRCA1 '\n",
      "               '185delAG, BRCA1 5382insC, and BRCA2<<< 6174delT>>>, account '\n",
      "               'for the great majority of inherited cancer risk due to BRCA1 '\n",
      "               'and BRCA2 (7).',\n",
      "               'In comparison with studies of BRCA2 in non-AJ populations, a '\n",
      "               'second contributing factor may be the location of BRCA2<<< '\n",
      "               '6174delT>>> in the BRCA2 “ovarian cancer cluster region,” '\n",
      "               'where mutations may be associated with higher ovarian cancer '\n",
      "               'risks (23).',\n",
      "               'les who chose to participate provided a blood or buccal sample '\n",
      "               'from which DNA was extracted and genotyped for BRCA1 185delAG, '\n",
      "               'BRCA1 5382insC and BRCA2<<< 6174delT>>> (SI Appendix, Table '\n",
      "               'S2).']}]\n"
     ]
    }
   ],
   "source": [
    "def top_papers_and_snippets(mentions, pyhgvs):\n",
    "    return [{\"pmid\": str(p[0]), \"points\": int(p[1]), \"snippets\": top_snippets(mentions, pyhgvs, p[0])}\n",
    "            for p in top_papers(mentions, pyhgvs)]\n",
    "\n",
    "for pyhgvs in [\"chr13:g.32363367:C>G\", \"chr17:g.43124027:ACT>A\", \"chr13:g.32340526:AT>A\"]:\n",
    "    print(pyhgvs)\n",
    "    pprint.pprint(top_papers_and_snippets(mentions, pyhgvs)[:3])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exported 1811 variants in 149 papers\n",
      "149 Papers and 1811 Variants exported\n"
     ]
    }
   ],
   "source": [
    "lit = {\n",
    "    \"date\": open(\"pubs-date.txt\").read().strip(),\n",
    "    \"papers\": articles[articles.pmid.isin(mentions.pmid)].set_index(\"pmid\", drop=False).to_dict(orient=\"index\"),\n",
    "    \"variants\": {pyhgvs: top_papers_and_snippets(mentions, pyhgvs) \n",
    "                 for pyhgvs in mentions.pyhgvs_Genomic_Coordinate_38.unique()}\n",
    "}\n",
    "\n",
    "with open(\"literature.json\", \"w\") as output:\n",
    "    output.write(json.dumps(lit, sort_keys=True))\n",
    "    \n",
    "print(\"Exported {} variants in {} papers\".format(\n",
    "    len(lit[\"variants\"].keys()), len(lit[\"papers\"].keys())))\n",
    "\n",
    "with open(\"literature.json\") as f:\n",
    "    lit = json.loads(f.read())\n",
    "print(\"{} Papers and {} Variants exported\".format(len(lit[\"papers\"]), len(lit[\"variants\"])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'pmid': '22962691',\n",
      "  'points': 40,\n",
      "  'snippets': [', in the BRCA2 gene, besides the exon 7 mutations described '\n",
      "               'above, only one mutation in exon 3, c.231T>G (p.Thr77Thr),16 '\n",
      "               'and two mutations in exon 18,<<< c.8165C>G>>> (p.Thr2722Arg)28 '\n",
      "               'and c.7992T>A (p.Ile2664Ile),16 have been reported to induce '\n",
      "               'exon skipping by altering splicing regulatory elements.',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'CA2 gene, besides the exon 7 mutations described above, only '\n",
      "               'one mutation in exon 3, c.231T>G (p.Thr77Thr),16 and two '\n",
      "               'mutations in exon 18, c.8165C>G <<<(p.Thr2722Arg>>>)28 and '\n",
      "               'c.7992T>A (p.Ile2664Ile),16 have been reported to induce exon '\n",
      "               'skipping by altering splicing regulatory elements.']},\n",
      " {'pmid': '18424508',\n",
      "  'points': 30,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'Of these, the exonic variant BRCA2 c.8162T→C in exon 18 '\n",
      "               'affects a position only three nucleotides upstream of the '\n",
      "               'mutation BRCA2 c.8165C→G (predicting<<< p.T2722R>>>), which is '\n",
      "               'known to cause exon skipping by disrupting several ESE '\n",
      "               'sites.20  Two variants, BRCA2 c.316+5G→C (IVS3+5G→C) and '\n",
      "               'c.7805G→C, at the last base of exon 16, induced strong effects '\n",
      "               'on splicing (figs 2A,B and 2C,D, respectively).',\n",
      "               'Of these, the exonic variant BRCA2 c.8162TRC in exon 18 '\n",
      "               'affects a position only three nucleotides upstream of the '\n",
      "               'mutation BRCA2<<< c.8165CRG>>> (predicting p.T2722R), which is '\n",
      "               'known to cause exon skipping by disrupting several ESE '\n",
      "               'sites.20 Two variants, BRCA2 c.316+5GRC (IVS3+5GRC) and '\n",
      "               'c.7805']},\n",
      " {'pmid': '20215541',\n",
      "  'points': 30,\n",
      "  'snippets': ['None of the five missense mutations directed to conserved ESE '\n",
      "               'significantly altered splicing, although variant<<< '\n",
      "               'c.8165C>G>>> (34) showed a weak exon 18 skipping in minigene '\n",
      "               'context, also detected by other authors (33, 35).  Six '\n",
      "               'positive splicing mutations showed simultaneou',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'None of the five missense mutations directed to conserved ESE '\n",
      "               'significantly altered splicing, although variant<<< '\n",
      "               'c.8165C>G>>> (34) showed a weak exon 18 skipping in minigene '\n",
      "               'context, also detected by other authors (33, 35). Six positive '\n",
      "               'splicing mutations showed simultaneous']},\n",
      " {'pmid': '18451181',\n",
      "  'points': 20,\n",
      "  'snippets': ['Two other variants (E2663V, D2723G) reached odds in favor of '\n",
      "               'disease causality of at least 100:1 and<<< T2722R>>> reached '\n",
      "               'odds of 93:1 ( Table 1).',\n",
      "               'In contrast, I2627F, L2647P, L2653P,<<< T2722R>>>, D2723H, '\n",
      "               'D2723G, D2723A, G2748D, R2784W, R3052W, and D3095E displayed '\n",
      "               'reduced HDR activity ( Fig. 3B).',\n",
      "               'The limits of these groups were defined by R2520Q (4.0-fold) '\n",
      "               'and<<< T2722R>>> (2.5-fold).']},\n",
      " {'pmid': '18607349',\n",
      "  'points': 20,\n",
      "  'snippets': ['Here, we tested a BRCA2 missense variant<<< T2722R>>> that was '\n",
      "               'recently reported to cause aberrant splicing and results in '\n",
      "               'deletion of exon 18 in the transcript22.',\n",
      "               'Nevertheless, the full-length<<< T2722R>>> BRCA2 did not '\n",
      "               'rescue ES cell viability, corroborating our conclusion that '\n",
      "               'the mutation is deleterious.  Figure 3: Analysis of BRCA2 '\n",
      "               'mutants affecting',\n",
      "               'ing our conclusion that the mutation is deleterious.  Figure '\n",
      "               '3: Analysis of BRCA2 mutants affecting splicing or domain '\n",
      "               'structure.  Figure 3  (a) BRCA2<<< T2722R>>> transcript '\n",
      "               'reveals skipping of exon 18 (Δ18) not detected for WT BRCA2.']},\n",
      " {'pmid': '20507642',\n",
      "  'points': 20,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'Fackenthal JD, Cartegni L, Krainer AR, Olopade OI: BRCA2<<< '\n",
      "               'T2722R>>> is a deleterious allele that causes exon skipping.']},\n",
      " {'pmid': '20522429',\n",
      "  'points': 20,\n",
      "  'snippets': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'In the BRCA2 gene, exon skipping has been associated with the '\n",
      "               'missense mutations c.8165C→G <<<(p.Thr2722Arg>>>) in exon 1824 '\n",
      "               'and c.6853A→G (p.Ile2285Val) in exon 12,25 but the molecular '\n",
      "               'bases of these splicing defects have not been described.  '\n",
      "               'Retrospectively,',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.']},\n",
      " {'pmid': '12145750',\n",
      "  'points': 10,\n",
      "  'snippets': ['71:625–631, 2002  Report  BRCA2<<< T2722R>>> Is a Deleterious '\n",
      "               'Allele That Causes Exon Skipping James D.',\n",
      "               'These predictions revealed that BRCA2<<< T2722R>>> (8393CrG), '\n",
      "               'which segregates with affected individuals in a family with '\n",
      "               'breast cancer, disrupts three potential ESE sites.',\n",
      "               'Of these, only one allele, BRCA2<<< T2722R>>>, segregated with '\n",
      "               'affected members of a family with breast cancer (family number '\n",
      "               '98-11).']},\n",
      " {'pmid': '16792514',\n",
      "  'points': 10,\n",
      "  'snippets': ['Am J Hum Genet 56:265–271. Fackenthal JD, Cartegni L, Krainer '\n",
      "               'AR, Olopade OI (2002) BRCA2<<< T2722R>>> is a deleterious '\n",
      "               'allele that causes exon skipping.']},\n",
      " {'pmid': '17899372',\n",
      "  'points': 10,\n",
      "  'snippets': ['Fackenthal JD, Cartegni L, Krainer AR et al (2002) BRCA2<<< '\n",
      "               'T2722R>>> is a deleterious allele that causes exon skipping.']},\n",
      " {'pmid': '17924331',\n",
      "  'points': 10,\n",
      "  'snippets': ['9R S1715R IVS12⫹2del21insA M18T A1623G (4987CrG) IVS11-1GrA '\n",
      "               'IVS18-1GrC BRCA2: R2659K (8204GrA) G2748D I2627F E2663V P3039P '\n",
      "               '(9345GrA) D2723G IVS15-1GrA<<< T2722R>>> IVS24-1GrC R2336H '\n",
      "               '(7235GrA) W2626C IVS21⫹4ArG L2653P D3095E IVS5-2ArG R2502C '\n",
      "               'IVS19⫹1GrA  FamilyHistory LLR  Co-Occurrence  Cosegregation  '\n",
      "               'Combined LL']},\n",
      " {'pmid': '12915465',\n",
      "  'points': 2,\n",
      "  'snippets': ['(  2002  ) BRCA2<<< T2722R>>> is a deleterious allele that '\n",
      "               'causes exon skipping.  Am.',\n",
      "               '(2002) BRCA2<<< T2722R>>> is a deleterious allele that causes '\n",
      "               'exon skipping.']}]\n"
     ]
    }
   ],
   "source": [
    "pprint.pprint(lit[\"variants\"][\"chr13:g.32363367:C>G\"])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
