{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BRCA Exchange Literature Search Export\n",
    "\n",
    "Ingest artifacts from the pipeline, wrangle, and generate literature.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import datetime\n",
    "import json\n",
    "import sqlite3\n",
    "import pprint\n",
    "import pandas as pd\n",
    "\n",
    "os.chdir(os.path.expanduser(\"/home/jovyan/data/pubmunch/crawl/\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ingest\n",
    "Ingest artifacts from various pieces of the pipeline indexed by normalized genomic hgvs and integer pmid "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "160 articles loaded from the articles sqlite database\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>articleId</th>\n",
       "      <th>externalId</th>\n",
       "      <th>source</th>\n",
       "      <th>publisher</th>\n",
       "      <th>origFile</th>\n",
       "      <th>journal</th>\n",
       "      <th>printIssn</th>\n",
       "      <th>eIssn</th>\n",
       "      <th>journalUniqueId</th>\n",
       "      <th>year</th>\n",
       "      <th>...</th>\n",
       "      <th>issue</th>\n",
       "      <th>page</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pmcId</th>\n",
       "      <th>doi</th>\n",
       "      <th>fulltextUrl</th>\n",
       "      <th>time</th>\n",
       "      <th>offset</th>\n",
       "      <th>size</th>\n",
       "      <th>chunkId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5018824701</td>\n",
       "      <td>PMID18824701</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Journal of clinical oncology : official journa...</td>\n",
       "      <td>0732-183X</td>\n",
       "      <td>1527-7755</td>\n",
       "      <td>8309333</td>\n",
       "      <td>2008</td>\n",
       "      <td>...</td>\n",
       "      <td>33</td>\n",
       "      <td>5393</td>\n",
       "      <td>18824701</td>\n",
       "      <td>2651073</td>\n",
       "      <td>10.1200/JCO.2008.17.8228</td>\n",
       "      <td></td>\n",
       "      <td>2019-03-19T20:57:21+0000</td>\n",
       "      <td>78</td>\n",
       "      <td>127342</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5012473589</td>\n",
       "      <td>PMID12473589</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Clinical cancer research : an official journal...</td>\n",
       "      <td>1078-0432</td>\n",
       "      <td>1078-0432</td>\n",
       "      <td>9502500</td>\n",
       "      <td>2002</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>3776</td>\n",
       "      <td>12473589</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>2019-03-19T20:57:51+0000</td>\n",
       "      <td>1155332769</td>\n",
       "      <td>73030</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5021673748</td>\n",
       "      <td>PMID21673748</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>European journal of human genetics : EJHG</td>\n",
       "      <td>1018-4813</td>\n",
       "      <td>1476-5438</td>\n",
       "      <td>9302235</td>\n",
       "      <td>2011</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>1052</td>\n",
       "      <td>21673748</td>\n",
       "      <td>3190263</td>\n",
       "      <td>10.1038/ejhg.2011.100</td>\n",
       "      <td></td>\n",
       "      <td>2019-03-19T20:58:12+0000</td>\n",
       "      <td>4038201791</td>\n",
       "      <td>88784</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5017493881</td>\n",
       "      <td>PMID17493881</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Biochimica et biophysica acta</td>\n",
       "      <td>0006-3002</td>\n",
       "      <td>0006-3002</td>\n",
       "      <td>0217513</td>\n",
       "      <td>2007</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>772</td>\n",
       "      <td>17493881</td>\n",
       "      <td></td>\n",
       "      <td>10.1016/j.bbapap.2007.03.018</td>\n",
       "      <td></td>\n",
       "      <td>2019-03-19T20:58:19+0000</td>\n",
       "      <td>5509507500</td>\n",
       "      <td>4372</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5009971877</td>\n",
       "      <td>PMID9971877</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Human molecular genetics</td>\n",
       "      <td>0964-6906</td>\n",
       "      <td>0964-6906</td>\n",
       "      <td>9208958</td>\n",
       "      <td>1999</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>413</td>\n",
       "      <td>9971877</td>\n",
       "      <td></td>\n",
       "      <td>10.1002/(sici)1097-0215(19980729)77:3&lt;354::aid...</td>\n",
       "      <td></td>\n",
       "      <td>2019-03-19T20:58:25+0000</td>\n",
       "      <td>5509510743</td>\n",
       "      <td>60799</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    articleId    externalId source publisher origFile  \\\n",
       "0  5018824701  PMID18824701         download            \n",
       "1  5012473589  PMID12473589         download            \n",
       "2  5021673748  PMID21673748         download            \n",
       "3  5017493881  PMID17493881         download            \n",
       "4  5009971877   PMID9971877         download            \n",
       "\n",
       "                                             journal  printIssn      eIssn  \\\n",
       "0  Journal of clinical oncology : official journa...  0732-183X  1527-7755   \n",
       "1  Clinical cancer research : an official journal...  1078-0432  1078-0432   \n",
       "2          European journal of human genetics : EJHG  1018-4813  1476-5438   \n",
       "3                      Biochimica et biophysica acta  0006-3002  0006-3002   \n",
       "4                           Human molecular genetics  0964-6906  0964-6906   \n",
       "\n",
       "  journalUniqueId  year   ...    issue  page      pmid    pmcId  \\\n",
       "0         8309333  2008   ...       33  5393  18824701  2651073   \n",
       "1         9502500  2002   ...       12  3776  12473589            \n",
       "2         9302235  2011   ...       10  1052  21673748  3190263   \n",
       "3         0217513  2007   ...        6   772  17493881            \n",
       "4         9208958  1999   ...        3   413   9971877            \n",
       "\n",
       "                                                 doi fulltextUrl  \\\n",
       "0                           10.1200/JCO.2008.17.8228               \n",
       "1                                                                  \n",
       "2                              10.1038/ejhg.2011.100               \n",
       "3                       10.1016/j.bbapap.2007.03.018               \n",
       "4  10.1002/(sici)1097-0215(19980729)77:3<354::aid...               \n",
       "\n",
       "                       time      offset    size  chunkId  \n",
       "0  2019-03-19T20:57:21+0000          78  127342  0_00000  \n",
       "1  2019-03-19T20:57:51+0000  1155332769   73030  0_00000  \n",
       "2  2019-03-19T20:58:12+0000  4038201791   88784  0_00000  \n",
       "3  2019-03-19T20:58:19+0000  5509507500    4372  0_00000  \n",
       "4  2019-03-19T20:58:25+0000  5509510743   60799  0_00000  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "connection = sqlite3.connect(\"file:text/articles.db?mode=ro\", uri=True)\n",
    "articles = pd.read_sql_query(\"SELECT * FROM articles\", connection)\n",
    "articles.pmid = articles.pmid.astype(str)\n",
    "print(\"{} articles loaded from the articles sqlite database\".format(articles.shape[0]))\n",
    "articles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total matched mentions: 10426\n",
      "After dropping duplicates: 5671\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pyhgvs_Genomic_Coordinate_38</th>\n",
       "      <th>pmid</th>\n",
       "      <th>snippets</th>\n",
       "      <th>points</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17:g.43092632:T&gt;A</td>\n",
       "      <td>10373534</td>\n",
       "      <td>1051-1863, using a Clontech transformer site-d...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17:g.43092632:T&gt;A</td>\n",
       "      <td>10373534</td>\n",
       "      <td>1051-1863, using a Clontech transformer site-...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17:g.43076488:C&gt;A</td>\n",
       "      <td>10373534</td>\n",
       "      <td>A mutation in close proximity has been compile...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr17:g.43074517:A&gt;C</td>\n",
       "      <td>10373534</td>\n",
       "      <td>Mutants&lt;&lt;&lt; S1497A&gt;&gt;&gt; and S1497T were generated...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr17:g.43074517:A&gt;C</td>\n",
       "      <td>10373534</td>\n",
       "      <td>Mutants&lt;&lt;&lt; S1497A&gt;&gt;&gt; and S1497T were generated...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  pyhgvs_Genomic_Coordinate_38      pmid  \\\n",
       "0         chr17:g.43092632:T>A  10373534   \n",
       "1         chr17:g.43092632:T>A  10373534   \n",
       "2         chr17:g.43076488:C>A  10373534   \n",
       "6         chr17:g.43074517:A>C  10373534   \n",
       "7         chr17:g.43074517:A>C  10373534   \n",
       "\n",
       "                                            snippets  points  \n",
       "0  1051-1863, using a Clontech transformer site-d...      10  \n",
       "1   1051-1863, using a Clontech transformer site-...      10  \n",
       "2  A mutation in close proximity has been compile...      10  \n",
       "6  Mutants<<< S1497A>>> and S1497T were generated...      10  \n",
       "7  Mutants<<< S1497A>>> and S1497T were generated...      10  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mentions = pd.read_csv(\"mentions-matched.tsv\", sep=\"\\t\", encoding=\"utf-8\")\n",
    "mentions.pmid = mentions.pmid.astype(str)\n",
    "print(\"Total matched mentions: {}\".format(mentions.shape[0]))\n",
    "mentions = mentions.drop_duplicates([\"pyhgvs_Genomic_Coordinate_38\", \"pmid\", \"snippets\"])\n",
    "print(\"After dropping duplicates: {}\".format(mentions.shape[0]))\n",
    "mentions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([['22962691', 40],\n",
       "       ['18424508', 30],\n",
       "       ['20215541', 30],\n",
       "       ['18451181', 20],\n",
       "       ['18607349', 20],\n",
       "       ['20507642', 20],\n",
       "       ['20522429', 20],\n",
       "       ['12145750', 10],\n",
       "       ['16792514', 10],\n",
       "       ['17899372', 10],\n",
       "       ['17924331', 10],\n",
       "       ['12915465', 2]], dtype=object)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def top_papers(mentions, pyhgvs):\n",
    "    \"\"\" Return (pmid, points) tuple sorted by points against pyhgvs \"\"\"\n",
    "    top = mentions[mentions.pyhgvs_Genomic_Coordinate_38 == pyhgvs] \\\n",
    "        .groupby([\"pyhgvs_Genomic_Coordinate_38\", \"pmid\"]) \\\n",
    "        .agg({\"points\": sum}) \\\n",
    "        .sort_values(\"points\", ascending=False)\n",
    "    return top.reset_index()[[\"pmid\", \"points\"]].values\n",
    "\n",
    "top_papers(mentions, \"chr13:g.32363367:C>G\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['BRCA2<<< T2722R>>> is a deleterious allele that causes exon skipping.',\n",
       " 'Of these, the exonic variant BRCA2 c.8162Tâ†’C in exon 18 affects a position only three nucleotides upstream of the mutation BRCA2 c.8165Câ†’G (predicting<<< p.T2722R>>>), which is known to cause exon skipping by disrupting several ESE sites.20  Two variants, BRCA2 c.316+5Gâ†’C (IVS3+5Gâ†’C) and c.7805Gâ†’C, at the last base of exon 16, induced strong effects on splicing (figs 2A,B and 2C,D, respectively).',\n",
       " 'Of these, the exonic variant BRCA2 c.8162TRC in exon 18 affects a position only three nucleotides upstream of the mutation BRCA2<<< c.8165CRG>>> (predicting p.T2722R), which is known to cause exon skipping by disrupting several ESE sites.20 Two variants, BRCA2 c.316+5GRC (IVS3+5GRC) and c.7805']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def top_snippets(mentions, pyhgvs, pmid):\n",
    "    \"\"\" Return list of top snippets for this variant and paper by points \"\"\"\n",
    "    paper = mentions[(mentions.pyhgvs_Genomic_Coordinate_38 == pyhgvs) & (mentions.pmid == pmid)]\n",
    "    return [snippet \n",
    "     for snippets in paper.sort_values(\"points\", ascending=False).snippets.values \n",
    "     for snippet in snippets.split(\"|\")][:3]\n",
    "\n",
    "top_snippets(mentions, \"chr13:g.32363367:C>G\", '18424508')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "def top_papers_and_snippets(mentions, pyhgvs):\n",
    "    return [{\"pmid\": str(p[0]), \"points\": int(p[1]), \"mentions\": top_snippets(mentions, pyhgvs, p[0])}\n",
    "            for p in top_papers(mentions, pyhgvs)]\n",
    "\n",
    "# for pyhgvs in [\"chr13:g.32363367:C>G\", \"chr17:g.43124027:ACT>A\", \"chr13:g.32340526:AT>A\"]:\n",
    "#     print(pyhgvs)\n",
    "#     pprint.pprint(top_papers_and_snippets(mentions, pyhgvs)[:3])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019-03-19 23:06:08.507579\n",
      "Remaining: 1000\r"
     ]
    }
   ],
   "source": [
    "variants = {}\n",
    "\n",
    "remaining = mentions.pyhgvs_Genomic_Coordinate_38.unique().shape[0]\n",
    "print(datetime.datetime.now())\n",
    "for pyhgvs in mentions.pyhgvs_Genomic_Coordinate_38.unique():\n",
    "    sys.stdout.write(\"Remaining: {}\\r\".format(remaining))\n",
    "    sys.stdout.flush()\n",
    "    variants[pyhgvs] = top_papers_and_snippets(mentions, pyhgvs)\n",
    "    remaining -= 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exported 1966 variants in 150 papers\n",
      "150 Papers and 1966 Variants exported\n"
     ]
    }
   ],
   "source": [
    "lit = {\n",
    "    \"date\": open(\"pubs-date.txt\").read().strip(),\n",
    "    \"papers\": articles[articles.pmid.isin(mentions.pmid)].set_index(\"pmid\", drop=False).to_dict(orient=\"index\"),\n",
    "    \"variants\": variants\n",
    "}\n",
    "\n",
    "with open(\"literature.json\", \"w\") as output:\n",
    "    output.write(json.dumps(lit, sort_keys=True))\n",
    "    \n",
    "print(\"Exported {} variants in {} papers\".format(\n",
    "    len(lit[\"variants\"].keys()), len(lit[\"papers\"].keys())))\n",
    "\n",
    "with open(\"literature.json\") as f:\n",
    "    lit = json.loads(f.read())\n",
    "print(\"{} Papers and {} Variants exported\".format(len(lit[\"papers\"]), len(lit[\"variants\"])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'mentions': [', in the BRCA2 gene, besides the exon 7 mutations described '\n",
      "               'above, only one mutation in exon 3, c.231T>G (p.Thr77Thr),16 '\n",
      "               'and two mutations in exon 18,<<< c.8165C>G>>> (p.Thr2722Arg)28 '\n",
      "               'and c.7992T>A (p.Ile2664Ile),16 have been reported to induce '\n",
      "               'exon skipping by altering splicing regulatory elements.',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'CA2 gene, besides the exon 7 mutations described above, only '\n",
      "               'one mutation in exon 3, c.231T>G (p.Thr77Thr),16 and two '\n",
      "               'mutations in exon 18, c.8165C>G <<<(p.Thr2722Arg>>>)28 and '\n",
      "               'c.7992T>A (p.Ile2664Ile),16 have been reported to induce exon '\n",
      "               'skipping by altering splicing regulatory elements.'],\n",
      "  'pmid': '22962691',\n",
      "  'points': 40},\n",
      " {'mentions': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'Of these, the exonic variant BRCA2 c.8162Tâ†’C in exon 18 '\n",
      "               'affects a position only three nucleotides upstream of the '\n",
      "               'mutation BRCA2 c.8165Câ†’G (predicting<<< p.T2722R>>>), which is '\n",
      "               'known to cause exon skipping by disrupting several ESE '\n",
      "               'sites.20  Two variants, BRCA2 c.316+5Gâ†’C (IVS3+5Gâ†’C) and '\n",
      "               'c.7805Gâ†’C, at the last base of exon 16, induced strong effects '\n",
      "               'on splicing (figs 2A,B and 2C,D, respectively).',\n",
      "               'Of these, the exonic variant BRCA2 c.8162TRC in exon 18 '\n",
      "               'affects a position only three nucleotides upstream of the '\n",
      "               'mutation BRCA2<<< c.8165CRG>>> (predicting p.T2722R), which is '\n",
      "               'known to cause exon skipping by disrupting several ESE '\n",
      "               'sites.20 Two variants, BRCA2 c.316+5GRC (IVS3+5GRC) and '\n",
      "               'c.7805'],\n",
      "  'pmid': '18424508',\n",
      "  'points': 30},\n",
      " {'mentions': ['None of the five missense mutations directed to conserved ESE '\n",
      "               'significantly altered splicing, although variant<<< '\n",
      "               'c.8165C>G>>> (34) showed a weak exon 18 skipping in minigene '\n",
      "               'context, also detected by other authors (33, 35).  Six '\n",
      "               'positive splicing mutations showed simultaneou',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'None of the five missense mutations directed to conserved ESE '\n",
      "               'significantly altered splicing, although variant<<< '\n",
      "               'c.8165C>G>>> (34) showed a weak exon 18 skipping in minigene '\n",
      "               'context, also detected by other authors (33, 35). Six positive '\n",
      "               'splicing mutations showed simultaneous'],\n",
      "  'pmid': '20215541',\n",
      "  'points': 30},\n",
      " {'mentions': ['Two other variants (E2663V, D2723G) reached odds in favor of '\n",
      "               'disease causality of at least 100:1 and<<< T2722R>>> reached '\n",
      "               'odds of 93:1 ( Table 1).',\n",
      "               'In contrast, I2627F, L2647P, L2653P,<<< T2722R>>>, D2723H, '\n",
      "               'D2723G, D2723A, G2748D, R2784W, R3052W, and D3095E displayed '\n",
      "               'reduced HDR activity ( Fig. 3B).',\n",
      "               'The limits of these groups were defined by R2520Q (4.0-fold) '\n",
      "               'and<<< T2722R>>> (2.5-fold).'],\n",
      "  'pmid': '18451181',\n",
      "  'points': 20},\n",
      " {'mentions': ['Here, we tested a BRCA2 missense variant<<< T2722R>>> that was '\n",
      "               'recently reported to cause aberrant splicing and results in '\n",
      "               'deletion of exon 18 in the transcript22.',\n",
      "               'Nevertheless, the full-length<<< T2722R>>> BRCA2 did not '\n",
      "               'rescue ES cell viability, corroborating our conclusion that '\n",
      "               'the mutation is deleterious.  Figure 3: Analysis of BRCA2 '\n",
      "               'mutants affecting',\n",
      "               'ing our conclusion that the mutation is deleterious.  Figure '\n",
      "               '3: Analysis of BRCA2 mutants affecting splicing or domain '\n",
      "               'structure.  Figure 3  (a) BRCA2<<< T2722R>>> transcript '\n",
      "               'reveals skipping of exon 18 (Î”18) not detected for WT BRCA2.'],\n",
      "  'pmid': '18607349',\n",
      "  'points': 20},\n",
      " {'mentions': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'Fackenthal JD, Cartegni L, Krainer AR, Olopade OI: BRCA2<<< '\n",
      "               'T2722R>>> is a deleterious allele that causes exon skipping.'],\n",
      "  'pmid': '20507642',\n",
      "  'points': 20},\n",
      " {'mentions': ['BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.',\n",
      "               'In the BRCA2 gene, exon skipping has been associated with the '\n",
      "               'missense mutations c.8165Câ†’G <<<(p.Thr2722Arg>>>) in exon 1824 '\n",
      "               'and c.6853Aâ†’G (p.Ile2285Val) in exon 12,25 but the molecular '\n",
      "               'bases of these splicing defects have not been described.  '\n",
      "               'Retrospectively,',\n",
      "               'BRCA2<<< T2722R>>> is a deleterious allele that causes exon '\n",
      "               'skipping.'],\n",
      "  'pmid': '20522429',\n",
      "  'points': 20},\n",
      " {'mentions': ['71:625â€“631, 2002  Report  BRCA2<<< T2722R>>> Is a Deleterious '\n",
      "               'Allele That Causes Exon Skipping James D.',\n",
      "               'These predictions revealed that BRCA2<<< T2722R>>> (8393CrG), '\n",
      "               'which segregates with affected individuals in a family with '\n",
      "               'breast cancer, disrupts three potential ESE sites.',\n",
      "               'Of these, only one allele, BRCA2<<< T2722R>>>, segregated with '\n",
      "               'affected members of a family with breast cancer (family number '\n",
      "               '98-11).'],\n",
      "  'pmid': '12145750',\n",
      "  'points': 10},\n",
      " {'mentions': ['Am J Hum Genet 56:265â€“271. Fackenthal JD, Cartegni L, Krainer '\n",
      "               'AR, Olopade OI (2002) BRCA2<<< T2722R>>> is a deleterious '\n",
      "               'allele that causes exon skipping.'],\n",
      "  'pmid': '16792514',\n",
      "  'points': 10},\n",
      " {'mentions': ['Fackenthal JD, Cartegni L, Krainer AR et al (2002) BRCA2<<< '\n",
      "               'T2722R>>> is a deleterious allele that causes exon skipping.'],\n",
      "  'pmid': '17899372',\n",
      "  'points': 10},\n",
      " {'mentions': ['9R S1715R IVS12â«¹2del21insA M18T A1623G (4987CrG) IVS11-1GrA '\n",
      "               'IVS18-1GrC BRCA2: R2659K (8204GrA) G2748D I2627F E2663V P3039P '\n",
      "               '(9345GrA) D2723G IVS15-1GrA<<< T2722R>>> IVS24-1GrC R2336H '\n",
      "               '(7235GrA) W2626C IVS21â«¹4ArG L2653P D3095E IVS5-2ArG R2502C '\n",
      "               'IVS19â«¹1GrA  FamilyHistory LLR  Co-Occurrence  Cosegregation  '\n",
      "               'Combined LL'],\n",
      "  'pmid': '17924331',\n",
      "  'points': 10},\n",
      " {'mentions': ['(  2002  ) BRCA2<<< T2722R>>> is a deleterious allele that '\n",
      "               'causes exon skipping.  Am.',\n",
      "               '(2002) BRCA2<<< T2722R>>> is a deleterious allele that causes '\n",
      "               'exon skipping.'],\n",
      "  'pmid': '12915465',\n",
      "  'points': 2}]\n"
     ]
    }
   ],
   "source": [
    "pprint.pprint(lit[\"variants\"][\"chr13:g.32363367:C>G\"])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
