{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Export\n",
    "Explorations using the intermediate files in correlate.py for various export formats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import json\n",
    "import sqlite3\n",
    "import pandas as pd\n",
    "\n",
    "os.chdir(os.path.expanduser(\"~/data/pubmunch/crawl/\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 569,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14044 articles downloaded with BRCA in the title or abstract\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>articleId</th>\n",
       "      <th>externalId</th>\n",
       "      <th>source</th>\n",
       "      <th>publisher</th>\n",
       "      <th>origFile</th>\n",
       "      <th>journal</th>\n",
       "      <th>printIssn</th>\n",
       "      <th>eIssn</th>\n",
       "      <th>journalUniqueId</th>\n",
       "      <th>year</th>\n",
       "      <th>...</th>\n",
       "      <th>issue</th>\n",
       "      <th>page</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pmcId</th>\n",
       "      <th>doi</th>\n",
       "      <th>fulltextUrl</th>\n",
       "      <th>time</th>\n",
       "      <th>offset</th>\n",
       "      <th>size</th>\n",
       "      <th>chunkId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5011585672</td>\n",
       "      <td>PMID11585672</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Trends in genetics : TIG</td>\n",
       "      <td>0168-9525</td>\n",
       "      <td>0168-9525</td>\n",
       "      <td>8507085</td>\n",
       "      <td>2001</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>S18</td>\n",
       "      <td>11585672</td>\n",
       "      <td></td>\n",
       "      <td>10.1016/s0168-9525(01)02451-9</td>\n",
       "      <td>https://linkinghub.elsevier.com/retrieve/pii/S...</td>\n",
       "      <td>2018-11-14T16:29:09+0000</td>\n",
       "      <td>78</td>\n",
       "      <td>5258</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5019688261</td>\n",
       "      <td>PMID19688261</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Breast cancer research and treatment</td>\n",
       "      <td>0167-6806</td>\n",
       "      <td>1573-7217</td>\n",
       "      <td>8111104</td>\n",
       "      <td>2010</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>575</td>\n",
       "      <td>19688261</td>\n",
       "      <td></td>\n",
       "      <td>10.1007/s10549-009-0501-3</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:29:16+0000</td>\n",
       "      <td>5048</td>\n",
       "      <td>100825</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5012228710</td>\n",
       "      <td>PMID12228710</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Science (New York, N.Y.)</td>\n",
       "      <td>0036-8075</td>\n",
       "      <td>1095-9203</td>\n",
       "      <td>0404511</td>\n",
       "      <td>2002</td>\n",
       "      <td>...</td>\n",
       "      <td>5588</td>\n",
       "      <td>1837</td>\n",
       "      <td>12228710</td>\n",
       "      <td></td>\n",
       "      <td>10.1126/science.297.5588.1837</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:29:40+0000</td>\n",
       "      <td>1436851129</td>\n",
       "      <td>148950</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5029369605</td>\n",
       "      <td>PMID29369605</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Genetika</td>\n",
       "      <td>0016-6758</td>\n",
       "      <td>0016-6758</td>\n",
       "      <td>0047354</td>\n",
       "      <td>2016</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>1215</td>\n",
       "      <td>29369605</td>\n",
       "      <td></td>\n",
       "      <td>10.1134/s102279541609012x</td>\n",
       "      <td>https://link.springer.com/article/10.1134%2FS1...</td>\n",
       "      <td>2018-11-14T16:30:43+0000</td>\n",
       "      <td>4458079378</td>\n",
       "      <td>22264</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5022084640</td>\n",
       "      <td>PMID22084640</td>\n",
       "      <td></td>\n",
       "      <td>download</td>\n",
       "      <td></td>\n",
       "      <td>Therapeutic advances in medical oncology</td>\n",
       "      <td>1758-8340</td>\n",
       "      <td>1758-8359</td>\n",
       "      <td>101510808</td>\n",
       "      <td>2011</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>257</td>\n",
       "      <td>22084640</td>\n",
       "      <td>3210467</td>\n",
       "      <td>10.1177/1758834011417039</td>\n",
       "      <td></td>\n",
       "      <td>2018-11-14T16:30:58+0000</td>\n",
       "      <td>6079002966</td>\n",
       "      <td>93229</td>\n",
       "      <td>0_00000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    articleId    externalId source publisher origFile  \\\n",
       "0  5011585672  PMID11585672         download            \n",
       "1  5019688261  PMID19688261         download            \n",
       "2  5012228710  PMID12228710         download            \n",
       "3  5029369605  PMID29369605         download            \n",
       "4  5022084640  PMID22084640         download            \n",
       "\n",
       "                                    journal  printIssn      eIssn  \\\n",
       "0                  Trends in genetics : TIG  0168-9525  0168-9525   \n",
       "1      Breast cancer research and treatment  0167-6806  1573-7217   \n",
       "2                  Science (New York, N.Y.)  0036-8075  1095-9203   \n",
       "3                                  Genetika  0016-6758  0016-6758   \n",
       "4  Therapeutic advances in medical oncology  1758-8340  1758-8359   \n",
       "\n",
       "  journalUniqueId  year   ...    issue  page      pmid    pmcId  \\\n",
       "0         8507085  2001   ...       10   S18  11585672            \n",
       "1         8111104  2010   ...        3   575  19688261            \n",
       "2         0404511  2002   ...     5588  1837  12228710            \n",
       "3         0047354  2016   ...       10  1215  29369605            \n",
       "4       101510808  2011   ...        6   257  22084640  3210467   \n",
       "\n",
       "                             doi  \\\n",
       "0  10.1016/s0168-9525(01)02451-9   \n",
       "1      10.1007/s10549-009-0501-3   \n",
       "2  10.1126/science.297.5588.1837   \n",
       "3      10.1134/s102279541609012x   \n",
       "4       10.1177/1758834011417039   \n",
       "\n",
       "                                         fulltextUrl  \\\n",
       "0  https://linkinghub.elsevier.com/retrieve/pii/S...   \n",
       "1                                                      \n",
       "2                                                      \n",
       "3  https://link.springer.com/article/10.1134%2FS1...   \n",
       "4                                                      \n",
       "\n",
       "                       time      offset    size  chunkId  \n",
       "0  2018-11-14T16:29:09+0000          78    5258  0_00000  \n",
       "1  2018-11-14T16:29:16+0000        5048  100825  0_00000  \n",
       "2  2018-11-14T16:29:40+0000  1436851129  148950  0_00000  \n",
       "3  2018-11-14T16:30:43+0000  4458079378   22264  0_00000  \n",
       "4  2018-11-14T16:30:58+0000  6079002966   93229  0_00000  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 569,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "connection = sqlite3.connect(\"file:text/articles.db?mode=ro\", uri=True)\n",
    "articles = pd.read_sql_query(\"SELECT * FROM articles\", connection)\n",
    "print(\"{} articles downloaded with BRCA in the title or abstract\".format(articles.shape[0]))\n",
    "articles.pmid = articles.pmid.astype(str)\n",
    "articles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 570,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20934 variants in BRCA Exchange database\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>pyhgvs_Genomic_Coordinate_38</th>\n",
       "      <th>pyhgvs_cDNA</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>norm_g_hgvs</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32314943A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32314943</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32314943:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.-764A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32315226G&gt;A</th>\n",
       "      <td>13</td>\n",
       "      <td>32315226</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr13:g.32315226:G&gt;A</td>\n",
       "      <td>NM_000059.3:c.-481G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32315490C&gt;T</th>\n",
       "      <td>13</td>\n",
       "      <td>32315490</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr13:g.32315490:C&gt;T</td>\n",
       "      <td>NM_000059.3:c.-217C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32315519C&gt;T</th>\n",
       "      <td>13</td>\n",
       "      <td>32315519</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr13:g.32315519:C&gt;T</td>\n",
       "      <td>NM_000059.3:c.-188C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32315532C&gt;T</th>\n",
       "      <td>13</td>\n",
       "      <td>32315532</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr13:g.32315532:C&gt;T</td>\n",
       "      <td>NM_000059.3:c.-175C&gt;T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Chr       Pos Ref Alt  \\\n",
       "norm_g_hgvs                                         \n",
       "NC_000013.11:g.32314943A>G   13  32314943   A   G   \n",
       "NC_000013.11:g.32315226G>A   13  32315226   G   A   \n",
       "NC_000013.11:g.32315490C>T   13  32315490   C   T   \n",
       "NC_000013.11:g.32315519C>T   13  32315519   C   T   \n",
       "NC_000013.11:g.32315532C>T   13  32315532   C   T   \n",
       "\n",
       "                           pyhgvs_Genomic_Coordinate_38            pyhgvs_cDNA  \n",
       "norm_g_hgvs                                                                     \n",
       "NC_000013.11:g.32314943A>G         chr13:g.32314943:A>G  NM_000059.3:c.-764A>G  \n",
       "NC_000013.11:g.32315226G>A         chr13:g.32315226:G>A  NM_000059.3:c.-481G>A  \n",
       "NC_000013.11:g.32315490C>T         chr13:g.32315490:C>T  NM_000059.3:c.-217C>T  \n",
       "NC_000013.11:g.32315519C>T         chr13:g.32315519:C>T  NM_000059.3:c.-188C>T  \n",
       "NC_000013.11:g.32315532C>T         chr13:g.32315532:C>T  NM_000059.3:c.-175C>T  "
      ]
     },
     "execution_count": 570,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants = pd.read_table(\"variants-normalized.tsv\", index_col=\"norm_g_hgvs\")\n",
    "print(\"{} variants in BRCA Exchange database\".format(variants.shape[0]))\n",
    "variants.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 571,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "442198 total mentions found\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>norm_c_hgvs</th>\n",
       "      <th>pmid</th>\n",
       "      <th>snippet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>norm_g_hgvs</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NC_000007.14:g.96709760A&gt;C</th>\n",
       "      <td>NM_006304.1:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000004.12:g.56655997A&gt;C</th>\n",
       "      <td>NM_001145459.1:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000004.12:g.56655997A&gt;C</th>\n",
       "      <td>NM_139212.3:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000004.12:g.56655997A&gt;C</th>\n",
       "      <td>NM_139211.4:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded β sheet with OB2β1 and OB2β2, as w...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000007.14:g.96709760A&gt;C</th>\n",
       "      <td>NM_006304.1:c.4T&gt;G</td>\n",
       "      <td>12228710</td>\n",
       "      <td>ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      norm_c_hgvs      pmid  \\\n",
       "norm_g_hgvs                                                   \n",
       "NC_000007.14:g.96709760A>C     NM_006304.1:c.4T>G  12228710   \n",
       "NC_000004.12:g.56655997A>C  NM_001145459.1:c.4T>G  12228710   \n",
       "NC_000004.12:g.56655997A>C     NM_139212.3:c.4T>G  12228710   \n",
       "NC_000004.12:g.56655997A>C     NM_139211.4:c.4T>G  12228710   \n",
       "NC_000007.14:g.96709760A>C     NM_006304.1:c.4T>G  12228710   \n",
       "\n",
       "                                                                      snippet  \n",
       "norm_g_hgvs                                                                    \n",
       "NC_000007.14:g.96709760A>C  ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "NC_000004.12:g.56655997A>C  ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "NC_000004.12:g.56655997A>C  ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "NC_000004.12:g.56655997A>C  ee-stranded β sheet with OB2β1 and OB2β2, as w...  \n",
       "NC_000007.14:g.96709760A>C  ee-stranded ␤ sheet with OB2␤1 and OB2␤2, as w...  "
      ]
     },
     "execution_count": 571,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# mentions = pd.read_table(\"mentions-normalized.tsv\", index_col=\"norm_g_hgvs\").drop_duplicates([\"pmid\", \"snippet\"])\n",
    "mentions = pd.read_table(\"mentions-normalized.tsv\", index_col=\"norm_g_hgvs\")\n",
    "mentions.pmid = mentions.pmid.astype(str)\n",
    "print(\"{} total mentions found\".format(mentions.shape[0]))\n",
    "mentions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 572,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "55657 mentions of variants in BRCA Exchange\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>pyhgvs_Genomic_Coordinate_38</th>\n",
       "      <th>pyhgvs_cDNA</th>\n",
       "      <th>norm_c_hgvs</th>\n",
       "      <th>pmid</th>\n",
       "      <th>snippet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>norm_g_hgvs</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32316461A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32316461:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>tions: in EPCAM, c.412C&gt;T p.(Arg138*), and coe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32316461A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32316461:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>EPCAM, c.412C&gt;T p.(Arg138*), and coexisting mu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32316461A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32316461:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>tions: in EPCAM, c.412C&gt;T p.(Arg138*), and coe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32316461A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32316461:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>EPCAM, c.412C&gt;T p.(Arg138*), and coexisting mu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NC_000013.11:g.32316461A&gt;G</th>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr13:g.32316461:A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>22006311</td>\n",
       "      <td>nd in-frame deletions were classified as delet...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Chr       Pos Ref Alt  \\\n",
       "norm_g_hgvs                                         \n",
       "NC_000013.11:g.32316461A>G   13  32316461   A   G   \n",
       "NC_000013.11:g.32316461A>G   13  32316461   A   G   \n",
       "NC_000013.11:g.32316461A>G   13  32316461   A   G   \n",
       "NC_000013.11:g.32316461A>G   13  32316461   A   G   \n",
       "NC_000013.11:g.32316461A>G   13  32316461   A   G   \n",
       "\n",
       "                           pyhgvs_Genomic_Coordinate_38         pyhgvs_cDNA  \\\n",
       "norm_g_hgvs                                                                   \n",
       "NC_000013.11:g.32316461A>G         chr13:g.32316461:A>G  NM_000059.3:c.1A>G   \n",
       "NC_000013.11:g.32316461A>G         chr13:g.32316461:A>G  NM_000059.3:c.1A>G   \n",
       "NC_000013.11:g.32316461A>G         chr13:g.32316461:A>G  NM_000059.3:c.1A>G   \n",
       "NC_000013.11:g.32316461A>G         chr13:g.32316461:A>G  NM_000059.3:c.1A>G   \n",
       "NC_000013.11:g.32316461A>G         chr13:g.32316461:A>G  NM_000059.3:c.1A>G   \n",
       "\n",
       "                                   norm_c_hgvs      pmid  \\\n",
       "norm_g_hgvs                                                \n",
       "NC_000013.11:g.32316461A>G  NM_000059.3:c.1A>G  30103829   \n",
       "NC_000013.11:g.32316461A>G  NM_000059.3:c.1A>G  30103829   \n",
       "NC_000013.11:g.32316461A>G  NM_000059.3:c.1A>G  30103829   \n",
       "NC_000013.11:g.32316461A>G  NM_000059.3:c.1A>G  30103829   \n",
       "NC_000013.11:g.32316461A>G  NM_000059.3:c.1A>G  22006311   \n",
       "\n",
       "                                                                      snippet  \n",
       "norm_g_hgvs                                                                    \n",
       "NC_000013.11:g.32316461A>G  tions: in EPCAM, c.412C>T p.(Arg138*), and coe...  \n",
       "NC_000013.11:g.32316461A>G  EPCAM, c.412C>T p.(Arg138*), and coexisting mu...  \n",
       "NC_000013.11:g.32316461A>G  tions: in EPCAM, c.412C>T p.(Arg138*), and coe...  \n",
       "NC_000013.11:g.32316461A>G  EPCAM, c.412C>T p.(Arg138*), and coexisting mu...  \n",
       "NC_000013.11:g.32316461A>G  nd in-frame deletions were classified as delet...  "
      ]
     },
     "execution_count": 572,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Prune mentions down to only those correlated to a variant in BRCA Excchange\n",
    "variant_mentions = pd.merge(variants, mentions, left_index=True, right_index=True)\n",
    "print(\"{} mentions of variants in BRCA Exchange\".format(variant_mentions.shape[0]))\n",
    "variant_mentions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 573,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unpacking 18909 of 55657 snippets with multiple phrases seaparated by '|'\n"
     ]
    }
   ],
   "source": [
    "# Some of the snippets are multiple mentiones separated by | so unpack these,\n",
    "# but limit to 5 as some have as many as 168!\n",
    "print(\"Unpacking {} of {} snippets with multiple phrases seaparated by '|'\".format(\n",
    "    variant_mentions[variant_mentions.snippet.str.contains(\"\\|\")].shape[0], variant_mentions.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 574,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "48941 after expanding and de-duplicating snippets\n"
     ]
    }
   ],
   "source": [
    "# https://stackoverflow.com/questions/17116814/pandas-how-do-i-split-text-in-a-column-into-multiple-rows/21032532\n",
    "\n",
    "# Resest index so each row id is unique vs. norm_g_hgvs\n",
    "df = variant_mentions.reset_index()\n",
    "# df = variant_mentions[variant_mentions.snippet.str.contains(\"\\|\")].iloc[0:100].reset_index()\n",
    "\n",
    "# Generate a new dataframe splitting each snippet segment into its own row\n",
    "# Limit to max of 3 as some of them have > 100\n",
    "snippets = df.apply(lambda x: pd.Series(x.snippet.split(\"|\")[:3]), axis=1).stack()\n",
    "\n",
    "# To line up with the original index\n",
    "snippets.index = snippets.index.droplevel(-1)\n",
    "\n",
    "# Join back to the original dataframe replaceing the old \"snippet\" columnd\n",
    "snippets.name = \"snippet\"\n",
    "del df[\"snippet\"]\n",
    "exploded = df.join(snippets).drop_duplicates([\"pyhgvs_Genomic_Coordinate_38\", \"snippet\"]).set_index(\"pyhgvs_Genomic_Coordinate_38\")\n",
    "print(\"{} after expanding and de-duplicating snippets\".format(exploded.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 575,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>norm_g_hgvs</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>pyhgvs_cDNA</th>\n",
       "      <th>norm_c_hgvs</th>\n",
       "      <th>pmid</th>\n",
       "      <th>snippet</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pyhgvs_Genomic_Coordinate_38</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr13:g.32316461:A&gt;G</th>\n",
       "      <td>NC_000013.11:g.32316461A&gt;G</td>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>tions: in EPCAM, c.412C&gt;T p.(Arg138*), and coe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:g.32316461:A&gt;G</th>\n",
       "      <td>NC_000013.11:g.32316461A&gt;G</td>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>EPCAM, c.412C&gt;T p.(Arg138*), and coexisting mu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:g.32316461:A&gt;G</th>\n",
       "      <td>NC_000013.11:g.32316461A&gt;G</td>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>30103829</td>\n",
       "      <td>in EPCAM, c.412C&gt;T p.(Arg138*), and coexisting...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:g.32316461:A&gt;G</th>\n",
       "      <td>NC_000013.11:g.32316461A&gt;G</td>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>22006311</td>\n",
       "      <td>nd in-frame deletions were classified as delet...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:g.32316461:A&gt;G</th>\n",
       "      <td>NC_000013.11:g.32316461A&gt;G</td>\n",
       "      <td>13</td>\n",
       "      <td>32316461</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>NM_000059.3:c.1A&gt;G</td>\n",
       "      <td>22006311</td>\n",
       "      <td>BRCA1 p.M1I,&lt;&lt;&lt; p.M1V&gt;&gt;&gt;, and p.C61G and TP53 ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             norm_g_hgvs  Chr       Pos Ref  \\\n",
       "pyhgvs_Genomic_Coordinate_38                                                  \n",
       "chr13:g.32316461:A>G          NC_000013.11:g.32316461A>G   13  32316461   A   \n",
       "chr13:g.32316461:A>G          NC_000013.11:g.32316461A>G   13  32316461   A   \n",
       "chr13:g.32316461:A>G          NC_000013.11:g.32316461A>G   13  32316461   A   \n",
       "chr13:g.32316461:A>G          NC_000013.11:g.32316461A>G   13  32316461   A   \n",
       "chr13:g.32316461:A>G          NC_000013.11:g.32316461A>G   13  32316461   A   \n",
       "\n",
       "                             Alt         pyhgvs_cDNA         norm_c_hgvs  \\\n",
       "pyhgvs_Genomic_Coordinate_38                                               \n",
       "chr13:g.32316461:A>G           G  NM_000059.3:c.1A>G  NM_000059.3:c.1A>G   \n",
       "chr13:g.32316461:A>G           G  NM_000059.3:c.1A>G  NM_000059.3:c.1A>G   \n",
       "chr13:g.32316461:A>G           G  NM_000059.3:c.1A>G  NM_000059.3:c.1A>G   \n",
       "chr13:g.32316461:A>G           G  NM_000059.3:c.1A>G  NM_000059.3:c.1A>G   \n",
       "chr13:g.32316461:A>G           G  NM_000059.3:c.1A>G  NM_000059.3:c.1A>G   \n",
       "\n",
       "                                  pmid  \\\n",
       "pyhgvs_Genomic_Coordinate_38             \n",
       "chr13:g.32316461:A>G          30103829   \n",
       "chr13:g.32316461:A>G          30103829   \n",
       "chr13:g.32316461:A>G          30103829   \n",
       "chr13:g.32316461:A>G          22006311   \n",
       "chr13:g.32316461:A>G          22006311   \n",
       "\n",
       "                                                                        snippet  \n",
       "pyhgvs_Genomic_Coordinate_38                                                     \n",
       "chr13:g.32316461:A>G          tions: in EPCAM, c.412C>T p.(Arg138*), and coe...  \n",
       "chr13:g.32316461:A>G          EPCAM, c.412C>T p.(Arg138*), and coexisting mu...  \n",
       "chr13:g.32316461:A>G          in EPCAM, c.412C>T p.(Arg138*), and coexisting...  \n",
       "chr13:g.32316461:A>G          nd in-frame deletions were classified as delet...  \n",
       "chr13:g.32316461:A>G          BRCA1 p.M1I,<<< p.M1V>>>, and p.C61G and TP53 ...  "
      ]
     },
     "execution_count": 575,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exploded.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Variants by pyhgvs_Genomic_Coordinate_38 as array of mentions\n",
    "# variant_mentions_dict = {\n",
    "#     k: v.to_dict(orient=\"records\")\n",
    "#     for k, v in variant_mentions.reset_index(level=0)\n",
    "#                 .set_index(\"pyhgvs_Genomic_Coordinate_38\")\n",
    "#                 .groupby([\"pyhgvs_Genomic_Coordinate_38\"])}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Variants by pyhgvs_Genomic_Coordinate_38 by pmid with all snippets in a list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 576,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined = exploded.groupby([\"pyhgvs_Genomic_Coordinate_38\", \"pmid\"])[\"snippet\"].apply(lambda s: s.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 577,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pmid\n",
       "17453335    [Two novel BRCA2 germline mutations <<<(G4X>>>...\n",
       "24010542    [_5797delTA  11B  c.4409_4410delTA c.4415_4418...\n",
       "24312913    [G, 461delTC [41]  4817A>G, 8477delAGA [41]  —...\n",
       "27225819    [  95.96%   3.239   447.681   98.46%   BRCA2+ ...\n",
       "28751759    [65Leu) VUS, which was also present in this sa...\n",
       "Name: snippet, dtype: object"
      ]
     },
     "execution_count": 577,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined[\"chr13:g.32316470:G>T\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 581,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'17453335': ['Two novel BRCA2 germline mutations <<<(G4X>>> and 3783del10) are reported here for the first time.',\n",
       "  'Each of the five BRCA2 mutations is observed only once; three are frameshift (3783del10, 4637delTA, 5950delCT), one is nonsense <<<(G4X>>>), and one is a single base substitution that results in aberrant splicing (7235G\\xa0>\\xa0A, skipping of exons 12–13) [15].',\n",
       "  'h.gov/ Intramural_research/Lab_transfer/Bic/Member/ BRCA1_mutation_database.html). The two novel BRCA2 mutations described here for the first time are<<< G4X>>> and 3783del10, located in exons 2 and 11, respectively.',\n",
       "  'Each of the five BRCA2 mutations is observed only once; three are frameshift (3783del10, 4637delTA, 5950delCT), one is nonsense <<<(G4X>>>), and one is a single base substitution that results in aberrant splicing (7235G > A, skipping of exons 12–13) [15].',\n",
       "  'The two novel BRCA2 mutations described here for the first time are<<< G4X>>> and 3783del10, located in exons 2 and 11, respectively.'],\n",
       " '24010542': ['_5797delTA  11B  c.4409_4410delTA c.4415_4418delAGAA c.4767delA  11A  10  c.3555_3564del10 c.3919delG  9  p.Ser780X  7  c.2808_2811delAAAC c.2830delA <<< p.Gly4X>>>  5  21 23 20 22  8  c.1796_1800del5  3  1  6  c.658_659delGT  4  2  Fig. 1.'],\n",
       " '24312913': ['G, 461delTC [41]  4817A>G, 8477delAGA [41]  —   Greek (South eastern Europe)  5382insC and G1738R   [92]   G5331A, 3819delGTAAA [180]   8984delG [35],<<< G4X>>> and 3783del10 [92]  5382insC and G1738R   [92], 3782del10, 4512insT [180]   Slovenia  c.1193C>A [86]   c.181T>G, c.1687C>T, 5382insC,   c.844_850dupT',\n",
       "  '8142del5 [28], exons 5–14 del Exons 1–17 del 5382insC [175]  c.6589delA [178, 179]  c.6589delA [178, 179]  4817A>G, 8477delAGA [41]  —  8984delG [35],<<< G4X>>> and 3783del10 [92]  5382insC and G1738R [92], 3782del10, 4512insT [180]  c.5101C>T, c.5433 5436delGGAA [86] c.7806−2A>G, c.5291C>G, c.3975 3978dupTGC'],\n",
       " '27225819': ['  95.96%   3.239   447.681   98.46%   BRCA2+   c.1447_1450ins4; p.Ala483AlaValfs   Path   #CQ3 TS   377.2   42.18%   227   120.384   57.25%   BRCA2+  <<< c.10G>T>>>; p.Gly4Ter   Path   #CQ5 TS   811.4   66.65%   400   173.040   89.41%   BRCA2+   c.5909C>A; p.Ser1970Ter   Path   #CQ7 TS   319.23   74.47%   105   3',\n",
       "  '  3.239   447.681   98.46%   BRCA2+   c.1447_1450ins4; p.Ala483AlaValfs   Path   #CQ3 TS   377.2   42.18%   227   120.384   57.25%   BRCA2+   c.10G>T;<<< p.Gly4Ter>>>   Path   #CQ5 TS   811.4   66.65%   400   173.040   89.41%   BRCA2+   c.5909C>A; p.Ser1970Ter   Path   #CQ7 TS   319.23   74.47%   105   32.248   10.',\n",
       "  'lT; p.Phe829Leufs c.1175_1214del40; p.Leu392Glnfs c.3756_3759delGTCT; p.Ser1253Argfs c.8764_8764delA; p.Ser2922Valfs c.1447_1450ins4; p.Ala483AlaValfs<<< c.10G>T>>>; p.Gly4Ter c.5909C>A; p.Ser1970Ter c.7617 1 1G>T c.6059_6062del4; p.Glu2020Valfs c.7251_7252delCA; p.His2417Glnfs c.9676delT; p.Tyr3226Ilefs c.4183C>T; p.Gln1395Ter c.5095C>T; p.Arg1699Trp  TABLE 3.',\n",
       "  '829Leufs c.1175_1214del40; p.Leu392Glnfs c.3756_3759delGTCT; p.Ser1253Argfs c.8764_8764delA; p.Ser2922Valfs c.1447_1450ins4; p.Ala483AlaValfs c.10G>T;<<< p.Gly4Ter>>> c.5909C>A; p.Ser1970Ter c.7617 1 1G>T c.6059_6062del4; p.Glu2020Valfs c.7251_7252delCA; p.His2417Glnfs c.9676delT; p.Tyr3226Ilefs c.4183C>T; p.Gln1395Ter c.5095C>T; p.Arg1699Trp  TABLE 3.'],\n",
       " '28751759': ['65Leu) VUS, which was also present in this sample.  Laboratory 16, using an HRM pre-screening strategy, failed to identify the BRCA2 nonsense mutation<<< c.10G>T>>>; p.(Gly4*) in one sample.',\n",
       "  '2Glnfs*5) NA 40 bp deletion (frameshift) 2 BRCA1 (NM_007294.3); BRCA2 (NM_000059.3) NA NA Control sample–no pathogenic mutations 3 BRCA2 (NM_000059.3)<<< c.10G4T>>>; p.(Gly4*) NA Point mutation (nonsense) 4 BRCA1 (NM_007294.3) c.442-?_547þ ?del, NA LGR: deletion (exon 8) 5 BRCA2 (NM_000059.3) c.5909C4A; p.(Ser197',\n",
       "  '465Leu) VUS, which was also present in this sample. Laboratory 16, using an HRM pre-screening strategy, failed to identify the BRCA2 nonsense mutation<<< c.10G4T>>>; p.(Gly4*) in one sample.']}"
      ]
     },
     "execution_count": 581,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "literature = {\n",
    "    \"papers\": articles[articles.pmid.isin(variant_mentions.pmid)].set_index(\"pmid\", drop=False).to_dict(orient=\"index\"),\n",
    "    \"variants\": {\n",
    "        k: {kk: vv[0] for kk, vv in v.unstack().transpose().iterrows()}\n",
    "        for k, v in combined.groupby(\"pyhgvs_Genomic_Coordinate_38\")},\n",
    "}\n",
    "with open(\"literature-05-12-2018-v1.json\", \"w\") as output:\n",
    "    output.write(json.dumps(literature))\n",
    "    \n",
    "literature[\"variants\"][\"chr13:g.32316470:G>T\"]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
