<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<base href="http://www.coursehero.com/file/1484690/Wangdis/"><div style="background:#fff;border:1px solid #999;margin:-1px -1px 0;padding:0;"><div style="background:#ddd;border:1px solid #999;color:#000;font:13px arial,sans-serif;font-weight:normal;margin:12px;padding:8px;text-align:left">This is Google&#39;s cache of <a href="http://www.coursehero.com/file/1484690/Wangdis/" style="text-decoration:underline;color:#00c">http://www.coursehero.com/file/1484690/Wangdis/</a>. It is a snapshot of the page as it appeared on Jul 6, 2013 00:35:24 GMT. The <a href="http://www.coursehero.com/file/1484690/Wangdis/" style="text-decoration:underline;color:#00c">current page</a> could have changed in the meantime. <a href="http://support.google.com/websearch/bin/answer.py?hl=en&amp;p=cached&amp;answer=1687222" style="text-decoration:underline;color:#00c">Learn more</a><br>Tip: To quickly find your search term on this page, press <b>Ctrl+F</b> or <b>âŒ˜-F</b> (Mac) and use the find bar.<br><br><div style="float:right"><a href="http://webcache.googleusercontent.com/search?q=cache:nJSM0MGoEngJ:www.coursehero.com/file/1484690/Wangdis/&amp;hl=en&amp;gl=us&amp;prmd=ivns&strip=0" style="text-decoration:underline;color:#00c">Full version</a></div>
<div>&nbsp;</div></div></div><div style="position:relative">

<!--[if lt IE 9 ]>
<html class="ltie9" lang="en" xmlns="http://www.w3.org/1999/xhtml" xmlns:nitro="http://www.bunchball.com/nitro">
<![endif]-->
<html lang="en" xmlns="http://www.w3.org/1999/xhtml" xmlns:nitro="http://www.bunchball.com/nitro">
<head>
					
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		
	
		
		
	
		
	
			
			
	
		
	

	<!--[if (gte IE 6)&(lte IE 8)]>
		<script type="text/javascript" src="/javascript/selectivizr-min.js">
		</script>
	<![endif]-->

	
		
		
	
	
		<title>LSU : ETD 01072004 : Wang_dis</title>



</head>


<body class="">
<div>






			<div id="ch-header-wrapper" class="ch-wrapper">
			<div class="ch-container" id="ch-navigation-container">
							<h1 id="brand">
	<a href="/">
		Course Hero
	</a>
</h1>
<div id="ch-nav-container">
			<ul id="accountentry">
			<li>
									<a class="ch-signup-btn" href="/registerForm.php" rel="nofollow"><i class="icon-lock"></i> Sign up</a>
							</li>
			<li>
									<a class="ch-login-btn" href="/login.php" rel="nofollow"><i class="icon-lock"></i> Log-in</a>
							</li>
		</ul>
		<ul id="ch-primary-nav">
				<li class="">
			<a href="/?no_redir=1">Home</a>
					</li>
				<li class="global-sitenav-withsubmenu ">
			<a href="/study-materials/">Study Documents</a>
							<ul id="ch-primary-nav-documents">
				
									<li><a href="/study-materials/">Find Documents</a></li>
									<li><a href="/textbooks/">Textbook Help</a></li>
								</ul>
					</li>
				<li class="global-sitenav-withsubmenu ">
			<a href="/tutors/">Tutors</a>
							<ul id="ch-primary-nav-qa">
				
									<li><a href="/qa_ask_question.php">Ask a Question</a></li>
									<li><a href="/become-a-tutor/">Become a Tutor</a></li>
									<li><a href="/qa/about_tutors.php">About Tutors</a></li>
								</ul>
					</li>
				<li class="global-sitenav-withsubmenu ">
			<a href="/courses/">Courses</a>
							<ul id="ch-primary-nav-courses">
				
									<li><a href="/courses/">Find Courses</a></li>
									<li><a href="/courses/edit_course_page.php?course_id=new">Create a Course</a></li>
									<li><a href="/courses/about.php">About Courses</a></li>
								</ul>
					</li>
				<li class="global-sitenav-withsubmenu ">
			<a href="/flashcards/">Flashcards</a>
							<ul id="ch-primary-nav-flashcards">
				
									<li><a href="/flashcards/browse/">Find Flashcards</a></li>
									<li><a href="/flashcards/create.php">Create Flashcards</a></li>
									<li><a href="/flashcards/">About Flashcards</a></li>
								</ul>
					</li>
				<li class="global-sitenav-withsubmenu ">
			<a href="/contact.php">Support</a>
							<ul id="ch-primary-nav-contact">
				
									<li><a href="/contact.php">Contact Us</a></li>
									<li><a href="/faq.php">FAQ</a></li>
								</ul>
					</li>
			</ul>
</div>

					</div>
	</div>
	

<div itemprop="mainContentOfPage" itemscope itemtype="http://schema.org/CreativeWork" class="ch-wrapper" id="ch-content-wrapper">
	<div class="ch-vlight-blue-bg ch-top-fold-banner">
		<div class="ch-container">
			<div class="ch-row">
				<div class="ch-column-12">
					<div class="ch-row">
						<h1 class="ch-column-12 ch-page-heading">
							Register now to access 7 million high quality study materials
							<a class="ch-what-tooltip">(What's Course Hero?)
								
								Course Hero is the premier provider of high quality online educational resources. With millions of study documents, online tutors, digital flashcards and free courseware, Course Hero is helping students learn more efficiently and effectively. Whether you're interested in exploring new subjects or mastering key topics for your next exam, Course Hero has the tools you need to achieve your goals.
								
							</a>
						</h1>
					</div>
					<div class="ch-row">
						<div class="ch-column-7">
							<div class="ch-doc-image-thumbnail">
								<div class="ch-doc-image-thumbnail-view ch-box-center">
																			
																	</div>
															</div>
							<div class="ch-doc-summary">
								<h3 itemprop="name">										Wang_dis
																	</h3>
								                                    								        Course: ETD 01072004, Fall 2008<br/>
                                    								School: LSU<br/>
																Rating: 
								<div id="doc-rating" class="doc-rating">
																		<div class="filled star">&nbsp;</div>
																		<div class="filled star">&nbsp;</div>
																		<div class="filled star">&nbsp;</div>
																		<div class="filled star">&nbsp;</div>
																											<div class="empty star">&nbsp;</div>
																	</div>
								<br/>
															</div>
						</div>
						<div class="ch-column-5" id="ch-doc-preview">
							<h4 class="ch-fl">Document Preview</h4>
							
							<div class="ch-clear"></div>
							<div class="ch-doc-preview-contents">
																										<p>OF ATTACHMENT OLIGONUCLEOTIDE PROBES TO POLYMER BIOCHIPS AND ITS APPLICATION FOR THE DETECTION OF POINT MUTATIONS
A Dissertation Submitted to the Graduate Faculty of the Louisiana State University and Agricultural and Mechanical College in partial fulfillment of the requirements for the degree of Doctor of Philosophy in The Department of Chemistry
By Yun Wang B.S., Shaanxi Normal University, 1987 M.S., Xian...</p>
																								</div>
							<a href="/registerForm.php?reg_only=1" class="ch-medium-orange-btn registerForm-link" id="ch-register-link">Register Now</a>
						</div>
						<div class="ch-clear"></div>
					</div>
				</div>
				<div class="ch-clear"></div>
			</div>
		</div>
	</div>
	    <div class="ch-gap"></div>
<div class="ch-container">
<div class="ch-row">
<div class="doc-search-bar doc-search-bar-1 ch-column-12 ch-vlight-blue-bg">
    <div class="ch-form-row">
    
        
        
                
        
        
        <a id="stx_submit_btn1" class="ch-medium-blue-btn search-btn-1 ch-fr" href="#">Search</a>
        
        
    
</div>
</div>
</div>
</div>



	<div class="ch-gap"></div>
    <div id="most_popular_materials_from_subject" class="ajax_load" data-params="school_type=HIGHER&school_name=LSU&dept_acro=ETD&course_num=01072004&db_filename=1484690"></div>
	<div class="ch-container ch-brand-promotion-panel-container">
		<div class="ch-row">
            <div id="what_course_hero_offers" class="ajax_load"></div>
			<div class="ch-clear"></div>
		</div>
	</div>
	
	<div class="ch-container">
		<div class="ch-row ch-tutorwidget-container">
			<div class="ch-column-12">
                <div id="qa_ask_question_widget" class="ajax_load" data-params="heading_inside=<strong>Can't find the right document?</strong> Our tutors are available 24/7 to answer your questions."></div>
            </div>
			<div class="ch-clear"></div>
		</div>
	</div>
	
	<div class="ch-container">
		<div class="ch-row">
            <div id="horizontal_register_now" class="ajax_load"></div>
			<div class="ch-clear"></div>
		</div>
	</div>
	
	<div class="ch-container">
		<div class="ch-row">
			<div class="ch-column-12 ch-document-excerpt ch-vlight-grey-bg ch-p10">
				<h4>Unformatted Document Excerpt</h4>
									<a href="/">Coursehero</a> >> <a href="/sitemap/states/Louisiana/">Louisiana</a> >>
					<a href="/sitemap/schools/65-LSU/">LSU</a> >>
					<a href="/sitemap/schools/65-LSU/departments/151772-ETD/">ETD 01072004</a>
								<p>Course Hero has millions of student submitted documents similar to the one<br/>below including study guides, practice problems, reference materials, practice exams, textbook help and tutor support.</p>
				<div style="margin-top:7px">
					Course Hero has millions of student submitted documents similar to the one
					below including study guides, practice problems, reference materials, practice exams, textbook help and tutor support.
				</div>
				<div title="Sign up now to view the formatted text." style="margin-top:7px;">
					                    
					  						OF ATTACHMENT OLIGONUCLEOTIDE PROBES TO POLYMER BIOCHIPS AND ITS APPLICATION FOR THE DETECTION OF POINT MUTATIONS
A Dissertation Submitted to the Graduate Faculty of the Louisiana State University and Agricultural and Mechanical College in partial fulfillment of the requirements for the degree of Doctor of Philosophy in The Department of Chemistry
By Yun Wang B.S., Shaanxi Normal University, 1987 M.S., Xian Jiaotong University, 1990 May, 2004
Acknowledgements
First and foremost, I thank my advisor, Dr. Steven A. Soper, for all of his guidance, insight, instruction, and endless patience throughout my graduate research, especially during my dissertation writing. I would like to particularly thank my committee members, Dr. Roberts P. Hammer, Dr. Robin L. McCarley, Dr. Andrew W. Maverick and Dr. Arthur L. Penn, for their invaluable discussions and critical comments. I would also like to thank Dr. Paul S. Russo and Dr. Steven F. Watkins for their encouragement and supports during the past years. My doctoral research would not have undergone smoothly without the valuable support provided by all the past and present members in Dr. Sopers group. I am particularly thankful to Dr. Wieslaw Stryjewski for his computer programming expertise that is definitely indispensable in operation and data acquisition of the near IR scanner and Dr. Shize Qi for fabricating the embossing tools. I extend many thanks to my fellow students, especially Dr. Emanuel Waddell, Dr. Yichuang Xu, Li Zhu, Michelle Galloway, Angela Jones, Ronderick Sinville, Catherine Situma, Vera Verdree, Andre Adams, Ya Liang, Harrison Musyimi, Shawn Llopis, Anne Abubuafo, and Hamed Shadpour for their helpful discussions and assistances in the course of this work. I am thankful to Drs. Jifeng Chen, Masahiko Hashimoto, Mateusz Hupert, and Makgorzata Witek for their supports. Finally, I would like to thank my family. My grandparents, Enzhu Li and Guifang Yang, have encouraged me through their diligence and integrity in my childhood. Thanks to my father and mother, Zhigang Wang and Yaxin Li, for their standing behind me throughout every challenge in my life. Thanks to my husband, Tao Wang, for his caring and support through this process and my daughter, Xinhui (Connie) Wang, for allowing me to escape from many many hours that were supposed to be quality family times. Thanks to my brother, Shaopeng Wang, for his sibling friendship and suggestions derived from his own Ph.D. research.
ii
Table of Contents Acknowledgements.. List of Tables... List of Figures. List of Schemes.. Abstract............... Chapter 1. DNA Microarrays..
Introduction... Nucleotides  Structural Units of Nucleic Acids Double-Helical Structure and Denaturation / Renaturation of DNA.. Genetic Codes and Gene Mutations Chemical Synthesis of Nucleic Acids. The Concept of DNA Microarrays.. Fabrication of DNA Microarrays Photochemical Deprotection Method Deposition of Presynthesized Oligonucleotide Probes or PCR Products. Applications of DNA Microarrays. Limitations and Improvements for Array-based Techniques. Microfluidic Arrays. Research Focus. References . ii vi vii x xi 1 1 1 3 5 9 10 13 13 15 18 25 26 28 28 34 34 35 37 39 42 44 44 45 47 49 49 49 50 52 52
Chapter 2. Laser-Induced Near-IR Fluorescence Detection
Introduction... Jablonskis Diagram Near-Infrared Fluorescence. Near-IR Labeling Dyes... Confocal Fluorescence Microscope and Scanner... Excitation Sources.. Lamps Lasers Single Photon Avalanche Diode Detectors. Research Focus. Experimental Section Reagents and Materials... Instrumentation... Optimization of Excitation Laser Power......... Near-IR Dye Photo-Stability on Solid Surface... iii
Results and Discussions Excitation Laser Power... The Photo-Stability of IRD800 Dyes.. Conclusions... References.
53 53 55 56 57 60 60 63 63 64 65 65 66 66 67 67 68 69 70 70 70 75 76 79 79 80 80 82 82 83 84 86 86
Chapter 3. Two-Dimensional PMMA Oligonucleotide Arrays.
Introduction... Experimental Section Chemicals and Reagents. PMMA Surface Modification. Surface Contact Angle Measurements Surface Amine Density... Preparation of PMMA Oligonucleotide Arrays.. Optimization of Coupling Time of Oligonucleotide Probes to Surface.. Spotting Concentration of Oligonucleotide Probe Solution... 3-32P Terminal Labeling and Immobilization of Oligonucleotide Probes. Hybridization Optimization Fluorescence Detection and Data Analysis. Results and Discussions ... PMMA Surface Modification and Oligonucleotide Coupling Chemistry.. Step 1  Aminolysis.. Step 2  Activation Step 3  Coupling.. Step 4 - Capping and Reduction... 3-32P End Terminal Labeling of Oligonucleotide Probes Optimization of Hybridization PMMA-Surface Amine Density on Hybridization Productivity... The Effect of Buffer Concentration.. Hybridization Kinetics.. Capture Sensitivity of Oligonucleotide Arrays... Stability of Linkage Chemistry. . Conclusions.. References 
Chapter 4. Microfluidic PMMA Oligonucleotide Array and Its Application for Point Mutation Detection.
Introduction... Experimental Section Materials and Reagents... Instrumentation... PMMA Chip Fabrication and Assembly of Microfluidic Channels... Dynamic Hybridization in PMMA Microfluidic Channels LDR Products Detection in PMMA Microfluidic Zip-Code Arrays.. Results and Discussions
89 89 93 93 94 95 97 97 99
iv
Hybridization Kinetics in Microfluidic Channels... Detecting K-ras Mutations Using PMMA Microfluidic Arrays. Conclusions... References .
99 101 103 104
Chapter 5. UV-Patterning of Oligonucleotide Arrays on Poly (methyl methacrylate) and Polycarbonate Surfaces..
Introduction... Experimental Section Materials and Chemicals  Instrumentation... Deposition of Oligonucleotide Probes on Pristine Polymer Surfaces  Deposition of Oligonucleotide Probes on UV-Patterned Polymer Surfaces .. Functional Assays of Polymer Surface-Immobilized Oligonucleotide Probes.. Results and Discussion.. Deposition of Oligonucleotide Probes on Pristine Polymer Surfaces. Deposition of Oligonucleotide Probes on UV-Patterned PC and PMMA Surfaces Conclusions.. Future Work. References
106 106 111 111 112 112 112 113 113 113 124 127 128 128 132 132 134 134 135 137 138 139 141
Chapter 6. Summary and Future Work...
Summary... Future Work.. Surface Modification.. Electro-pumping. UV-patterning. References.
Appendix. Letters of Permission... Vita
v
List of Tables
1.1. The relationships between the codons and the amino acids. 7 1.2. Gene expression high-density microarrays...... 22
2.1. The structures of some representative near-IR fluorescent dyes and their absorption and emission properties.................................... 40 2.2. The characteristics of some commercial fluorescence scanners.. 2.3. Characteristics of commonly used lasers. 3.1. Sequences of oligonucleotide probes and dye-labeled complements.. 3.2. SNR values in different hybridization buffers. 4.1. The sequences of oligonucleotides used in these experiments 5.1. The sequences of oligonucleotides.. 43 46 63 82 93 111
vi
List of Figures
1.1. The chemical structures of the five nucleotide triphosphates and the two pentose sugars... 2
1.2. The double strand structure of nucleic acids ... 4 1.3. The transcription of the -globin gene to mRNA ... 1.4. The scheme of phosphotriester and phosphite-triester chain assembly for solid-phase synthesis of oligonucleotides .. 7
9
1.5. cDNA microarray scheme ....... 12 1.6. Light directed oligonucleotide synthesis ..................... 14 1.7. In situ synthesis of oligonucleotide probes on aminated polypropylene substrate for utilization in solid-phase hybridization analysis.. 1.8. Diagram of linker chemistry employed for attachment of oligonucleotide probes to glass silanized surface at their 5 amino-modified termini.. 1.9. Coupling of amine-terminated (A) or dialdehyde-terminated (B) oligonucleotide probes to polyacrylamide gel matrix containing aldehyde or amino functional groups..
15
17
17
1.10. An array designed for sequencing.. 19 1.11. Gene expression analysis using a cDNA microarray..... 21 1.12. A tiling array designed to detect a specific sequence 1.13. Minisequencing analysis of genotypes.. 23 24
2.1. Jablonski energy level diagram... 36 2.2. Position of near-IR regions in the electromagnetic spectrum. 2.3. Absorbance and fluorescence of biomolecules in relation to visible and near-IR fluorophores 2.4. The confocal principle in epi-illumination microscope.. 2.5. Schematic representation of a typical laser source. 37
38 42 45
2.6. The near-IR fluorescence scanner optical (A) and instrument set up (B)... 50
vii
2.7. The optimization of excitation laser power. 53 2.8. The photostability of IRD800 on a PMMA substrate. 56
3.1. The effect of aminolysis time on the surface amine density ... 73 3.2. The effect of N-lithioethylenediamine concentration on the surface amine density . 3.3. The conversion efficiency of surface amine groups to aldehyde groups as a function of the length of treatment time in 5% glutaric dialdehyde solution 3.4. The effect of oligonucleotide probe solution concentration on the amount of surface immobilized probes as indicated by the hybridization fluorescence signal-to-noise ratio (SNR)... 3.5. The length of the incubation time on the amount of surface immobilized oligonucleotide probes 3.6. The fluorescence intensity produced from a hybridization assay as a function of aminolysis time 3.7. The hybridization kinetics on flat PMMA slides. 3.8. Fluorescence intensity versus concentration of dye-labeled oligonucleotides in the hybridization solution.. 3.9. Stability of the oligonucleotide-PMMA coupling chemistry as a function of denaturation /hybridization cycles... 4.1. Schematic of LDR reactions and hybridizations to zip codes. 4.2. Picture of the Ni molding master fabricated using LIGA and a topographical layout of one of the hybridization microfluidic networks.. 4.3. Hybridization kinetics in PMMA microchip channels 4.4. LDR reactions of K-ras c12.2 v mutant and wild type DNAs with different ratios of mutant to wild type. 4.5. Mutational analysis of 1:10,000 c12.2 V mutant to wild-type LDR products using the zip-code PMMA microchip array 5.1. Fluorescence images of oligonucleotide on PMMA (A) and PC (B) slides
74
75
76
77
81 83
84
85 90
96 100
101
102 115
viii
5.2. The normalized hybridization SNR for oligonucleotide probes non-covalently immobilized to pristine PMMA surfaces (solid squares) and for oligonucleotide probes covalently immobilized to chemically modified PMMA surfaces (solid circles).... 117 5.3. The images of hybridized C6 amino 34-mer probes covalently-immobilized on chemically-modified PMMA surfaces (A) and non-covalently-immobilized on pristine PMMA surfaces (B) 5.4. The effect of oligonucleotide probe types on SNR values for (A) pristine PMMA and (B) pristine PC. ... 5.5. Proposed adsorption mechanism of amine modified oligonucleotide probes onto pristine PMMA surfaces.. 5.6 Aminolysis of PC main chain with primary amines
119
120
121 122
5.7. The catalytic mechanism of EDC in amide bond formation 124 5.8. UV-patterning of oligonucleotide probes on PC and PMMA surfaces... 6.1. The schematic set up of the electro-pumping experiment... 125 136
ix
List of Schemes
3.1. Flow chart of oligonucleotide probe coupling chemistry to PMMA surfaces. 71
3.2. Schematic of surface amine group measurement using sulfo-SDTB reagent.. 72
x
Abstract
This dissertation is on the fabrication of polymer-based microfluidic arrays for the detection of genetic mutations. Poly(methyl methacrylate) (PMMA) was chosen as one of the polymer substrate materials due to its low background levels, low adsorption of biomolecules, and low assembly temperatures. The surface modification of polymer substrates for covalent attachment of oligonucleotide probes, the construction of fluidic channels/arrays, and hybridization kinetics in coupled microfluidic channels and arrays will be covered. As an example of the application, point mutation detection using immobilized arrays constructed in microfluidic devices will be demonstrated. The PMMA surface was derivatized with N-lithioethylenediamine solution to introduce amine groups, which were utilized for the covalent immobilization of terminal amino modified oligonucleotide probes via the homo-bi-functional linker molecule. The coupling bonds formed were stable enough to withstand multiple denaturation/rehybridization cycles. To overcome the drawbacks associated with conventional 2-D flat microarrays, such as long hybridization times and large sample consumption, oligonucleotide arrays were constructed into the microfluidic channels hot embossed into PMMA substrate. With the use of these fluidic channels, we observed increased hybridization kinetics as compared to that on the flat arrays. Another benefit is that the channel-attached oligonucleotide probes allowed the detection of target concentrations down to pM levels. Specially designed oligonucleotide probes, which had similar melting temperatures, were printed in microfluidic channels. Low-abundant point mutations in K-ras genes were successfully detected by using a ligase detection reaction (LDR) combined with the microfluidic hybridization. Near-IR laser-induced fluorescence was used for the detection of surface conducted bioanalytical reactions and high detection sensitivity was obtained.
xi
In addition, preliminary work was also conducted on direct photo-patterning using deep ultraviolet (UV) light for immobilizing oligonucleotides onto poly(methyl methacrylate) and polycarbonate substrates. Well-defined spots were observed on PMMA surfaces. However, the proper dose of UV-exposure and the appropriate EDC concentration need to be optimized in future work to increase the contrast on the immobilization efficiency between the exposed regions and un-exposed regions.
xii
Chapter 1 DNA Microarrays
Introduction
With the completion of the Human Genome Project and the understanding of relations between genetic mutations and diseases, various techniques have been developed for the detection of nucleic acid sequence variations. A new technology, DNA microarrays, has attracted tremendous interests among biologists because of their potential to handle a large number of samples simultaneously; thus providing high throughput of sequence information on the biological samples [1-2]. These microarrays have been used by researchers for gene expression analysis [3], mutation detection [4], and genomic analysis [5]. In this chapter, the basic concepts of nucleic acid chemical structures, the fabrication, and application of DNA microarrays will be described. Nucleotides  Structural Units of Nucleic Acids Nucleotides are the building blocks of nucleic acids. These structural units consist of three essential components: a nitrogenous base (a heterocyclic ring of carbon and nitrogen atoms), a pentose sugar (a five-carbon sugar in ring form), and phosphate groups. Deoxyribonucleic acid (DNA) and ribonucleic acid (RNA) are polymers of nucleotides linked as a chain through phosphodiester bonds. There are five types of bases in nucleic acids; adenine (A) and guanine (G) are purine derivatives (fused five- and six-membered rings as shown in Figure 1.1); cytosine (C), thymine (T), and uracil (U) are pyrimidine derivatives (six-membered rings as shown in Figure 1.1). Both
1
DNA and RNA contain the bases A, C, and G; only DNA contains the base T, whereas RNA contains the base U.
NH 2
O
N 7
N 7 6 5 4 3 N 1 NH 2 NH 2
-O
6 5 4 3 N 1 N 2
8 O P OO O P OO O P OO CH 2 4' H H 3' OH H 5' 9 N
O
-O
O O P OO
O P OO 5' CH 2 4' H H 3' OH H
8 9 N
P O-
O H 2' H 1'
O H 2' H 1'
2'-deoxyguanosine 5'-triphosphate (dGTP)
NH 2
2-doxyadenosine 5 triphosphate (dATP)
O H3C 4 5
4 5 6 O
-O
3NH 2 O
3N
6
O O P OO O P OO 5' CH 2 4' H H 3' OH 1 N 2 O
-O
O P OO
O P OO
O P OO 5' CH 2 4' H H 3' OH
1 N
P O-
O H H 1' 2'H
O H H 1 2'H
'
2'-deoxycytidine 5'-triphosphate (dCTP)
2'-deoxythymidine 5'-triphosphate (dTTP)
O 4 5 6 1 N 3NH 2 O
O
-O
O O P OO
O P OO 5' CH 2 4' H H 3' OH
P O-
O H 1'
2'H OH
uridine 5'-triphosphate (UTP)
HO
5' 4'
H H 3' OH
OH O H
HO O
OH H H OH OH H H
1'
2' H
H
2-deoxyribose ribose Figure 1.1. The chemical structures of the five nucleotide triphosphates and the two pentose sugars. Two types of pentose sugars are found in nucleic acids, 2-deoxyribose in DNA and ribose in RNA with the difference being the presence or absence of 2-hydroxyl groups as illustrated in Figure 1.1. The nitrogenous base is linked to position 1 on the pentose ring by a glycosidic bond
2
from N1 for pyrimidines or N9 for purines and such a unit is defined as a nucleoside. A nucleotide is a phosphate ester of a nucleoside. The most common site of esterification in naturally occurring nucleotides is the phosphate group attached to C-5 of the sugar. More than one phosphate group can be linked to the 5 position in the nucleotides. Each nitrogen or carbon atom of the ring structures is designated by an unprimed number on nitrogenous bases and primed number on the sugar rings. Double-Helical Structure and Denaturation / Renaturation of DNA DNA consists of an exceedingly long polynucleotide chain with a backbone consisting of an alternating series of sugar and phosphate residues. The 5 position of one pentose ring is connected to the 3 position of the next pentose ring via a phosphate group, as shown in Figure 1.2. The nitrogenous bases stick out from the backbone. Normally, DNA molecules exist in double-stranded structures [6], Watson and Crick proposed that the two polynucleotide chains in the double helix associate by hydrogen bonding between the nitrogenous bases. Figure 1.2 demonstrates that G can hydrogen bond specifically with C with the formation of 3 hydrogen bonds, while A can bond specifically with T with 2 hydrogen bonds formed. These connections are described as base pairing, and the paired bases are said to be complementary. The two strands are oriented in an antiparallel fashion to one another with the negatively charged sugarphosphate backbone on the outside and the stacking of base pairs in the interior of the helix. The non-covalent forces that stabilize the double helix can be disrupted by heating or by exposure to low salt concentrations. The process of strand separation is called denaturation or melting. The midpoint of the temperature range over which 50% of the duplex strands of DNA are separated is called the melting temperature (Tm). The G/C contents of the sequence, the size of the strand, the degree of mismatching between the sequences, the salt concentration and the
3
presence of denaturant in the buffer solutions, all have effects on the thermostability of the duplex structure and hence the melting temperature Tm.
O O OP O O O OP O
O OP O
O
O O OP O
O O OP O
O O OP O
O O OP O
O O OP O
Figure 1.2. The double stranded structure of nucleic acids [7]. The two strands are aligned antiparallel to each other and connected by the specific base pairing between G-C and A-T. The thermostability of a duplex is proportional to the number of hydrogen bonds formed between the two strands [8], thus, Tm is proportional to the length of the strands. In addition, G/C base pairs produce more hydrogen bonds than A/T pairs, therefore, Tm is proportional to the number of G/C base pairs and normally increases ~0.4oC for every 1% increase in G/C content [7]. The existence of mismatches between the two stands destabilizes the structures, thus, the Tm
4
for perfect matched duplexes is generally higher than that with mismatched bases. Since the phosphate-sugar backbones are negatively charged, the presence of salt in the solution tends to reduce the repellent interactions between the pairing strands and hence Tm increase with salt concentrations to a certain point. The presence of denaturing reagents, such as formamide, will affect Tm with the Tm reduced by 0.65oC for every percent of the denaturing reagent in aqueous hybridization buffers [9]. The denaturation of DNA duplex structure is a reversible process and the two separated complementary strands can reform into a double helix under appropriate conditions. This process is named renaturation. Renaturation can be extended to allow any two complementary nucleic acid sequences from different sources to anneal with each other to form a duplex structure and this process is called hybridization. The reaction takes place in two stages. First, single strands of DNA in the solution encounter one another by chance; if their sequences are complementary, the two strands base pair to generate a short double-helical region. Then the region of base pairing extends along the molecule by a zipper-like effect to form a lengthy duplex molecule [10]. Genetic Codes and Gene Mutations A gene is defined as a sequence of DNA that controls the production of a specific protein. However, DNA is not the direct template for protein synthesis. Rather, the template for protein synthesis is RNA (ribonucleic acid) [11]. The flow of genetic information in normal cells is:
DNA Transcription RNA Translation Protein
RNA polymerase takes instructions from a DNA template, which contains regions called promoter sites that specifically bind RNA polymerase and determine where transcription begins [12]. RNA polymerase proceeds along the DNA template and transcribes one of its stands until a terminator sequence is reached. Nascent RNA spontaneously dissociates from RNA polymerase
5
when the extended strand is terminated. The transcribed DNA strand is called a template strand or antisense strand. The other DNA strand that bears the same sequence as the mRNA is called the coding strand or sense strand. The transcribed mRNA serves as the templates for protein synthesis. The relation between the sequence of bases in DNA (or its mRNA transcript) and the sequence of amino acids in proteins is called the genetic code [13]. The genetic code is read from a fixed starting point in a non-overlapping trinucleotide sequence, which is called a codon. Depending on the starting point there are three possible ways of translating a nucleotide sequence into protein and these are called reading frames. Written in the conventional 5-3 direction, the nucleotide sequence of the DNA strand that codes for a protein corresponds to the amino acid sequence of the protein written in the direction from N-terminus to C-terminus. In bacteria, polypeptide chains are encoded by a continuous array of triplet codons in DNA. However, most eukaryotic genes are discontinuous, that is, the coding regions (exons) are disrupted by un-coding regions (introns). For example, the gene for the  chain of hemoglobin is composed of three amino acid coding sequences interrupted by two intervening sequences with a length of 550 base pairs and 120 base pairs. Thus, the -globin gene is split into three coding sequences as illustrated in Figure 1.3. When the -Globin gene is transcribed into mRNA, the intervening sequences (introns) in the primary transcript are excised and the coding sequences (exons) are simultaneously linked by a precise splicing enzyme to form the mature mRNA. Regions that are removed from the primary transcript are called introns (the intervening sequences), whereas those that are retained in the mature RNA are called exons [14].
6
exon 240 120 500
intron 550 -Globin gene 250
Primary transcript
-Globin mRNA
Figure 1.3. The transcription of the -globin gene to mRNA [15]. There are 64 codons with only 61 assigned to amino acids. The other three triplets are termination codons or stop codons that end protein synthesis. The relationships between the codons and the amino acids are listed in Table 1.1 [7]. Because there are 20 amino acids and 61 triplets that code for them, many amino acids are designated by more than one triplet. Table 1.1. The relationships between the codons and the amino acids [7]
Genetic code Amino acid Genetic code Amino acid Genetic code Amino acid Genetic code Amino acid Genetic code Amino acid Genetic code Amino acid
UUU UUC UUA UUG UCU UCC UCA UCG UAU UAC UGU
Phe Phe Leu Leu Ser Ser Ser Ser Tyr Tyr Cys
UGC UGG CUU CUC CUA CUG AUU AUC AUA AUG GUU
Cys Trp Leu Leu Leu Leu lle lle lle Met Val
GUC GUA GUG CCU CCC CCA CCG CAU CAC CAA CAG
Val Val Val Pro Pro Pro Pro His His Gln Gln
CGU CGC CGA CGG ACU ACC ACA ACG AAU AAC AAA
Arg Arg Arg Arg Thr Thr Thr Thr Asn Asn Lys
AAG AGU AGC AGA AGG GCU GCC GCA GCG GAU GAC
Lys Ser Ser Arg Arg Ala Ala Ala Ala Asp Asp
GAA GAG GGU GGC GGA GGG UAA UAG UGA
Glu Glu Gly Gly Gly Gly Stop Stop Stop
7
A specific gene always lies on a certain chromosome and an eukaryotic chromosome contains thousands of genes. The human genome includes approximately 36,000 genes and three billion base pairs [15]. A gene may exist in alternative forms that determine the expression of some particular characteristics (phenotype). The forms of the gene are called alleles. When an organism has two identical alleles of a gene, it is said to be homozygous; if the alleles are different, the organism is heterozygous [7]. Any change in the sequence of DNA in a genome is called a mutation and it is directly associated with a change in phenotype (visible or otherwise measurable properties). The organism carrying the altered gene is called a mutant; an organism carrying the normal gene is called wild-type. Mutations occur as the result of normal cellular operations or random interactions with the environment. Such mutations are called spontaneous. Spontaneous mutations that inactivate gene function occur in bacteria at a rate of 10-5  10-6 events per locus per generation [7]. The occurrence of mutations can be increased by treatment with certain compounds, the mutagens (with the ability to either modify directly a particular base of DNA or to become incorporated into the nucleic acid), and the caused changes are referred to as induced mutations. Mutations can be divided into two general classes [16]: point mutations and rearrangement mutations. In point mutations, only a single base pair is involved. If the point mutation changes a codon to represent a different amino acid, it is called a missense mutation. If a point mutation creates one of the three termination codons, which causes premature termination of protein synthesis, it is called a nonsense mutation. In rearrangement mutations, a large region of a gene is affected. The simplest type of rearrangements is insertions or deletions of a stretch of sequence, which can change the reading frame for the entire subsequent sequence. A change of
8
this sort is called a frameshift. If a mutation involves the deletion of the whole gene, it is called a null mutation. Chemical Synthesis of Nucleic Acids The development of recombinant DNA techniques [17], DNA sequencing [18], and the use of oligonucleotide primers as hybridization probes [18] and as primers for the synthesis of cDNAs, created the necessity of synthesizing short fragments of DNAs. The fragments of nucleic acids can be chemically synthesized. Two of the chemical synthesis routes for polynucleotides are depicted in Figure 1.4.
2) DMTrO O base 2 DMTrO O O P O Et3 NH O
te r
O
base 2
CI CI
os
ph
o tr
O OP O
ie s
O
O
base 1
ph
O P 2) DMTrO O base 2 DMTrO O H 3 CO P N 3) capping 4) I2 oxidation O H 3 CO P O O O base 1
- tr ie
DMTrO
O
base 1
HO 1)acid P O P
O
base 1
OCOCH 2CH 2CONH
O
base 2
Figure 1.4. The schema of phosphotriester and phosphite-triester chain assembly for solid-phase synthesis of oligonucleotides [19]. As shown in Figure 1.4, the DNA is synthesized from 3' to 5' with the first base on the support prior to oligonucleotide synthesis. The first 5-O-protected deoxyribonucleoside is attached to an amino-substituted solid substrate through a 3-O-succinate functional group, which can be easily hydrolyzed after chain assembly. Each base thereafter is delivered in the
ph os ph i te st er
O P
9
form of a phosphorylated deoxyribonucleoside, which is protected on the 5' end with dimethoxytrityl (DMTr). The two commonly used chain assembly strategies are, the phosphotriester or phosphite-triester. The acidic terminal deprotection step, the DMTr is removed by exposure to acid, normally trichloroacetic acid (TCA), is identical for both routes. In the phosphotriester route the 3-O-phosphorylated monomer requires activation by a coupling agent and the coupling time is reduced to a few minutes by use of an extra catalyst. In the phosphite-triester route an active 3-O-phosphitylated monomer is prepared as a stable phosphoramidite but is converted to a highly active phosphitylating agent by an acidic catalyst. The intermediate dinucleoside phosphite is oxidized to the more stable phosphate (Pv) before chain extension. Cycles of chain assembly are continued until the required length is obtained and then the oligonucleotide is cleaved from the support, followed by removing the protecting groups on exocyclic bases and the purification of the deprotected oligonucleotide products. The Concept of DNA Microarrays A DNA microarray is a hybridization-based assay in which the affinity of surfaceimmobilized DNA probes for binding their complements in solution is utilized to detect and quantify targets. It is a high throughput technology with ordered arrays of oligonucleotides or DNA molecules with known sequences attached to the solid support using in situ synthesis [20] or deposition of pre-prepared probes [21]. DNA microarray technology evolved from Southern blot techniques, in which radioactive-labeled nucleic acid probes were used to detect membraneimmobilized nucleic acid targets blotted from gel-fractionation [22]. Modern DNA microarrays emerged with the replacement of membrane materials with impervious solid substrates such as glass. In addition, the probes instead of the targets are arrayed onto the solid surfaces. The flatness, rigidity, and transparency of glass supports
10
improved image acquisition and processing because the locations of the probes are much better defined than they are on a flexible membrane. The use of solid substrates makes it possible to build high-density oligonucleotide arrays using photolithographic chemistry [20]. In addition, the hybridization rate is enhanced on flat surfaces since the labeled targets do not need to diffuse into and out of the pores in membrane materials [23]. The surface-immobilized probes, which are defined by B. Phimister [24] as the nucleic acids with known sequences, are used to interrogate the unknown targets in solution [25]. Since only the targets with complementary sequences to the probes can be retained on the surface after stringent washing, the sequence information of the targets can be deduced. There are two variants of DNA microarrays in terms of the length of the arrayed probes; cDNA arrays and oligonucleotide arrays. cDNA arrays contain long single-stranded probes with kilobase length and this format of DNA microarrays are generally used in RNA expression analysis [3]. These surface arrayed probes may come from genomic DNA or cloned DNA fragments, which are amplified with the polymerase chain reaction (PCR) [26]. The preparation and application of c-DNA microarrays for RNA expression analysis are illustrated in Figure 1.5. The PCR products are printed onto solid substrate and the slides are denatured in distilled water for 2 mins at 90oC to prepare cDNA arrays [3]. Then, total RNA from the test and reference sample is reverse transcribed into complementary DNAs and labeled with different fluorescent labels, for example, Cy3- and Cy5dUTP, using a single round of reverse transcription and allowed to mix and hybridize to the cDNA arrays. After hybridization and washing, the slide is scanned using two different wavelengths corresponding to the dyes used, and the intensity of the same spot in both channels
11
is compared. The relative expression level of the gene in the sample relative to the reference is obtained by the ratio of the fluorescence measurement [27].
Figure 1.5. cDNA microarray scheme. Templates for genes of interest are obtained and amplified by PCR. The purified products are printed onto coated glass microscope slides using a computer-controlled, high-speed robot. Then total RNA from both the test and reference sample is fluorescently labeled with either Cy3- or Cy5-dUTP using a single round of reverse transcription and allowed to hybridize under stringent conditions. Laser excitation of the incorporated targets yields an emission with characteristic spectra, which is measured using a scanning confocal laser microscope. Data from a single hybridization experiment is viewed as a normalized ratio (that is, Cy3/Cy5) in which significant deviations from 1 (no change) are indicative of increased (&gt;1) or decreased (&lt;1) levels of gene expression relative to the reference sample [28]. Oligonucleotide arrays consist of an array of oligonucleotide (16~80-mer) or peptide nucleic acid (PNA) probes synthesized either in situ (on-chip) [29] or by conventional synthesis followed by on-chip immobilization. This format can be used for both expression analysis and sequence analysis [30]. The fabrication of oligonucleotide arrays will be introduced in the following section. To determine the hybrids formed in the hybridization reaction, it is necessary to label the target with an indicator molecule. The labels used include radioisotopes [31], fluorescent dyes
12
[32], or colorimetric labels [33]. The conventional and most sensitive labels are radioisotopes such as 32P, 33P, 3H, or 14C. The detection limit of radiolabeled hybridization is &lt;1pg to 1-10 fg of target DNA. The detection limit of the non-radioactively labeled hybridization is between 1 to 5 pg of the target nucleic acids [34]. However, due to their health hazard and the cost for handling and waste disposal, radioisotope labels are being gradually replaced by fluorescent dyes [35]. The diversity of fluorescent dyes makes it possible to label the targets with various colored dyes and to simultaneously detect different targets. Fortunately, the emergence of PCR [26] and other amplification techniques [36] make the amplification of biological samples possible. Amplification generates specific targets with high signal/noise ratios and permits the use of less sensitive nonisotopic reporters in DNA analysis. Fabrication of DNA Microarrays There are two alternatives to produce DNA microarrays: in situ synthesis [37] or deposition of presynthesized oligonucleotide sequences or DNA fragments onto solid substrates [28].
Photochemical Deprotection Method [37]
One method for the in situ syntheses of oligonucleotide probes is photochemical deprotection, which was developed at Affymetrix (www.affymetrix.com). The synthesis scheme is illustrated in Figure 1.6. A light-directed synthesis technique is employed to construct highdensity DNA probe arrays. Synthetic linkers with photo-chemically removable protecting groups are attached to glass substrates and light is directed to specific areas on the surface through a photo-lithographic mask to generate free hydroxyl groups in the photo-deprotected regions. Then, hydroxyl-protected deoxynucleotides are incubated with the surface and chemical coupling occurs at those sites that have been illuminated in the preceding step. Next, light is
13
directed to different regions of the substrate by a new mask, and the chemical cycle is repeated [38-39]. Rounds of illumination and coupling are repeated until the desired set of products is obtained. With this method, arbitrary polynucleotide sequences at specific locations on the array can be synthesized in a minimum number of chemical steps [20]. For example, the complete set of 4N polynucleotides of length N can be synthesized in only 4 x N cycles. Since photolithography is used, the process can be miniaturized to generate high-density arrays of oligonucleotide probes. Currently the individual features of the probes are 24 m x 24 m and with this feature size 65536 probes can be constructed on a 1.28 cm x 1.28 cm chip. The feature size can be expected to be reduced to 10 m x 10 m with a corresponding density of &gt; 106 sequences/cm2 [36]. For the in situ synthesis, oligonucleotide probes are generally tethered to the surface through their 3 end positions.
Figure 1.6. Light directed oligonucleotide synthesis. A solid support is derivatized with a covalent linker molecule terminated with a photolabile protecting group. Light is directed through a mask to deprotect and activate selected sites, and protected nucleotides couple to the activated sites. The process is repeated, activating different sets of sites and coupling different bases allowing arbitrary DNA probes to be constructed at each site [37].
14
In situ synthesis of oligonucleotide probes from monomers is not limited to glass slides; polypropylene substrates were aminated by exposure to ammonia plasma generated by radiofrequency plasma discharge to serve as a solid-phase support for oligonucleotide synthesis (Figure 1.7.) [40]. Oligonucleotides synthesized on the surface of the polymers remain attached following deprotection and can be used directly for hybridization.
CH 3 C CH 2 n H CH 3 C CH 2 n H CH 3 C CH 2 n H
ammonia plasma CH 3 C CH 2 n H DMTr O NH 2 C CH 2 n H H N Base 1 CH 3 C CH 2 n H
NCCH 2 CH 2 O P
O N(iPr) DMTr
N NN activation
I2 oxidation O Base 1
O CH 3 C CH 2 n H OPO NH C CH 2 n H CH 3 C CH 2 n H
Figure 1.7. In situ synthesis of oligonucleotide probes on aminated polypropylene substrate for utilization in solid-phase hybridization analysis. The polymer surface is first aminated by exposure to ammonia plasma generated by radiofrequency plasma discharge followed by phosphoramidite activation to yield a base stable amidate bond after oxidation with iodine.
Deposition of Presynthesized Oligonucleotide Probes or PCR Products
In this format, a computer controlled collection of pin devices pick up small drops of solution from the multi-well plates and print them to the glass or other substrate surfaces. The
15
first array produced with such a method was developed in Browns laboratory at Stanford University (cmgm.stanford.edu/pbrown/index.html). Microscope slides are coated with polylysine, amino silane or amino-reactive silanes [41] to enhance both the hydrophobicity of the slide and the adherence of the deposited DNA. When double-stranded PCR products are deposited, the deposited probe is rendered single-stranded by heat or alkali conditions [42]. In most cases, DNA samples are immobilized to the surface by ultraviolet irradiation to form covalent bonds between the thymidine residues in the DNA and the positively charged amine groups on the silane slides [42]. With this probe coupling format, some of the bases will be constrained and become inaccessible in the following hybridization process. Compared to the oligonucleotide probes which are immobilized at single terminal end, long probes are probably less efficient in the hybridization [28]. After fixation, residual amines on the slide surface are reacted with succinic anhydride to reduce the positive charge at the surface to minimize nonspecific adsorption. For chemically modified oligonucleotide probes, various methods have been developed for the end attachment of the probes to the solid supports [32]. For example, pre-synthesized 5amino-modified oligonucleotides can be attached to amino-silanized glass microscope slides with homo-bi-functional crosslinkers, such as 1,4-phenylene diisothiocyanate (PDC) (Figure 1.8) [32], carbodiimide-mediated end-attachment of 5'-phosphate and 5'-NH2 modified nucleic acids to, respectively, amino and carboxyl magnetic beads [43]. Beattie and Fowler [44] have utilized streptavidin-coated magnetic beads to sequester 5-biotinylated oligonucleotides, while Rasmussen et al. [45] employed carbodimide to condense the 5 phosphate group of the oligomer to microtiter plate surface modified with a secondary amino group. Another approach utilizes a poly T tail incorporated into the 3 terminus of the oligonucleotide and the tailed oligonucleotide
16
is then UV-crosslinked to the membranes [46]. Alternatively, unmodified oligonucleotides can be attached electrostatically to polylysine-treated or amino-silanized glass slides [47].
S OCH 3 H Si O Si CH 2 HN C N 3 OCH 3 amino silane
S O H H N C N CH 2 O P O TTTTTTTTTTTTTTTATAACCCGGAATCCT 6 O amino modification spacer hybridization sequence
Figure 1.8. Diagram of linker chemistry employed for attachment of oligonucleotide probes to glass silanized surface at their 5 amino-modified termini [32]. In addition to flat surfaces, presynthesized, purified oligonucleotides have been immobilized on functionalized polyacrylamide gel matrix [23] [48]. Oligodeoxynucleotides for immobilization were synthesized with 3-terminal 3-methyluridine, activated by oxidizing with NaIO4 to produce dialdehyde groups for coupling with amine groups of the gel [49], or synthesized with 3-terminal amino C6 linker modifications for coupling to aldehyde containing gel pads as illustrated in Figure 1.9. A.
gel matrix O C NR 2 O C H N O 1.5'-Oligonucleotide-p-Link-NH 2 glass substrate R = H or CH 3 2. XBH 3 X = NaCN, Py or (CH3 )3N O C NR 2 O C H N Link-p-Oligonucleotide-5' NH
B.
p-oligonucleotide-5' O U me O gel matrix O C NR 2 O C H N NH 2 1. C O C O O C NR 2 O C
Figure 1.9. Coupling of amine-terminated (A) or dialdehyde-terminated (B) oligonucleotide probes to polyacrylamide gel matrix containing aldehyde or amino functional groups.
substrate
cross linker PDC
glass substrate
H N
O-p-oligonucleotide -5' N O U me
glass substrate R =H or CH 3
2. XBH 3 X = NaCN, Py or (CH 3 )3N
glass substrate
17
In the presence of a reducing reagent, XBH3 (sodium cyanoborohydride, pyridine-borane and trimethylamine-borane complexes), the oligonucleotide probes are immobilized to the gel matrix by the formation of covalent secondary amine bonds. Both the aldehyde and the amino matrices provide fast attachment in high yields, ~97%, ~74%, respectively [50]. The slightly lower efficiency of the amino matrix is probably caused by partial elimination of the 3dialdehyde moiety as pH rises in the course of reduction [50]. The probe loading density on the polyacrylamide gel pad has been estimated to be ~ 4.6 x 107 - 6.44 x 109 probes/m2 [48]. The loading density at the densest packing on glass surfaces is ~ 6.02 x 106 probes/m2 using the in situ synthesis method [51]. Oligonucleotide probes with their short chains and single points of constraint at each chain end may well be a far more accessible probe for hybridization. However, the variations in melting temperature due to AT-GC composition, and the reduction in specificity due to truncating the number of nucleotides from hundreds to as few as twenty are the disadvantages of using short-chain probes [28]. Applications of DNA Microarrays One of the early proposed applications for DNA microarrays was for sequencing by hybridization (SBH) in which a complete set of oligonucleotide species with N nucleotides in length (4N) are arrayed onto a solid substrate and used for large-scale, hybridization-based sequence analysis of target DNA [23] [52]. However, this wan not realized because of some inherent difficulties. To illustrate this, a complete set of three nucleotide probes for SBH is depicted in Figure 1.10. The probe sites that would hybridize to the illustrated 17-mer DNA target are highlighted with black circles. However, exchange of the first guanine with the first cytosine in the target DNA would give exactly the same result on the DNA array because of the
18
repeating TT sequence around these bases. Thus, it is difficult to analyze sequences containing short repetitive elements. Although this target could be uniquely identified using longer probes, even longer repeating sequences in genomic DNA might occur sufficiently often so that the chance of sequencing totally unknown targets is eliminated. Currently, the applications of DNA microarrays are focused in two areas: 1) determination of expression levels of genes [30] [5354]; and 2) identification of sequence variants [55-56].
Figure 1.10. An array designed for sequencing. All possible 3-mer probes (43=64) were arrayed on the substrate surfaces. The probe sites that would hybridize to the depicted 17-mer DNA target are highlighted with black circles. The number of n-mer probes that will hybridize to an m-mer target is m-n+1-r, where r is the number of repeats. In the illustration, there are three repeats (shown by connecting arrows) and so there are 17-3+1-3 = 12 hybridized probe sites [57]. Genes, which code for proteins, are transcribed into messenger RNAs (mRNAs) in the cell nucleus. The mRNAs in turn are translated into proteins by ribosomes in the cytoplasm. Gene expression analysis is used to probe the mRNA content of cells at a certain stage or under particular environment conditions. mRNAs are prone to being destroyed because of the ubiquitous presence of RNA-digesting enzymes. They are thus reverse-transcribed back into more stable DNA forms. The products of this reaction are called complementary DNAs (cDNAs) because their sequences are the complements of the original mRNA sequences. In order to detect
19
cDNAs, a reporter molecule is needed to label the cDNAs to identify their presence. The reporters currently used in comparative hybridization to microarrays are fluorescent dyes. Both oligonucleotide arrays and cDNA microarrays can be used for gene expression analysis. When oligonucleotide arrays are used for gene expression analysis, each gene must be represented by several (typically 20) different oligonucleotide probes because of the differences in hybridization properties and reduced hybridization specificity inherent in such short probes [58]. A redundancy of probes, the use of multiple oligonucleotides of different sequences designed to hybridize to different regions of the same RNA, were designed to improve signal-tonoise ratios, the accuracy of RNA quantitation, the dynamic range and to mitigate crosshybridization effects and reduce the rate of false positives and miscalls [55]. In addition, each oligonucleotide on the chip has a partner adjacent to it that differs at just one central base, which serves as an internal control for hybridization specificity. Thus, each gene has 40 features on an oligonucleotide chip, while it requires only one feature on a cDNA fragment microarray. Most microarray-based expression analyses do not measure absolute levels of RNA but rather compare RNA levels between two samples. As shown in Figure 1.11, mRNAs from vegetative and sporulating yeast cells are used to prepare fluorescently labeled cDNAs by reverse transcription in the presence of fluorescently labeled nucleotide precursors, a red fluorescent dye label for the mRNA from sporulating yeast and a green fluorescent dye label for the mRNA from the vegetative yeast cells. The two fluorescently labeled cDNAs are then mixed and hybridized with a DNA microarray in which each yeast gene is represented as a distinct spot of DNA. After hybridization, fluorescence from each dye is measured separately on each spot and these measurements are used to determine the ratio and in turn the relative abundance of the sequence of each specific gene in the two mRNA or DNA samples. For example, the greater
20
relative abundance of the TEP1 mRNA in the sporulating cells results in a high ratio of redlabeled to green-labeled copies of the corresponding cDNA.
vegetative yeast cell
sporulating yeast cell
Figure 1.11. Gene expression analysis using a cDNA microarray. mRNA samples from vegetative and sporulating yeast cells are compared. The total pool of mRNA from each cell population is used to prepare fluorescently labeled cCNA by reverse transcription in the presence of fluorescently labeled nucleotide precursors, a red fluor for the mRNA from sporulating yeast and a green fluor for the mRNA from the vegetative yeast cells. The two fluorescently labeled cDNAs are then mixed and hybridized with a DNA microarray in which each yeast gene is represented as a distinct spot of DNA. The cDNA sequences representing each individual transcript hybridize specifically with the corresponding gene sequence in the array. Thus, the relative abundance in sporulating as compared with vegetative yeast cells of the transcripts from each gene is reflected by the ratio of red to green fluorescence measured at the array element representing that gene. For example, the greater relative abundance of the TEP1 mRNA in the sporulating cells results in a high ratio of red-labeled to green-labeled copies of the corresponding cDNA. Affymetrix GeneChip arrays (www.affymetrix.com) and slide arrays from Incyte (www.synteni.com) are the commercially available microarray-based formats for expression analysis. The performance comparisons of these two formats are listed in Table 1.2. [59]. At the moment, both of these are relatively expensive and fairly limited in their applications for different expression analyses. 21
Table 1.2. Gene expression high-density microarrays [59].
Another primary application for DNA microarrays is identification and genotyping of mutations and polymorphisms. For the genotyping of known DNA variants, oligonucleotide arrays containing specific probes for each allele at many loci are used [5]. Allele specific hybridization detection for known genomic DNA sequence variations was first reported in 1989 [46]. Probes complementary to six HLA-DQA alleles as well as nine mutations in HBB (encoding -globin) were spotted onto nylon filters and incubated with biotin-labeled PCR products. A Colorimetric assay was used to detect the identity of the genotypes. With the
22
development of advanced oligonucleotide array manufacturing processes, arrays of 1480 oligonucleotide probes synthesized in situ by photolithographic-based processes were designed to detect 37 known mutations in the coding region of CFTR, as well as all possible singlenucleotide substitutions using fluorescence readouts [60].
Figure 1.12. A tiling array designed to detect a specific sequence. The probes on the same row have identical sequences except they differ on the center nucleotide. The four circled probes are tiled complements of the desired 12-mer target, and all four would hybridize if the illustrated target DNA is present. The probes in each row that are not circled are single-base variations of the circled probe and serve as hybridization controls [57]. A tiling array, as exemplified in Figure 1.12, in which each nucleotide in the target sequence is covered by four complementary oligonucleotide probes that differ only by an A, T, C or G substitution at the central position to provide the best discrimination for hybridization specificity, was synthesized at high density to scan a target sequence for unknown mutations [37] [60]. When it was necessary to screen both strands of N bp DNA target, 8N probes were needed to interrogate all possible single nucleotide substitutions. In Figure 1.12, only the probes corresponding to the four underlined nucleotides are shown. Oligonucleotide arrays are also utilized to scan a sequence of interest for all possible changes using enzyme-dependent mini-sequencing approaches (Figure 1.13) [61]. The oligonucleotide detection primers are attached through a 5 linkage to the array to leave an 23
exposed 3-OH group. These immobilized primers anneal to their template sequences immediately adjacent to the mutant nucleotide positions. A mixture of all four dideoxyribonucleoside triphosphates, each labeled with a different fluorescent dye, is used in subsequent enzymatic primer extension reactions to extend the hybridized primer probes [62]. The identity of the extended ddNTPs is determined through fluorescent microscopy and used to assign the identity of the target nucleotide extended from the 3 end of each probe. A minisequencing array designed to interrogate both target strands of N bp for all possible sequence changes minimally consists of 2N overlapping probes.
3 3 3 3
Oligonucleotide array Hybridize with unlabeled target
TddA
CddG
C ddG
TddA
Mini-sequencing analysis
Figure 1.13. Mini-sequencing analysis of genotypes. Oligonucleotide detection primers are attached through a 5 linkage to the array to leave an exposed 3-OH group. These surfaceimmobilized primers anneal to the sequence target immediately adjacent to the nucleotide of analysis. Fluorescently tagged ddNTPs are used in subsequent enzymatic primer extension reactions to extend the hybridized primer probes by one base. The identity of the extended ddNTPs is used in genotype analysis.
24
Limitations and Improvements for Array-based Techniques DNA microarrays can be configured to detect sequence variations at many different sites simultaneously and as such, can be used for diagnostic applications [63-65]. These arrays consist of thousands of short oligonucleotide probes (~20mers) with known sequence attached to twodimensional solid surfaces such as glass via siloxane chemistry or electrostatically to poly-lysine coated slides [51]. Some of the limitations associated with microarrays include the difficulty to simultaneously assay sequence tracts with localized regions of high G/C and A/T content (despite the use of buffer systems such as those containing tetramethylammonium chloride) [6667], the extended hybridization times required to elicit an optimal response (diffusion controlled passive hybridization), the instability of the linkage chemistry (hydrolytically unstable) and the large amount of sample required to interrogate the array. Due to the similarities in the hybridization thermodynamics (single base mismatched versus fully matched duplexed DNA with the mutant DNA being a minority) ambiguities result making it difficult to detect lowabundant point mutations using allele specific probes in an array format. A number of approaches have been proposed to normalize hybridization signals on all the arrayed probes. For example, chemically modified oligonucleotides have been exploited to endow the DNA duplexes with similar stability independent of their AT/GC ratio content and peptide nucleic acid arrays [68-69], uridine ribonucleoside triphosphates [70], and other modified triphosphates [71] were reported to promote the hybridization specificity and efficiency. Electrical fields have also been used to alter target affinity to specific probes and a sophisticated means of normalizing hybridization signal by coupling electrophoresis-based technology with hybridization has been reported [72-73].
25
Other approaches have been made to enhance the specificity of array hybridization. One involves coupling enzymatic ligation reactions to the hybridization scheme to maximize the destabilizing effect of mismatches at the end of duplexes [74]. Biotinylated duplex probes containing single-stranded 5-base 3' overhangs were immobilized on streptavidin-coated magnetic beads. A labeled single-stranded target was hybridized to the single-stranded region of the duplex probes and ligated in the presence of ligase enzyme if perfect match exist at the junction points. This provided enhanced discrimination of perfectly matched targets from those containing mismatches. Another involves the contiguous stacking hybridization approach [48] in which unlabeled target was hybridized to the oligonucleotide array and followed by the subsequent hybridization of sets of fluorescently tagged 5-base oligonucleotides to the immobilized duplex. Sequence information is derived from the identity of the short oligonucleotides, which hybridize to the target-probe complex. Microfluidic Arrays Although microarray technology is a very powerful tool, it still requires extensive sample preparation before the sample solution is applied to the probe array for hybridization analysis and detection [75]. A DNA array-based diagnosis system has to combine all steps necessary to perform analysis when applied in the field for clinical disease diagnosis. Integration of several different processes on a single chip to form a system for sample pretreatment, separation, and detection is the gist of the lab-on-a-chip concept [76]. The placement of a DNA array inside microfluidic channels to create microfluidic hybridization array channels can be an integral part to realize the potential of such microanalysis systems. Compared to conventional systems, the small sample volume required by the microfluidic array makes the sample manipulation more consistent and uniform across the surface of the
26
array, and hybridization sensitivity is expected to improve when the hybridization sample solution is oscillated inside the microfluidic channel [75]. The integration of the DNA array with the sample preparation unit is more efficient when the DNA array is in the microfluidic environment. Several reports have appeared, which have discussed microarrays and microfluidics on a single platform. Fan and co-workers used paramagnetic beads as a solid support for the
immobilization of target DNA via biotin-streptavidin linkages [77]. A microfluidic device constructed from glass was used to capture the beads magnetically and then interrogate them with fluorescently labeled probes. They were able to demonstrate that the hybridization signal reached a maximum in several seconds using pneumatic pumping and the beads containing the target DNA could be interrogated up to 12 times before observing a loss of signal. Benoit et al. developed three-dimensional flow through arrays in glass chips by fabricating circular channels (10 m id) that were 0.5-2.0 mm in height [78]. The authors were able to show detection limits for a hybridization assay of 40 pM for the target, which was a 15-mer labeled with 5carboxyfluorescein. Liu and co-workers fabricated a plastic (polycarbonate, PC) chip that
integrated PCR and a hybridization array to a monolithic device [79]. The microchips were fabricated using a CO2 laser and assembled by thermal and adhesive tape bonding. The
hybridization probes were tethered to the surface of the PC using a proprietary technique into a chamber that measured 1.2 mm x 5.0 mm x 0.25 mm. Using a 200 bp PCR generated product from E. coli, they determined that approximately 1 hr was sufficient for detection of the hybridization event. One potential problem associated with the use of PC is that it can generate a significant amount of autofluorescence, which can potentially lower detection limits when using fluorescence readout.
27
Research Focus
This research will focus on the development of polymer-based oligonucleotide arrays aimed at gene mutation diagnosis. A protocol for building oligonucleotide arrays on flat PMMA surfaces and into microfluidic channels hot-embossed in PMMA substrates will be established. The hybridization results of near-IR dye-labeled oligonucleotides to its complement probes immobilized to PMMA surfaces in microfluidic channels will be monitored with a home-built near-IR fluorescence scanner. The approaches to transport oligonucleotide solutions through the microfluidic devices will be researched. The application of PMMA microfluidic arrays for detecting low abundant mutations in K-ras gene fragments (c12.2) will be covered. Finally, the possibility of photo-patterning with UV-light for the immobilization of oligonucleotide molecules on polymer substrates will be discussed.
References
1. 2. 3. 4. 5. Marshall, A., Hodgson, J., Nat. Biotechnol., 1998, 16(1), 27-31. Ramsay, G., Nat. Biotechnol., 1998, 16(1), 40-44. Schena, M., Shalon, D., Davis, R.W., Brown, P.O., Science, 1995, 270(5235), 467-470. Marx, J., Science, 2000, 289(5485), 1670-1672. Wang, D.G., Fan, J.B., Siao, C.J., Berno, A., Young, P., Sapolsky, R., Ghandour, G., Perkins, N., Winchester, E., Spencer, J., Kruglyak, L., Stein, L., Hsie, L., Topaloglou, T., Hubbell, E., Robinson, E., Mittmann, M., Morris, M.S., Shen, N.P., Kilburn, D., Rioux, J., Nusbaum, C., Rozen, S., Science, 1998. 280(5366), 1077-1082. Watson, J.D., Crick, F.H., Nature, 1953, 248, 737-738. Lewin, B., Genes. 1997, Oxford University Press, Oxford. Durrant, I., in Methods in Molecular Biology, Harwood, A.J. Editor. 1996, Humana Press, Totowa, New Jersey, 155-168.
6. 7. 8.
28
9.
Durrant, I., Cunningham, M. in Gene Probes 1: A Practical Approach, Hames, B.D., Higgins, S.J., Editor. 1995, IRL Press, Oxford, 189-210. Marmur, J., Rownd, R., Schildkraut, L., Prog. Nucleic Acid Res., 1963, 1, 232-300. Brenner, S., Jacob, F., Meselson, M., Nature, 1961, 190, 576-581. Jacob, F., Monod, J., J. Mol. Biol., 1961, 3, 318-356. Crick, F.H.C., Barnett, L., Brenner, S., Watts-Tobin, R.J., Nature, 1961, 192, 1227-1232. Tilghman, S.M., Tiemeier, D.C., Seidman, J.G., Peterlin, B.M., Sullivan, M., Maizel, J.V., Leder, P., Proc. Natl. Acad. Sci. USA, 1978, 75(2), 725-729. Stryer, L., Biochemistry, 4th ed. 1995, W.H. Freeman and Co., New York. Benzer, S., Champe, S.P., Proc. Natl. Acad. Sci. USA, 1961, 47, 403-416. Davies, J.E., Gassen, H.G., Angew. Chem. Int. Edit., 1983, 22(1), 13-31. Bankier, A.T., Barrell, B.G., Techniques in nucleic acid biochemistry, 1983, Elsevier Scientific Publishers Ireland. Gait, M.J., An Introduction to Modern Methods of DNA Synthesis, in Oligonucleotide synthesis a practical approach, Gait, M.J. Editor, 1984, IRL Press, Oxford Washington DC. Fodor, S.P., Read, J.L., Pirrung, M.C., Stryer, L., Lu, A.T., Solas, D., Science, 1991, 251, 767-773. Blanchard, A.P., Kaiser, R.J., Hood, L.E., Biosensors &amp; Bioelectronics, 1996, 11, 687690. Southern, E.M., J. Mol. Biol., 1975, 98(3), 503-517. Khrapko, K.R., Lysov, Y.P., Khorlyn, A.A., Shick, V.V., Florentiev, V.L., Mirzabekoc, A.D., FEBS Letters, 1989, 256(1-2), 118-122. Phimister, B., Nat. Genet., 1999, 21, 1-1. Maskos, U., Southern, E.M., Nucleic Acids Res., 1993, 21, 2267-2268. Saiki, R.K., Gelfand, D.H., Stoffel, S., Scharf, S.J., Higuchi, R., Horn, G.T., Mullis, K.B., Erlich, H.A., Science, 1988, 239(4839), 487-491. Schulze, A., Downward, J., Nat. Cell Biol., 2001, 3(8), E190-E195.
10. 11. 12. 13. 14.
15. 16. 17. 18.
19.
20.
21.
22. 23.
24. 25. 26.
27.
29
28.
Duggan, D.J., Bittner, M., Chen, Y.D., Meltzer, P., Trent, J.M., Nat. Genet., 1999, 21, 10-14. Winzeler, E.A., Schena, M., Davis, R.W., Methods Enzymol., 1999, 306, 3-18. Lockhart, D.J., Dong, H.L., Byrne, M.C., Follettie, M.T., Gallo, M.V., Chee, M.S., Mittmann, M., Wang, C.W., Kobayashi, M., Horton, H., Brown, E.L., Nat. Biotechnol., 1996, 14(13),1675-1680. Southern, E.M., J. Mol. Biol., 1976, 98, 503-517. Guo, Z., Guilfoyle, R.A., Thiel, A.J., Wang, R., Smith, L.M., Nucleic Acids Res., 1994, 22, 5456-5465. Nickerson, D.A., Kaiser, R., Lappin, S., Stewart, J., Hood, L., Landegren, U., Proc. Natl. Acad. Sci. USA, 1990, 87, 8923-8927. Bej, A.K., Mahbubani, M.H., Miller, R., DiCesare, J., Haff, L., Atlas, R.M., Mol. Cell. Probes, 1990, 4, 353-365. Lakowicz, J.R.,Thompson, R.B., Proc. SPIE-Int. Soc. Opt. Eng., 1993, 1885, 480. Wu, D.Y., Wallace, R.B., Itakura, K., Genomics, 1989, 4(4), 560-569. Lipshutz, R.J., Fodor, S.P., Gingeras, T.R., Lockhart, D.J., Nat. Genet., 1999, 21, 20-24. McGall, G.H., Barone, A.D., Diggelmann, M., Fodor, S.P.A., Gentalen, E., Ngo, N., J. Amer. Chem. Soc., 1997, 119(22), 5081-5090. Pirrung, M.C., Fallon, L., McGall, G., J. Org. Chem., 1998, 63(2), 241-246. Matson, R.S., Rampal, J.B., Coassin, P.J., Anal. Biochem., 1994, 217, 306-310. Schena, M., Shalon, D., Heller, R., Chai, A., Brown, P.O., Davis, R.W., Proc. Natl. Acad. Sci. USA, 1996, 93(20), 10614-10619. Cheung, V.G., Morley, M., Aguilar, F., Massimi, A., Kucherlapati, R., Childs, G., Nat. Genet., 1999, 21, 15-19. Lund, V., Schmid, R., Rickwood, D., Hornes, E., Nucleic Acids Res., 1988, 16, 1086110880. Beattie, K.L., Fowler, R.F., Nature, 1991, 352, 548-549.
29. 30.
31. 32.
33.
34.
35. 36. 37. 38.
39. 40. 41.
42.
43.
44.
30
45.
Rasmussen, S.R., Larsen, M.R., Rasmussen, S.E., Anal. Biochem., 1991, 198(1), 138142. Saiki, R.K., Walsh, P.S., Levenson, C.H., Erlich, H.A., Proc. Natl. Acad. Sci. USA, 1989. 86(16), 6230-6234. Ganachaud, F., Elaissari, A., Pichot, C., Laayoun, A., Cros, P., Langmuir, 1997, 13(4), 701-707. Yershov, G., Barsky, V., Belgovskiy, A., Kirillov, E., Kreindlin, E., Ivanov, I., Parinov, S., Guschin, D., Drobishev, A., Dubiley, S., Mirzabekov, A., Proc. Natl. Acad. Sci. USA, 1996, 93(10), 4913-4918. Khrapko, K.R., Lysov, Y.P., Khorlin, A.A., Ivanov, I.B., Yershov, G.M., Vasilenko, S.K., Florentiev, V.L, &amp; Mirzabekov, A.D., DNA Sequencing, 1991, 1, 375-388. Timofeev, E., Kochetkova, S., Mirzabekov, A, and Florentiev, V., Nucleic Acids Res., 1996, 24, 3142-3148. Southern, E., Mir, K., Shchepinov, M.S., Case-Green, S.C., Southern, E.M., Nat. Genet., 1999, 21, 5-9. Bains, W., Smith, G.C., J. Theor. Biol., 1988, 135(3), 303-307. Zhu, H., Cong, J.P., Mamtora, G., Gingeras, T., Shenk, T., Proc. Natl. Acad. Sci. USA, 1998, 95(24), 14470-14475. de Saizieu, A., Certa, U., Warrington, J., Gray, C., Keck, W., Mous, J., Nat. Biotechnol., 1998, 16(1), 45-48. Hacia, J.G., Nat. Genet., 1999, 21, 42-47. Chakravarti, A., Nat. Genet., 1999, 21, 56-60. Whitaker, T.J., Jacobson, K.B., DNA arrays: preparation and application, in Encyclopedia of Analytical Chemistry: applications, theory, and instrumentation, Meyers, R.A., Editor, 2000, New York, Wiley, 1219-1235. Johnston, M., Curr. Biol., 1998, 8(5), R171-174. Bowtell, D.D.L., Nat. Genet., 1999, 21, 25-32. Cronin, M.T., Fucini, R.V., Kim, S.M., Masino, R.S., Wespi, R.M., Miyada, C.G., Hum. Mutat., 1996, 7(3), 244-255.
46.
47.
48.
49.
50.
51.
52. 53.
54.
55. 56. 57.
58. 59. 60.
31
61.
Pastinen, T., Kurg, A., Metspalu, A., Peltonen, L., Syvanen, A.C., Genome Res., 1997, 7(6), 606-614. Nikiforov, T.T., Rendle, R.B., Goelet, P., Rogers, Y.H., Kotewicz, M.L., Anderson, S., Trainor, G.L., Knapp, M.R., Nucleic Acids Res., 1994, 22(20), 4167-4175. Consolandi, C., Castiglioni, B., Bordoni, R., Busti, E., Battaglia, C., De Bellis, G., Minerva Biotechnol., 2001, 13(4), 261-268. De Benedetti, V.M.G., Biglia, N., Sismondi, P., De Bortoli, M., Int. J. of Biol. Marker, 2000, 15(1), 1-9. Dimitrjevic, B., Jugoslav. Med. Biohem., 2001, 20, 65-71. Chee, M., Yang, R., Hubbell, E., Berno, A., Huang, X.C., Stern, D., Winkler, J., Lockhart, D.J., Morris, M.S., Fodor, S.P.A., Science, 1996, 274(5287), 610-614. Hacia, J.G., Brody, L.C., Chee, M.S., Fodor, S.P.A., Collins, F.S., Nat. Genet., 1996, 14(4), 441-447. Weiler, J., Gausepohl, H., Hauser, N., Jensen, O.N., Hoheisel, J.D., Nucleic Acid Res., 1997, 25(14), 2792-2799. Arlinghaus, H.F., Kwoka, M.N., Jacobson, K.B., Anal. Chem., 1997, 69(18), 3747-3753. Hacia, J.G., Woski, S.A., Fidanza, J., Edgemon, K., Hunt, N., McGall, G., Fodor, S.P.A., Collins, F.S., Nucleic Acids Res., 1998, 26(21), 4975-4982. Nguyen, H.K., Bonfils, E., Auffray, P., Costaglioli, P., Schmitt, P., Asseline, U., Durand, M., Maurizot, J.C., Dupret, D., Thuong, N.T., Nucleic Acids Res., 1998, 26(18), 42494258. Sosnowski, R.G., Tu, E., Butler, W.F., OConnell, J.P., Heller, M.J., Proc. Natl. Acad. Sci. USA, 1997, 94, 1119-1123. Edman, C.F., Raymond, D.E., Wu, D.J., Tu, E. Sosnowski, R.G. Butler, W.F., Nerenberg, M. Heller, M.J., Nucleic Acids Res., 1997, 25, 4907-4914. Broude, N.E., Sano, T., Smith, C.L., Cantor, C.R., Proc. Natl. Acad. Sci. USA, 1994, 91, 3072-3076. Liu, Y.J., Rauch, C.B., Anal. Biochem., 2003, 317(1), 76-84. Manz, A., Graber, N., Widmer, H.M., Sensors, and Actuators B-Chemical, 1990, 1(1-6), 244-248.
62.
63.
64.
65. 66.
67.
68.
69. 70.
71.
72.
73.
74.
75. 76.
32
77.
Fan, Z.H., Mangru, S., Granzow, R., Heaney, P., Ho, W., Dong, Q.P., Kumar, R., Anal. Chem., 1999, 71(21), 4851-4859. Benoit, V., Steel, A., Torres, M., Lu, Y.Y., Yang, H.J., Cooper, J., Anal. Chem., 2001, 73(11), 2412-2420. Liu, Y.J., Rauch, C.B., Stevens, R.L., Lenigk, R., Yang, J.N., Rhine, D.B., Grodzinski, P., Anal. Chem., 2002, 74(13), 3063-3070.
78.
79.
33
Chapter 2 Laser-Induced Near-IR Fluorescence Detection*
Introduction
Fluorescence is the emission of photons occurring when molecules return from excited singlet states to the ground state and this detection technique has wide applications in biology and biochemistry [1-2]. In order for the fluorescence to occur, an external excitation light source is needed to promote the chromophore molecules from the ground state to the excited states. With the development of laser technology, more and more lasers have been used as the excitation light source instead of conventional incandescent light sources due to the spectral purity and the variable wavelengths they provide [3]. In fluorescence detection, not only the emissions from sample targets but the sample matrix with similar spectral emissions and the scatterings from the light source will contribute to the photons collected by the detector. These are the background sources that degrade the detection limit. The fluorescence measurement in the near-IR region (600-1000 nm) is proposed to reduce the background because the intensity of the scattered light is inversely proportional to the fourth power of the excitation wavelength and only a few classes of molecules exhibit significant absorption or fluorescence in these spectral areas. Therefore, high detection sensitivity is expected. It was reported that single molecule detection limits have been realized by labeling target molecules with near-IR tricarbocynine fluorescing dyes [4-5]. In this chapter, the basic principles of fluorescence and the performance of a home-built near-IR fluorescence scanner will be discussed. * Reproduced in part with permission from Analytical Chemistry, volume 72, number 24, pages 5907-5917, 2000.
Copyright 2000 American Chemical Society.
34
Jablonskis Diagram Fluorescence is the emission of photons by atoms or molecules whose electrons are transiently excited to a higher energy level by a radiant light source. The related absorption and emission processes are graphically depicted by the Jablonski diagram as illustrated in Figure 2.1 [6]. The stack of horizontal lines in the graph shows a series of increasing energy levels in which S0 represents the electronic ground state, S1, and S2 denotes the first and second singlet excited states, respectively, and T1 is the first triplet excited state. At each of these electronic energy states there exist a number of subdivisions of vibrational energy levels, which is denoted by o, 1, 2, 3 .When a fluorescent molecule absorbs photons of the appropriate wavelength, some of the electrons in the ground state are excited to higher electronic energy levels at various vibrational states and this process can be completed on a 10-15 s timescale. Generally, these excited state electrons relax to the lowest vibrational level of S1 in 10-12 s or less and from there they return to the ground state (S0) with the accompanying emission of photons. This process is called fluorescence emission and the period from absorption of light to emission of fluorescence has a timescale of 10-8 to 10-12 s. Besides the fluorescence emission, it is also possible for the electrons in the S1 state to undergo a spin conversion (intersystem crossing) to move into the first excited triplet state (T1) and return to the ground state with the accompanying photon emission known as phosphorescence. As a consequence of being spin-forbidden, the period between excitation and phosphorescence emission is longer than that for fluorescence and can last fractions of a second to several minutes. The vertical lines in the graph denote transitions that occur on a very short timescale and can be considered as virtually instantaneous while the nonvertical lines depict transitions that occur on a longer timescale.
35
S2
3 2 1 0
Vibrational relaxation 3 2 1 0
Internal conversion
S1
Intersystem crossing Absorption 10-15s Phosphorescence 10-3-102 s
3 2 1 T1 0
S0
3 2 1 0
Absorption
Fluorescence
Fluorescence 10-9-10-12 s
Phosphorescence
Figure 2.1. Jablonski energy level diagram. The electronic energy levels of various ground and excited states, the timescales of the absorption, conversion, and emission are depicted in the graph [6]. As depicted in Figure 2.1, some energy is lost before the fluorescence occurs, and thus, the emitted fluorescent photons typically exhibit a longer wavelength than the excitatory photons. The difference in wavelength or energy between the excitation and emission photons is called the Stokes Shift. The shift can range from just a few to several hundred nanometers depending on the particular fluorescent molecule. Phosphorescence is generally shifted to longer wavelengths relative to fluorescence because T1 is lower in energy than S1. For a given population of molecules in an excited state, not all will return to the ground state by a fluorescence emission. The quantum yield (Q) is the number of photons emitted relative to the number of photons absorbed. It can be expressed as the ratio of the emissive rate of the fluorophore () to the total depopulation rate, which is the sum of the rate of emissive radiation () and the nonradiative decay (knr) [7]:
36
Q=
  + k nr
(2.1)
The quantum yield can be close to unity if the radiationless decay rate is much smaller than the rate of radiative decay, that is, knr &lt;&lt; . The quantum yield is dependent on the molecule structures and the environment. For example, rhodamines have quantum efficiencies approaching unity and thus display bright emission. The quantum yield is typically higher at lower temperatures and in more viscous solvents. Near-Infrared Fluorescence When fluorescence is used as the readout mode, the surrounding host will invariably generate a signal due to autofluorescence from impurities. The adoption of near-IR fluorescence technique has been demonstrated as a viable strategy to reduce the contributions of these sources to the background [8-10].
Gamma X-ray UV visible nearIR infraIR Far-far IR Microwave TV, FM, Radiofrequency
10-10
10-9
10-8
10-7
10-6
10-5
10-4
10-3
10-2
10-1 100
101
102
103 m
650
1100
2500 nm
Figure 2.2. Position of the near-IR region in the electromagnetic spectrum [11]. As shown in Figure 2.2, near-IR is defined as the regions of the electromagnetic spectrum with wavelengths between 650-2500 nm [11]. It is further divided into two ranges of 650-1100 nm and 1100-2500 nm based on the different solid-state detectors that are used for detection. For example, silicon avalanche diode detectors have a range from 650 to 1100 nm, and lead sulfidebased detectors from 1100 to 2500 nm.
37
Typical absorption and fluorescence regions for biomolecules and fluorophores are illustrated in Figure 2.3 [11]. As can be seen, most biomolecules and impurities absorb and fluoresce between 190 and 650 nm which coalesce with the absorption/emission regions (400650 nm) of most commonly used fluorophores [10-11]. Thus, fluorescence detections in the UVVisible regions are susceptible to biological interferences from the sample matrix. In contrast, a limited number of components show intrinsic fluorescence in the near-IR region, therefore, nearIR fluorescence has the potential to offer reduced background matrix interferences compared to UV-Visible fluorescence [12-14]. In addition, the spectral interferences, Rayleigh scatterings, are inversely proportional to 4. As a result, high detection sensitivity is expected for near-IR fluorescence due to background reduction. It has been demonstrated that improved detection limits were obtained with the use of near-IR fluorescence detection compared to visible excitation [12]. A detection limits of 10-12 M [13] or even 10-18 M [14] for polymethine dyes and even single molecules for near-IR tricarbocyanine fluorescing dyes, IR-132 and IR-140 [4-5], have been reported. Visible Fluorophores Near-IR Dyes Biomolecules 200 400 600 800 1000 nm
Figure 2.3. Absorbance and fluorescence of biomolecules in relation to visible and near-IR fluorophores [11]. The development of inexpensive near-IR diode lasers (780-830 nm) and photodiode detectors for the near-IR spectral region provide lower initial capital investment as well as enhanced reliability of near-IR instrumentation. The smaller footprint affordable by solid state
38
optics and the power supplies associated with them make the near-IR instrumentation easily portable to allow flexibility for dynamic needs of the lab [15]. There are many applications for near-IR fluorescence detection in the bioanalytical field. These include DNA sequencing [16-18], DNA fragment analysis with capillary [19] or slab gel electrophoresis using two-dimensional scanners [20], fluorescence lifetime instrumentation [2122] and fluorescence microscopy [23]. Near-IR Labeling Dyes With fluorescence detection, a fluorescent chromophore is normally coupled either covalently or non-covalently to a non-fluorescent molecule and the fluorescence emitted is detected. This is true for near-IR fluorescence detection since few biomolecules fluoresce intrinsically in this region [24]. Although several hundred infrared absorbing and emission dyes are now in existence [25], few of them can be used directly for the labeling of biomolecules due to the lack of proper functional groups and their poor water solubility. At present, IRDye700 and IRDye800 series near-IR dyes with functional groups that can be conjugated to DNAs and proteins are commercially available (LI-COR, Lincoln, NE) and the development of new water soluble dyes are still in progress [26-27]. The structures and the related characteristics of several near-IR dyes are listed in Table 2.1. As listed in Table 2.1, near-IR dyes generally come from the cyanine, phthalocyanine and naphthalocyanine classes [25]. Tricarbocyanine-based compounds have been one of the most widely investigated near-IR dyes for covalent and noncovalent labeling of biomolecules. They possess large molar absorptivities (&gt; 105 M-1cm-1) and display favorable solubility in aqueous solvents [20]. However, they have poor chemical and photo-chemical stabilities and low
39
quantum yields in aqueous media. Thus, their utility in ultra-sensitive applications is limited [4] [28]. Phthalocyanine and naphthalocyanine compounds are alternatives to the cyanine-based near-IR fluorophores. They show superior chemical and photo-chemical stabilities and have large extinction coefficients (&gt;105 M-1cm-1) as well as favorable quantum yields [29]. The absorbance and fluorescence maxima can easily be altered by varying the substituents around the ring or by changing the metal center as shown in Table 2.1. Functionalization of the periphery of the dye with charged groups (SO3-, CO2-, PO32-, etc.) or attachment of a polar group to the metal center induces water-solubility of phthalocyanine dyes [30]. Two new water-soluble phthalocyanine dyes containing isothiocyanate [31] or carboxylic [32] functional groups for covalent labeling of oligonucleotides were recently reported. A series of near-IR fluorescent dyes with similar absorption and emission maxima but distinctive fluorescence lifetimes were also developed for fluorescence lifetime-based detection [9] [33]. Table 2.1. The structures of some representative near-IR fluorescent dyes and their absorption and emission properties. Extinction Quantum abs em coefficient efficiency Dyes Structures (nm) (nm)  (M-1cm-1)  (%)
N=C=S
Tricarbocyanine Dyes
NH
X O NH
765
794
&gt;105
poor
O 3S
X=H, I, Br, Cl, F
O 3S
40
Table 2.1. (continued). Dyes Structures
S
abs em (nm) (nm)
Extinction coefficient  (M-1cm-1)
Quantum efficiency  (%)
N
N
N
N
S
NMN
Phthalocyanine Dyes
S
N
N
600 ~ 700
700 ~ 780
&gt;105
favorable
L M - metal center Zn, Sn, Al, Si, etc. S - water soluble groups, SO 3 -, CO 2-, PO 3 2 L - reactive group, -N=C=S
O
IRD 800
N
N
787
SO 3
812
275000
15
OH
N
N
IRD 700
SO 3 OH
681
712
170000
47.7
The commercial near-IR dyes, IRDye700 and IRDye800 series, with carbodiimide (CDI) active groups that can covalently be bonded to thymine (T) and guanine (G) on single and double stranded nucleic acids, were designed to have maximum absorption wavelengths closely
41
matched to commercially available diode lasers with exciting wavelengths of 685 nm and 785 nm, respectively (www.licor.com). The labeled nucleic acids can be used as probes for DNA or RNA hybridization experiments to detect DNA and RNA in low abundance. Confocal Fluorescence Microscope and Scanner The optical composition of a confocal fluorescence microscope is illustrated in Figure 2.4. In an epi-illumination microscope, the incident light is reflected by a dichroic mirror and focused by an objective onto the detection plane and the resulting fluorescence from the excitation is collected by the same objective and passed through the dichroic mirror and detected by a photon detector.
Detector
Confocal aperture
Dichroic mirror Light source
Objective
Focal plane Specimen
Figure 2.4. The confocal principle in an epi-illumination microscope. The photons from the excitation light source are reflected by the dichroic mirror and focused by the objective onto the sample. The emitted photons from both the focal plane and out-of-focus voxel are collected by the same objective and transmitted through the dichroic mirror. Only the photons from the focal plane can pass the confocal aperture located in front of the detector and be detected while others from the out-of-focus voxel will be eliminated by the aperture. Thus, in-depth resolution is obtained [34].
42
To prevent fluorescence emissions or reflections from regions above and below the focal plane from contributing to the collected photons, an optical aperture is inserted into the optical path as illustrated in Figure 2.4. Only emission from the in-focus voxel is able to pass unimpeded through the confocal aperture and detected by the photon detector. This endows the confocal microscope with significant axial resolution and the capability for direct non-invasive serial optical sectioning of intact and even living specimens [34]. In a fluorescence microscope, the illumination of the specimen is limited to a single point (&lt;0.1 mm in diameter). To increase the area of detection, a fluorescence scanner was developed to produce a complete image of relatively larger regions, for example, several cm2 [35]. The first commercial confocal laser scanning fluorescence microscopy systems were produced in the mid 1980s by Bio-Rad Microscience Ltd. More choices are now available for use specifically with DNA microarray techniques. Several commercial fluorescence scanners are listed in Table 2.2. Table 2.2. The characteristics of some commercial fluorescence scanners for reading DNA microarrays Fluorescence Dynamic Scanning Manufacturer Light source Detection limit scanner range time SH-YAG laser DNA 8 mins (532 nm) Microarray Agilent 104 for 25 mm x 75 0.05 fluor/m2 HeNe laser (633 Scanner Technologies mm glass slide nm) 1.0 fmole fluorescein and 2 mins for 20 HEX end-labeled Amersham FluorImager Ar+ laser 5 10 mm x 25 mm DNA (23-mer) Biosciences 595 488 nm, 514 nm slide primer in 15% Polyacrylamide gel GenePix 532 nm, 4000B Axon 5 min for 22 x 0.1 fluor/m2 104 635 nm Instruments Microarray 73 mm slide Cy 3 and Cy 5 Scanner Array WORxe Applied 0.1 fluor/m2 Automated 350-700 nm &gt;104 NA Precision, LLC Cy 3 and Cy 5 Biochip Reader
43
These scanners have large dynamic ranges and low detection limits with some even equipped with dynamic auto-focusing for multiple slide loading capacity. However, most of them use near-Visible excitation and fluorescence emission. Microscope scanners can be divided into two fundamentally distinct types: image-plane scanning and object-plane scanning [34]. In the first format, the two-dimensional (2D) optical image of the specimen is focused onto the image plane of an electronic imaging device, such as the faceplate of a video camera image tube, or the 2D semiconductor imaging array of a charge coupled device (CCD) camera. After simultaneous exposure of all areas of the electronic imaging device to the incoming optical image, the information is read out in serial electronic form by the raster scanning of the faceplate or the sequential charge transfer of the CCD array as a series of sequential image lines [34]. In the second format, the 2D optical image of the specimen is not formed within the microscope. An objective of high numerical aperture is used to focus a beam of light into the microscopes focal plane within the specimen. In the epi-illumination configuration, the same objective is used in conjunction with an appropriate beam splitter or dichroic mirror to image the reflected light or fluorescence emission from the specimen onto a photodiode or photomultiplier tube (PMT) light detector. The scanning rate is often significantly slower than conventional video. However, the acquired image is easy to digitize on the fly and accumulate in a digital image memory, from which it may be read out and displayed at video rates or subjected to subsequent digital image processing [34]. Excitation Sources
Lamps
The high-pressure Xenon arc lamp is one of the most versatile excitation light sources utilized for fluorescence before the advent of lasers. It provides continuous light output from 270 44
to 700 nm with a number of sharp lines near 450 nm [36]. The high pressure mercury lamp is more intense but only emits at specific wavelengths. Flash lamps typically produce lamp pulses of a few nanoseconds and the spectral output depends on the type of gas that is used. Nitrogen and hydrogen are the most commonly used gases when shorter wavelengths are needed [7].
Lasers
Laser is an acronym for Light Amplification by Stimulated Emission of Radiation. The components of a typical laser source are shown schematically in Figure 2.5.
Figure 2.5. Schematic representation of a typical laser source [37]. It is composed of a lasing medium, a pair of mirrors, and pumping source. The pumped emissions from the lasing medium is amplified and collimated between the pair of mirrors and a portion of the amplified light is released by a partially transmitting mirror to serve as the laser beam. It consists of a lasing medium, a pair of mirrors, and a pumping source. The lasing medium is activated by the pumping source, which can be the radiation from an external source or an electrical current, or an electrical discharge. The radiation produced by the lasing action is caused to pass back and forth through the medium numerous times between the pair of mirrors and during the passages, the radiation is amplified and collimated. One of the mirrors is made partially transparent so that a fraction of the beam is transmitted rather than reflected. The output of a laser can be characterized as being coherent, collimated and monochromatic. That is, the emitted photons have specific quantum mechanically allowed single wavelength and move in the same direction and in phase with one another.
45
Table 2.3. Characteristics of commonly used lasers Laser Types Laser Names Lasing Medium 512 % He 1% Ne Ar+ KrF* XeCl* CO2, He, N2 Pure N2 Nd3+/Y3Al5O12 0.1% Ti3+/Al2O3 GaAs ZnSe Ga 0.5 In 0.5P In 0.2Ga0.8 Ga (1-x) AlxAs In(1-x)GaxAsyP(1-y) PbEuSeTe PbSSe PbSnTe PbSnSe Tuning Range or Principal Emission  632.8 nm 1.152 m 3.391 m 488 nm 514.5 nm 248 nm 308 nm 9.6 m 10.6 m 337.1 nm 1.064 m 650-1100 nm 904 nm 525 nm 670 nm 980 nm 620-895 nm 1100-1165 nm 3.3-5.8 m 4.2-8.0 m 6.3-29.0 m 8.0-29.0 m Output Power
He-Ne Argon Ion Gas Laser Excimer Carbon Dioxide Nitrogen Solid Laser Nd:YAG Ti: Sapphire Gallium Arsenide Diode Laser Diode Laser Lead Salt Diode Lasers
1-5 mW mW-W 1-100 W Up to 10 kW 250 kW peak power ~ 100W ~W
~mW
~mW
The lasing medium is the heart of a laser. The emitted wavelength of a laser is defined by the lasing medium and the cavity length of the laser, which may range from sub-millimeters to meters [36]. The lasing medium varies from simple monatomic gases to solid-state materials. Three major categories of lasers are commonly identified as gas lasers, solid-state lasers, and semiconductor lasers and an array of the commonly used lasers are listed in Table 2.3 [3]. These lasers cover most of the electromagnetic spectrum and some of them have the capacity to operate in both pulsed-mode or continuous wavelength (CW) mode. Of these, diode
46
lasers will be introduced in more detail because of its relevant applications in near-IR fluorescence detection. The lasing medium in a diode laser is a p-n junction integrated in a complex multilayer structure composed of semiconductors with variable compositions. The p-n junction is created by mounting parallel layers of p-type and n-type semiconductors at a separation distance of about 1
m. A population inversion and stimulated emission are effected with the application of a
forward biasing potential across the p-n junction (a negative voltage is applied to the n-type semiconductor and a positive voltage is applied to the p-type material). By placing layers with lower refractive indices adjacent to the p-n junction, a cavity is formed to provide light amplification. The relations between the emitted wavelength () and the length of the cavity (L) can be expressed as L= n/2   [3]. Most diode lasers have wavelengths in the near-IR region and some of the more popular wavelengths are 680, 780, 850, and 1550 nm [38]. The output power of a near-IR laser ranges from a few milli-watts to tens of milli-watts. Single Photon Avalanche Diode Detectors Single photon avalanche diode detectors (SPAD) are semiconductor-based photodetectors. When a piece of donor-doped (n-type) semiconductor material and another piece of acceptor-doped (p-type) semiconductor material are put into contact, a p-n junction is formed. Because of diffusion, electrons flow from the n region to the p region and holes from p to n region, leaving the immobile donor ions and acceptor ions behind to build up an internal electric field around the p-n junction. This potential barrier prevents further gross migration of the carriers. When a reverse bias is applied by an external circuit, almost all the majority mobile carriers in an area close to the p-n junction are swept by the bias electric field until the internal
47
electric field built up by the donors and acceptors reaches a sufficient level to cancel the bias field at the junction. Such an area is called the depletion area. If the minority carriers, that is, holes in n-type materials and electrons in p-type materials, happen to wander into the vicinity of the p-n junction, they are immediately accelerated by the internal field and migrate towards the opposite regions to form the drift current crossing the junction. The drift current under reverse bias is very small, generally being of the order of microamperes (A). However, the drift current can be greatly increased upon illumination with optical radiation. The current that is produced is proportional to the incident light intensity and exhibits excellent linearity over six to seven decades of incident radiation power [39]. By increasing the reverse bias voltage across the p-n junction, the field in the depletion area can be increased to a point at which carriers (electrons or holes) that are accelerated across the depletion area can gain enough kinetic energy to stimulate new electrons from the valence to the conduction band. These second-generation carriers move under the influence of the field and can create third-generation carriers. This process is referred to as avalanche multiplication. The current is given by [40];
i = eM (
I ) h
(2.2)
where, e is the energy of the electron;  is the quantum efficiency, which is defined as the probability of the excitation of a carrier by an incident photon; h is Plancks constant, h is the energy of a single photon; I is the optical beam intensity; and M the number of new carriers generated for each primary carrier created by the absorbed photon with typical values ranging from 20 to 100. The avalanche process greatly enhances the sensitivity of the detector. However, arbitrarily large values of M, which depends on the applied field, should be avoided, since the avalanche multiplication also contributes additional noise.
48
SPAD detectors used for near-IR studies are often cooled to below room temperature in order to reduce the noise. Noise is also proportional to the active surface area. In the case of SPADs, the surface areas are typically in the neighborhood of 10-100 m in diameter. The small active surface area also significantly reduces the detectors sensitivity. Thus, high-powered laser source may be needed for accurate quantification.
Research Focus
As stated previously, most of the commercially available DNA microarray readers use the excitation and emission wavelengths that fall within the visible regions. Therefore, a near-IR fluorescence scanner was built in this Lab and the operation optimization of the near-IR scanner was conducted in this Chapter. The appropriate excitation laser power was determined by monitoring the fluorescence intensity and the background with the excitation laser power, the proper scanning steps and integration times per pixel were also evaluated. Finally, the photostability of near-IR dyes was examined at the optimized excitation laser power.
Experimental Section
Reagents and Materials
A 5 terminal amine modified oligonucleotide probe (5-C6 Amino TTT TTT TTT TTT TTT GTC GTT TTA CAA CGT CGT G-3) used for covalent immobilization onto poly(methylmethacrylate) (PMMA) substrate surfaces was obtained from the Midland Certified Reagent Company (Midland, TX). Its near-IR dye-labeled complement, M13 Fwd (-29)-IRD 800 (5-IRD 800 CAC GAC GTT GTA AAA CGA C-3), was obtained from LI-COR Biosciences (Lincoln, NE). Twenty x SSPE buffer (a saline phosphate buffer containing 0.2 M sodium phosphate, 2.98 M NaCl and 0.02 M EDTA with pH ~ 7.4) was from Aldrich (Milwaukee, WI)
49
and used without purification. PMMA sheets (1.0 mm thick) were purchased from GoodFellow (Berwyn, PA).
Instrumentation
A near-IR fluorescence scanner was set up in this lab for the measurement of fluorescence originating from oligonucleotide array surfaces. The optical set up and the instrumentation are shown in Figure 2.6.
SPAD
Condensing lens
SPAD
Band pass filter centered at 825 nm Long pass filter cut on 820 nm Pinhole 2.0 mm
x/y micro-tanslational stage
Diode laser
Line filter Dichroic mirror
sample
Diode laser 780 nm Microscope objective 40 x, NA 0.85
Array surface
B A Figure 2.6. The near-IR fluorescence scanner optical (A) and instrumental set up (B). The near-IR scanner consisted of a diode laser (PicoQuant GmbH, Model 800, Berlin, Germany), counting electronics (PicoQuant GmbH, Model SPC 430, Berlin, Germany) and a single photon avalanche diode (SPAD, EG&amp;G Optoelectronics, Model SPCM-PQ, Vaudreuil, Canada). The components were mounted with the aid of a mounting cube and lens tubes purchased from ThorLabs (Newton, NJ) and configured in an epi-illumination format. The diode laser operated at a wavelength of 780 nm with 7.5 mW of average power. An integrated optics set was provided with the laser to produce an elliptically shaped collimated output. The laser
50
excitation beam was passed through a 780 nm line filter (Omega Optical, 780DF10, Brattleboro, VT), reflected by a dichroic mirror (Omega Optical, 795DRLP) and focused onto the array surface using a 40x high numerical aperture (NA) microscope objective (Nikon, Natick, MA, NA=0.85). The 1/e2 spot size of the excitation beam at the array surface was measured to be 3 m (minor axis) by 5 m (major axis). The fluorescence emission was collected by the same microscope objective, transmitted through the dichroic, a circular aperture set at 2.0 mm (clear diameter) and finally through a filter stack consisting of a long-pass filter (cut on wavelength = 820 nm, Newport Corporation, Irvine, CA) and a band-pass filter centered at 825 nm (825RDF30, Omega Optical). After passing through the filters, the fluorescence was sent through a
condensing lens (01LAG111/076, Melles Griot) and focused onto a single photon avalanche diode (SPAD). The entire fluorescence detector was mounted on a X/Y micro-translational stage, which was controlled by stepper motors interfaced to a PC computer. Two bipolar stepper motors interfaced to the PC using STP-100 stepper motor controller boards obtained from Pontech, Inc. (Upland, CA) drove the X and Y directions of the micro-translational stages. Each STP-100 was equipped with the RS-485 interface allowing full duplex, multi-drop communication with the host computer. A PP232-485f interface (Pontech, Inc.) was used to convert RS485 into the PC's RS232 protocol. Hall sensors were used to monitor the travel limits of the X/Y stages. The smallest step resolution of these stages was 12.7 m with a scan range of 4 cm in both the X and Y coordinates. The scanner operated by taking a single step and then acquiring the fluorescence data for a software-selectable integration period (10 ms - 10 s). The data acquisition software was written in Visual Basic and consisted of several control and data acquisition functions such as, recording the position of the scanning head, streaming
51
data to the hard drive and providing real-time visualization of the acquired images. During a typical experiment, 4  12 MB of data were generated and stored in chunks of 2 GB. After imaging the array, the data was compressed one scan line at a time and subsequently assembled into one contiguous file.
Optimization of Excitation Laser Power
A near-IR dye-labeled oligonucleotide (M13 Fwd (-29)-IRD800) was dissolved in 5x SSPE buffer to make a 10 pM stock solution. The prepared solution was spotted onto aminatedPMMA surfaces in a 1 x 4 pattern (0.2 L/spot) and the slides were dried in darkness. The prepared slides were then scanned at different excitation laser powers (from 0.32 W to 0.8 mW) at a resolution of 101.6 m/step and the average fluorescence intensity of the dye-labeled oligonucleotide spots as well as the background intensity of the aminated-PMMA surfaces were extracted at each excitation power using the acquisition software. The integration time used for photon collection was 0.1 s in all cases.
Near-IR Dye Photo-Stability on Solid Surfaces
The 5 terminal amine modified oligonucleotide probe was covalently immobilized at its 5-end on chemically modified PMMA surfaces (see Chapter 3 for details). Then, the slide was hybridized in a 10 nM dye-labeled complement solution, M13 FWD (-29)-IRD 800 in 5x SSPE, 0.1 % SDS, at room temperature for 30 mins. The slide was washed twice in 2x SSPE, 0.1 % SDS solution for 15 mins. The prepared slide was then scanned under the near-IR scanner repeatedly (laser output power of 50 W) without changing position. The fluorescence intensity coming from the same pixel was extracted from each scan and normalized to the intensity value of the first scan.
52
Results and Discussions
Excitation Laser Power
Figure 2.7 shows the effects of excitation laser power on fluorescence emission intensity and on the background intensity of PMMA oligonucleotide arrays.
40000 35000
fluorescence intensity from hybridized duplex fluorescence intensity from aminated-PMMA surface
800 700 600
SNR
30000 25000 20000 15000 10000 5000 0 0.0 0.2 0.4 0.6 0.8
500
arbitrary intensity
SNR
400 300 200 100 0 0.0 0.2 0.4 0.6 0.8
laser power (mw)
laser power (mw)
A
B
Figure 2.7. The optimization of excitation laser power. The fluorescence intensities coming from the dye-labeled oligonucleotide arrays and the intensities from PMMA substrate (A) and the signal to noise ratio (SNR) (B) at various excitation laser powers. 10 pM IRD800 dye labeled oligonucleotide sequences (19-mer) in 5x SSPE, 0.1% SDS buffer solution were spotted onto aminated PMMA surfaces and dried in darkness. The slides were scanned at various excitation laser power ranges from 0.32 W to 0.8 mW (a newly-spotted slide was used for each scan) with a scanning step of 101.6 m. The fluorescence intensity was averaged on four spots of each slide.
As can be seen, the intensity of the background increases linearly with the excitation laser power in the range from 0.32 W to 0.8 mW, while the fluorescence increases linearly at lower excitation values and remained constant after 0.25 mW. In this spectral region, the background mainly originates from the scattering of excitation light, which is proportional to the incident light intensity. Although the fluorescence is also proportional to the power of the excitation light as illustrated in the formula [39] [41]; FI = Iex   Q t N (2.3)
53
where, FI is the fluorescence intensity (photon counts), Iex is the excitation light intensity (photons cm-2 s-1),  is the molar extinction coefficient (M-1cm-1 at the excitation wavelength),  is a constant with a value of 3.8 x 10-21 M cm3, Q is the fluorescence quantum yield, t is the time interval, and N is the total number of fluorescent molecules in the ground state. The fluorescence intensity is assumed to be linearly related to the illumination intensity (Iex) based on two assumptions; 1) the illumination intensity is sufficiently low such that any given molecule has a very low probability of absorbing a photon on the time-scale of the fluorescent lifetime, 2) the product of the molar extinction coefficient, fluorophore concentration and path length is sufficiently low such that the excitation light is not significantly attenuated as it passes through the sample (inner filter effect) [38]. In the case of the fluorescence detection on a solid substrate, in which the illumination and the emission is limited to a monolayer on the sample surface, it is valid to assume that inner filtering will be absent. If the absorption rate of a single molecule is low relative to the fluorescent lifetime, then the number of dye molecules in the ground state (No) is approximately equal to the total number of dye molecules (N). However, as the excitation intensity is increased, more and more molecules will possibly be promoted onto the excited states and No &lt;&lt; N. This will lead to a population inversion of fluorophore molecules and the invalidation of the first assumption, so that the emitted fluorescence no longer follows the linearity rule. The fluorescence emission will increase linearly with Iex at low intensities and approaches a constant value when the population inversion occurs. The maximum number of photons that can be emitted by a single dye molecule per second is equal to the fluorescence quantum efficiency divided by the fluorescent lifetime [38].
54
With the near-IR scanner and the near-IR 800 dye combination, the fluorescence saturation was reached at a laser power of ~0.25 mW. However, the maximum signal-to-noise ratio ( SNR = signal  background background ) was observed at 0.05 mW (50 W) as shown in Figure 2.7 B.
The detection limit of the near-IR scanner at the optimized excitation laser power was deduced as follows: 0.2 L of a 10 pM dye-labeled oligonucleotide solution was spotted onto the aminated PMMA surface. The formed fluorescent spot had a diameter of ~ 1.2 mm as examined under the scanner. The calculated number of fluorophore molecules per m2 is therefore ~ 1 molecule and at this target surface density, the signal to noise ratio was 700 at an excitation laser power of 50 W. Therefore, based on the assumption that the linearity relationship exists in the detection range, the detection limit can be extended to be ~ 0.003 fluors per m2 at SNR =2 level. Compared to most of the commercial detectors listed in Table 2.2, which have detection limits ranging from 0.05 to 1 fluors per m2, our near-IR scanner is ten times more sensitive. Although the detection limit could be further increased by increasing the integration time for the photo collection, this will be at the cost of increasing the length of scanning times. Normally, we set the integration time per pixel at 0.1 s. The fluorescence intensity was also affected by the scanning steps and the smaller scanning steps tend to produce more intense fluorescence intensities. The most commonly used scanning steps in our experiment were 101.6 and 50.8 m, depending on the scanning sample size and the image quality required.
The Photo-Stability of IRD800 Dyes
The photo-stability of the near-IR dye, IRD800, was examined by repeatedly scanning the slide on which dye-labeled oligonucleotides had been deposited and the extracted fluorescence intensity at the same image pixel compared. The normalized fluorescence
55
intensities are listed in Figure 2.8. The constancy of the curve indicates that almost no photobleaching occurred during scans at these laser fluences.
1.2
1.0
normalized intensity
0.8
0.6
0.4
0.2
0.0 0 1 2 3 4 5 6 7 8 9 10 11 12
repeated times
Figure 2.8. The photostability of IRD800 on a PMMA substrate. The PMMA-substrate immobilized oligonucleotide probes were hybridized to its dye-labeled oligonucleotide complements and the resultant dye-labeled duplexes were scanned at an excitation laser power of 50 W. The fluorescence intensity on the same pixel was extracted for each scan and normalized to the first scan.
Conclusions
In this chapter, the basic concepts of near-IR fluorescence detection and the performance of our near-IR fluorescence scanner were discussed. The fluorescence signal-to-noise ratio was maximized at excitation laser power of 50 W and at this excitation output, the detection limit of the scanner is around 0.003 fluorophore molecule/m2 and a good photo-stability of near-IR dye IRD-800 was observed. The fluorescence intensity does not degrade noticeably even after 11 repeated scans.
56
References
1. 2. 3. Yeung, E.S., Sepaniak, M.J., Anal. Chem., 1980, 52(13), 1465-1487. O'Donnell, C.M., Stephen, C. S., Anal. Chem., 1979, 51, 33A-40A. Van Hecke, G.R., A guide to lasers in chemistry, 1998, Jones and Bartlett Publishers, Boston. Soper, S., Mattingly, Q., Vegunta, P., Anal. Chem., 1993, 65, 740-747. Lee, Y.H., Mans, R.G., Smith, B.W., Winefordner, J.D., Anal. Chem., 1994, 66, 41424149. McCarroll, M.E., Warner, I.M., Agbaria, R.A., Electronic Absorption and Luminescence: Introduction, in Encyclopedia of Analytical Chemistry, Meyers, R.A., Editor, 2000, 10259-10280. Lakowicz, J.R., Principles of fluorescence spectroscopy, 1999, Plenum Press, New York. McWhorter, S., Soper, S.A., Electrophoresis, 2000, 21(7), 1267-1280. Flanagan, J.H., Owens, C.V., Romero, S.E.,Waddell, E., Kahn, S.H., Hammer, R.P., Soper, S.A., Anal. Chem., 1998, 70(13), 2676-2684. Soper, S.A., McGown, L.B., Warner, I.M., Anal. Chem., 1994, 66(12), R428-R444. Raghavachari, R., Near-Infrared Applications in Biotechnology, 2001, Marcel Dekker Inc., New York, 2-3. Williams, D.C., Soper, S. A., Anal. Chem., 1995, 67, 3427 - 3432. Imasaka, T., Yoshitake, A., Ishibashi, N., Anal. Chem., 1984, 56, 1077-1079. Kawabata, Y., Imasaka, T., Ishibashi, N., Talanta, 1986, 33, 281-283. Middendorf, L., Amen, J., Bruce, R., Draney,D., DeGraff, D., Gewecke, J., Grone, D., Humphrey, P., Little, G., Lugade, A., Narayanan, N., Oommen, A., Osterman, H., Peterson, R., Rada, J., Raghavachari, R., Roemer, S. Near-infrared fluorescence instrumentation for DNA analysis. in Near-infrared Dyes for High Technology Applications, 1998, The Netherlands: Kluwer Academic, Amsterdam. Soper, S.A., Flanagan, J.H., Legendre, B.L., Williams, D.C., Hammer, R.P., IEEE J. Sel. Top. Quant., 1996, 2(4),1129-1139.
4. 5.
6.
7. 8. 9.
10. 11.
12. 13. 14. 15.
16.
57
17.
Legendre, B.L., Moberg, D.L.,Williams, D.C., Soper, S.A., J. Chromatogr. A, 1997, 779(1-2), 185-194. Lassiter, S.J., Stryjewski, W., Legendre, B.J., Erdmann, R., Wahl, M., Wurm, J., Peterson, R., Middendorf, L.,Soper, S.A., Anal. Chem., 2000, 72(21), 5373-5382. Owens, C.V., Davidson, Y.Y., Kar, S,, Soper, S.A., Anal. Chem., 1997, 69(6), 1256-1261. Shealy, D.B., Lipowska,M., Lipowski, J., Narayanan, N., Sutter, S.,Strekowskil, P. G., Anal. Chem., 1995, 67(2), 247-251. Soper, S.A., Legendre, B.L., Williams, D.C., Anal. Chem., 1995, 67(23), 4358-4365. Flanagan, J.H., Romero, S., Legendre, B.L., Hammer, R.P., Soper, S.A., Abstr. Pap. Am. Chem. S., 1996, 211, 68. Schrock, E., duManoir, S., Veldman, T., Schoell, B., Wienberg, J., FergusonSmith, M.A., Ning, Y., Ledbetter, D.H., BarAm, I., Soenksen, D. Garini, Y., Ried, T., Science, 1996, 273(5274), 494-497. Patonay, G., Antoine, M.D., Anal. Chem., 1991, 63, 321A-327A. Matsuoka, M., Infrared Absorbing Dyes, 1990, Plenum Press, New York. Flanagan, J.H., Khan, S.H., Menchen, S., Soper, S.A., Hammer, R.P., Bioconj. Chem., 1997, 8(5), 751-756. Middendort, L.R., Bruce, J.C., Bruce, R.C., Eckles, R.D., Grone, D.L., Roemer, S.C., Sloniker, G.D., Steffens, D.L., Sutter, S.L., Brumbaugh, J.A., Patonay, G., Electrophoresis, 1992, 13(8), 487-494. Soper, S.A., Legendre, B.L., Williams, D.C., Appl. Spectrosc., 1998, 52(1), 1-6. Wheeler, B.L., Nagasubramanian, G., Bard, A.J., Schechtman, L.A., Dininny, D.R., Kenney, M.E., J. Amer. Chem. Soc., 1984, 106(24), 7404-7410. Sharman, W.M., Kudrevich, S.V., vanLier, J.E., Tetrahedron Lett., 1996, 37(33), 58315834. Hammer, R.P., Owens, C.V., Hwang, S.H., Sayes, C.M., Soper, S.A., Bioconj. Chem., 2002, 13(6), 1244-1252. Su, G.F., Verdree, V., Hammer, R.P., Soper, S.A., Abstr. Pap. Am. Chem. S., 2002, 224, 057-ORGN Part 2.
18.
19. 20.
21. 22.
23.
24. 25. 26.
27.
28. 29.
30.
31.
32.
58
33.
Lieberwirth, U., Arden-Jacob, J., Drexhage, K.H., Herten, D.P., Muller, R., Neumann, M., Schulz, A., Siebert, S., Sagner, G., Klingel, S., Sauer, M., Wolfrum, J., Anal. Chem., 1998, 70(22), 4771-4779. Sheppard, C.J.R., Shotton, D.M., Confocal Laser Scanning Microscopy, 1997, SpringerVerlag New York Inc. 3-13. Zlatanova, J., Mirzabekov, A., Gel-Immobilized Microarrays of Nucleic Acids and Proteins, in DNA Arrays: Methods and Protocols, Rampal, B.J., Editor, 2001, Human Press Inc., Totowa, NJ, 17-38. Waddell, E., Wang, Y., Stryjewski, W., McWhorter, S., Henry, A. C., Evans, D., McCarley, R. L., Soper, S. A., Anal. Chem., 2000, 72, 5907 - 5917. Skoog, D.A., Holler, F. J., Nieman, T. A., Principles of instrumental analysis. 5th ed. 1998, Saunders College Pub, Philadelphia. Imasaka, T., Talanta, 1999, 48(2), 305-320. Becker, P.L., Quantitative Fluorescence Measurements, in Chemical Analysis-A Series of Monographs on Analytical Chemistry and Its Applications, Winefordner, J.D., Editor, 1996, John Wiley &amp; Sons, Inc., New York, 1-28. Yu, F.T.S., Yang, X., in Introduction to optical engineering, 1997, Cambridge University Press, New York, NY, 89-98. Mathies, R.A., Stryer, L., Single-molecular fluorescece detection: A feasibility study using phycoerythrin, in Application of Fluorescence in the Biomedical Sciences, Talor, D.L., Waggoner, A.S., Murphy, R.F., Lannin, F., Birge, R.R., Editor, 1986, Alan R. Liss, New York, 129-140.
34.
35.
36.
37.
38. 39.
40.
41.
59
Chapter 3 Two-Dimensional PMMA Oligonucleotide Arrays*
Introduction
Polymers have long been used as substrates for immobilization of DNA molecules in nucleic acid assays [1-4]. For example, nitrocellulose or nylon membrane filters were utilized for trapping DNA samples in Southern blots [5], and dot or slot blotting techniques for DNA sequence detection [6]. Polystyrene and polypropylene have been the polymer substrates widely used as non-covalent [7-8], covalent [9-10], and in-situ synthesis [11] supports of oligonucleotide probes. These surface-immobilized oligonucleotide probes were used for typing of single nucleotide polymorphisms either by direct hybridization [9] [12] or the involvement of enzymatic reactions [13]. For example, short (12-30 mer) oligonucleotide probes were immobilized in polystyrene micro-wells and hybridized to PCR-amplified DNA targets with its 3end facing the suspect position in the target. If a perfect match existed between the 3 end of the probe and the detection position in the target, the appropriate labeled dideoxynucleoside triphosphate (ddNTP) will be added to the 3 end of the probe in the presence of a polymerase enzyme. The identity of the mutation could be deduced by examining the added ddNTPs [8]. The use of polymeric materials as the solid substrate for nucleic acid-based assays has been promoted by the recent development of micro-analysis systems [14-15]. Compared to glass substrates, there are more alternative approaches for the fabrication of polymer-based devices.
Reproduced in part with permission from Analytical Chemistry, volume 75, number 5, pages 1130-1140, 2003. Copyright 2003 American Chemical Society.
*
60
For instance, injection molding, hot-embossing, or UV-ablation, provide the flexibility of mass production of polymer-based micro-devices at low cost. In a recent paper, four different types of plastic substrates, polystyrene (PS), polycarbonate (PC), poly(methylmethacrylate) (PMMA), and polypropylene (PP), were evaluated for array fabrication [16]. It was found that at both the excitation laser wavelengths of 532-nm and 635-nm, PC had the highest intrinsic fluorescence and PMMA had the lowest; PP is the most hydrophobic and PMMA is the least hydrophobic. Three oligonucleotide probe immobilization procedures, Reacti-Bind (www.piercenet.com), CTAB [8], and Surmodics [17] were compared as for their efficiency in attaching oligonucleotide probes on these four substrates. Reacti-Bind involves the use of a Reacti-Bind DNA coating solution developed and commercialized by Pierce Biotechnology, Inc. for the immobilization of unmodified DNA onto plastic micro-wells without surface pretreatment. CTAB involves the use of a cationic reagent, cetyltrimethylammonium bromide (CTAB), in the oligonucleotide probe solution to facilitate the adsorption of probes onto polymer surfaces. Surmodics involves the coating of the polymer surface with a Surmodics surface treatment solution (developed at Motorola Life Science Division, Northbrook, IL) followed by 2 min of UV irradiation and the immobilization of oligonucleotide probes were realized by incubation of the probe phosphate solution on the activated surface overnight. It was reported [16] that oligonucleotide attachment using ReactiBind procedure was successful only on PC substrates, resulting in a hybridization reaction with a limit of detection (LOD) of 6.25 nM; the use of the amine-modified oligonucleotide probes in the CTAB procedure had better immobilization results than the unmodified probes and the lowest LOD for the hybridization of surface immobilized probes was 0.2 nM. The Surmodics procedure was reported to be the most efficient of these three methods for attaching oligonucleotide probes
61
to the above four polymer substrates with the lowest LOD (0.012 nM). Most of these procedures dealt with a non-covalent, passive adsorption of oligonucleotide probes to the polymer substrate surface. In DNA array fabrication, a covalent bonding scheme is more desirable due to its controllable and reproducible immobilization characteristics as compared to the non-covalent mechanism [18]. In the former scheme, oligonucleotide probes will form a monolayer of molecules on the surface through covalent bond formation, which is important for obtaining reproducible and quantitative detection of the hybridization signals, whereas for noncovalent immobilization multi-layers may form. However, it is extremely difficult, if not impossible, for oligonucleotides to couple directly to most pristine polymer surfaces because of the lack of appropriate surface functional groups. Therefore, it is necessary to modify the polymer surface with reactive functional groups that are prone to initiate the attachment of oligonucleotides. Nevertheless, the development of routine, simple, well-defined surface modification protocols for polymers is still in its infancy [19-20]. In this chapter, we report on the immobilization chemistry for covalent attachment of oligonucleotide probes to chemically-modified PMMA substrates. PMMA was the material of choice due mainly to both the presence of functional groups (methyl ester) for potential modification and its extensive use for microfluidic devices [15] [21-22]. The following sections will discuss the introduction of primary amines onto the polymer surfaces through aminolysis, optimization of surface amine group density and covalent attachment of terminal aminomodified oligonucleotide probes. Hybridization performance of the surface immobilized probes, robustness of the coupling chemistry and the detection limit of the PMMA oligonucleotide arrays will also be covered.
62
Experimental Section
Chemicals and Reagents Oligonucleotide probes were purchased from Midland Certified Reagent Company (Midland, TX) or IDT (Coralville, IA). Near-IR dye-labeled oligonucleotide molecules were from LI-COR Biotechnology (Lincoln, NE). Their sequences and their theoretical melting temperatures (Tm) to full complements are listed in Table 3.1. Table 3.1. Sequences of oligonucleotide probes and dye-labeled complements Oligonucleotides Oligo A Oligo B DA013002 Sequences (53) *C6 Amino - TTTTTTTTTTTTGTCGTTTTACAACGTCGTG GTC GTT TTA CAA CGT CGT GTT TTT TTT TTT TTT T *C7 Amino IRD800  TTT TTT TTT TTT TTT CAC GAC GTT GTA AAA CGA CT -*C7 Amino Tm (oC) 56.6 59.6 61.3
M13 FWD (-29) IRD 800 C ACG ACG TTG TAA AAC GAC 48.9 IRD 800 Zip code 3 CAGCACCTGACCATCGATCGCAGC - *C7 Amino 65.5 Zip code 11 TGCGGGTACAGCACCTACCTTGCG - *C7 Amino 66.7 cZip 3 GCT GCG ATC GAT GGT CAG GTG CTG IRD41 67.8 cZip 11 CGC AAG GTA GGT GCT GTA CCC GCA IRD41 67.8 * C6 and C7 indicate the number of carbon atoms in the linker molecule between the terminal amino group and the actual sequence PMMA sheets (CQ-grade, 1.2-mm thick, GoodFellow, Berwyn, PA) were used as substrates throughout these experiments. Chemicals used for the substrate modification and hybridization assays included N-lithioethylenediamine, ethylenediamine, 50 wt-% glutaric dialdehyde, sodium borohydride, sodium cyanoborohydride (5.0 M solution in aqueous ~1 M sodium hydroxide), perchloric acid and 20x SSPE buffer, which contains 0.2 M phosphate buffer, pH ~ 7.4, 2.98 M NaCl and 0.02 M EDTA (all from Aldrich Chemicals, Milwaukee, WI). The Sulfo-succinimidyl-4-O-(4, 4dimethoxytrityl)-butyrate) reagent (sulfo-SDTB, Pierce
63
Chemicals, Rockford, IL) was used to determine the amine group surface density on the modified PMMA surface. Standard solutions used were: hybridization solution  10 nM M13 Fwd (-29)-IRD 800 in 5x SSPE and 0.1% SDS; wash buffer  2x SSPE and 0.1 % SDS; capping solution  NaBH4, 0.25 g dissolved in 1 mL of 95% ethanol and diluted to 100 mL with 0.1 M phosphate buffer (pH 6.1). All the solutions were made in doubly distilled water (ddH2O, 18 M-cm), (Barnstead / Thermolyne NANO-pure Infirmity, Dubuque, IA). -32P ddATP (10 mCi/mL) was from Amersham Bioscience (Piscataway, NJ). 3-end terminal modification enzyme, Terminal Deoxynucleotidyl Transferase (TdT, 5 U/L), and 5x buffer were from USB (Cleveland, Ohio). PMMA Surface Modification (see Scheme 3.1on page 71) The PMMA slides were treated in N-lithioethylenediamine solution in a glove box filled with argon. In the process, the reagent solution was prepared by dissolving the Nlithioethylenediamine salt in a proper volume of ethylenediamine solvent for approximately 40 mins under magnetic agitation. The reagent concentration was altered by varying the relative amounts of the solid salt and the solvent. Next, the PMMA slides were soaked in this solution for 5 to 30 mins to form amine-terminated surfaces (PMMA-NH2). After amination was complete, the slides were rinsed in isopropanol (IPA), removed from the glove box, and thoroughly rinsed with ddH2O. The aminated slides were then soaked in 5% glutaric dialdehyde in 0.5 M phosphate buffer (pH 6.4) containing 10 L/mL sodium cyanoborohydride at room temperature for 1 hr to convert the PMMA surface to an aldehyde-terminated surface (PMMA-CHO). The finished slides were cleaned in ddH2O, dried with N2, and stored in a desiccator.
64
Surface Contact Angle Measurements Hydrophobicity properties of both pristine and the amine-terminated PMMA surfaces were measured with a VCA 2000 Video Contact Angle System and the integrated software (AST Products, Inc. Billerica, MA). A small ddH2O droplet (2 L) was dispensed onto the surface and the droplet profile was recorded with a CCD camera 2 mins after dispensation. The image was captured into the computer and the contact angle of water on the PMMA surface was calculated with the computer software. Surface Amine Density Surface amine densities of the aminated PMMA slides were derived from the amount of released 4, 4-dimethoxytrityl cation from developing the reaction products between sulfo-SDTB reagent [sulfo-succinimidyl-4-O-(4, 4-dimethyoxytrityl)-butyrate] and the primary amine groups over the aminated slide in acid solution. To form the reaction product, the aminated PMMA slides were soaked in 0.1 mM sulfo-SDTB in 50 mM sodium bicarbonate buffer (pH 8.5) at room temperature for 1 hr, rinsed thoroughly with ddH2O, and dried in air. Then, the slides were developed in 1 mL 35% perchloric acid for 5 mins such that the 4, 4-dimethoxytrityl cation on the surface was completely dissolved into the perchloric acid. The absorbance of the resultant pink solution was measured with a UV-VIS spectrometer (Amersham Pharmacia, New York, NY) at a wavelength of 498 nm. The total number of amine groups was determined from the absorbance, the extinction coefficient (70,000 M-1cm-1) of the released 4, 4-dimethoxytrityl cation and the total solution volume (1 mL). The surface amine density was then calculated from the total amine-group numbers divided by the exposed surface area (not corrected for surface roughness). The density reported was averaged from five samples.
65
Preparation of PMMA Oligonucleotide Arrays The 5 terminal amino-modified oligonucleotide probe A (Table 3.1) was first dissolved in 200 mM phosphate buffer (pH 8.3) to a final concentration of 2 M. The prepared oligonucleotide solution was then spotted (0.2 L/spot) onto the modified PMMA slide following a paper template using a micropipette. The spotted slides were kept in a humidified chamber at 37oC for 1.5 hrs, then soaked in capping solution (see Chemicals and Reagents) for 5 mins to remove the unbound oligonucleotides and to deactivate the unreacted aldehyde groups, and rinsed with ddH2O. The slides were finally dried and stored in a desiccator for the hybridization analysis. Optimization of Coupling Time of Oligonucleotide Probes to Surface The 3 terminal amino-modified oligonucleotide probe, DA013002, (see Table 3.1 for sequences), was 5 end-labeled with IRD800 (LI-COR, Lincoln, NE). A concentration of 1 M solution was prepared by dissolving 0.1 nmol oligonucleotide DA013002 in 100 L 0.2 M phosphate buffer (pH 8.3). The resultant solution was arrayed onto the PMMA-CHO surfaces in a 2 x 2 pattern, 0.5 L /spot. The slides were incubated at room temperature in darkness in a humidified chamber and two of the slides were removed from the container at time intervals of 15, 30, 60, 120, 240, 510, and 1440 mins and were soaked in capping solution for 10 mins. Then, the slides were washed in 2x SSPE washing for buffer 15 mins twice and scanned under the near-IR scanner (37 W laser power) at room temperature. The average fluorescence intensities at each time interval was extracted with the data acquisition software and plotted against the coupling time.
66
Spotting Concentration of Oligonucleotide Probe Solution Oligonucleotide probe B containing a 3 terminal amino group was dissolved in 0.2 M phosphate buffer (pH 8.3) to make 4.7 mM stock solutions, from which a sequential dilution was made to prepare a concentration series of 470, 47, 4.7, 0.47, and 0.047 M solutions. The dilution series were spotted onto the PMMA-CHO slides, 0.2 L/spot, and two spots for each concentration. The spotted slide was maintained at 37oC in a humidified chamber for 4 hrs followed by capping the residual surface aldehyde functional groups for 10 mins in capping solution. After that, the slide was hybridized in 10 pM dye-labeled complements (M13 Fwd (-29)IRD800) in 5x SSPE, 0.1 % SDS buffer solution at 37oC overnight. The cleaned slide was scanned under the near-IR scanner with the excitation laser output power set at 51.4 W and step size of 101.6 m with integration time per pixel of 0.1 s. 3-32P Terminal Labeling and Immobilization of Oligonucleotide Probes The 5terminal amino-modified oligonucleotide probe A (see Table 3.1) was 3 endlabeled with [-32P]ddATP (Amersham, 5000 Ci/mmol, 10 mCi/mL). A 100 L labeling reaction containing 46.8 nmol oligonucleotides, 20 L 5x TdT buffer (500 mM sodium cacodylate, 10 mM CoCl2 and 1 mM 2-mercaptoethanol), 14 L terminal deoxynucleotidyl transferase (TDT, USB, Cleveland, OH, 5 U/L), and 10 L [-32P]ddATP was incubated at 37oC for 1 hr. The labeling mixture was purified on a MicroSpinTM G-25 Column (Amersham Pharmacia Biotech, Piscataway, NJ), and the concentration of oligonucleotide probe solution was determined by the absorbance at 260 nm on a UV/Visible Spectrometer (Ultrospec 4000, Pharmacia Biotech, Piscataway, NJ ). The purified products were dried down using an Eppendorf Concentrator 5301 (Westbury, NY) and resuspended in 200 mM sodium phosphate buffer (pH 8.3) to make a concentration of 10 M solution. One L of the resultant solution was 67
applied to 5 mm x 5mm PMMA-CHO slides at 37oC for 2 hrs. The slides were then washed thoroughly with 1% NH3H2O followed by copious amounts of water. As a control, 1 L of the labeled oligonucleotide solution was also applied to 5 mm x 5 mm pristine PMMA slides and dried in air. Bound oligonucleotides were quantified by comparing the radioactivity values coming from the PMMA-CHO and pristine PMMA slides detected on a Beckman LS6500 counter (GMI, Inc. Albertville, MN) by soaking the slides in 5 mL scintillation corktail (ScientiSafeTM EconoF, Fisher, Pittsburgh, PA). Hybridization Optimization To study the effect of amine surface density on hybridization efficiency, PMMA slides were cut into 12 mm x 12 mm pieces and derivatized in 0.13 M N-lithioethylenediamine solution for different lengths of time (each group had 5 pieces), and oligonucleotide probe (Oligo A) couplings were performed as described above. The prepared oligonucleotide slides were hybridized in 10 nM dye-labeled complements M13 Fwd (-29)-IRD 800 in 5x SSPE, 0.1 % SDS buffer solution at 37oC for 3 hrs (in Bambino hybridization oven, Model 230300, Boekel Scientific, Feasterville, PA). Two 15 min washes were performed in 15 mL washing buffer (2x SSPE, 0.1% SDS) each on a VWR DS-500 orbital shaker (VWR Scientific Products, West Chester, PA). Hybridization signals were quantified by fluorescence intensity detected by the home-built near-IR scanner from the slide surfaces (excitation laser power 0.24 mW, motor step at 101.6 m/step with integration time of 0.1 s). To compare the buffer concentration on the hybridization productivity, oligonucleotide arrays containing two different fragment sequences (Zip codes 3 and 11) were prepared by spotting 100 M phosphate solution containing each oligonucleotide probe onto the PMMACHO surfaces, 0.2 L/spot at 37oC for 1.5 hrs. Two buffer concentrations, 5x SSPE, 0.1 % SDS
68
and 1x SSPE, 0.025% SDS, were used for preparation of hybridization solutions. The hybridization solution containing two types of dye-labeled sequences, cZip 3 and cZip 11, which are complementary to surface-immobilized Zip code 3 and Zip code 11, respectively. The capped oligonucleotide-array slides were hybridized either in 20 nM (cZip 3 + cZip 11) in 5x SSPE, 0.1% SDS or in 20 nM (cZip 3 + cZip 11) in 1x SSPE, 0.025% SDS buffers at 50oC for 1 hr. Finally, the PMMA pieces were washed as described above and hybridization signals were determined by fluorescence measurements. To study the hybridization kinetics, 2 M oligonucleotide (Oligo A) phosphate solutions were applied onto flat aldehyde-terminated PMMA surfaces in a 2 x 2 pattern, 0.2 L/spot. After capping of the residual surface aldehyde groups, PMMA slides were incubated in a 10 nM hybridization solution at 37oC for 5, 30, 60, 90, 150, 210, 240, and 300 mins, respectively. After incubation, the PMMA slides were washed and scanned using the near-IR scanner. Fluorescence Detection and Data Analysis After hybridization and washing, the dried PMMA slides were scanned directly under the near-IR scanner. Fluorescence images were generated by scanning a laser (diode laser lasing at 780 nm, 7.5 mW output, 1/e2 spot size was measured to be 3 m at minor axis and 5 m at major axis) beam over the sample surface in a raster pattern and monitoring the resulting fluorescence (through a long pass 820 nm and band pass 825 nm filter stack) in a pixel-by-pixel fashion. The spatial resolution of the scan had several choices, 203.2, 101.6, 50.8, 25.4, 12.7 m per step and the appropriate value was chosen according to the sample size and the resolution required. Normally, 101.6 m /step was used. The collection time at each pixel was set at 0.1 s. The fluorescence events at each pixel (as measured by a single photon avalanche diode detector) were stored in a multi-channel analyzer. After scanning, the fluorescence image was reconstructed
69
from the digitized pixel intensities using HarpPlayer software written in Visual Basic 6.0. The fluorescence intensity across the slide surface was also extracted with the same software. For signal-to-noise ratio calculations, the home-made software (HarpPlayer) was used to calculate the peak intensity on the center of each spot image as the fluorescence intensity. The signal-to-noise value was calculated as the fluorescence intensity level over the background divided by the square-root of the background. The intensity coming from the PMMA substrate around the hybridized spots was extracted as background.
Results and Discussions
PMMA Surface Modification and Oligonucleotide Coupling Chemistry A protocol that permits end-attachment of terminal amino-modified oligonucleotide probes to PMMA surfaces adopted in this work is shown in Scheme 3.1. Four major steps were involved in this scheme: 1) aminolysis; 2) activation; 3) coupling and 4) capping/reduction. Optimization and performance of each step will be discussed in detail in the following sections.
Step 1  Aminolysis
As shown in Scheme 3.1, part of the surface methyl-ester functional groups was replaced by N-lithioethylenediamine in the aminolysis step. With the introduction of these amine groups, the PMMA surface changed from hydrophobic to hydrophilic as evidenced by the contact angle change from 70o  74o on pristine PMMA to 39o  45o on aminated surfaces after 5-30 mins of aminolysis. This characteristic surface change was easily observed by a homogeneous ddH2O film formation on the treated surface during rinse, which did not appear on the pristine slides. The contact angle further decreased with increases of the aminolysis time, for example, the contact angle dropped to 27o on the slide when it was aminated for 10 hrs. However, with the increase of treatment time, the PMMA slide lost its transparency. As with longer contact of the
70
polymer substrate with the reaction solutions, it was possible for the solvent and reagent to penetrate into the bulk substrate, therefore, the treatment will not be limited to the surface but the bulk as well, and the integrity of the substrate will be affected.
H2N H2N H2N
OCH 3 CO
OCH 3 CO
OCH 3 CO
OCH 3 CO
LiHN H2N
NH 2 NH 2
NH CO
NH CO
NH CO
OCH 3 CO
pristine PMMA surface
aminated PMMA surface
1. aminolysis
O HC OHC CHO NaBH 3 CN NH H2N
O HC
H 2 N oligonucleotide probe NH 3. coupling
2. activation
NH CO
NH CO
NH CO
OCH 3 CO
aldehyde-terminated PMMA surface
oligo N HC
O HC NaBH 4
oligo NH HOH 2 C
NH
H2N
NH
4. capping/reduction
NH
H2N
NH
NH CO
NH CO
NH CO
OCH 3 CO
NH CO
NH CO
NH CO
OCH 3 CO
PMMA surface
PMMA surface
Scheme 3.1. Flow chart of oligonucleotide probe coupling chemistry to PMMA surfaces. Monitoring Surface Amine Group Density
71
As these surface amine groups serve as the starting point for the covalent attachment of oligonucleotide probes, their surface density is crucial to provide a sufficient number of capture sites. Therefore, the surface amine group density was monitored and calculated.
OCH 3 O O CON
SO 3Na
H2N
C O CH 2
3
OCH 3
O sulfo-SDTB MW 605.6
OCH 3
C O CH 2
3
NH
OCH 3
perchloric acid
PMMA
O C
PMMA
OC H 3CO C OCH 3 NH
CH 2
3
OH
PMMA 4,4'-dimethoxytrityl cation  max = 498 nm  = 70,000 M -1 cm -1
Scheme 3.2. Scheme of surface amine group measurement using sulfo-SDTB reagent. The amount of the amine groups introduced to the PMMA surface was monitored using a reagent sulfo-SDTB (Pierce Chemicals, Rockford, IL). As illustrated in Scheme 3.2, the reagent forms stable bonds with the surface amine groups at alkaline conditions [23]. When developed in acidic solutions, 4, 4-dimethoxytrityl cation, which has a high absorbance at 498 nm with an
72
extinction coefficient of 70,000 M-1cm-1, is liberated. The corresponding surface amine amount can be deduced by measuring the absorbance of the developed solution and the slide surface area.
9
amine group surface density x10 mole/cm
2
8
-9
7
6
5
4
3
2 0 100 200 300 400 500
aminolysis time (min)
Figure 3.1. The effect of aminolysis time on the surface amine density. PMMA slides were treated in 0.13 M N-lithioethylenediamine solution at room temperature for different lengths of time ranging from 5 mins to 7 hrs. The surface amine group density was evaluated using sulfoSDTB reagent by measuring the released dimethoxytrityl cation at 498 nm following acid treatment. Figure 3.1 shows the dependence of the surface amine density on aminolysis time when treated with 0.13M N-lithioethylenediamine solution. The curve shows typical saturation characteristics, that is, the density increases with time up to ~ 180 mins and then reaches a plateau at a surface concentration of approximately (7.68  0.5) x 10-9 mol/cm2. This saturation density is almost one order of magnitude larger than that found on glass substrates [24]. We suspect that the modification chemistry is not limited to activation of only surface exposed groups, but underlying layers of the polymer as well, which provides an apparent higher 73
concentration for surface amine groups compared to glass [19]. As the amination time increased from 5 to 180 mins, the amine surface density increased by 2.5-fold. However, it was noticed that if the slides were soaked in the N-lithioethylenediamine solution over 30 mins, the slide became less transparent due to erosion of PMMA substrates by the chemical reagents.
6
amine group surface density (x10 mole/cm )
2 -9
5
4
3
2
1
0 0.0 0.2 0.4 0.6 0.8 1.0 1.2
N-lithioethylenediamine concentration (M)
Figure 3.2. The effect of N-lithioethylenediamine concentration on the surface amine density. Various N-lithioethylenediamine solutions ranging from 0.1 to 1.1 M was prepared by dissolving the N-lithioethylenediamine salt in ethylenediamine solvent in the Ar glove box. 1.2 x 6 x 6 mm3 PMMA slides were treated in the solution series for 5 mins. The surface amine density was measured using sulfo-SDTB reagent by measuring the released cation at 498 nm following acid treatment. Figure 3.2 shows the influence of N-lithioethylenediamine concentration on the surface amine group density for a fixed reaction time of 5 mins. From the figure, the density increased linearly with the concentration lower than 0.25 M and remained constant at concentrations above 0.4 M. The amine surface density was determined to be (4.79  0.55) x 10-9 mole/cm2 for slides treated in 0.4 M N-lithioethylenediamine solution at room temperature for 5 mins.
74
Step 2  Activation
1.4
1.2
1.0
amine surface density -9 2 (x10 mole/cm )
0.8
0.6
0.4
0.2 0 10 20 30 40 50 60
treatment time (min)
Figure 3.3. The conversion efficiency of surface amine groups to aldehyde groups as a function of the length of treatment time in 5% glutaric dialdehyde solution. The PMMA-NH2 slides were soaked in 5% glutaric dialdehyde solution for different lengths of time at room temperature, and the surface amine group density was measured using sulfo-SDTB reagent (for details see Experimental Section). After aminolysis, the slides were soaked in 5 % glutaric dialdehyde phosphate buffer solution (0.5 M, pH 6.2) with 10 L/mL cyanoborohydride solution added. The glutaric dialdehyde reacted with the surface amine groups to form a Schiffs base and the added cyanoborohydride reduced the unstable imine bond formed between the surface amine group and the dialdehyde to a stable secondary amine while leaving the free aldehyde group intact [25]. The conversion efficiency of surface amine groups to aldehyde groups was monitored using sulfoSDTB. The residual amine group density on the slides was treated in 5% glutaric dialdehyde solution for different lengths of time as shown in Figure 3.3. As can be seen, the conversion of
75
surface amine groups to aldehyde functional groups was fast with the conversion completed typically in less than 30 mins at room temperature. The conversion efficiency of surface amine groups to aldehyde groups was determined to be approximately 76% as estimated by the ratio of the surface amine density before and after being fully converted in glutaric dialdehyde solution for 30 mins.
Step 3  Coupling
10
SNR
1 0.01 0.1 1 10 100 1000 10000
concentration(M)
Figure 3.4. The effect of oligonucleotide probe solution concentration on the amount of surface immobilized probes as indicated by the hybridization fluorescence signal-to-noise ratio (SNR). Oligonucleotide arrays were prepared by spotting oligonucleotide solutions with different concentrations on the same chemically modified PMMA slide (0.2 L/spot) and two spots for each concentration. The fluorescently tagged complementary strand was hybridized to the surface-immobilized oligonucleotide arrays, and the hybridization intensity for each concentration detected by fluorescence scanning (see Experimental Section for details). Figure 3.4 shows the dependence of the amount of surface immobilized probes on the concentration of the oligonucleotide spotting solution as measured by monitoring the
76
fluorescence intensity emanating from surface-immobilized probes, which were bound to the surface by depositing different concentrations of probe solution.
35000
30000
fluorescence intensity
25000
20000
15000
10000
5000 0 200 400 600 800 1000 1200 1400 1600
coupling time (min)
Figure 3.5. The length of the incubation time on the amount of surface immobilized oligonucleotide probes. As described in the experimental section, the oligonucleotide probe, DA013002, was amino-modified at its 3 end and dye-labeled at its 5 end. The solution was incubated on a PMMA-CHO surface at room temperature for different lengths of time. The fluorescence of the spotted oligonucleotide probes were examined under the near-IR scanner after the slide was washed thoroughly (see Experimental Section for details). Assuming no fluorescence quenching due to inner filter effects on the surface, the fluorescence intensity should be proportional to the amount of duplexes formed, which is proportional to the amount of the oligonucleotide probes immobilized onto the surface. As can be seen, similar SNR values were obtained in the logarithmic graph in Figure 3.4 at all concentrations except the lowest concentration (0.047 M). This can be explained by the limited amount of immobilization sites for the oligonucleotide probes. Monolayer of oligonucleotide probes will be attached to the substrate surface through the terminal amino group. For a 0.2 L 77
spotting volume, the base of the droplet on the surface has a diameter around 1.32 mm as estimated from the image of the slide after hybridization. It is deduced that the amount of oligonucleotide probes existing in the 0.2 L droplet far exceed the number of surface functional groups covered by the probe spot. Therefore, the amount of the immobilized oligonucleotide probes should be governed by the surface functional group density and the steric hindrance between the neighboring molecules instead of the concentration of the oligonucleotides. As for the relative smaller SNR value at the lowest concentration, it may be attributed to the kinetic rule governing chemical reactions-the higher the concentration, the quicker the reaction will be completed, and if sufficient long reaction time is allowed, there maybe no difference exist among all these tested concentrations. The time needed for the 1 M oligonucleotide probe solution to saturate the PMMA surface functional groups was monitored by incubating the oligonucleotide probe DA013002, which is 3 amino-modified and 5 dye-labeled, solution on an aldehyde-terminated PMMA surface (PMMA-CHO) for different lengths of time and the resultant fluorescence intensity measured after the slide was washed to remove unbound probes. The results are illustrated in Figure 3.5. From the graph, the fluorescence intensity increases with the length of incubation time and saturates around 4 hrs. Since the fluorescence intensity is proportional to the amount of fluorescent molecules, we came to the conclusion that the coupling reaction of 1 M oligonucleotide probes on the PMMA-CHO surface is complete around 4 hrs. Slight reduction in the fluorescence was observed for longer incubation times (&gt; 8 hr). This may be attributed to oxidation of the fluorescent molecules caused by longer exposure of the solution to the ambient environment.
78
Step 4 - Capping and Reduction
After the coupling of oligonucleotide probes to the surface, the PMMA slide was capped in a capping solution at room temperature for 10 mins. Several goals were completed in this process. The unattached probes were washed off the surface; the less stable Schiffs base was reduced to a more stable secondary amine by sodium borohydride; the residual aldehyde groups on the surface were also reduced to hydroxyl groups, which does not have high reactivity toward the nucleotide bases in the oligonucleotide sequences; and the negatively charged hydroxyl groups may produce repellent interactions to the negatively charged oligonucleotide backbones, therefore, minimizing non-specific adsorption of dye-labeled sequences to the surface.
3-32P End Terminal Labeling of Oligonucleotide Probes
Terminal Deoxynucleotidyl Transferase (TdT) catalyzes a template-independent, repetitive addition of deoxynucleoside triphosphates to the 3'-hydroxyl termini of single-stranded or double-stranded DNA molecules with the release of inorganic pyrophosphate groups [26]. In our work, we choose - 32P ddATP to make sure only one isotopic molecule was added to each oligonucleotide probe. The recovery efficiency of the oligonucleotide probe was about 57.91% after labeling and purification. The amount of surface immobilized oligonucleotide probes was evaluated using the 32Plabeled oligonucleotide A, which is 5 amino-modified (see experimental section for details). By comparing the radioactivity counting on the PMMA-CHO and pristine PMMA slide surface and with the knowledge of the amount of oligonucleotide probes on pristine PMMA slide, the immobilized oligonucleotide probe on aldehyde PMMA surface was estimated to be 0.2 pmol for the 1 L spotting volume of 10 M oligonucleotide probe solution. The calculated probe surface density corresponding to the above value was approximately 1.065 x 105 probes/m2. Compared
79
to the PMMA surface amine density (~ 4.623 x 107 /m2) and the aldehyde conversion efficiency of 76%, approximately 1% of the functional groups were used for the attachment of oligonucleotide probes (with a length of 34-mer). Optimization of Hybridization The solid-phase-immobilized oligonucleotide probes have a different hybridization characteristic from that observed in solution. The tethering of oligonucleotide probes at one end to a surface is expected to affect its formation of duplexes with a target in solution [27]. Normally, a spacer was added between the solid substrate and the actual oligonucleotide sequence to reduce interferences of the support on the hybridization behavior of immobilized oligonucleotides. It has been shown that the length, charge, hydrophobicity and solvation of the spacer molecules are important in affecting duplex yields [28]. In Guos paper, it was reported that when oligonucleotide probes were directly coupled to glass substrates, no hybridization signal was observed. It was noticed that the hybridization yield increased strongly with the addition of poly dT spacers of 15 nucleotides between the probe and the substrate surface [24]. Therefore, such a spacer was integrated into the oligonucleotide probes in our experimental work as well. In the following sections, the surface amine group density, the buffer concentration of hybridization solution, and the length of hybridization time on the hybridization yields will be discussed.
PMMA-Surface Amine Density on Hybridization Productivity
Increasing the number of functional amine sites on the PMMA may not necessarily result in increased loading of complements or a higher hybridization signal on the surface following hybridization due to steric considerations when using fluorescence readout [24] [28]. Therefore,
80
we carried out experiments in which oligonucleotide probes were tethered to sheet PMMA and an IRD800-labeled oligonucleotide complement hybridized to the array as a function of the amination time, which was used to control the PMMA surface amine density. The results of this study are shown in Figure 3.6.
3.5
fluorescence arbitrary intensity (x 10 )
4
3.0
2.5
2.0
1.5
1.0
0.5 0 100 200 300 400 500
aminolysis time (min)
Figure 3.6. The fluorescence intensity produced from a hybridization assay as a function of aminolysis time. Probe A was immobilized on PMMA surfaces, which were treated in the 0.13 M n-lithioethylenediamine solution at room temperature for different lengths of time ranging from 5 min to 5 hrs. The PMMA-NH2 slides were soaked in 5% glutaric dialdehyde solution for 3 hrs and 2 M oligonucleotide A in phosphate buffer was spotted onto the slides. The slides were hybridized in 10 nM M13 Fwd-IRD 800 dye labeled primer, 5x SSPE, 0.1 % SDS solution at 37oC for 1 hr and the fluorescence intensity on the oligonucleotide probes were extracted as detailed in Experimental Section. As can be seen, the fluorescence intensity did not show a direct correlation with amination time and therefore, surface amine density (see Figure 3.1). Instead, the intensity decreased monotonically with amination time in the first 60 min and subsequently plateaued. As the amination time increased from 5 to 180 min, the amine surface density increased by 2.5-fold
81
whereas the fluorescence intensity decreased by 4.5-fold. Two reasons may be responsible for these observations; (1) self-quenching of fluorophores as they are brought into close proximity on the surface [28]; (2) steric hindrance of surface bound oligonucleotide probes at higher surface densities [27]. Therefore, short amination times (in the range of 5-30 min) were chosen to modify the PMMA surface, which resulted in maximum fluorescence signals following hybridization.
The Effect of Buffer Concentration
In Table 3.2 are listed the SNR values coming from the PMMA surface-immobilized oligonucleotide probes hybridized in dye-labeled complementary sequences prepared in different buffer solutions, 5x SSPE and 1x SSPE (see Experimental Section for details). Table 3.2. SNR values in different hybridization buffers Oligonucleotide probes Hybridization buffer Zip 3 Zip 11 1 x SSPE 654.72  78.57 912.58  155.14 5 x SSPE 1497.9  224.68 1953.75  312.6 From the Table, both oligonucleotide probes (Zip 3 and Zip 11) gave higher SNR values in 5x SSPE buffer compared to 1x SSPE buffer, which indicated that higher buffer concentrations promote better hybridization yields. In higher buffer medium, the negatively charged oligonucleotide backbones were effectively shielded by the salt to make it easier for the complementary sequences to form stable duplexes [29].
Hybridization Kinetics
Heterogeneous hybridization, in which the molecular recognition is between a singlestrand DNA or RNA immobilized to a solid support, and the labeled single strand targets in solution, has a slower hybridization rate compared to the solution phase (homogeneous) case [29-30]. Figure 3.7 shows the normalized fluorescence intensities acquired from four 82
hybridization assays carried out on PMMA slides with hybridization times ranging from 5 mins to 5 hrs. In this format, the time required for the fluorescence intensity to saturate was approximately 3 hrs.
1.4
normalized fluorescence intensity
1.2
1.0
0.8
0.6
0.4
0.2 0 50 100 150 200 250 300
time (min)
Figure 3.7. The hybridization kinetics on flat PMMA slides. 2 M oligonucleotide phosphate solutions were spotted onto the modified flat PMMA surface in a 2 x 2 pattern, 0.2 L/spot. After capping of the residual surface aldehyde groups, PMMA slides were soaked in a 10 nM hybridization solution at 37oC for different lengths of time ranging from 5 mins to 5 hrs. The average fluorescence intensity was extracted from each spot and normalized to the highest fluorescence intensity at each length of time. Capture Sensitivity of Oligonucleotide Arrays Figure 3.8 shows the detection sensitivity of oligonucleotide arrays in a logarithmic scale. The PMMA slides with oligonucleotide arrays were hybridized in 10 pM - 100 nM dye-labeled complement solutions at 37 oC for 30 min. The SNR on the oligonucleotide spots were calculated and plotted against the solution concentration. As can be seen from the graph, the SNR increased linearly with concentration up to ~10 nM and tended to saturate above that due to the limited
83
number of capture sites. At the lowest concentration investigated (10 pM), we observed a SNR of 2.05  0.28.
10000
1000
SNR
100
10
1 0.01 0.1 1 10 100 1000
hybridization solution concentration (nM)
Figure 3.8. Fluorescence intensity versus concentration of dye-labeled oligonucleotides in the hybridization solution. Each assay was hybridized in different concentrations of the complement (M13 Fwd (-29) IRD800) at 37oC for 30 min. The arrays were scanned with the near-IR scanner and the fluorescence intensity extracted from each spot with the average values plotted for each concentration value. However, the detection concentration could be further decreased by lengthening the hybridization time. For example, when the oligonucleotide slide was allowed to hybridize in 1 pM M13 Fwd (-29) IRD800, 5 x SSPE, 0.1 % SDS buffer solution overnight, the SNR value was calculated to be 18.06  2.79. Thus, the surface-immobilized probes serve as trapping sites to enrich the dye-labeled complement in solution to further lower the detection limit. Stability of Linkage Chemistry We used arrays spotted onto PMMA sheets to examine the stability of the linkage chemistry toward typical denaturing conditions. The DNA duplexes on the surface carrying a 84
fluorescent label were denatured and rehybridized and the fluorescence intensity examined after each cycle. The hybridized PMMA slides were denatured in 1:1 formamide: NaOH (0.5%) at 95
o
C for ~10 min. After each denaturation step, the slide was scanned using the fluorescence
microscope to examine for the absence of fluorescence on the surface. Two different linkage chemistries were investigated, imine-bond between the amine and surface tethered aldehyde and the resultant secondary amine formed after reduction of the imine linkage. We found that after two cycles of denaturation/hybridization, the imine-linkage produced hybridization signals that were reduced to ~5% of their original intensities (data not shown). However, when the iminelinkage was reduced to a secondary amine, the PMMA-based arrays did not show appreciable degradation of signal even after 12 cycles (see Figure 3.9).
1.2 1.1 1.0 0.9
normalized intensity
0.8 0.7 0.6 0.5 0.4 0.3 0.2 0.1 0.0 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15
times
Figure 3.9. Stability of the oligonucleotide-PMMA coupling chemistry as a function of denaturation /hybridization cycles. The average hybridization fluorescence intensity (circles) from five different spots of an array are shown. The aminated PMMA slides were soaked in 5% glutaric dialdehyde solution with 10 L/mL sodium cyanoborohydride added and the capping step finished in NaBH4 for 15 min. Between each hybridization step, the denatured slides were imaged with the fluorescence scanner (squares) to ensure that all labeled oligonucleotide had been removed from the slide. Denaturation was carried out by heating the slide to 95oC in formamide/NaOH (0.5%) for ~ 10 min. The probes were complements to the M13 Fwd (-29) IRD 800 oligonucleotide. 85
This result was attributed to the selective reduction by sodium cyanoborohydride of the imine bond. The sodium cyanoborohydride converts the less stable imine to a more stable secondary amine. When the oligonucleotide slides were capped in sodium borohydride, residual aldehyde groups were transformed to primary alcohols and also the imine bonds between the oligonucleotide probes and the cross-linker were reduced (see Scheme 3.1). We noticed that following capping using sodium borohydride to produce the alcohol of the reactive aldehyde, very little non-specific adsorption of solution complement to the surface was observed as well. This should result in improved imaging and signal-to-noise in the readout phase of the assay.
Conclusions
In this chapter, a method for attachment of oligonucleotide probes to chemically modified PMMA substrates through covalent bond formation was established. The linkage chemistry consisted of the use of a dialdehyde bi-functional linker, forming an imine bond between the surface amine group and the amine-terminated oligonucleotide probe, which was subsequently reduced to a secondary amine. Favorable linkage chemistry stability was obtained, which allowed re-interrogation of the array. The hybridization performance of the surface-immobilized probes was also evaluated. These surface-immobilized probes were efficient in capturing dyelabeled complements in solution as low as 1 pM target concentration.
References
1. 2. 3. 4. Polsky-Cynkin, R., Parsons, G.H., Allerdt, L., Landes, G., Davis, G., Rashtchian, A., Clin Chem, 1985, 31, 1438-1443. Langdale, J.A., Malcolm, A.D.B., Gene, 1985, 36(3),. 201-210. Langdale, J.A., Malcolm, A.D.B., Bioche. Soc. T., 1984, 12(4), 693-694. Nagata, Y., Yokota, H., Kosuda, O., Yokoo, K., Takemura, K., Kikuchi, T., FEBS Lett., 1985, 183(2), 379-382. 86
5. 6. 7. 8. 9. 10. 11. 12. 13.
Southern, E.M., J. Mol. Biol., 1975, 98(3), 503-517. Schuster, V., Matz, B., Wiegand, H., Traub, B., Kampa, D., Neumannhaefelin, D., J. Infect. Dis., 1986, 154, 309-314. Nickerson, D.A., Kaiser, R., Lappin, S., Stewart, J., Hood, L., Landegren, U., Proc. Natl. Acad. Sci. USA, 1990, 87, 8923-8927. Nikiforov, T.T., Rogers, Y.H., Anal. Biochem., 1995, 227(1), 201-209. Rasmussen, S.R., Larsen, M.R., Rasmussen, S.E., Anal. Biochem., 1991, 198(1),138-142. Kohsaka, H., Taniguchi, A., Richman D.D., Carson, D.A., Nucleic Acids Res., 1993, 21(15), 3469-3472. Matson, R.S., Rampal, J., Pentoney, S.L., Anderson, P.D., Coassin, P., Anal. Biochem., 1995, 224(1), 110-116. Matson, R.S., Rampal, J.B., Coassin, P.J., Anal. Biochem., 1994, 217, 306-310. Nikiforov, T.T., Rendle, R.B., Goelet, P., Rogers, Y.H., Kotewicz, M.L., Anderson, S., Trainor, G.L., Knapp, M.R., Nucleic Acids Res., 1994, 22, 4167-4175. Manz, A., Harrison, D.J., Fettinger, J.C., Verpoorte, E. Ludi, H., and Widmer, H.M. Integrated electroosmotic pumps and flow manifolds for total chemical analysis systems. in Transducers '91. Dig. Tech. Papers, 1991. Soper, S.A., Ford, S.M., Qi, S., McCarley, R.L., Kelly, K., Murphy, M.C., Anal. Chem., 2000, 72(19), 642A-651A. Liu, Y.J., Rauch, C.B., Anal. Biochem., 2003, 317(1), 76-84. Duran, L., Swanson, M. J., Amos, R.A., Hu, S.P.J., Guire, P. E., Reagent and method for attaching target moleules to a surface. 1999, USA. Schena, M., DNA Microarrays: A practical approach, 1999, Oxford University press, Oxford. Henry, A.C., Tutt, T.J., Galloway, M., Davidson, Y.Y., McWhorter, C.S., Soper, S.A., McCarley, R.L., Anal. Chem., 2000, 72, 5331-5337. Waddell, E., Wang, Y., Stryjewski,W., McWhorter, S., Henry, A. C., Evans, D., McCarley, R. L., Soper, S. A., Anal. Chem., 2000, 72, 5907 - 5917.
14.
15. 16. 17. 18. 19. 20.
87
21. 22. 23. 24. 25. 26. 27. 28. 29.
Ford, S.M., Kar, B., McWhorter, S., Davies, J., Soper, S.A., Klopf, M., Calderon, G., Saile, V., J. Microcolumn Sep., 1998, 10, 413-422. Ford, S.M., Davies, J., Kar, B., Qi, S.D., McWhorter, S., Soper, S.A., Malek, C.K., J. Biomech. Eng-T. ASME, 1999, 121(1), 13-21. Gaur, R.K., Gupta, K.C., Anal. Biochem., 1989, 180(2), 253-258. Guo, Z., Guilfoyle, R.A., Thiel, A.J., Wang, R., Smith, L.M., Nucleic Acids Res., 1994, 22, 5456-5465. Hermanson, G.T., Bioconjugate techniques, 1996, Academic Press, San Diego, 169-173. Roychoudhury, R., Jay, E., Wu, R., Nucl. Acids. Res., 1976, 3, 863-877. Southern, E., Mir, K., Shchepinov, M.S., Case-Green, S.C., Southern, E.M., Nat. Genet., 1999, 21, 5-9. Shchepinov, M.S., Case-Green, S.C., Southern, E.M., Nucleic Acids Res., 1997, 25, 1155-1161. Asim, K.B., Nucleic Acid Hybridizations: Principles and Strategies, in Nucleic Acid Analysis: Principles and Bioapplications, Dangler, A.C., Editor, 1996, Wiley-Liss, Inc., New York, 1-29. Gillespie, D., Spiegelman, S., J. Mol. Biol., 1965, 12, 829-842. Britten, R.J., Davidson, E.H., Hybridization strategy, in Nucleic Acid Hybridization: A Practical Approach, Hames, B.D. and Higgins, S.J., Editor, 1985, IRL Press, Washington DC.
30. 31.
88
Chapter 4 Microfluidic PMMA Oligonucleotide Array and Its Application for Point Mutation Detection*
Introduction
With the completion of the Human Genome Project, an array of diagnostic markers has been produced for predicting or monitoring the course of treatments for genetic-based diseases. For example, sickle cell anemia has been linked to a single base mutation in the -globin gene (substitution of an A in the wild-type with a T at the second position of the sixth codon of the  chain gene in the mutated form). Many diseases have also been determined to carry several different mutations in different genes, all of which can be used for the diagnosis of that phenotype. For example, colorectal adenomas and cancers have been determined to possess point mutations in the K-ras genes (19 different point mutations), which can occur early in the development of the disease in nearly 30-50% of patients [1-4]. One of the analytical challenges associated with detecting genetic mutations for diagnostics is that the mutation of interest (mutant DNA) may be present in low copy number (minority) in a mixed population of higher copy number wild-type or normal DNA. For
example, even at the primary tumor site for many cancers, the normal stromal cell content can be as high as 70%. Therefore, if the mutation is present in only one of the two chromosomes of a tumor cell, the mutated DNA can be present in as little as 15% of the DNA found in a sample for the target gene [5]. This number goes down precipitously if the sampling is done away from the primary tumor site.
* Reproduced in part with permission from Analytical Chemistry, volume 75, number 5, pages 1130-1140, 2003. Copyright 2003 American Chemical Society.
89
In order to detect low abundant mutations for diagnostics it becomes necessary to include a step into the assay that can distinguish with high sensitivity wild-type from mutant DNA. One such technique is the ligase detection reaction (LDR) coupled to a primary PCR reaction [6-13]. A diagram of the PCR/LDR technique is depicted in Figure 4.1.
12
13
61
Genomic DNA
PCR amplification 12 13 3 5
PCR Products
Discriminating Probe Ligate
5 3
X
cZ2
Ligate
Common Probe
LDR Products
cZ1
3
GT C
Hybridization
F
5
3
Hybridization
AT C
F 5
GT G GT G G G
F
F
Universal Zip Code Array
A A A A A A Z2 Solid Support
Z1
Figure 4.1. Schematic of LDR reactions and hybridizations to zip code arrays. The target region in the genomic DNA is first amplified using PCR technique. Then, discriminating probe corresponding to the specific mutation point and the dye-labeled common probe are hybridized to the target flanking the detection point. The two oligonucleotide probes become joined (ligated) in the presence of a mutation at the junction point. The complements of the surface-immobilized probe on the upstream (5end) of the discriminating probe guide the ligated product to the specific position on the array surface and the detection of the fluorescence on the specific position indicates the presence of a specific mutation. If no mutation is present at the junction point, the flanking probe will not be ligated and no fluorescence signal is detected. Following the PCR reaction, which contains the section of the gene that possesses the point mutations, the amplicon is mixed with two kinds of LDR probes that flank the particular
90
point mutation of interest (the common probe and the discriminating probes). The discriminating probes contain a base at its 3 end that coincides with the single base mutation site. If there is a mismatch, ligation of the two probes does not occur. However, a perfect match results in the successful ligation of the two probes and produces a product that can be analyzed in a variety of fashions. The advantages of the PCR/LDR assay are that it can be configured to do highly multiplexed assays, uses a thermally stable ligase enzyme (Tth ligase) to linearly amplify the LDR product to aid in detection and LDR can potentially detect 1 mutant DNA in &gt;1,000 copies of normal DNA. Recently, universal zip-code arrays have been developed for screening many single base mutations in low abundance and can be used as a universal platform for molecular recognition [5] [14-15]. The array format (see Figure 4.1) uses small probes that serve as zip codes (24-mers with similar Tm values) and contain unique sequences targeted for a specific mutation to be analyzed. The PCR/LDR uses discriminating and common probes similar to that described above. However, the allele-specific probes contains on its 5 end, a zip code complement that directs the ligation product to a particular site on the array. The common probe contains a fluorescent dye on its 3 end. If the mutation is present, LDR ligates the two probes together and generates a fluorescence signal at that particular element of the array. Even in the absence of the mutation, the discriminating probes hybridize to their complements, but do not generate a fluorescence signature. The challenge in this assay format is that unligated discriminating probes compete with ligated product for a fixed number of sites at the particular element of the array. In order to enhance the signal, Gerry and co-workers used gel pads at each element of the array to add a third dimension to increase the loading level of zip code probes [5]. The attractive feature of these universal arrays is that they can be used to detect a variety of different mutations by
91
simply appending the correct zip code complement sequence to the discriminating probe used in the LDR step. DNA microarrays can be configured to detect sequence variations at many different sites simultaneously and as such, can be used for diagnostic applications [16-19]. Some of the limitations associated with microarrays include the extended hybridization times required to elicit an optimal response (diffusion controlled passive hybridization) and the large amount of sample required to interrogate the array. In this chapter, we wish to report the development of universal arrays in polymer-based microfluidic chips for the detection of low abundant mutations. The material of choice for the fluidic device was poly(methylmethacrylate) (PMMA), since we have previously outlined procedures for covalently attaching oligonucleotide probes to the surface of this material [20-21]. The attractive feature of merging microarray technology with microfluidic platforms include the reduced amount of sample required to address each element of the array, the enhanced mass transport to the array surface to reduce analysis time, the ability to monitor several samples simultaneously using multi-channel chips and the closed architecture of the microfluidic channel reducing the potential for contamination. In this work, a PCR/LDR assay was carried out on K-ras genes to detect the presence of point mutations possessing clinical relevance for diagnosing colorectal cancers. The array
containing the zip codes were printed into microfluidic channels hot embossed from PMMA using pre-synthesized oligonucleotides. Following production of the array, the microchip was assembled with a cover plate and the array accessed using dynamic hybridization in the microfluidic channel. The reduced hybridization kinetics in the microfluidic channel and the
92
high selectivity (low amounts of mutant DNA in high copy number wild-type DNA) for the hybridization assay will be demonstrated.
Experimental Section
Materials and Reagents The sequences and their melting temperatures of the oligonucleotides used in LDR reactions and as the immobilized oligonucleotide probes are listed in Table 4.1. All these oligonucleotides were from Integrated DNA Technologies (IDT, Coralville, IA) except the dyelabeled primers, which were ordered from LI-COR Biotechnology (Lincoln, NE). Table 4.1. The sequences of oligonucleotides used in these experiments Oligonucleotides C6 amino 34-mer M13 Fwd (-29)-IRD800 cZip11-K-ras c12.2V (cZip11) K-ras c12 Com-2 (Com-2) Zip code 1 Sequences 5-3 *C6 Amino-TTT TTT TTT TTT TTT GTC GTT TTA CAA CGT CGT G IRD 800 C ACG ACG TTG TAA AAC GAC CGC AAG GTA GGT GCT GTA CCC GCA AAA CTT GTG GTA GTT GGA GCT GC *p-TGG CGT AGG CAA GAG TGC CT-IRD41 Tm(oC) 56.6 48.9 74.5 69.7
TGC GAC CTC AGC ATC GAC CTC AGC-*C7 66.4 Amino CAG CAC CTG ACC ATC GAT CGC AGC -*C7 65.5 Zip code 3 Amino GAC CAC CTT GCG ATC GGG TAC AGC-*C7 66.8 Zip code 5 Amino TGC GGG TAC AGC ACC TAC CTT GCG -*C7 Zip code 11 66.7 Amino *p- phosphorylate modification at terminal end *C6, C7- the linker molecules between amino modification and oligonucleotide sequences PMMA used as the universal array and microfluidic chip substrates were purchased from GoodFellow (Berwyn, PA). The PCR products of K-ras-c12.2 wild-type (WT) and V mutant (c12.2 V), were kind gifts from Dr. Baranys group at Cornell Medical School and the stock
93
solution came in as ~ 100 ng DNA/3 L. Ampligase DNA ligase and 10x reaction buffer (200 mM Tris-HCl, pH 8.3, 250 mM KCl, 100 mM MgCl2, 5 mM NAD, and 0.1% Triton X-100) were purchased from Epicentre Technologies (Madison, WI). Near-IR fluorescence dye-labeled DNA sizing ladders (50~350-bps), 5.5 % gel matrix, 10x TBE buffers (0.89 M Tris-Base, 0.89 M Boric Acid, 20 mM EDTA, pH 8.3), and blue loading buffer were from LI-COR Biosciences (Lincoln, NE). Other chemicals were from Aldrich-Sigma and used without further purification. Instrumentation A LI-COR Model 4200 DNA Sequencer [22] was used for the LDR product examination. A solid-state dual-near-infrared-laser excitation source scanning device was equipped with one of the diode lasers lasing at 780 nm and the other at 680 nm. The detector was a solid-state silicon avalanche diode and was mounted at Brewsters angle (56o) with respect to the incident light coming from the laser diode beam at the point of observation (the gel between the glass plates). The band pass regions of the detectors were set at 820 nm and 720 nm. A belt drove the laser and detector assembly back and forth to scan the electrophoretic gel apparatus. Ordinary soda-lime glass plates were used to cast the thin (0.2 mm) polyacrylamide gels for separation and 1x TBE buffer (pH = 8.3) was used. A Near-IR Fluorescence Scanner was used for the detection of hybridization events on both flat PMMA substrates and in microfluidic channels. The collection time was set at 0.1 s per pixel. The scanning was performed at room temperature and the scanning step was selected according to the sample size and the resolution quality needed. The smallest step used was 12.7 m and the most commonly used step was 101.6 m.
94
PMMA Chip Fabrication and Assembly of Microfluidic Channels The PMMA chip mold insert contained four sets of microfluidic channels (50 m in depth, 500 m in width and 0.7 cm in length), which were fabricated via the LIGA (a German acronym for x-ray lithography, electrodeposition and molding) process. Similar protocols to fabricate a mold insert have been reported elsewhere [23-25] and the process adopted in this work is outlined as follows. The x-ray mask was constructed on a Au (500)/Cr(100) coated 25 m polyimide membrane with 6 m eletrodeposited gold absorbers using UV lithography and a sulfite-based gold electrolyte solution. The PMMA resist sheet was then bonded to a 306 stainless steel substrate (12 cm in diameter and 6.25 mm in thickness) and fly cut to the desired thickness. Next, the PMMA was exposed with 1.3 GeV x-rays at a critical wavelength (c = 7.4 ) using the synchrotron at the Center for Advanced Microstructures and Devices at the Louisiana State University (LSU-CAMD). A beryllium foil of 175 m in thickness was used as a preabsorber. The absorbed energy at the PMMA bottom was set at 3000 J/cm2 and the ratio of the energy absorbed at top to that at the bottom was 2.2. After development in GG developer (GG developer used for removing x-ray exposed PMMA consisting of 2-(2-butoxyethoxy)ethanol 12 vol. parts; morpholine 4 vol. parts; 2-aminoethanol 1 part; and water 3 parts) and rinsed, nickel was electrodeposited into the PMMA developed areas from a sulfamate-based electrolyte. Finally, lapping the deposited nickel surface followed by stripping the remaining PMMA completed the mold insert fabrication. The device topology is depicted in Figure 4.2. Fluid access channels (DNA sample and wash buffers) were 100 m in width and 50 m in depth. Both channels merged into one common channel and emptied into the hybridization chamber, which measured 500 m in width, 50 m in depth, and 6.7 mm in length (volume = 168 nL).
95
This mold insert was then utilized as the master to replicate PMMA chips using a hot embosser (PHI Precision Press, City of Industry, CA). The PMMA plate (5 mm thickness from GoodFellow) was cut into  = 5 inch discs and dried at 85oC overnight. The disc was placed at the bottom of the chamber that was evacuated and was embossed against the molding die at a pressure of 3.2-3.6 bars at 135oC for 4.5 mins. After demolding at the same temperature, the disc was post-baked under a weight at 80oC for 8-10 hrs. Finally, the disc was cut into  = 3 inch chips and small holes of 0.76 mm in diameter were mechanically drilled through the reservoir sites for liquid inlet and outlet connectors.
Molding Die
Waste
DNA Target
Wash Buffer
Topographical Layout of Chip
Figure 4.2. Picture of the Ni molding master fabricated using LIGA and a topographical layout of one of the hybridization microfluidic networks. There are two microchannels for feeding the hybridization chamber: one to deliver the DNA sample to the hybridization chamber and the other, wash buffer. The feed channels were 100 m in width and 50 m in depth. The hybridization chamber was 500 m in width and 50 m in depth. All dark lines represent raised structures on the molding die. The large horizontal rectangles flanking the fluidic channels were placed on the molding die to assist in bonding of the cover plate to the substrate.
96
The surface of the PMMA microfluidic chip was activated following the same procedures developed for flat PMMA slides (see Chapter 3). Oligonucleotide probes were spotted in the main channels with the help of a capillary tube ( = 105/41 m) with four spots per channel (&lt;0.1 L/spot with a diameter of ~ 400 m). After the deposition of oligonucleotide probes (5 amino-modified oligonucleotide probe C6-amino 34-mer) and the deactivation of residual surface aldehyde groups was completed, another 250 m thick PMMA cover plate (40 mm x 40 mm) was directly bonded to the structured surface by clipping between two glass slides and heated in a GC oven at 110oC for 15 min to seal the channels. Dynamic Hybridization in PMMA Microfluidic Channels After the PMMA chip was assembled, two 30-cm capillary tubes ( = 375 / 75 m) were sealed with 5-min epoxy to the outlet and inlet of the chip. Ten nM dye-labeled complement sequences of M13 Fwd (-29)-IRD 800 in 5x SSPE, 0.1% SDS buffer solution was pumped through the channel using a N2 pneumatic cylinder pump for a specific length of time. After washing the channel with washing buffer, the chip was scanned under the near-IR scanner. LDR Products Detection in PMMA Microfluidic Zip-Code Arrays LDRs were executed in a total volume of 80 L in 0.5-mL polypropylene microtubes. The reaction cocktail consisted of 4 L of 1 M c-zip 11 discriminating primer, 4 L of 1 M com-2 fluorescently-labeled primer, 1 L of 200 mM DTT, 1 L of 10 mM NAD and various ratios of mutant (K-ras c12.2v) and wild type PCR products with the total DNA mass set at 0.1 pmol for each reaction. The concentration ratio of the mutant to wild-type DNA was adjusted in the LDR cocktail to 0:1 (mutant:wild type, control) to 1:10,000. The LDR cocktail was taken to a final volume of 80 L by adding 2x reaction buffer. The LDR mix was then pre-heated to 94
o
C for 2 min followed by the addition of 1 L of Ampligase enzyme (5 units/L, Epicentre). The 97
reaction mix was subjected to 40 LDR thermal cycles using the following temperatures; 94 oC for 30 s; 65 oC for 2 min. To test the fidelity of the LDR reaction, slab gel electrophoresis was run on an aliquot of each reaction (1 L of reaction mixture mixed with 1 L of loading dye and 1 L then loaded into individual wells of the slab gel). The electrophoresis was accomplished using a cross-linked polyacrylamide gel (5.5% gel matrix, LI-COR Biotechnology, Lincoln, NE). The gel was polymerized between two soda-lime glass plates (21 cm x 47 cm) and placed in an automated DNA sequencer (LI-COR 4200, LI-COR Biotechnology, Lincoln, NE). electrophoresis was typically run at 1 500 V for 2 hrs. Zip code oligonucleotide probes 1, 3, 5, and 11 (see Table 4.1 for sequences) were dissolved separately in 0.2 M phosphate buffer (pH 8.3) to a final concentration of 0.1 mM. The arrays were micro-printed on the activated PMMA using a micropipette. The volume deposited was less than 0.1 L of the probe, which resulted in a spot of approximately 400 m in diameter. The spot for each probe was placed along the length of the hybridization chamber. Following deposition of the four probes and capping, the chip was assembled by placing a cover plate over the microfluidic channels, clamping the assembly together and thermally annealing using the procedure described previously. The microchip was then flushed with a rinsing buffer using a syringe pump (Harvard PHD Syringe Pumps, Holliston, MA). The LDR product was pumped through the hybridization chamber at a volumetric flow rate of 1 L/min with the chip heated to ~50 oC for 2 mins. This was followed by a subsequent wash with the hybridization buffer (2.0 L/min) for 15 mins and finally imaging of the chamber with the near-IR laser-induced fluorescence scanner. We could estimate the capture efficiency (CapE) of solution complements in the microfluidic chip from knowledge of the residence time of the complement above its appropriate The
98
probe (t), the diffusion coefficient of LDR product (D), the height of the hybridization chamber (h) and the input concentration (Cin) using;
CapE =
&lt; Cout &gt; [h  2 Dt ] =1 Cin h
(4.1)
where &lt;Cout&gt; is the average output concentration and t is calculated from the linear velocity and the size of the array spot (400 m). Inserting appropriate values into equation 1 (h = 50 m; t = 0.61 s; D = 43 x 10-8 cm2/s) [26] gave a value of 0.14 for CapE using our microchip.
Results and Discussions
Hybridization Kinetics in Microfluidic Channels
Hybridization assays were performed in hot-embossed PMMA microfluidic channels and compared to hybridization signals obtained from flat PMMA slides. Figure 4.3 shows the normalized fluorescence intensities obtained from four oligonucleotide spots in the fluidic channel with hybridization incubation times ranging from 24 s to 40 min. The data suggested that the fluorescence intensity saturated at or before our first data point in the microfluidic format and remained nearly constant for longer hybridization times. As a comparison, the time required for the fluorescence intensity to saturate was much longer (~3 hrs) on flat PMMA slides (see Chapter 3). The forced flow of target solution over the probe-tethered surface provides enhanced mass transfer of targets to the surface immobilized probes reducing hybridization time. In addition, the small volume of the hybridization chamber reduces the diffusional distances required for target to reach the surface probes.
99
1.0
normalized fluorescenc e intens ity
0.8
0.6
0.4
0.2
0.0 0.4 10 15 25 30 40
hybridization time (min)
Figure 4.3. Hybridization kinetics in PMMA microchip channels. Oligonucleotide probes C6 amino 34-mer was immobilized to the bottom of PMMA microfluidic channels, 4 spots/per channel. Ten nM M13 Fwd (-29)-IRD800 in 5x SSPE, 0.1% SDS buffer solution was pumped through the channel for different lengths of time via a syringe pump and washed with wash buffer. The chip was scanned under the near-IR scanner, the fluorescence intensity was normalized to the highest value and the average values on four spots were plotted.
An additional advantage associated with the PMMA microfluidic arrays is that the required assembly temperature is much lower than that required for glass [27]. Therefore, the oligonucleotide probes can be coupled to the surface prior to chip assembly provided that the probes do not suffer from chemical degradation during heat annealing. We found no appreciable differences between the fluorescence signal generated from arrays situated in microfluidic chips and those placed on planar PMMA surfaces, indicating little if any chemical degradation of the spotted probes during device assembly (data not shown).
100
Detecting K-ras Mutations Using PMMA Microfluidic Arrays
75-mer
LDR product
50-mer
Com-2
0 1 2 3 4 5 6 7 Figure 4.4. LDR reactions of K-ras c12.2 v mutant and wild-type DNAs with different ratios of mutant to wild type. Lane 1 - control (no mutant); lane 2 - 1:1 (mutant:wild-type); lane 3 - 1:10; lane 4 - 1:100; lane 5 - 1:1 000; and lane 6 - 1:10 000. Lanes 0 and 7 contained a single stranded DNA mass sizing ladder with the 50 and 75mer fragments labeled in the figure.
As a demonstration of the use of microfluidic arrays for the detection of low-abundant point mutations, 24-mer zip codes were spotted into hot-embossed PMMA microfluidic channels. A LDR was used to generate ligation products containing the zip code complement. The LDRs were carried out in a series of solutions consisting of c-zip 11 K-ras c12.2v (the discriminating probe carrying the complementary sequence to zip code 11 at its 5-end), K-ras c12 com-2 (common probe containing fluorescent label at its 3-end and phosphorylated at its 5-end), and the K-ras c12.2 v mutant and wild-type PCR products. When the ligated dye-labeled LDR product and unligated discriminating primer were pumped through the channel it was captured by the appropriate zip code. The uncaptured product was removed by flushing with a wash buffer. 101
The fluorescent signal detected at a specific zip code position indicates the presence of the corresponding mutation in the examined sample. Figure 4.4 shows a slab gel image of the LDR products taken from a DNA sequencer using fluorescence detection. As can be seen in the control lane 1, only the com-2 primer was detected without LDR product present in that lane. The LDR product could be detected at mutant: wild-type ratios of 1:1 and 1:10, but was not visible in the gel tracks at lower ratios of mutant: wild-type DNA. The LDR product was not visible at these lower ratios due to the low amount of LDR product generated. We then ran a control sample through the chip and no obvious hybridization signal was detected. Then, we interrogated the 1:10,000 mutant: wild-type LDR reaction using the PMMA microfluidic device, which contained zip codes 1, 3, 5, and 11. The results are illustrated in Figure 4.5.
Zip-code 1 Zip-code 3 Zip-code5 Zip-code11
25 320 615 910 1205 1500
Intensity Scale (cps) Figure 4.5. Mutational analysis of 1:10,000 c12.2 V mutant to wild-type LDR products using the zip-code PMMA microchip array. The hybridization solution was pumped through the chip using a syringe pump at a flow rate of 1 L/min (50oC hybridization temperature). The chip was scanned using the near-IR scanner. As can be seen from the image displayed in Figure 4.5, a very intense signal appeared on zip code 11, which is the appropriate zip code for this particular mutation. Therefore, in spite of a large excess of wild-type DNA with a small amount of ligation product, the mutation could be detected. The hybridization of LDR products in the microchannel was conducted at 50 oC, at
102
which no cross-hybridization was observed due to high stringency of hybridization. The number of ligated products captured at the appropriate spot of the array is hard to estimate, but the upper limit can be determined from the surface concentration of functional amine sites at the optimized amination time (3.4 x 10-9 mol/cm2), the ratio of mutant to wild-type sequences (1:10,000), the number of LDR thermal cycles (40) and the capture efficiency (amount of solution complements reaching the surface, CapE = 0.14). Using these values, we estimated 2.8 x 10-13 mol/cm2 of dye molecules detected in the assay. If the spot on the array was 400 m in diameter, then the mass of dye comprising this signal would be 4.5 fmol if all potential sites were occupied. This hybridization signal was produced from a LDR solution with an approximate concentration of ligated products (hybridization solution) equal to 16 pM, well above our concentration detection limit for this hybridization assay (see Figure 3.8). The high sensitivity is due in part to enhanced rates of complement capture produced by the microfluidic format, the low levels of non-specific adsorption of DNA to the functionalized polymer surface and the use of near-IR fluorescence, which results in low background levels generated from the array substrate and reagents used in the assay [21] [28]. As can be seen from equation 1, the shallow height of the hybridization chamber was also critical so as to provide high capture efficiency for low concentration targets.
Conclusions
We have successfully demonstrated the ability to detect point mutations in certain gene fragments using universal zip code arrays placed into PMMA microfluidic channels fabricated using hot-embossing. Our experiments indicated the ability to detect one mutant DNA in 10,000 normal sequences. Compared to a planar surface array, arrays constructed in microfluidic channels were found to display enhanced hybridization kinetics. In addition, the low assembly temperature required for the device allowed pre-spotting the probes into an open channel format
103
and then assembling the cover plate to the device with no noticeable degradation in hybridization signal. This format will allow the use of conventional spotting machines to construct the arrays in the micro-channels and then accessing the arrays using micro-fluidics. The universal array format adopted here using zip code capture elements can serve as a viable platform for detecting a number of different mutations in different gene fragments due to the high stability of the linkage chemistry and the ability to effectively remove hybridized complements from the surface.
References
1. 2. 3. 4. 5. 6. 7. 8. 9. 10. 11. 12. Rothschild, C.B., Brewer, C.S., Loggie, B., Beard, G.A., Triscott, M.X., J. Immunol. Methods, 1997, 206, 11-19. Otori, K., Oda, Y., Sugiyama, K., Hasebe, T., Mukai, K., Fujii, T., Tajiri, H., Yoshida, S., Fukushima, S., Esumi, H., GUT, 1997, 40(5), 660-663. Chiang, J.M., Cancer Lett., 1998, 126, 179-185. Andersen, S.N., Lovig, T., Breivik, J., Lund, E., Gaudernack, G., Meling, G.I., Rognum, T.O., Scand. J. Gastroentero., 1997, 32(1), 62-69. Gerry, N.P., Witowski, N.E., Day, J., Hammer, R.P., Barany, G., Barany, F., J. Mol. Biol., 1999, 292(2), 251-262. Barany, F., Proc. Natl. Acad. Sci. USA, 1991, 88(1), 189-193. Wei, Q., Barany, F., Wilson, V.L., Mol. Biol. Cell, 1992, 3, A22-A22. Wiedmann, M., Barany, F., Batt, C.A., Appl. Environ. Microb., 1993, 59(8), 2743-2745. Wiedmann, M., Wilson, W.J., Czajka, J., Luo, J.Y., Barany, F., Batt, C.A., PCR Methods Appl., 1994, 3, S51-S64. Day, D.J., Apeiser, P.W., White, P.C., Barany, F., Genomics, 1995, 29(1), 152-162. Luo, J.Y., Bergstrom, D.E., Barany, F., Nucleic Acids Res., 1996, 24(15), 3071-3078. Khanna, M., Park, P., Zirvi, M., Cao, W.G., Picon, A., Day, J., Paty, P., Barany, F., Oncogene, 1999, 18(1), 27-38.
104
13. 14.
Tong, J., Cao, W.G., Barany, F., Nucleic Acids Res., 1999, 27, 788-794. Favis, R., Barany, F., Mutation detection in K-ras, BRCA1, BRCA2, and p53 using PCR/LDR and a universal DNA microarray, in Circulating Nucleic Acids in Plasma or Serum Annals of the New York Academy of Sciences, Anker, P. Stroun, M., Editor, 2000, New York Academy of Sciences,New York, 39-43. Favis, R., Day, J.P., Gerry, N.P., Phelan, C., Narod, S., Barany, F., Nat. Biotechnol., 2000, 18(5), 561-564. Consolandi, C., Castiglioni, B., Bordoni, R., Busti, E., Battaglia, C., De Bellis, G., Minerva Biotechnol., 2001, 13(4), 261-268. De Benedetti, V.M.G., Biglia, N., Sismondi, P., De Bortoli, M., Int. J. of Biol. Marker, 2000, 15(1), 1-9. Dimitrjevic, B., Jugoslav. Med. Biohem., 2001, 20, 65-71. Gupta, P.K., Roy, J.K., Prasad, M., Current Science, 1999, 77(7), 875-884. Henry, A.C., Tutt, T.J., Galloway, M., Davidson, Y.Y., McWhorter, C.S., Soper, S.A., McCarley, R.L., Anal. Chem., 2000, 72, 5331-5337. Waddell, E., Wang, Y., Stryjewski, W., McWhorter, S., Henry, A.C., Evans, D., McCarley, R. L., Soper, S. A., Anal. Chem., 2000, 72, 5907 - 5917. Middendorf, L.R., Bruce, R.C., Brumbaugh, J. A., Grone, D.L., Jang, G.Y., Richterich, P., Hoeltke, H. J., Williams, R.J., Peralta, J. M., Proc. SPIE, 1994, 2388, 44-52. Ford, S.M., Davies, J., Kar, B., Qi, S.D., McWhorter, S., Soper, S.A., Malek, C.K., J. Biomech. Eng-T. ASME, 1999, 121(1), 13-21. Soper, S.A., Ford, S.M., Qi, S., McCarley, R.L., Kelly, K., Murphy, M.C., Anal. Chem., 2000, 72(19), 642A-651A. Qi, S., Liu, X., Ford, S., Barrows, J., Thomas, G., Kelly, K., McCandless, A., Lian, K., Goettert, J., Soper, S. A., Lab Chip, 2002, 2, 88-95. Stellwagen, N.C., Magnusdottir, S., Gelfi, C., Righetti, P.G., Biopolymer, 2001, 58(4), 390-397. Reyes, D.R., Iossifidis, D., Auroux, P.A., Manz, A., Anal. Chem., 2002, 74, 2623 - 2636. Williams, D.C., Soper, S. A., Anal. Chem., 1995, 67, 3427 - 3432.
15. 16. 17. 18. 19. 20. 21. 22. 23. 24. 25. 26. 27. 28.
105
Chapter 5 UV-Patterning of Oligonucleotide Arrays on Poly (methyl methacrylate) and Polycarbonate Surfaces
Introduction
DNA microarrays employ thousands of single-stranded DNAs or oligonucleotides with known sequences, identified as probes, attached to solid substrate surfaces to interrogate the sequence information of target samples based on molecular recognition governed by WatsonCrick base pairing rules [1]. Due to its high throughput characteristics, they have become an indispensable tool for gene expression analysis [2], gene discovery [3], disease diagnosis [4], drug discovery [5], toxicological research [6], micro-organism detection, and single nucleotide polymorphism analysis [7]. To construct an array, knowledge of the probes, selection of the solid substrate, and the linkage chemistry used to attach the probes to the substrate surface are essential. DNA probes are classified in one of two categories according to the length of the arrayed sequences, short (12-30 bases) oligonucleotide probes or long cDNA or PCR product (&gt;100 bases) probes. Glass has been the most commonly used substrate for microarrays because of its wide-spread availability and well-known coupling chemistries. In recent years, polymers have become more attractive as solid substrates for arrays due to their wide choice in physical and chemical properties and advantages in fabrication of micro-analytical devices [8-10]. However, selection of linkage chemistries on polymers, which are substrate and probe dependent, is less developed than that for glass. Typically, long probes which have thread-like structures are cross-linked to the substrate surface along the chain at several random points using UV-irradiation. While for short probes, which generally have a rod-like shape, end attachment is preferred to retain their capability to
106
hybridize to complementary sequences. To realize this, modifications of the probes and solid surfaces are generally required for covalent end attachment. There are two approaches for DNA microarray constructions, in situ synthesis or postdeposition. In the first format, the nucleotide building blocks of the DNA probes are assembled onto a solid substrate in a step-by-step manner using a light-directed synthesis procedure, which has been employed for the manufacture of high density arrays [11-14]. A planar glass or fused silica substrate is first covalently modified with a silane reagent to provide hydroxyalkyl groups, which serve as the initial synthesis sites. These sites are extended with linker groups possessing terminal photolabile-protecting groups, which can be activated with light for the addition of nucleoside phosphoramidites. The monomers are also protected at their 5 (or 3) position with a photolabile group and the cycles of photo-deprotection and nucleotide addition are repeated to build the desired length of sequences for two-dimensional arrays. Due to the less than 100% efficiency of the photo-deprotection chemistry, the yield of the full-length product decreases dramatically with increases in the number of synthesis cycles. Therefore, in situ synthesis is more applicable for the construction of short oligonucleotide probe arrays (up to about 25 bases). In the second method, cDNAs, PCR products, or presynthesized oligonucleotides are deposited and immobilized onto specific locations of solid substrates. For example, DNA molecules are cross-linked to polylysine-derivatized glass surfaces [15] or the 5 aminomodified PCR products can be immobilized to aldehyde-containing surfaces [16-17]. For short oligonucleotide probes, some derivatization of the probe is necessary for the probe immobilization. Typically, a primary amine, thiol, or carboxylic acid functionality is attached to the oligonucleotide probe during the initial or final round of chemical synthesis to yield a 3 or 5 modification. For example, Guo et al. [18] reported that pre-synthesized 5-amino-modified
107
oligonucleotides can be attached to SiO2-modified silanized glass microscope slides with homobi-functional cross-linkers, such as 1, 4-phenylene diisothiocyanate (PDC). Compared to glass substrates, several polymer substrates have been investigated for the attachment of oligonucleotide probes. Among them, polystyrene (PS) microtiter plates are the most widely reported polymer substrate for the immobilization of DNA molecules [19]. Immobilization can be realized by passive adsorption [20], UV light [21], or biotin labeled DNA probes attached to streptavidin-coated microtiter plates [22]. In passive immobilization, DNA probes in high-cation buffer are incubated directly in micro-wells for probe attachment [19-20] [23]. Chemically modified polystyrene micro-well plates are also commercially available for the immobilization of oligonucleotide probes. One such product is the NucleoLink micro-well plate from Nalge Nunc International. The micro-well surface is amine-modified and capable of reacting with 5 phosphate groups on the DNA molecules in the presence of a carbodiimide (www.nunc.nalgenunc.com). Other polymer substrates, such as polypropylene (PP),
poly(methylmethacrylate) (PMMA), and polycarbonate (PC) have also been used. For example, a PP support was aminated in ammonia plasma to serve as a solid-phase support for in situ oligonucleotide probe synthesis [24], and PMMA was aminated in N-lithioethylenediamine solution and a homo-bi-functional crosslinker molecule to conjugate amino terminal modified oligonucleotide probes to the surface [25]. Several coating cocktails have been developed as well to facilitate the immobilization of oligonucleotides to polymer substrates. Reacti-Bind DNA coating solution is a commercial product developed by Pierce that is applicable to many kinds of polymer surfaces for DNA probe immobilizations. Liu, et al. [26] reported its application for attachment of oligonucleotide probes onto four polymer substrates, PS, PC, PMMA, and PP. Promising results were obtained only for
108
the PC substrate using this cocktail. Surmodics surface activation technology for DNA array manufacture has been recently developed at Motorola [27]. The cocktail contains photo-reactive groups for use in attaching the reagent composition to the solid surface and it also contains components that attract and covalently bind nucleic acid molecules. It was reported to be successful for the construction of DNA microarrays on PS, PC, PMMA, and PP substrates [26]. In the development of DNA probe immobilization techniques, covalent attachment of DNA molecules to solid substrates is preferred to non-covalent methods due to the controllable and reproducible immobilization characteristics [21], especially when developing methods that enable or improve the likelihood of end-point attachment of a synthetic oligonucleotide to a surface. End-point attachment (i.e., with the nucleic acid sequence attached through one or the other terminal nucleotide) is desirable because the entire length of the sequence is available for hybridization to another nucleic acid sequence, which is present in solution. This is particularly advantageous for the detection of single base pair changes requring stringent hybridization conditions. UV-photo modification of polymer substrates is an easy and quick process, which could potentially be used for polymer surface modifications. Individual polymers absorb UV radiation within specific wavelength regions where photo-oxidation is likely to occur [28]. Different wavelength maxima for PC photo-oxidation have been reported, which are 295 nm [28] and 264 nm [29]. The wavelength maximum for PMMA is between 290-315 nm. Photo-oxidation produces a variety of physical and chemical changes. Chain scission and cross-linking are the general reactions that take place, often accompanied by formation of oxygen-containing groups [30]. Also, photo-oxidation is primarily a surface reaction, suggesting that the effect of UV
109
irradiation may not extend into the polymer bulk to a significant extent. Permeation of oxygen into the polymer bulk could also be restricted within surface layers. Although several papers have been published in which UV irradiation are employed to modify polymer surfaces, none of them have dealt with the selective immobilization of oligonucleotide molecules on UV-patterned polymer substrates [31]. For example, UV (220 nm) irradiation was used to increase surface hydrophilicity and charge density on PC [31] or PMMA [32] substrates. When exposed to UV light in air, the surfaces of the polymers are oxidized, which leads to the introduction of functional groups and contributes to the increase of surface hydrophilicity. These functional groups consist of peroxides, hydroxyl, carboxylic, or carbonylic groups [29-30]. UV-irradiation was reported to produce carboxylate groups on polycarbonate (PC) surfaces and the carboxylated surfaces in PC microchannels were used as an immobilization bed in the SPRI process (Solid-Phase Reversible Immobilization) for DNA purification prior to gel electrophoretic sorting [33]. In this Chapter, we will report the use of UV broad band irradiation for the potential surface modification of PMMA and PC substrates for the spatially selective immobilization of oligonucleotide probes. To the best of our knowledge, this is the first report describing a photopatterning process for linking oligonucleotides to PMMA or PC substrates, both of which are in widespread use for fabricating micro-fluidic devices [8] [25] [31] [34-40]. To investigate the influence of pre-existing surface functional groups that may result from photo-chemical reactions triggered by ambient UV exposure [41], oligonucleotide probes were directly deposited onto the pristine polymer surfaces and their interaction with the polymer surface monitored through the hybridization of these probes to their dye-labeled complements. In separate experiments, oligonucleotide probe solutions were deposited onto the polymer surfaces, which were UV (193-
110
254 nm) patterned through a perforated metallic mask and subsequently examined by hybridization-based assays with fluorescence detection.
Experimental Section
Materials and Chemicals PMMA and PC sheets (both 1.0 mm thickness) were purchased from GoodFellow (Berwyn, PA). The oligonucleotide probes used in these studies, their sequences and providers are listed in Table 5.1. Chemicals such as 2-(N-morpholino)ethane sulfonic acid (MES), 1-ethyl3-(3-dimethylaminopropyl) carbodiimide hydrochloride (EDC or EDAC), 20x SSPE stock buffer containing 0.2 M phosphate, pH ~ 7.4, 2.98 M NaCl and 0.02 M EDTA, were obtained from Sigma-Aldrich and used without further purifications. Table 5.1. The oligonucleotide probe sequences and their providers. Oligonucleotides C6-amino 34-mer C6-amino 19-mer 30-dT 49-mer 15-dT 34-mer Zip code 1 Zip code 3 M13 Fwd (-29)IRD 800 cZip 3 cZip 11 Sequences (5-3) *C6 Amino -TTT TTT TTT TTT TTT GTC GTT TTA CAA CGT CGT G *C6 Amino- GTC GTT TTA CAA CGT CGT G TTT TTT TTT TTT TTT TTT TTT TTT TTT TTT GTC GTT TTA CAA CGT CGT G TTT TTT TTT TTT TTT GTC GTT TTA CAA CGT CGT G TGC GAC CTC AGC ATC GAC CTC AGC-*C7 Amino CAG CAC CTG ACC ATC GAT CGC AGC -*C7 Amino IRD 800 C ACG ACG TTG TAA AAC GAC IRD800 GCT GCG ATC GAT GGT CAG GTG CTG IRD800 CGC AAG GTA GGT GCT GTA CCC GCA Sources IDT IDT IDT IDT IDT IDT LI-COR LI-COR LI-COR
*C6, C7- the linker molecules between amino modification and oligonucleotide sequences
111
Instrumentation Contact angles were measured with a VCA 2000 Video Contact Angle System and the integrated software (AST Products, Inc. Billerica, MA). The UV station was equipped with a broad band deep UV light bulb (500 W DUV, Model UXM-501 MA) and was purchased from Ushio America (Cypress, CA). Radiation intensities at a distance of 1 cm from the source were measured to be 15 mW/cm2. Selective photo-patterning was performed using a perforated brass sheet mask with 500 m diameter through holes, 600 m apart. The near-IR scanner has been described in detail in Chapters 2 and 3. Deposition of Oligonucleotide Probes on Pristine Polymer Surfaces The PMMA and PC slides were ultrasonically cleaned in a water bath for 2 mins and rinsed with isopropanol (IPA) and ddH2O thoroughly followed by drying with pressurized air. The synthetic oligonucleotide probes were dissolved in 0.2 M phosphate buffer (pH 8.36) to make 100 M stock solutions. Then, the oligonucleotide solution was manually spotted with a micropipette onto the appropriately prepared PMMA and PC surfaces in a 3 x 3 pattern and the slides placed in a humidified chamber maintained at 37oC overnight. The slides were finally washed with ddH2O to remove excess oligonucleotide probes. Deposition of Oligonucleotide Probes on UV-Patterned Polymer Surfaces The PC and PMMA (20 mm x 20 mm x 1 mm) slides were cleaned in IPA ultrasonically for 2 mins and dried with pressurized air. The polymer surfaces were then selectively exposed using the through-hole metallic mask under deep UV light for 10 mins to receive a dose of 9000 mJ/cm2. After exposure, the slides were rinsed thoroughly with methanol and ddH2O and dried with pressurized air. A 100 M oligonucleotide probe (C6-amino 34-mer) solution was prepared in 0.1 M MES (pH 5.86) buffer containing 5 mM EDC. Five L of the prepared solution was
112
spotted onto the UV-exposed slides. Then, the slides were kept at 37oC in a humidified chamber overnight and washed in ddH2O to remove the excess oligonucleotide probe solution. Functional Assays of Polymer Surface-Immobilized Oligonucleotide Probes The prepared oligonucleotide slides were soaked in hybridization solutions (dye-labeled complement, M13 Fwd (-29)-IRD 800, in 5x SSPE, 0.1 % SDS for a specific length of time followed by washing in a washing buffer (2x SSPE, 0.1 % SDS) for 15 mins twice. Then, the slide was scanned with the near-IR fluorescence scanner and the hybridization signal was extracted from the spots using the in-house developed software (see Chapter 2 for details). The scanner settings used in this Chapter were; output power of the excitation laser lasing at 780 nm was 44 W, scanning resolution was either 101.6 m/step or 50.8 m/step, and the integration time per step was 0.1 s.
Results and Discussions
Deposition of Oligonucleotide Probes on Pristine Polymer Surfaces Prior to immobilization, surface properties of the pristine polymer substrates, such as background fluorescence levels and hydrophobicity were characterized. Since the background level affects the detection limit in fluorescence detection, the background signal intensity on the pristine PMMA and PC surfaces was first analyzed from images using the near-IR scanner. The hydrophobicity of the substrate surface affects the spreadability of the spotted solution and hence the spot definition on the surface. Therefore, water contact angle measurements were performed to evaluate the surface hydrophobicities of PMMA and PC. At the excitation laser wavelength of 780 nm, PMMA showed an intrinsic background level of ~2,000 cnts per pixel, while PC displayed a level five times higher that of PMMA (~10,000 cnts per pixel). The background levels at other excitation wavelengths have also been
113
reported; at 532 nm, PC produced a fluorescence level 16 times higher than PMMA, and at 635 nm, it was 6 times higher [26]. Longer excitation wavelengths seem to suppress the background levels of both polymers. The reduced background at longer excitation wavelengths can be attributed to the fact that few materials fluoresce in the near-IR regions and the Rayleigh radiation is proportional to the inverse of the fourth-power of the incident wavelength. The contact angle measurements indicated that PC was more hydrophobic than PMMA. The sessile water contact angle on pristine PC was 81-84o while that on pristine PMMA was 70-74o. To examine the adsorption behavior of oligonucleotide probes to the pristine polymer surfaces, hybridization assays of dye-labeled complements were performed for both pristine PMMA and PC slides on which spots of oligonucleotide probes in phosphate buffers had been incubated overnight. The fluorescence images are shown in Figure 5.1. The slides were prepared by spotting 100 M oligonucleotide probe C6-amino 34-mer to the surfaces in a 3 x 3 pattern (0.2 L/spot), incubated at 37oC overnight and hybridized to a 10 nM solution containing the dye-labeled complement (M13 Fwd (-29)-IRD800) in 5x SSPE, 0.1% SDS, at 60oC for 1 hr. In the case of PMMA, the array generated weak fluorescence signals against a uniform background. The image exhibited low SNR, non-uniform in the intensity levels and poor spot definition. The SNR values were used to avoid the background differences from slide to slide. Typical SNR values for PMMA slides were 82 for these hybridization conditions. As a comparison, the covalent immobilization described in Chapter 3 produced much higher SNR for PMMA using similar conditions (~1500, see Table 3.2). In contrast to PMMA, PC demonstrated much higher SNR (~1400) and well-defined, more uniform spots, although the PC background was five times higher than PMMA.
114
A
20000 18000 16000 14000
B
200000
arbitrary intensity
12000 10000 8000 6000 4000 2000 0 0.0 0.5 1.0 1.5 2.0 2.5 3.0 3.5 4.0 4.5
arbitrary intensity
150000
100000
50000
0 0 1 2 3 4 5 6
distance (mm)
distance (mm)
C D Figure 5.1. Fluorescence images of oligonucleotide probes deposited on pristine PMMA and PC slides. The near-IR fluorescence images are shown in panels A (PMMA) and B (PC). The intensity profiles crossing the top three spots are shown in panels C (PMMA) and D (PC). Both slides were prepared by spotting 100 M oligonucleotide probe C6-amino 34-mer in 200 mM phosphate solution to native slide surfaces in a 3 x 3 pattern (0.2 L/spot), incubated at 37oC overnight followed by hybridization in 10 nM dye-labeled complements (M13 Fwd (-29)-IRD 800) in5x SSPE, 0.1% SDS, at 60oC for 1 hr. We noted that the melting point of an oligonucleotide duplex (Tm) formed on the solid substrate might be different from that in the liquid phase. Since the hybridization was conducted at a temperature of 60oC, which was higher than the theoretical melting point for the perfectly matched duplex in solution (Tm = 48.9oC), as can be seen from Figure 5.1, intense fluorescence spots are observed from the PC slide and spots with moderate intensity from the PMMA slide. To verify that these fluorescent signals are coming from specific hybridization of the dye-labeled complements to the surface-immobilized probes, oligonucleotide probes, Zip code 1 and Zip
115
code 3, which possess sequences non-complementary to the dye-labeled M13 Fwd (-29)-IRD800 were immobilized onto the pristine polymer surfaces. After incubating the 10 nM M13 Fwd (-29)IRD800 hybridization solutions to these surface-immobilized probes, no fluorescent signals were observed on these spotted probes (data not shown). Therefore, the fluorescent spots in Figure 5.1 are from the specific hybridization of the dye-labeled M13 Fwd (-29)-IRD800 to the surfaceimmobilized probe C6-amino 34-mer. High temperatures (close to Tm) are generally believed to ensure high stringency for hybridization and reduce nonspecific adsorption of dye-labeled fragments to the surface [42]. Therefore, only perfect matched sequences would be retained by the surface immobilized probes and those mismatched sequences would be removed during washing steps at these temperatures. By conducting the hybridization at 60oC, not only the stringency of hybridization was achieved but the interaction strength of the oligonucleotide probes with the pristine polymer surfaces was also examined indirectly. After incubating the oligonucleotide solution on pristine PC surfaces overnight, the adsorbed probes could withstand being immersed in 60oC or even 75oC solution for one hour followed by thoroughly washing in buffer and not suffer significant loss in the hybridization signal. These results show that the interactions between the oligonucleotide probes and the pristine PC surfaces are very strong. We next examined the effects of oligonucleotide probe concentration, presence of terminal amine groups and probe molecular structures on the hybridization signal. Figure 5.2 shows the dependence of the normalized hybridization signal-to-noise ratio (SNR) on the concentration of the probe, C6-amino 34-mer, incubated on the pristine PMMA surfaces and hybridized to a 10 nM dye-labeled complement solution. The normalized SNR at various probe concentrations is plotted in Figure 5.2 (solid squares). As a comparison, the normalized SNR on
116
the PMMA surface with covalently-immobilized probes formed from various spotting concentrations of oligonucleotide probes to the chemically-modified surfaces is shown in Figure 5.2 (solid circles).
1.4
non-covalent immobilization covalent immobilization
1.2
1.0
normalized SNR
0.8
0.6
0.4
0.2
0.0 0 250 500 750 1000 1250
probe concentration (M)
Figure 5.2. The normalized hybridization SNR for oligonucleotide probes non-covalently immobilized to pristine PMMA surfaces (solid squares) and for oligonucleotide probes covalently immobilized to chemically modified PMMA surfaces (solid circles). Two M, 10 M, 100 M, and 1000 M oligonucleotide probe (C6-amino 34-mer) phosphate solution was spotted onto pristine PMMA slides which were kept in a humidified chamber at 37oC overnight and the slides were hybridized at 60oC for 1 hr. For covalent couplings, oligonucleotide probe dilution series of 0.047, 0.47, 4.7, 47, and 470 M was spotted onto PMMA-CHO slides at 37oC for 4 hrs. The slides were hybridized in 10 pM dye-labeled complements at 37oC overnight (solid circle) and scanned with the near-IR scanner. It has been reported that fluorescence intensity is approximately proportional to the amount of fluorophores present on the surface at low probe concentrations before the probe
117
density becomes too high so that fluorescence quenching may occur [18] [25]. Thus, the fluorescence intensity from hybridization of the immobilized oligonucleotide probes to their dyelabeled complements can be used as an indicator of the extent of probe immobilization. As can be seen, both the SNR values on pristine (solid squares) and on chemically modified PMMA surfaces (solid circles) initially increase and then tend to saturate with oligonucleotide probe concentration. That is, the oligonucleotide probes show monolayer adsorption characteristics on both pristine and chemically modified PMMA substrates although they have different immobilization mechanisms. On the pristine polymer surfaces, the oligonucleotide probes were believed to be immobilized via hydrophobic and/or electrostatic interactions [43] and this process is governed by a Langmuir adsorption mechanism. Therefore, the amount of oligonucleotide probes immobilized on pristine PMMA surfaces increase with the probe concentration as indicated by the SNR increase and saturated after the surface active sites have been taken up. On the chemically modified surfaces, however, the probes are covalently attached to the surface functional groups via monolayer coverage, and the number of oligonucleotide probes allowed to incorporate is limited by the functional group density and steric hindrance [18]. The hybridization image for the non-covalently- or covalently-immobilized C6 amino 34mer on the PMMA substrate is compared in Figure 5.3. As can be seen, the oligonucleotide probes have better spot definition on chemically modified PMMA surfaces than the noncovalently immobilized probes on pristine surfaces. In Figure 5.3, the non-covalentlyimmobilized probes show irregular shapes and patchy hybridization images. In all cases, the covalently-immobilized probes showed higher hybridization intensity than the non-covalently immobilized probes under similar hybridization conditions.
118
A.
35000 30000
B.
6000
5000
arbitrary intensity
arbitrary intensity
25000 20000 15000 10000
4000
3000
2000
5000 0 0 1 2 3 4 5 6 7
1000 0.0 0.5 1.0 1.5 2.0 2.5 3.0 3.5 4.0
distance (mm)
distance (mm)
D C Figure 5.3. The images of hybridized C6 amino 34-mer probes covalently-immobilized on chemically-modified PMMA surfaces (A) and non-covalently-immobilized on pristine PMMA surfaces (B). The intensity profiles crossing the spots are illustrated in panels C (covalentlyimmobilized) and D (non-covalently immobilized). A 4.7 M oligonucleotide probe (0.2 L/spot) was incubated on PMMA-CHO surfaces for 4 hrs and hybridized in 10 nM dye-labeled complement solution at 37oC for 1 hr; 2 M oligonucleotide probe phosphate solution (0.2 L/spot) was incubated on pristine PMMA surfaces overnight and hybridized in 10 nM dyelabeled complements at 60oC for 1 hr.
Figure 5.4 shows the effect of oligonucleotide probe type, including (A) C6-amino 34mer, (B) C6-amino 19-mer, (C) 30-dT 49-mer, and (D) 15-dT 34-mer on the hybridization assay. The first two probes, A and B, contain terminal amine modifications, probes A and D contain 15 and probe C a 30 poly-dT nucleotide spacer (for sequences see Table 5.1).
119
2200
60
2000
pristine PMMA
50
pristine PC
1800 1600
40
1400 1200
SNR
SNR
A B C D
30
1000 800 600
20
10
400 200
0
0 A B C D
oligonucleotide probe
oligonucleotide probe
B Figure 5.4. The effect of oligonucleotide probe types on SNR values for (A) pristine PMMA and (B) pristine PC. One hundred M oligonucleotide Probe A (C6-amino 34-mer), Probe B (C6amino 19-mer), Probe C (30-dT 49-mer), and Probe D (15-dT 34-mer) were spotted onto pristine surfaces and the slides were incubated at 37oC overnight. The spotted slides were hybridized in 10 nM M13 Fwd (-29)-IRD 800 in 5x SSPE, 0.1 % SDS hybridization solution at 60oC for 1 hr. the cleaned and dried slides were scanned under the scanner at laser power of 44 W with scanning step of 50.8 m/step, and integration time of 0.1 s per pixel. As can be seen from Figure 5.4, the presence of the terminal amine group was essential for oligonucleotide probe adsorption to the pristine PMMA surface. For example, the two oligonucleotide probes with amine groups incorporated into their 5 terminus demonstrated hybridization signals (SNR&gt;14), whereas the two probes without amine groups showed negligible hybridization signals (SNR&lt;2). The use of amine-modified oligonucleotide probes in the presence of CTAB (a cationic detergent) resulted in better immobilization efficiency on pristine polymer substrates compared to the use of unmodified oligonucleotide probes was also observed in a recent publication by Liu and coworkers [26]. From Figure 5.4, we could also observe that the incorporation of a poly-dT spacer between the hybridization sequence and the terminal amine groups also increased the hybridization signal. The SNR value of oligonucleotide Probe A (C6-amino 34-mer containing 15 dT spacer) was higher than that on Probe B (C6-amino 19-mer containing the 19-mer hybridization sequence only). Clearly, the spacer sequence
A
120
improved hybridization efficiency as reported by Guo and co-workers for end-point attachment of oligonucleotide probes [18]. Based on the above observations, we propose an electrostatic interaction mechanism for terminal amine modified oligonucleotide probe attachment to pristine PMMA surfaces as depicted in Figure 5.5. The terminal amine group in oligonucleotide probes may be positively charged at the pH value used here (pH = 8.36), and thus, the oligonucleotide probe was docked at its 5 end to the PMMA substrate surface, which is negatively charged [44]. As shown in Figure 5.5, the incorporation of a poly-dT spacer between the recognition sequence and the immobilized end increases the distance of the actual hybridization sequence from the substrate surface so that steric hindrance and substrate interferences are reduced, allowing the dye-labeled targets easy access to the surface-immobilized targets [18]. For non-amine terminal modified oligonucleotide probes, the lack of such positively charged terminus leads to undetectable immobilization onto pristine PMMA surfaces.
Recognition sequence Amine groups Poly-dT tail
Pristine PMMA Figure 5.5. Proposed adsorption mechanism of amine-modified oligonucleotide probes onto pristine PMMA surfaces. In contrast, all four immobilized oligonucleotide probes (see Figure 5.4) show relatively high SNR values on pristine PC surfaces in spite of the absence or presence of terminal amine
121
groups. In addition, the absence of the poly-dT spacer showed little effect on the hybridization signal. It has been reported that aminolysis of the polycarbonate backbone with primary amines can modify the polymer surface to introduce hydrophilic properties without affecting the bulk polymer. As shown in Figure 5.6, the aromatic carbonate linkage is readily cleaved by primary amines to form urethane groups [45]. The cleavage of the polycarbonate backbone by the amines result in the formation of two fragments, one terminated by a urethane group and the other terminated by a phenolic hydroxyl group. The reaction takes place at a relatively low temperature and is not limited to primary aliphatic amines. With the proper conditions, the cleavages occur at only a few points so that relatively large fragments are produced. These fragments are large enough to retain their original positions within the polymer matrix.
CH 3 C CH 3 O OCO CH 3 C CH 3
O
O
RNH 2
O
CH 3 C CH 3
OH
O RHN C O
CH 3 C CH 3
O
Figure 5.6. Aminolysis of PC main chain with primary amines [45]. R = polynucleotide in the case on pristine PC surfaces. The above process may explain the adsorption of oligonucleotide probes on pristine PC surfaces. When the oligonucleotide probes were incubated with the pristine PC surface, the terminal amine functional groups of the oligonucleotide probes may incorporate into the PC surface through a mechanism similar to that illustrated in Figure 5.6. In the absence of terminal amine groups, the oligonucleotide probes may be immobilized to the pristine PC surfaces through their secondary amines in the poly-dT tails or other amine groups in nitrogenous bases
122
comprising the oligonucleotide probes, since the aminolysis of the PC backbone is not limited to only primary amines [45]. If the amines in the nitrogenous bases in the hybridization probes were used as the immobilization points, the hybridization yield of the immobilized probes may experience inaccessibility during base pair formation in the hybridization steps. The immobilization of oligonucleotide probes on other polymer substrates, such as polystyrene plates, hydrophobic interactions between the oligonucleotide probes and the hydrophobic regions of the polymer have been reasoned to be the major underlying mechanism for immobilization and the presence of reagents, like low concentrations of EDC [46], NaCI [47], or cationic detergent CTAB [48] are needed to decrease electrostatic repulsion between the negative phosphate-sugar backbone in the oligonucleotide probe and the negatively charged groups on the polystyrene surface for the enhancement of the binding of oligonucleotide molecules onto polymer substrates [46]. In the case of PC, an oligonucleotide probe in 200 mM phosphate buffer without added components, such as CTAB, EDC or NaCI, could be immobilized to pristine PC surfaces with the slide showing strong hybridization signals. According to the reactions illustrated in Figure 5.6, we propose that covalent bonds were formed for oligonucleotide probes immobilization onto pristine PC surfaces. In addition, the surfaces of polycarbonate are hydrophobic (water droplet contact angle of 81-84o), which prevents the spread of the spotted oligonucleotide probe solution and facilitates the fabrication of high density arrays with well-defined boundaries. Thus, it is possible to use pristine PC for oligonucleotide array fabrication, eliminating the necessity of surface modification steps or modification to the oligonucleotide probes.
123
Deposition of Oligonucleotide Probes on UV-Patterned PC and PMMA Surfaces In this section, the polymer surfaces were exposed to deep-UV light through a metal mask and then an oligonucleotide solution containing 1-ethyl-3-(3-dimethylaminopropyl) carbodiimide hydrochloride (EDC) was spotted onto these UV-patterned PC and PMMA substrates. EDC was utilized here because it is a widely accepted water soluble crosslinking agent for forming amide bonds between primary amines and carboxylate groups [49]. Its reaction mechanism is illustrated in Figure 5.7.
H 3C
NCN EDC
CH 3 NH CI CH 3
O R HO carboxylate containing molecule
R O O H 2N R' R CH 3 H3C N N NH H CH 3 O-Acylisourea active intermediate O R' N H amide bond formation
H2O
O H 3C N H N H CH 3 NH CH 3
O R HO regenerated carboxylate compound
isourea by-product
Figure 5.7. The catalytic mechanism of EDC in amide bond formation [49]. The carbodiimide first reacts with carboxylate-containing substrates to form a highly reactive, O-acylisourea intermediate. This active species can then react with primary amine containing oligonucleotide probes to form an amide bond between the probes and the polymer substrate, which may contain carboxylate groups produced from UV-irradiation [50]. In addition, water in the aqueous solution may compete with the above reaction to cleave the activated ester intermediate, forming an isourea and regenerating the carboxylate group [48]. EDC is also a very 124
poor cationic detergent that may hinder the passive adsorption of oligonucleotides to solid substrates at concentrations above its critical micelle concentration (cmc), which has been estimated to be 300-500 mM [20] [51]. As a result, the oligonucleotide probe solutions with an EDC concentration lower than the cmc was adopted for the covalent immobilization of oligonucleotide probes to the polymer surfaces in these studies.
A
190000
B
5000
4500
arbitrary intensity
arbitrary intensity
0.0 0.5 1.0 1.5 2.0 2.5 3.0
180000
4000
170000
3500
160000
3000
2500
distance (mm)
0.0
0.5
1.0
1.5
2.0
2.5
3.0
distance (mm)
C D Figure 5.8. UV-patterning of oligonucleotide probes on PC and PMMA surfaces. One mm-thick PC and PMMA slides were UV-irradiated through a metal mask in a UV station at room temperature for 10 mins. Five l of the oligonucleotide probe (C6-amino 34-mer) 100 M in 0.1 M MES buffer containing 5 mM EDC was spotted onto the surface of the UV-patterned slides. The dispensed slides were kept at 37oC for 8 hrs, then, washed in ddH2O and dried in pressurized air. The slides were hybridized in a 10 nM hybridization solution at 60oC for 1 hr. The rinsed slides were scanned at an excitation laser power of 44 W and a step resolution of 50.8 m. Illustrated in the figure are the near-IR images of hybridized UV-patterned slides (A) PC and (B) PMMA; the hybridization intensity profiles crossing the photo-patterned oligonucleotide probes (C) PC and (D) PMMA. The spots size on PC is ~ 0.76 mm and on PMMA is ~0.71 mm. The incomplete spot on the right bottom corner of the PMMA slide was caused by the intercrossing of the spot formed by the 5 L probe solution deposited onto the surfaces and the exposed areas.
125
After the hybridization and washing steps, the slides were rinsed with 0.2 N NaOH for 15 mins to remove non-covalently bound oligonucleotides [24]. The fluorescence signals acquired from the UV-exposed PC and PMMA substrates following hybridization to their fluorophorelabeled complement solution are illustrated in Figure 5.8. Both slides were exposed for 10 mins through the metal mask. The fluorescence intensity contrast between the UV-exposed and non-exposed regions indicated that more oligonucleotide probes were immobilized in the UV-exposed regions for both polymers. Since EDC is an indispensable reagent for covalent bond formation between carboxylate groups and amine groups and the hybridized slides were rinsed thoroughly with 0.2 N NaOH, we can contribute the immobilization of oligonucleotide probes to the UV-exposed regions as resulting from a covalent bond. However, the UV-exposed PC slide demonstrated significant amounts of nonspecific adsorption and the spot definition was not well defined. Although the hybridization intensity of oligonucleotide probes was comparable to those on pristine PC surfaces, the SNR value on the exposed region was only ~ 26 due to the high level of nonspecific adsorption. In contrast, the oligonucleotide probes were well-defined on the UV-treated PMMA slide. In this experiment, the exposed region does not show material loss or surface roughness changes as examined using a profilometer (data not shown) for both PMMA and PC slides. Further detail work will be needed in the future for a better understanding of the relations between the exposure dose and the introduced functional groups density and the appropriate dose to improve the discriminating immobilization of oligonucleotide probes on the exposed and nonexposed regions.
126
Conclusions
Oligonucleotide probes were found to demonstrate strong adsorption characteristics to pristine PC surfaces as indicated by the high yield of specific hybridization signals to its dyelabeled complements. The immobilization can be attributed to the covalent bond formation through the aminolysis of the PC backbone with the terminal amine functional groups or secondary amines in the poly-dT tails. When the pristine PC was UV-irradiated through a metal mask and oligonucleotide probe solution containing EDC was incubated for a specific length of time, comparable hybridization intensity was obtained on UV-treated regions as those on pristine PC surfaces, however, the SNR was much lower due to the existence of nonspecific adsorptions to these slides. In addition, the oligonucleotide probe has well-defined spot boundary on pristine PC. Thus, it is more efficient to use pristine PC directly for the construction of DNA microarrays, eliminating the surface modification step which is normally an indispensable step, in most microarray constructions. As for pristine PMMA, only oligonucleotide probes with terminal amine modifications can be moderately immobilized to the PMMA surface through electrostatic interactions. The hybridization intensity is less homogeneous and the boundary of the spotted probe not well defined. When the oligonucleotide solution containing EDC was incubated onto the UVpatterned PMMA surface, well defined oligonucleotide spot, which was defined by the metal mask, was formed. The low background indicated minimized non-specific adsorption at non-UV exposed regions and the hybridization signal of the oligonucleotide probe was basically uniform. However, the proper dose of UV-exposure needs to be optimized to increase the loading of covalently immobilized oligonucleotide probes so that a strong contrast of hybridization intensity between the exposed regions and un-exposed regions will result.
127
Future Work
The proper dose of the UV irradiation on PMMA substrate to introduce appropriate surface functional group density for the immobilization of oligonucleotide probes will be fathomed in future work. As an example of the application, universal zip code arrays will be photon patterned onto the PMMA surfaces and perpendicular microfluidic arrays will be superimposed on the arrayed surfaces so that parallel, quick hybridization-based assays can be implemented.
References
1. 2. 3. 4. 5. 6. Watson, J.D., Crick, F.H., Nature, 1953, 248, 737-738. Bassett, D.E., Eisen, M.B., Boguski, M.S., Nat. Genet., 1999, 21, 51-55. Lemieux, B., Aharoni, A., Schena, M., Mol. Breeding, 1998, 4(4), 277-289. Marx, J., Science, 2000, 289(5485), 1670-1672. Housman, D., Ledley, F.D., Nat. Biotechnol., 1998, 16(6), 492-493. Nuwaysir, E.F., Bittner, M., Trent, J., Barrett, J.C., Afshari, C.A., Mol. Carcinogen, 1999, 24, 153-159. Marth, G.T., Korf, I., Yandell, M.D., Yeh, R.T., Gu, Z.J., Zakeri, H., Stitziel, N.O., Hillier, L., Kwok, P.Y., Gish, W.R., Nat. Genet., 1999, 23(4), 452-456. Soper, S.A., Ford, S.M., Qi, S., McCarley, R.L., Kelly, K., Murphy, M.C., Anal. Chem., 2000, 72(19), 642A-651A. Becker, H., Fabrication of Polymer Microfluidic Devices, in Biochip Technology, Cheng, J., Kricka, L. J., Editor, 2001, Harwood Academic Publishers, London. Martynova, L., Locascio L.E., Gaitan, M., Kramer, G.W., Christensen, R.G., MacCrehan, W.A., Anal. Chem., 1997, 69(23), 4783-4789. Fodor, S.P., Read, J.L., Pirrung, M.C., Stryer, L., Lu, A.T., Solas, D., Science, 1991, 251, 767-773.
7.
8.
9.
10.
11.
128
12.
Pease, A.C., Solas, D., Sullivan, E.J., Cronin, M.T., Holmes, C.P., Fodor, S.P.A., Proc. Natl. Acad. Sci. USA, 1994, 91, 5022-5026. McGall, G.H., Barone, A.D., Diggelmann, M., Fodor, S.P.A., Gentalen, E., Ngo, N., J. Amer. Chem. Soc., 1997, 119(22), 5081-5090. Lipshutz, R.J., Fodor, S.P., Gingeras, T.R., Lockhart, D.J., Nat. Genet., 1999, 21, 20-24. DeRisi, J.L., Iyer, V.R., Brown, P.O., Science, 1997, 278(5338), 680-686. Schena, M., Shalon, D., Davis, R.W., Brown, P.O., Science, 1995, 270(5235), 467-470. Schena, M., Shalon, D., Heller, R., Chai, A., Brown, P.O., Davis, R.W., Proc. Natl. Acad. Sci. USA, 1996, 93(20), 10614-10619. Guo, Z., Guilfoyle, R.A., Thiel, A.J., Wang, R., Smith, L.M., Nucleic Acids Res., 1994, 22, 5456-5465. Nagata, Y., Yokota, H., Kosuda, O., Yokoo, K., Takemura, K., Kikuchi, T., FEBS Lett., 1985, 183(2), 379-382. Nikiforov, T.T., Rogers, Y.H., Anal. Biochem., 1995, 227(1), 201-209. Schena, M., DNA Microarrays: A practical approach. 1999, Oxford University press, Oxford. Luneberg, E., Jensen, J.S., Frosch, M., J Clin Microbiol, 1993, 31, 1088-1094. Hirayama, H., Tamaoka, J., Horikoshi, K., Nucl. Acids. Res., 1996, 24, 4098-4099. Matson, R.S., Rampal, J.B., Coassin, P.J., Anal. Biochem., 1994, 217, 306-310. Wang, Y., Vaidya, B., Farquar, H. D., Stryjewski, W., Hammer, R. P., McCarley, R. L., Soper, S. A., Cheng, Y.-W., Barany, F., Soper, S.A, Anal. Chem., 2003, 75(5), 11301140. Liu, Y.J., Rauch, C.B., Anal. Biochem., 2003, 317(1), 76-84. Duran, L., Swanson, M. J., Amos, R.A., Hu, S.P.J., Guire, P. E., Reagent and method for attaching target moleules to a surface, 1999, USA. Hirt, R.C., Searle, N.Z., Schmitt, R.G., Soc. Plastics Engs., Trans., 1961,1, 21. Rabek, J.F., Polymer photodegradation : mechanisms and experimental methods. 1995, Chapman &amp; Hall, London.
13.
14. 15. 16. 17.
18.
19.
20. 21.
22. 23. 24. 25.
26. 27.
28. 29.
129
30.
Hawkins, W.L., Polymer degradation and stabilization. in Polymers/Properties and Applications, Harwood, H.J., editor, Vol. 8., 1984, Springer-Verlag, New York. Liu, Y.J., Ganser, D., Schneider, A., Liu, R., Grodzinski, P., Kroutchinina, N., Anal. Chem., 2001, 73, 4196 - 4201. Johnson, T.J., Waddell, E.A., Kramer, G.W., Locascio, L.E., Applied Surface Science, 2001, 181, 149-159. Xu, Y., Vaidya, B., Patel, A. B., Ford, S. M., McCarley, R. L., Soper, S. A., Anal. Chem., 2003, 75, 2975 - 2984. Ford, S.M., Davies, J., Kar, B., Qi, S.D., McWhorter, S., Soper, S.A., Malek, C.K., J. Biomech. Eng-T. ASME, 1999, 121(1), 13-21. Soper, S.A., Henry, A.C., Vaidya, B., Galloway, M., Vabuyele, M., McCarley, R.L., Anal. Chim. Acta, 2002, 470, 87-99. Williams, D.C., Soper, S. A., Anal. Chem., 1995, 67, 3427 - 3432. Liu, Y.J., Rauch, C.B., Stevens, R.L., Lenigk, R., Yang, J.N., Rhine, D.B., Grodzinski, P., Anal. Chem., 2002, 74(13), 3063-3070. Anderson, R.C., Su, X. , Bogdan, G.J. , and Fenton, J., Nucl. Acids. Res., 2000, 28, e60. Xu, J.D., Locascio, L., Gaitan, M., Lee, C.S., Anal. Chem., 2000, 72(8), 1930-1933. Fan, Z.H., Mangru, S., Granzow, R., Heaney, P., Ho, W., Dong, Q.P., Kumar, R., Anal. Chem., 1999, 71(21), 4851-4859. Guillet, J.E., Polymer Photophysics and Photochemistry: an introduction to the study of photoprocesses in macromolecules. 1985, Cambridge, Cambridge University Press, U.K.. Ross, J., ed. Nucleic Acid Hybridization - Essential Techniques, 1998, John Wiley &amp; Sons, New York. Schwarz, A., Rossier, J.S., Roulet, E., Mermod, N., Roberts, M.A., Girault, H.H., Langmuir, 1998, 14(19), 5526-5531. Galloway, M., Stryjewski, W., Henry, A., Ford, S.M., Llopis, S., McCarley, R.L., Soper, S.A., Anal. Chem., 2002, 74(10), 2407-2415. Caldwell, J.R., Jackson, W. J. Jr., Polymer Preprints, Am. Chem. Soc., Div. Polym. Chem., 1967, 8, 687-692.
31.
32.
33.
34.
35.
36. 37.
38. 39. 40.
41.
42.
43.
44.
45.
130
46.
Nikiforov, T.T., Rendle, R.B., Kotewicz, M.L., Rogers, Y.H., PCR Methods Appl., 1994, 3(5), 285-291. Nikiforov, T.T., Rendle, R.B., Goelet, P., Rogers, Y.H., Kotewicz, M.L., Anderson, S., Trainor, G.L., Knapp, M.R., Nucleic Acids Res., 1994, 22, 4167-4175. Gilles, M.A.H., A.Q.; Borders, C.L., Anal. Biochem., 1990, 184(2), 244-248. Hermanson, G.T., Bioconjugate techniques. 1996, Academic Press, San Diego, 169-173. Williams, A., Ibraihim, I.T., J. Amer. Chem. Soc., 1981, 103(24), 7090-7095. Nikiforov, T.T., Rendle, R.B., Goelet, P., Rogers, Y.H., Kotewicz, M.L., Anderson, S., Trainor, G.L., Knapp, M.R., Nucleic Acids Res., 1994, 22(20), 4167-4175.
47.
48. 49. 50. 51.
131
Chapter 6 Summary and Future Work
Summary
The focus of this research was to develop polymer-based oligonucleotide arrays in microfluidic channels for gene mutation detection. Poly (methylmethacrylate) (PMMA) was selected as the polymer substrate because of the existence of surface functional groups and established modification methods as well [1-2] as well as its wide applications in micro-fluidic devices [3-6]. Near-IR fluorescence was used to examine the hybridization results on polymer-based oligonucleotide arrays. In addition, preliminary results of oligonucleotide immobilization on native and UV patterned PMMA and polycarbonate (PC) substrates were discussed. In Chapter 1, basic concepts of DNA structural units; nucleotides, double-stranded structures of DNA molecules and the denaturation/rehybridization of the duplex structures, genetic codes and genetic mutations were introduced. DNA microarrays, the array substrates, array construction methods-in situ synthesis and deposition procedures, applications and limitations of DNA microarrays were reviewed as well. In Chapter 2, concepts of fluorescence detection, such as a Jablonskis diagram, excitation light source and detector, and confocal microscope/scanner were reviewed. In particular, near-IR fluorescence detection and related dyes were emphasized. Finally, the operation of a home-built near-IR fluorescence scanner was optimized to maximize signal to noise ratio (SNR) and to minimize dye photo-bleaching. At micro-watts level of laser excitation power, the near-IR IRD800 dye-labeled primer on PMMA surfaces showed good photo-stability during repeated scanning runs up to 11 scans. The detection limit of the scanner was found to be around 0.003 dye-labeled primer /m2 at SNR=2. 132
In Chapter 3, a protocol of covalent end-immobilization of oligonucleotide probes to chemically modified PMMA surfaces was established. The PMMA surface was aminated using N-lithioethylenediamine solution to introduce primary amine groups followed by covalent attachment of amine-terminated oligonucleotide probes via a homo-bi-functional cross-linker, glutaric-dialdehyde. Construction parameters, such as the surface amine group density, oligonucleotide probe concentration and coupling time were optimized. It was found that the covalent immobilization of probes could be realized at an oligonucleotide concentration as low as 10-8 M and the detection limit of oligonucleotide array-based analysis at the pM level. Moreover, the coupling chemistry was stable enough to withstand 12 denaturation /rehybridization cycles without significant signal loss. In Chapter 4, low-density oligonucleotide arrays were assembled into micro-fluidic channels hot-embossed in PMMA to allow the detection of low-abundant mutations in gene fragments (K-ras) that carry point mutations with high diagnostic value for colorectal cancers. The fabrication of a mold insert for the production of polymer micro-chip structures, the hotembossing of PMMA chips and the ligation detection mechanisms were introduced. The low glass transition temperature (Tg = 105oC) of PMMA made it is possible to spot the oligonucleotide probes on the bottom of the micro-fluidic channel before sealing the PMMA micro-fluidic devices and the performance of the immobilized probes was not degraded following hybridization. Improved hybridization kinetics in micro-fluidic oligonucleotide arrays was observed. The saturation of the surface immobilized probes with its dye-labeled complements could be completed within a minute. In Chapter 5, preliminary work on the immobilization of oligonucleotide probes onto pristine or UV-patterned PC/PMMA surfaces was investigated. It was found that oligonucleotide
133
probes with terminal amine modifications or poly-dT tails covalently bound to the pristine PC surfaces and high hybridization SNR values were obtained on these surface-immobilized probes. Moderate adsorption of amine-modified oligonucleotide probes was observed on pristine PMMA surfaces and the spot boundary was not well-defined and hybridization signals were patchy. Deep UV-patterning using a through-hole mask showed promising results on PMMA substrates. Oligonucleotide probes immobilized on the UV-activated regions showed clear boundaries defined by the mask. However, the proper dose of UV-exposure still needs to be optimized in future work to increase the probe density. It was also noted that UV-exposure of PC surfaces does not improve oligonucleotide probe immobilization compared to the pristine material. Extensive non-specific adsorption and ill-defined spot boundaries were found on UVpatterned PC slides.
Future Work
Surface Modification Although the PMMA surface has been successfully modified with N-
lithioethylenediamine solutions to introduce primary amine groups, the reagents involved are moisture and air sensitive, which demand extra cautions in carrying out the modification steps. A simple, user friendly alternative surface modification method is desired. Plasma dry etching seems like an ideal candidate. There are articles dealing with plasma etching to introduce amines [7-8] or other functional groups [9] to polypropylene surfaces. With the plasma method, the surface modification can be easily expanded to other polymer substrates so that more choices could be made on selecting the appropriate materials for micro-fluidic device fabrication and application.
134
Electro-pumping As described in Chapter 4, a syringe pump was connected to the PMMA micro-fluidic channels for fluid transport. This external format is less efficient in performing micro-scale genetic analysis because of larger sample volumes and difficulty to integrate into the polymer chips. Thus, it is necessary to develop an integrated pumping system for effective transportation and manipulation of aqueous solutions in the microfluidic channels. Electro-pumping is an established method for transporting liquid samples in glass- or quartz-based micro-channels [10]. In this work, we attempted to use an electric field to transport sample solutions in the hot-embossed PMMA channels for the hybridization based analysis. As shown in Figure 6.1, two-millimeter-diameter holes were mechanically drilled through the three reservoir positions on the hot-embossed PMMA micro-fluidic chip. The surface activation and oligonucleotide immobilization chemistries were similar to those described in Chapters 3 and 4. The 250 m-thick PMMA cover plate was heat annealed to the microchip to seal the channels and one piece of film heater (Flexible Heater KHLV-0502/5-P, Omegalux, Stamford, CT) was attached to the bottom of the PMMA cover plate for controlling the hybridization channel temperature. To begin with, the microfluidic channel was filled with 0.2x SSPE buffer and two Pt electrodes were inserted in the reservoirs spanning the hybridization channel. A 600 V voltage was applied across the channel (200 V/cm) and the current was monitored via a multimeter, which was paralleled to a 10 K resistor integrated in the circuit. After equilibrating the system for 45 mins, the three reservoirs were cleaned and 10 l of 0.2x buffer and 10 l dye-labeled complements (M13 Fwd (-29) IRD800, LI-COR) were filled inside the two reservoirs flanking the main channel with the rest of the reservoir filled with buffer. The heat controller was used to
135
set the channel temperature to 55oC. Either the cathode or the anode platinum electrode was inserted inside the reservoir with the dye-labeled DNA samples and after connecting the external electric field for 1 hr, the solution in the opposite end reservoir was examined under the near-IR fluorescence scanner to check the existence of fluorescence. In both cases, the fluorescence signal was observed in the buffer end solution. However, no hybridization signal was observed in the microfluidic channel, which means even though the dye-labeled DNA samples was transported through the microfluidic channel from one end to the other, the transported DNA targets were not captured by the immobilized probes.
Figure 6.1. The schematic set up of the electro-pumping experiment. Two millimeter diameter holes were mechanically drilled through the PMMA chip to serve as the reservoirs for buffer and DNA samples. The PMMA microchip was surface activated using the wet chemistry method described in Chapter 3 and oligonucleotide probes were immobilized to the bottom of the main channel. A 0.25 mm cover plate was annealed to the microchip to seal the channel and a film heater was attached to the bottom of the cover plate to regulate the temperature. The reservoirs were first filled with 0.2x SSPE buffer to equilibrate the system, then the reservoirs were cleaned of the residual solutions. The dye labeled complement solutions were pup in the positive or negative reservoirs and the rest reservoirs filled with 0.2x SSPE buffer. The 600-V external power supply was turned on to perform the electro-pumping process.
There are two possible electro-driving forces [11] that can be used in the transportation of oligonucleotide molecules through the micro-fluidic channel, the electrophoretic flow of
136
negatively charged oligonucleotide molecules in the electric field from negative end to positive end and the electro-osmotic flow (EOF) of the bulk fluidic solution due to the existence of the double-electric layer on the channel walls and the application of an external electric field. The direction of the EOF is substrate material dependent. In microfluidic channels hot-embossed into PMMA, there exists a weaker EOF with the direction from anode (positive) to cathode (negative) and the EOF is less pHdependent than that in glass-based channels [3] [12]. The direction of the EOF was revered, from negative to positive after the micro-fluidic channel was aminated [2]. When the aminated surface was converted to aldehyde terminated and the residual aldehyde group was reduced with sodium borohydride after the coupling of oligonucleotide probes, it is expected that the channel surface will be negatively charged under basic buffer medium and the normal direction of EOF in the channel will be recovered. In this case, the direction of the EOF in the PMMA oligonucleotide microfluidic channels will be opposite to that of the electrophoretic movement of the oligonucleotide samples. In the future work, these issues, such as magnitude of the electrophoretic migration and electro-osmotic flow in the capped PMMA micro-fluidic channels, and which pumping force is more appropriate for transportation of the dye-labeled oligonucleotide samples through the PMMA microfluidic arrays needs to be addressed. The capability of the dye-labeled complementary sequences to hybridize under the influence of an electric field to the immobilized oligonucleotide probes also needs to be examined. UV-patterning In the future, the UV-patterning work will be focused on PMMA substrates. The effects of the exposure dose of deep UV irradiation on the surface functional group density and the amount of immobilized oligonucleotide probes as indicated by the intensity of the hybridization
137
signals needs to be monitored closely. An optimal exposure dose will be determined based on the maximum hybridization intensity contrast between the exposed and non-exposed regions. As an application example, UV-patterning of universal zip code arrays coupled to microfluidic channels will be constructed for genetic mutation detection.
References
1. 2. Karandikar, B., Puschett, J., Abstr. Pap. Am. Chem. S., 1989, 197, 126. Henry, A.C., Tutt, T.J., Galloway, M., Davidson, Y.Y., McWhorter, C.S., Soper, S.A., McCarley, R.L., Anal. Chem., 2000, 72, 5331-5337. Galloway, M., Stryjewski, W., Henry, A., Ford, S.M., Llopis, S., McCarley, R.L., Soper, S.A., Anal. Chem., 2002, 74(10), 2407-2415. Ford, S.M., Davies, J., Kar, B., Qi, S.D., McWhorter, S., Soper, S.A., Malek, C.K., J. Biomech. Eng-T. ASME, 1999, 121(1), 13-21. Qi, S., Liu, X., Ford, S., Barrows, J., Thomas, G., Kelly, K., McCandless, A., Lian, K., Goettert, J., Soper, S. A., Lab Chip, 2002, 2, 88-95. Soper, S.A., Ford, S.M., Qi, S., McCarley, R.L., Kelly, K., Murphy, M.C., Anal. Chem., 2000, 72(19), 642A-651A. Chu, T.J., Caldwell, K.D., Weiss, R.B., Gesteland, R.F., Pitt, W.G., Electrophoresis, 1992, 13(3), 105-114. Matson, R.S., Rampal, J., Pentoney, S.L., Anderson, P.D., Coassin, P., Anal. Biochem., 1995, 224(1), 110-116. Hirotsu, T., Nugroho, P., J. Appl. Polym. Sci., 1997, 66, 1049-1057. Harrison, D.J., Flurik, S.K., Fan, Z.H., Effenhauser, C.S., Manz, A., Science, 1993, 261(5123), 895-897. Zanzucchi, P, J., McBride, S, E., Burton, C.A., Cherukuri, S,C., Apparatus and Methods for Controlling Fluid Flow in Microchannels, 1997, US patent No. 5,632,876, David Sarnoff Research Center, Inc., Princeton, NJ. Galloway, M., Soper, S.A., Electrophoresis, 2002, 23(21)